{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Representations, LDA and topics in CORD-19\n",
    "\n",
    "In this notebook, representations of the papers will be calculated based on their text content.\n",
    "Then, from these representations, a modeling of topics will be carried out using the LDA method.\n",
    "\n",
    "Finally, the most relevant papers for each topic will be determined using the PageRank scores of each paper."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!pip install -q -r requirements.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from risotto.references import load_papers_from_metadata_file, build_papers_reference_graph, paper_as_markdown\n",
    "from fastprogress.fastprogress import progress_bar\n",
    "\n",
    "from pathlib import Path\n",
    "\n",
    "import scispacy\n",
    "import en_core_sci_sm\n",
    "import networkx as nx\n",
    "from collections import defaultdict\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Loading paper dataset and re-generating the graph of papers and the corresponding PageRank."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cord19_dataset_folder = \"./datasets/CORD-19-research-challenge\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div>\n",
       "        <style>\n",
       "            /* Turns off some styling */\n",
       "            progress {\n",
       "                /* gets rid of default border in Firefox and Opera. */\n",
       "                border: none;\n",
       "                /* Needs to be in here for Safari polyfill so background images work as expected. */\n",
       "                background-size: auto;\n",
       "            }\n",
       "            .progress-bar-interrupted, .progress-bar-interrupted::-webkit-progress-bar {\n",
       "                background: #F44336;\n",
       "            }\n",
       "        </style>\n",
       "      <progress value='57366' class='' max='57366' style='width:300px; height:20px; vertical-align: middle;'></progress>\n",
       "      100.00% [57366/57366 08:46<00:00]\n",
       "    </div>\n",
       "    "
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "papers, _ = load_papers_from_metadata_file(cord19_dataset_folder)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "G = build_papers_reference_graph(papers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pageranks = nx.pagerank(G)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Paper representations\n",
    "\n",
    "In order to build a representation for each paper, the following libraries will be used:\n",
    "\n",
    "- spaCy: https://spacy.io/\n",
    "- scispaCy: https://allenai.github.io/scispacy/\n",
    "\n",
    "The language model named`en_core_sci_sm` will be used, which has been trained with a corpus of biomedical text with a vocabulary of more than 100.000 words.\n",
    "In case of needing a model with a larger vocabulary, there are some others available."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the biomedical language pipeline\n",
    "nlp = en_core_sci_sm.load()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'be'"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Select a paper to showcase spacy's features\n",
    "sample_paper = list(pageranks.keys())[0]\n",
    "sample_text = \"\\n\".join([ paragraph[\"text\"] for paragraph in sample_paper._file_contents[\"body_text\"]])\n",
    "sample_text\n",
    "\n",
    "doc = nlp(sample_text, disable=[\"tagger\", \"parser\", \"ner\"])\n",
    "doc[17].lemma_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The document tokenized by the `spacy` pipeline is displayed.\n",
    "An interesting thing about using `spacy` with the pretrained language model is that it automatically computes document and token representations vectors.\n",
    "It's a pending task to find out which language model architecture it's used to compute those vectors.\n",
    "\n",
    "A relevant aspect that influences downstream tasks is the number of out--of--vocabulary (OOV) tokens.\n",
    "The following cell makes a quick inspection over a sample paper counting the number of OOV tokens.\n",
    "A continuación, se realizará una iteración sobre los tokens para detectarlos."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "virus found\n",
      "virus found\n",
      "virus found\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(336, 4.801371820520148)"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "num_oov = 0\n",
    "for token in doc:\n",
    "    if token.is_oov and token.string != \"\\n\":\n",
    "        if token.string.endswith(\"virus\"):\n",
    "            print(token, \"not found\")\n",
    "        num_oov += 1\n",
    "    else:\n",
    "        if token.string.endswith(\"virus\"):\n",
    "            print(token, \"found\")\n",
    "num_oov, 100 * num_oov / len(doc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genetic\n",
      "information\n",
      "RNA\n",
      "virus\n",
      "organize\n",
      "efficiently\n",
      "Practically\n",
      "nucleotide\n",
      "genome\n",
      "utilize\n",
      "protein-coding\n",
      "sequence\n",
      "cis-acting\n",
      "signal\n",
      "translation\n",
      "RNA\n",
      "synthesis\n",
      "RNA\n",
      "encapsidation\n",
      "genome\n",
      "expression\n",
      "strategy\n",
      "group\n",
      "positive-strand\n",
      "RNA\n",
      "+\n",
      "RNA\n",
      "virus\n",
      "produce\n",
      "subgenomic\n",
      "sg\n",
      "mRNAs\n",
      "review\n",
      "Miller\n",
      "Koev\n",
      "2000\n",
      "replication\n",
      "genomic\n",
      "RNA\n",
      "mRNA\n",
      "viral\n",
      "replicase\n",
      "supplement\n",
      "generation\n",
      "sg\n",
      "transcript\n",
      "express\n",
      "structural\n",
      "auxiliary\n",
      "protein\n",
      "encode\n",
      "downstream\n",
      "replicase\n",
      "gene\n",
      "genome\n",
      "Sg\n",
      "mRNAs\n",
      "+\n",
      "RNA\n",
      "virus\n",
      "3′-co-terminal\n",
      "genomic\n",
      "RNA\n",
      "different\n",
      "mechanism\n",
      "synthesis\n",
      "virus\n",
      "brome\n",
      "mosaic\n",
      "virus\n",
      "initiate\n",
      "sg\n",
      "mRNA\n",
      "synthesis\n",
      "internally\n",
      "full-length\n",
      "minus\n",
      "strand\n",
      "RNA\n",
      "template\n",
      "Miller\n",
      "et\n",
      "al.\n",
      "1985\n",
      "exemplify\n",
      "red\n",
      "clover\n",
      "necrotic\n",
      "mosaic\n",
      "virus\n",
      "RCNMV\n",
      "rely\n",
      "premature\n",
      "termination\n",
      "minus\n",
      "strand\n",
      "synthesis\n",
      "genomic\n",
      "RNA\n",
      "template\n",
      "follow\n",
      "synthesis\n",
      "sg\n",
      "plus\n",
      "strand\n",
      "truncate\n",
      "minus\n",
      "strand\n",
      "template\n",
      "Sit\n",
      "et\n",
      "al.\n",
      "1998\n",
      "Members\n",
      "order\n",
      "Nidovirales\n",
      "include\n",
      "coronavirus\n",
      "arteriviruses\n",
      "evolve\n",
      "unique\n",
      "mechanism\n",
      "employ\n",
      "discontinuous\n",
      "RNA\n",
      "synthesis\n",
      "generation\n",
      "extensive\n",
      "set\n",
      "sg\n",
      "RNAs\n",
      "review\n",
      "Brian\n",
      "Spaan\n",
      "1997\n",
      "Lai\n",
      "Cavanagh\n",
      "1997\n",
      "Snijder\n",
      "Meulenberg\n",
      "1998\n",
      "Nidovirus\n",
      "sg\n",
      "mRNAs\n",
      "differ\n",
      "fundamentally\n",
      "viral\n",
      "sg\n",
      "RNAs\n",
      "3′-co-terminal\n",
      "5′-co-terminal\n",
      "genome\n",
      "Figure\n",
      "1A\n",
      "5′\n",
      "common\n",
      "leader\n",
      "sequence\n",
      "65–221\n",
      "nucleotide\n",
      "derive\n",
      "5′\n",
      "end\n",
      "genomic\n",
      "RNA\n",
      "attach\n",
      "3′\n",
      "sg\n",
      "RNA\n",
      "mRNA\n",
      "body\n",
      "model\n",
      "forward\n",
      "explain\n",
      "co-transcriptional\n",
      "fusion\n",
      "non-contiguous\n",
      "part\n",
      "nidovirus\n",
      "genome\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "Figure\n",
      "1B\n",
      "C\n",
      "Central\n",
      "model\n",
      "short\n",
      "transcription-regulating\n",
      "sequence\n",
      "TRSs\n",
      "present\n",
      "3′\n",
      "end\n",
      "leader\n",
      "5′\n",
      "end\n",
      "sg\n",
      "RNA\n",
      "body\n",
      "region\n",
      "genomic\n",
      "RNA\n",
      "TRS\n",
      "copy\n",
      "mRNA\n",
      "connect\n",
      "leader\n",
      "body\n",
      "Spaan\n",
      "et\n",
      "al.\n",
      "1983\n",
      "Lai\n",
      "et\n",
      "al.\n",
      "1984\n",
      "Synthesis\n",
      "sg\n",
      "mRNAs\n",
      "initially\n",
      "propose\n",
      "prime\n",
      "free\n",
      "leader\n",
      "transcript\n",
      "base-pair\n",
      "complementary\n",
      "TRS\n",
      "region\n",
      "full-length\n",
      "minus\n",
      "strand\n",
      "extend\n",
      "subsequently\n",
      "sg\n",
      "plus\n",
      "strand\n",
      "Figure\n",
      "1B\n",
      "Baric\n",
      "et\n",
      "al.\n",
      "1983\n",
      "1985\n",
      "model\n",
      "base\n",
      "report\n",
      "sg\n",
      "minus\n",
      "strand\n",
      "present\n",
      "coronavirus-infected\n",
      "cell\n",
      "Lai\n",
      "et\n",
      "al.\n",
      "1982\n",
      "subsequent\n",
      "discovery\n",
      "molecule\n",
      "Sethna\n",
      "et\n",
      "al.\n",
      "1989\n",
      "result\n",
      "reconsideration\n",
      "initial\n",
      "leader-primed\n",
      "transcription\n",
      "model\n",
      "Sawicki\n",
      "Sawicki\n",
      "1995\n",
      "propose\n",
      "alternative\n",
      "model\n",
      "Figure\n",
      "1C\n",
      "discontinuous\n",
      "step\n",
      "occur\n",
      "minus\n",
      "instead\n",
      "plus\n",
      "strand\n",
      "RNA\n",
      "synthesis\n",
      "model\n",
      "minus\n",
      "strand\n",
      "synthesis\n",
      "attenuate\n",
      "copy\n",
      "body\n",
      "TRS\n",
      "plus\n",
      "strand\n",
      "template\n",
      "nascent\n",
      "minus\n",
      "strand\n",
      "TRS\n",
      "complement\n",
      "3′\n",
      "end\n",
      "transfer\n",
      "leader\n",
      "TRS\n",
      "attach\n",
      "mean\n",
      "TRS\n",
      "TRS\n",
      "base\n",
      "pair\n",
      "RNA\n",
      "synthesis\n",
      "reinitiated\n",
      "complete\n",
      "sg\n",
      "minus\n",
      "strand\n",
      "add\n",
      "complement\n",
      "genomic\n",
      "leader\n",
      "sequence\n",
      "Subse\n",
      "quently\n",
      "sg\n",
      "minus\n",
      "strand\n",
      "template\n",
      "sg\n",
      "mRNA\n",
      "synthesis\n",
      "presence\n",
      "leader\n",
      "complement\n",
      "3′\n",
      "end\n",
      "allow\n",
      "use\n",
      "RNA\n",
      "signal\n",
      "direct\n",
      "genome\n",
      "synthesis\n",
      "full-length\n",
      "minus\n",
      "strand\n",
      "site-directed\n",
      "mutagenesis\n",
      "TRSs\n",
      "arterivirus\n",
      "equine\n",
      "arteritis\n",
      "virus\n",
      "EAV\n",
      "show\n",
      "previously\n",
      "base\n",
      "pair\n",
      "sense\n",
      "leader\n",
      "TRS\n",
      "antisense\n",
      "body\n",
      "TRSs\n",
      "crucial\n",
      "sg\n",
      "mRNA\n",
      "synthesis\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999a\n",
      "base\n",
      "pair\n",
      "step\n",
      "nascent\n",
      "strand\n",
      "transfer\n",
      "process\n",
      "essential\n",
      "model\n",
      "outline\n",
      "Figure\n",
      "1\n",
      "EAV\n",
      "genomic\n",
      "RNA\n",
      "contain\n",
      "sequence\n",
      "match\n",
      "leader\n",
      "TRS\n",
      "precisely\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "den\n",
      "Boon\n",
      "et\n",
      "al.\n",
      "1996\n",
      "Pasternak\n",
      "et\n",
      "al.\n",
      "2000\n",
      "suggest\n",
      "leader\n",
      "body\n",
      "TRS\n",
      "similarity\n",
      "necessary\n",
      "sufficient\n",
      "strand\n",
      "transfer\n",
      "occur\n",
      "gain\n",
      "insight\n",
      "cis-acting\n",
      "signal\n",
      "regulate\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "perform\n",
      "comprehensive\n",
      "site-directed\n",
      "mutagenesis\n",
      "study\n",
      "EAV\n",
      "leader\n",
      "body\n",
      "TRSs\n",
      "nucleotide\n",
      "TRS\n",
      "5′-UCAACU-3′\n",
      "substitute\n",
      "alternative\n",
      "nucleotide\n",
      "analysis\n",
      "reveal\n",
      "numb\n",
      "strike\n",
      "similarity\n",
      "process\n",
      "copy-choice\n",
      "RNA\n",
      "recombination\n",
      "occur\n",
      "RNA\n",
      "virus\n",
      "leader\n",
      "TRS\n",
      "play\n",
      "target\n",
      "role\n",
      "translocation\n",
      "nascent\n",
      "strand\n",
      "body\n",
      "TRS\n",
      "nucleotide\n",
      "appear\n",
      "fulfil\n",
      "diverse\n",
      "position-specific\n",
      "base-specific\n",
      "function\n",
      "addition\n",
      "sequence\n",
      "leader\n",
      "body\n",
      "junction\n",
      "sg\n",
      "mRNAs\n",
      "produce\n",
      "mutant\n",
      "provide\n",
      "strong\n",
      "evidence\n",
      "discontinuous\n",
      "minus\n",
      "strand\n",
      "extension\n",
      "model\n",
      "dissect\n",
      "EAV\n",
      "RNA\n",
      "synthesis\n",
      "routinely\n",
      "use\n",
      "full-length\n",
      "cDNA\n",
      "clone\n",
      "van\n",
      "Dinten\n",
      "et\n",
      "al.\n",
      "1997\n",
      "infectious\n",
      "EAV\n",
      "RNA\n",
      "vitro\n",
      "transcribe\n",
      "Following\n",
      "transfection\n",
      "RNA\n",
      "baby\n",
      "hamster\n",
      "kidney\n",
      "BHK-21\n",
      "cell\n",
      "intracellular\n",
      "RNA\n",
      "isolate\n",
      "analyse\n",
      "northern\n",
      "blot\n",
      "hybridization\n",
      "RT\n",
      "PCR\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999a\n",
      "difference\n",
      "transfection\n",
      "efficiency\n",
      "total\n",
      "virus-specific\n",
      "RNA\n",
      "genomic\n",
      "RNA\n",
      "sg\n",
      "mRNA\n",
      "isolate\n",
      "transfect\n",
      "cell\n",
      "culture\n",
      "somewhat\n",
      "variable\n",
      "accurate\n",
      "quantitation\n",
      "sg\n",
      "mRNA\n",
      "synthesis\n",
      "TRS\n",
      "mutant\n",
      "require\n",
      "internal\n",
      "standard\n",
      "transfection\n",
      "efficiency\n",
      "viral\n",
      "genomic\n",
      "RNA\n",
      "standard\n",
      "amplification\n",
      "dramatically\n",
      "affect\n",
      "TRS\n",
      "mutation\n",
      "prove\n",
      "case\n",
      "previously\n",
      "describe\n",
      "mutant\n",
      "L4\n",
      "B4\n",
      "LB4\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999a\n",
      "nucleotide\n",
      "TRS\n",
      "5′-UCAAC-3′\n",
      "replace\n",
      "sequence\n",
      "5′-AGUUG-3′\n",
      "leader\n",
      "TRS\n",
      "L4\n",
      "RNA7\n",
      "body\n",
      "TRS\n",
      "B4\n",
      "TRSs\n",
      "LB4\n",
      "mutant\n",
      "test\n",
      "independent\n",
      "experiment\n",
      "Intracellular\n",
      "RNA\n",
      "isolate\n",
      "14\n",
      "h\n",
      "post-transfection\n",
      "early\n",
      "prevent\n",
      "spread\n",
      "wild-type\n",
      "control\n",
      "virus\n",
      "non-transfected\n",
      "cell\n",
      "cycle\n",
      "analysis\n",
      "Transfection\n",
      "efficiency\n",
      "determine\n",
      "immunofluorescence\n",
      "assay\n",
      "Materials\n",
      "method\n",
      "vary\n",
      "10\n",
      "23\n",
      "datum\n",
      "show\n",
      "Prior\n",
      "RNA\n",
      "analysis\n",
      "isolate\n",
      "intracellular\n",
      "RNA\n",
      "correct\n",
      "transfection\n",
      "efficiency\n",
      "sample\n",
      "lane\n",
      "Figure\n",
      "2\n",
      "represent\n",
      "EAV-specific\n",
      "RNA\n",
      "approximately\n",
      "equal\n",
      "numb\n",
      "EAV-positive\n",
      "cell\n",
      "Phosphoimager\n",
      "quantitation\n",
      "reveal\n",
      "genomic\n",
      "RNA\n",
      "replication\n",
      "mutant\n",
      "L4\n",
      "B4\n",
      "LB4\n",
      "vary\n",
      "30\n",
      "Table\n",
      "difference\n",
      "reflect\n",
      "example\n",
      "slight\n",
      "influence\n",
      "RNA\n",
      "secondary\n",
      "structure\n",
      "change\n",
      "TRS\n",
      "region\n",
      "genomic\n",
      "RNA\n",
      "synthesis\n",
      "Remarkably\n",
      "genomic\n",
      "RNA\n",
      "level\n",
      "leader\n",
      "body\n",
      "TRS\n",
      "double\n",
      "mutant\n",
      "LB4\n",
      "affect\n",
      "10\n",
      "view\n",
      "result\n",
      "obtain\n",
      "pentanucleotide\n",
      "TRS\n",
      "mutant\n",
      "assume\n",
      "genomic\n",
      "RNA\n",
      "internal\n",
      "standard\n",
      "analysis\n",
      "mutant\n",
      "contain\n",
      "single\n",
      "nucleotide\n",
      "replacement\n",
      "leader\n",
      "TRS\n",
      "and/or\n",
      "RNA7\n",
      "body\n",
      "TRS\n",
      "numerous\n",
      "example\n",
      "regulatory\n",
      "RNA\n",
      "RNA\n",
      "interaction\n",
      "eukaryotic\n",
      "prokaryotic\n",
      "cell\n",
      "RNA\n",
      "virus\n",
      "Essential\n",
      "process\n",
      "translation\n",
      "replication\n",
      "encapsidation\n",
      "RNA\n",
      "virus\n",
      "genome\n",
      "frequently\n",
      "depend\n",
      "RNA\n",
      "RNA\n",
      "interaction\n",
      "high\n",
      "order\n",
      "RNA\n",
      "structure\n",
      "Regulation\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "+\n",
      "RNA\n",
      "virus\n",
      "RNA\n",
      "RNA\n",
      "interaction\n",
      "precedent\n",
      "tomato\n",
      "bushy\n",
      "stunt\n",
      "virus\n",
      "RNA\n",
      "element\n",
      "locate\n",
      "1000\n",
      "nucleotide\n",
      "upstream\n",
      "sg\n",
      "RNA2\n",
      "promoter\n",
      "base-pair\n",
      "promoter\n",
      "necessary\n",
      "sg\n",
      "RNA\n",
      "production\n",
      "Zhang\n",
      "et\n",
      "al.\n",
      "1999\n",
      "Similarly\n",
      "base\n",
      "pair\n",
      "interaction\n",
      "complementary\n",
      "sequence\n",
      "5′\n",
      "end\n",
      "potato\n",
      "virus\n",
      "X\n",
      "genomic\n",
      "RNA\n",
      "sequence\n",
      "upstream\n",
      "major\n",
      "sg\n",
      "RNA\n",
      "promoter\n",
      "require\n",
      "efficient\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "Kim\n",
      "Hemenway\n",
      "1999\n",
      "RCNMV\n",
      "intermolecular\n",
      "RNA\n",
      "RNA\n",
      "interaction\n",
      "require\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "Sit\n",
      "et\n",
      "al.\n",
      "1998\n",
      "Recently\n",
      "establish\n",
      "pivotal\n",
      "role\n",
      "interaction\n",
      "sense\n",
      "antisense\n",
      "RNA\n",
      "sequence\n",
      "life\n",
      "cycle\n",
      "EAV\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999a\n",
      "study\n",
      "role\n",
      "TRS\n",
      "nucleotide\n",
      "C2\n",
      "C5\n",
      "test\n",
      "substitute\n",
      "G.\n",
      "conclude\n",
      "base\n",
      "pair\n",
      "sense\n",
      "leader\n",
      "TRS\n",
      "antisense\n",
      "body\n",
      "TRS\n",
      "play\n",
      "crucial\n",
      "role\n",
      "nidovirus\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "take\n",
      "systematic\n",
      "approach\n",
      "perform\n",
      "extensive\n",
      "site-directed\n",
      "co-variation\n",
      "mutagenesis\n",
      "study\n",
      "entire\n",
      "leader\n",
      "TRS\n",
      "RNA7\n",
      "body\n",
      "TRS\n",
      "direct\n",
      "synthesis\n",
      "abundant\n",
      "EAV\n",
      "sg\n",
      "RNA\n",
      "nucleotide\n",
      "TRS\n",
      "5′-UCA\n",
      "ACU-3′\n",
      "replace\n",
      "possible\n",
      "nucleotide\n",
      "study\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999a\n",
      "mutation\n",
      "introduce\n",
      "leader\n",
      "TRS\n",
      "RNA7\n",
      "body\n",
      "TRS\n",
      "TRSs\n",
      "result\n",
      "54\n",
      "mutant\n",
      "construct\n",
      "mutant\n",
      "give\n",
      "unique\n",
      "e.g.\n",
      "BU1A\n",
      "refer\n",
      "mutant\n",
      "U\n",
      "change\n",
      "position\n",
      "1\n",
      "body\n",
      "TRS\n",
      "LU1A\n",
      "refer\n",
      "substitution\n",
      "leader\n",
      "TRS\n",
      "DU1A\n",
      "mean\n",
      "substitution\n",
      "combine\n",
      "double\n",
      "mutant\n",
      "construct\n",
      "sg\n",
      "RNA7\n",
      "quantitated\n",
      "phosphoimager\n",
      "scan\n",
      "hybridize\n",
      "gel\n",
      "correct\n",
      "genomic\n",
      "RNA\n",
      "lane\n",
      "outline\n",
      "Figure\n",
      "3\n",
      "show\n",
      "relative\n",
      "sg\n",
      "RNA7\n",
      "level\n",
      "54\n",
      "mutant\n",
      "compare\n",
      "RNA7\n",
      "level\n",
      "wild-type\n",
      "control\n",
      "selection\n",
      "11\n",
      "interest\n",
      "mutant\n",
      "analysis\n",
      "repeat\n",
      "time\n",
      "Figure\n",
      "4\n",
      "observe\n",
      "significant\n",
      "variation\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "comprehensive\n",
      "analysis\n",
      "effect\n",
      "TRS\n",
      "mutation\n",
      "considerably\n",
      "expand\n",
      "understand\n",
      "discontinuous\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "Remarkably\n",
      "effect\n",
      "single\n",
      "leader\n",
      "body\n",
      "TRS\n",
      "mutation\n",
      "base\n",
      "specific\n",
      "i.e.\n",
      "different\n",
      "nucleotide\n",
      "substitution\n",
      "position\n",
      "affect\n",
      "sg\n",
      "RNA7\n",
      "synthesis\n",
      "different\n",
      "extent\n",
      "example\n",
      "position\n",
      "1\n",
      "BU1A\n",
      "mutant\n",
      "retain\n",
      "44\n",
      "wild-type\n",
      "RNA7\n",
      "synthesis\n",
      "level\n",
      "BU1C\n",
      "BU1\n",
      "G\n",
      "mutant\n",
      "lose\n",
      "RNA7\n",
      "synthesis\n",
      "completely\n",
      "Conversely\n",
      "U1\n",
      "leader\n",
      "TRS\n",
      "change\n",
      "G\n",
      "RNA7\n",
      "synthesis\n",
      "completely\n",
      "abolish\n",
      "13\n",
      "wild-type\n",
      "level\n",
      "maintain\n",
      "LU1C\n",
      "position\n",
      "2\n",
      "BC2U\n",
      "mutant\n",
      "retain\n",
      "30\n",
      "wild-type\n",
      "RNA7\n",
      "synthesis\n",
      "level\n",
      "position\n",
      "2\n",
      "single\n",
      "mutant\n",
      "lose\n",
      "90\n",
      "wild-type\n",
      "RNA7\n",
      "synthesis\n",
      "example\n",
      "position\n",
      "6\n",
      "BU6C\n",
      "leave\n",
      "5\n",
      "wild-type\n",
      "RNA7\n",
      "synthesis\n",
      "BU6A\n",
      "produce\n",
      "high\n",
      "RNA7\n",
      "level\n",
      "imply\n",
      "position\n",
      "1\n",
      "2\n",
      "6\n",
      "certain\n",
      "mismatch\n",
      "duplex\n",
      "plus\n",
      "leader\n",
      "TRS\n",
      "minus\n",
      "body\n",
      "TRS\n",
      "U\n",
      "U\n",
      "BU1A\n",
      "BU6A\n",
      "C\n",
      "LU1C\n",
      "BC2U\n",
      "allow\n",
      "limit\n",
      "extent\n",
      "contrast\n",
      "mismatch\n",
      "allow\n",
      "position\n",
      "5\n",
      "single\n",
      "nucleotide\n",
      "substitution\n",
      "abolish\n",
      "RNA7\n",
      "synthesis\n",
      "completely\n",
      "Surprisingly\n",
      "body\n",
      "TRS\n",
      "U\n",
      "C\n",
      "substitution\n",
      "position\n",
      "1\n",
      "6\n",
      "BU1C\n",
      "BU6C\n",
      "result\n",
      "low\n",
      "level\n",
      "RNA7\n",
      "despite\n",
      "fact\n",
      "mutation\n",
      "allow\n",
      "formation\n",
      "G\n",
      "U\n",
      "base\n",
      "pair\n",
      "plus\n",
      "leader\n",
      "TRS\n",
      "provide\n",
      "U\n",
      "nucleotide\n",
      "minus\n",
      "body\n",
      "TRS\n",
      "provide\n",
      "G.\n",
      "hand\n",
      "position\n",
      "3\n",
      "4\n",
      "G\n",
      "U\n",
      "base\n",
      "pair\n",
      "show\n",
      "functional\n",
      "mutant\n",
      "LA3\n",
      "G\n",
      "LA4\n",
      "G\n",
      "form\n",
      "G\n",
      "U\n",
      "base\n",
      "pair\n",
      "G\n",
      "plus\n",
      "leader\n",
      "TRS\n",
      "U\n",
      "minus\n",
      "body\n",
      "TRS\n",
      "position\n",
      "3\n",
      "4\n",
      "single\n",
      "mutant\n",
      "produce\n",
      "reasonable\n",
      "level\n",
      "RNA7\n",
      "Taken\n",
      "finding\n",
      "suggest\n",
      "factor\n",
      "leader–\n",
      "body\n",
      "base\n",
      "pair\n",
      "play\n",
      "role\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "primary\n",
      "sequence\n",
      "secondary\n",
      "structure\n",
      "TRSs\n",
      "dictate\n",
      "strong\n",
      "base\n",
      "preference\n",
      "certain\n",
      "position\n",
      "analysis\n",
      "degree\n",
      "complementation\n",
      "double\n",
      "mutant\n",
      "provide\n",
      "strong\n",
      "support\n",
      "assumption\n",
      "TRS\n",
      "nucleotide\n",
      "2\n",
      "5\n",
      "6\n",
      "case\n",
      "DU6C\n",
      "RNA7\n",
      "level\n",
      "double\n",
      "mutant\n",
      "clearly\n",
      "high\n",
      "correspond\n",
      "single\n",
      "mutant\n",
      "mean\n",
      "base\n",
      "pair\n",
      "leader\n",
      "body\n",
      "TRS\n",
      "nucleotide\n",
      "involve\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "double\n",
      "mutant\n",
      "reach\n",
      "wild-type\n",
      "sg\n",
      "RNA7\n",
      "level\n",
      "double\n",
      "mutant\n",
      "position\n",
      "1\n",
      "3\n",
      "4\n",
      "mutant\n",
      "DU6C\n",
      "clear\n",
      "contradiction\n",
      "prediction\n",
      "base\n",
      "pair\n",
      "model\n",
      "RNA7\n",
      "synthesis\n",
      "significantly\n",
      "restore\n",
      "comparison\n",
      "value\n",
      "B\n",
      "D\n",
      "mutant\n",
      "Figure\n",
      "3\n",
      "show\n",
      "mutant\n",
      "e.g.\n",
      "position\n",
      "1\n",
      "mutant\n",
      "sg\n",
      "RNA7\n",
      "produce\n",
      "double\n",
      "mutant\n",
      "appear\n",
      "limit\n",
      "level\n",
      "allow\n",
      "body\n",
      "TRS\n",
      "mutation\n",
      "RNA7\n",
      "level\n",
      "double\n",
      "mutant\n",
      "leader\n",
      "mutant\n",
      "DU1C\n",
      "DA3\n",
      "G\n",
      "DA4\n",
      "G\n",
      "DU6C\n",
      "Clearly\n",
      "substitution\n",
      "restoration\n",
      "possibility\n",
      "leader\n",
      "body\n",
      "duplex\n",
      "formation\n",
      "restore\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "Apparently\n",
      "effect\n",
      "body\n",
      "TRS\n",
      "mutation\n",
      "level\n",
      "primary\n",
      "sequence\n",
      "secondary\n",
      "structure\n",
      "dominant\n",
      "duplex-restoring\n",
      "effect\n",
      "double\n",
      "mutation\n",
      "Body\n",
      "TRS\n",
      "mutant\n",
      "fall\n",
      "distinct\n",
      "type\n",
      "determine\n",
      "position\n",
      "chemistry\n",
      "substitution\n",
      "mutant\n",
      "type\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "impair\n",
      "mainly\n",
      "disruption\n",
      "leader–\n",
      "body\n",
      "TRS\n",
      "duplex\n",
      "effect\n",
      "compensate\n",
      "introduction\n",
      "correspond\n",
      "mutation\n",
      "leader\n",
      "TRS\n",
      "double\n",
      "mutant\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "restore\n",
      "compare\n",
      "correspond\n",
      "single\n",
      "mutant\n",
      "mutant\n",
      "2\n",
      "type\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "down-regulated\n",
      "consequence\n",
      "TRS\n",
      "duplex\n",
      "disruption\n",
      "disruption\n",
      "primary\n",
      "sequence\n",
      "secondary\n",
      "structure\n",
      "body\n",
      "TRS\n",
      "Obviously\n",
      "effect\n",
      "compensate\n",
      "mutate\n",
      "leader\n",
      "TRS\n",
      "correspond\n",
      "double\n",
      "mutant\n",
      "restoration\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "contrast\n",
      "finding\n",
      "body\n",
      "TRS\n",
      "mutant\n",
      "obtain\n",
      "leader\n",
      "TRS\n",
      "mutation\n",
      "appear\n",
      "determine\n",
      "level\n",
      "sg\n",
      "RNA7\n",
      "synthesis\n",
      "correspond\n",
      "double\n",
      "mutant\n",
      "Figure\n",
      "3\n",
      "effect\n",
      "mutation\n",
      "leader\n",
      "TRS\n",
      "dominant\n",
      "duplex-restoring\n",
      "effect\n",
      "double\n",
      "mutation\n",
      "suggest\n",
      "affect\n",
      "duplex\n",
      "formation\n",
      "indicate\n",
      "leader\n",
      "TRS\n",
      "probably\n",
      "additional\n",
      "sequence-specific\n",
      "function\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "addition\n",
      "participation\n",
      "TRS\n",
      "TRS\n",
      "base\n",
      "pair\n",
      "fact\n",
      "single\n",
      "leader\n",
      "TRS\n",
      "mutation\n",
      "position\n",
      "severely\n",
      "repress\n",
      "RNA7\n",
      "synthesis\n",
      "indicate\n",
      "base\n",
      "pair\n",
      "TRS\n",
      "nucleotide\n",
      "contribute\n",
      "sg\n",
      "RNA\n",
      "production\n",
      "respect\n",
      "significant\n",
      "leader\n",
      "TRS\n",
      "mutant\n",
      "high\n",
      "RNA7\n",
      "level\n",
      "LA3\n",
      "G\n",
      "LA4\n",
      "G\n",
      "form\n",
      "G\n",
      "U\n",
      "base\n",
      "pair\n",
      "maintain\n",
      "duplex\n",
      "observation\n",
      "leader\n",
      "TRS\n",
      "mutation\n",
      "rescue\n",
      "introduce\n",
      "complementary\n",
      "mutation\n",
      "body\n",
      "TRS\n",
      "body\n",
      "TRS\n",
      "mutation\n",
      "rescue\n",
      "correspond\n",
      "change\n",
      "leader\n",
      "TRS\n",
      "clearly\n",
      "illustrate\n",
      "U1A\n",
      "mutant\n",
      "restoration\n",
      "TRS\n",
      "base\n",
      "pair\n",
      "possibility\n",
      "RNA7\n",
      "synthesis\n",
      "double\n",
      "mutant\n",
      "DU1A\n",
      "significantly\n",
      "increase\n",
      "compare\n",
      "LU1A\n",
      "level\n",
      "BU1A\n",
      "restoration\n",
      "leader\n",
      "body\n",
      "duplex\n",
      "DU1A\n",
      "exert\n",
      "clear\n",
      "effect\n",
      "sg\n",
      "RNA7\n",
      "production\n",
      "compare\n",
      "LU1A\n",
      "effect\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "compare\n",
      "BU1A\n",
      "exemplify\n",
      "dominant\n",
      "nature\n",
      "mutation\n",
      "primary\n",
      "sequence\n",
      "body\n",
      "TRS\n",
      "contrast\n",
      "instance\n",
      "BC2U\n",
      "mutation\n",
      "probably\n",
      "affect\n",
      "duplex\n",
      "formation\n",
      "RNA7\n",
      "synthesis\n",
      "restore\n",
      "wild-type\n",
      "level\n",
      "DC2U\n",
      "double\n",
      "mutant\n",
      "result\n",
      "indicate\n",
      "strong\n",
      "base\n",
      "preference\n",
      "constraint\n",
      "body\n",
      "TRS\n",
      "position\n",
      "interpret\n",
      "base\n",
      "preference\n",
      "accurately\n",
      "necessary\n",
      "limit\n",
      "analysis\n",
      "double\n",
      "mutant\n",
      "mutant\n",
      "down-regulation\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "sequence\n",
      "change\n",
      "body\n",
      "TRS\n",
      "disruption\n",
      "leader\n",
      "body\n",
      "TRS\n",
      "duplex\n",
      "strict\n",
      "preference\n",
      "position\n",
      "1\n",
      "3\n",
      "4\n",
      "body\n",
      "TRS\n",
      "position\n",
      "1\n",
      "U\n",
      "substitution\n",
      "allow\n",
      "significant\n",
      "RNA7\n",
      "level\n",
      "∼40\n",
      "wild-type\n",
      "position\n",
      "3\n",
      "4\n",
      "U\n",
      "mutant\n",
      "retain\n",
      "15–20\n",
      "wild-type\n",
      "level\n",
      "position\n",
      "2\n",
      "5\n",
      "sequence\n",
      "constraint\n",
      "stringent\n",
      "substitution\n",
      "allow\n",
      ">\n",
      "20\n",
      "wild-type\n",
      "level\n",
      "DC2A\n",
      "DC2U\n",
      "reach\n",
      ">\n",
      "50\n",
      "position\n",
      "6\n",
      "body\n",
      "TRS\n",
      "U\n",
      "C\n",
      "allow\n",
      "double\n",
      "mutant\n",
      "produce\n",
      "50\n",
      "RNA7\n",
      "word\n",
      "functional\n",
      "EAV\n",
      "RNA7\n",
      "body\n",
      "TRS\n",
      "base\n",
      "analysis\n",
      "single\n",
      "nucleotide\n",
      "substitution\n",
      "describe\n",
      "U1(C/u/a)2A3A4C5(U/a/g)6\n",
      "wild-type\n",
      "nucleotide\n",
      "show\n",
      "upper\n",
      "case\n",
      "nucleotide\n",
      "allow\n",
      "50\n",
      "wild-type\n",
      "RNA7\n",
      "level\n",
      "show\n",
      "low\n",
      "case\n",
      "Remarkably\n",
      "TRS\n",
      "nucleotide\n",
      "A3\n",
      "A4\n",
      "C5\n",
      "conserve\n",
      "TRSs\n",
      "arteriviruses\n",
      "Snijder\n",
      "Meulenberg\n",
      "1998\n",
      "fact\n",
      "DC2U\n",
      "retain\n",
      "80\n",
      "RNA7\n",
      "synthesis\n",
      "correspond\n",
      "nicely\n",
      "presence\n",
      "U\n",
      "position\n",
      "arteriviruses\n",
      "recently\n",
      "Almazan\n",
      "et\n",
      "al.\n",
      "2000\n",
      "Thiel\n",
      "et\n",
      "al.\n",
      "2001\n",
      "infectious\n",
      "cDNA\n",
      "clone\n",
      "lack\n",
      "coronavirus\n",
      "Consequently\n",
      "study\n",
      "coronavirus\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "carry\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "defective\n",
      "interfere\n",
      "DI\n",
      "RNAs\n",
      "replicons\n",
      "carry\n",
      "body\n",
      "TRSs\n",
      "moderate\n",
      "level\n",
      "sg\n",
      "mRNAs\n",
      "produce\n",
      "presence\n",
      "helper\n",
      "virus\n",
      "system\n",
      "Joo\n",
      "Makino\n",
      "1992\n",
      "van\n",
      "der\n",
      "et\n",
      "al.\n",
      "1994\n",
      "perform\n",
      "body\n",
      "TRS\n",
      "mutagenesis\n",
      "study\n",
      "murine\n",
      "coronavirus\n",
      "MHV\n",
      "Joo\n",
      "Makino\n",
      "systematically\n",
      "mutagenized\n",
      "core\n",
      "MHV\n",
      "body\n",
      "TRS\n",
      "contrast\n",
      "result\n",
      "find\n",
      "21\n",
      "body\n",
      "TRS\n",
      "mutant\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "DI\n",
      "RNA\n",
      "genome\n",
      "abolish\n",
      "support\n",
      "normal\n",
      "level\n",
      "sg\n",
      "RNA\n",
      "production\n",
      "possible\n",
      "MHV\n",
      "TRS\n",
      "study\n",
      "tolerant\n",
      "single-nucleotide\n",
      "mismatch\n",
      "EAV\n",
      "sg\n",
      "RNA7\n",
      "TRS\n",
      "similar\n",
      "study\n",
      "van\n",
      "der\n",
      "et\n",
      "al.\n",
      "1994\n",
      "observe\n",
      "U\n",
      "C\n",
      "substitution\n",
      "position\n",
      "1\n",
      "3\n",
      "MHV\n",
      "body\n",
      "TRS\n",
      "maintain\n",
      "duplex\n",
      "change\n",
      "U\n",
      "base\n",
      "pair\n",
      "U\n",
      "G\n",
      "base\n",
      "pair\n",
      "reduce\n",
      "sg\n",
      "RNA\n",
      "level\n",
      "strongly\n",
      "substitution\n",
      "disrupt\n",
      "duplex\n",
      "van\n",
      "der\n",
      "et\n",
      "al.\n",
      "1994\n",
      "imply\n",
      "case\n",
      "EAV\n",
      "leader\n",
      "body\n",
      "TRS\n",
      "duplex\n",
      "formation\n",
      "factor\n",
      "determine\n",
      "coronavirus\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "limitation\n",
      "DI\n",
      "RNA\n",
      "system\n",
      "leader\n",
      "TRS\n",
      "mutagenized\n",
      "study\n",
      "body\n",
      "TRS-specific\n",
      "effect\n",
      "distinguish\n",
      "effect\n",
      "level\n",
      "leader\n",
      "body\n",
      "duplex\n",
      "formation\n",
      "recent\n",
      "study\n",
      "arterivirus\n",
      "coronavirus\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999a\n",
      "Baric\n",
      "Yount\n",
      "2000\n",
      "Sawicki\n",
      "et\n",
      "al.\n",
      "2001\n",
      "discontinuous\n",
      "minus\n",
      "strand\n",
      "extension\n",
      "model\n",
      "Figure\n",
      "1C\n",
      "gain\n",
      "grind\n",
      "model\n",
      "predict\n",
      "TRS-derived\n",
      "sequence\n",
      "form\n",
      "leader\n",
      "body\n",
      "junction\n",
      "sg\n",
      "mRNA\n",
      "copy\n",
      "body\n",
      "TRS\n",
      "leader\n",
      "TRS\n",
      "leader-primed\n",
      "transcription\n",
      "model\n",
      "predict\n",
      "opposite\n",
      "Figure\n",
      "1B\n",
      "determine\n",
      "origin\n",
      "leader\n",
      "body\n",
      "junction\n",
      "sg\n",
      "mRNAs\n",
      "help\n",
      "distinguish\n",
      "model\n",
      "wild-type\n",
      "situation\n",
      "EAV\n",
      "leader\n",
      "body\n",
      "TRSs\n",
      "identical\n",
      "consequently\n",
      "determine\n",
      "origin\n",
      "sg\n",
      "mRNA\n",
      "leader\n",
      "body\n",
      "junction\n",
      "problem\n",
      "overcome\n",
      "trace\n",
      "mutation\n",
      "introduce\n",
      "leader\n",
      "RNA7\n",
      "body\n",
      "TRS\n",
      "mutant\n",
      "retain\n",
      "ability\n",
      "produce\n",
      "mRNA7\n",
      "previous\n",
      "study\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999a\n",
      "find\n",
      "nucleotide\n",
      "2\n",
      "5\n",
      "mRNA7\n",
      "leader\n",
      "body\n",
      "junction\n",
      "sequence\n",
      "derive\n",
      "exclusively\n",
      "body\n",
      "TRS\n",
      "leader\n",
      "TRS\n",
      "show\n",
      "direct\n",
      "sequence\n",
      "RT\n",
      "PCR\n",
      "product\n",
      "obtain\n",
      "residual\n",
      "mRNA7\n",
      "produce\n",
      "mutant\n",
      "BC2\n",
      "G\n",
      "LC2\n",
      "G\n",
      "BC5\n",
      "G\n",
      "LC5\n",
      "G\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999a\n",
      "approach\n",
      "analyse\n",
      "mRNA7\n",
      "mutant\n",
      "BC2A\n",
      "BC2U\n",
      "transcript\n",
      "contain\n",
      "mutate\n",
      "nucleotide\n",
      "derive\n",
      "body\n",
      "TRS\n",
      "datum\n",
      "show\n",
      "Assuming\n",
      "crossover\n",
      "event\n",
      "occur\n",
      "leader\n",
      "body\n",
      "join\n",
      "map\n",
      "crossover\n",
      "position\n",
      "–1\n",
      "+\n",
      "2\n",
      "sg\n",
      "RNA\n",
      "junction\n",
      "sequence\n",
      "leave\n",
      "intrigue\n",
      "question\n",
      "crossover\n",
      "site\n",
      "map\n",
      "precisely\n",
      "word\n",
      "nucleotide\n",
      "+\n",
      "1\n",
      "junction\n",
      "sequence\n",
      "derive\n",
      "body\n",
      "TRS\n",
      "leader\n",
      "TRS\n",
      "position\n",
      "1\n",
      "mutant\n",
      "describe\n",
      "answer\n",
      "question\n",
      "Figure\n",
      "5\n",
      "strike\n",
      "result\n",
      "mRNA7\n",
      "mutant\n",
      "BU1A\n",
      "BU1\n",
      "G\n",
      "LU1C\n",
      "contain\n",
      "exclusively\n",
      "body\n",
      "TRS-derived\n",
      "nucleotide\n",
      "position\n",
      "+\n",
      "1\n",
      "mutant\n",
      "crossover\n",
      "site\n",
      "map\n",
      "precisely\n",
      "TRS\n",
      "nucleotide\n",
      "position\n",
      "–1\n",
      "+\n",
      "1\n",
      "mean\n",
      "complete\n",
      "leader–\n",
      "body\n",
      "junction\n",
      "sequence\n",
      "EAV\n",
      "sg\n",
      "mRNA\n",
      "body\n",
      "TRS\n",
      "derive\n",
      "hand\n",
      "sg\n",
      "RNAs\n",
      "mutant\n",
      "LU1A\n",
      "BU1C\n",
      "LU1\n",
      "G\n",
      "contain\n",
      "mix\n",
      "population\n",
      "leader\n",
      "TRS-\n",
      "body\n",
      "TRS-derived\n",
      "nucleotide\n",
      "position\n",
      "+\n",
      "1\n",
      "Figure\n",
      "5\n",
      "U\n",
      "LU1A\n",
      "C\n",
      "U\n",
      "BU1C\n",
      "G\n",
      "U\n",
      "LU1\n",
      "G\n",
      "Remarkably\n",
      "pattern\n",
      "correlate\n",
      "relative\n",
      "amount\n",
      "sg\n",
      "mRNA7\n",
      "produce\n",
      "mutant\n",
      "Figure\n",
      "3\n",
      "Mutants\n",
      "produce\n",
      "population\n",
      "sg\n",
      "RNAs\n",
      "mix\n",
      "respect\n",
      "origin\n",
      "nucleotide\n",
      "position\n",
      "+\n",
      "1\n",
      "leader\n",
      "body\n",
      "junction\n",
      "lose\n",
      "RNA7\n",
      "synthesis\n",
      "completely\n",
      "hand\n",
      "mutant\n",
      "contain\n",
      "exclusively\n",
      "body\n",
      "nucleotide\n",
      "position\n",
      "+\n",
      "1\n",
      "retain\n",
      "high\n",
      "level\n",
      "RNA7\n",
      "synthesis\n",
      "observation\n",
      "explain\n",
      "follow\n",
      "wild-type\n",
      "situation\n",
      "large\n",
      "majority\n",
      "crossover\n",
      "probably\n",
      "occur\n",
      "position\n",
      "–1\n",
      "+\n",
      "1\n",
      "lead\n",
      "body\n",
      "TRS-derived\n",
      "nucleotide\n",
      "position\n",
      "+\n",
      "1\n",
      "sg\n",
      "RNA\n",
      "low\n",
      "numb\n",
      "crossover\n",
      "place\n",
      "nucleotide\n",
      "+\n",
      "1\n",
      "+\n",
      "2\n",
      "result\n",
      "leader\n",
      "TRS-derived\n",
      "nucleotide\n",
      "position\n",
      "+\n",
      "1\n",
      "Mutants\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "block\n",
      "substitution\n",
      "position\n",
      "+\n",
      "1\n",
      "deficient\n",
      "crossover\n",
      "–1\n",
      "+\n",
      "1\n",
      "retain\n",
      "ability\n",
      "crossover\n",
      "+\n",
      "1\n",
      "+\n",
      "2\n",
      "detect\n",
      "sequence\n",
      "analysis\n",
      "Conversely\n",
      "position\n",
      "+\n",
      "1\n",
      "mutant\n",
      "retain\n",
      "reasonable\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "crossover\n",
      "occur\n",
      "position\n",
      "–1\n",
      "+\n",
      "1\n",
      "obscure\n",
      "minority\n",
      "crossover\n",
      "+\n",
      "1\n",
      "+\n",
      "2\n",
      "sequence\n",
      "electropherogram\n",
      "Alternatively\n",
      "position\n",
      "+\n",
      "1\n",
      "TRS\n",
      "mutation\n",
      "strongly\n",
      "interfere\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "force\n",
      "shift\n",
      "crossover\n",
      "site\n",
      "remain\n",
      "molecule\n",
      "believe\n",
      "present\n",
      "finding\n",
      "strongly\n",
      "support\n",
      "discontinuous\n",
      "minus\n",
      "strand\n",
      "extension\n",
      "model\n",
      "fact\n",
      "complete\n",
      "body\n",
      "TRS\n",
      "copy\n",
      "sg\n",
      "RNA\n",
      "difficult\n",
      "reconcile\n",
      "alternative\n",
      "model\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "genomic\n",
      "minus\n",
      "strand\n",
      "template\n",
      "prime\n",
      "free\n",
      "plus\n",
      "strand\n",
      "leader\n",
      "transcript\n",
      "contain\n",
      "leader\n",
      "TRS\n",
      "3′\n",
      "end\n",
      "Figure\n",
      "1B\n",
      "explain\n",
      "presence\n",
      "complete\n",
      "copy\n",
      "body\n",
      "TRS\n",
      "sg\n",
      "mRNA\n",
      "model\n",
      "assume\n",
      "3′–5′\n",
      "exonuclease\n",
      "activity\n",
      "trim\n",
      "free\n",
      "leader\n",
      "transcript\n",
      "prior\n",
      "extension\n",
      "sg\n",
      "mRNA\n",
      "Baker\n",
      "Lai\n",
      "1990\n",
      "Note\n",
      "single\n",
      "base\n",
      "pair\n",
      "leave\n",
      "hold\n",
      "trim\n",
      "leader\n",
      "molecule\n",
      "template\n",
      "enzymatic\n",
      "activity\n",
      "unprecedented\n",
      "+\n",
      "RNA\n",
      "virus\n",
      "exist\n",
      "yeast\n",
      "retrotransposon\n",
      "Ty5\n",
      "Ke\n",
      "et\n",
      "al.\n",
      "1999\n",
      "reverse\n",
      "transcription\n",
      "prime\n",
      "internal\n",
      "region\n",
      "tRNA\n",
      "system\n",
      "duplex\n",
      "remove\n",
      "single-stranded\n",
      "3′\n",
      "tail\n",
      "tRNA\n",
      "base-pair\n",
      "Ty5\n",
      "RNA\n",
      "Removal\n",
      "TRS\n",
      "3′\n",
      "end\n",
      "nidovirus\n",
      "leader\n",
      "base\n",
      "pair\n",
      "template\n",
      "energetically\n",
      "unfavourable\n",
      "RdRp\n",
      "Instead\n",
      "start\n",
      "elongation\n",
      "intact\n",
      "properly\n",
      "position\n",
      "leader\n",
      "primer\n",
      "disrupt\n",
      "newly\n",
      "form\n",
      "duplex\n",
      "degrade\n",
      "leader\n",
      "RNA\n",
      "reinitiate\n",
      "polymerization\n",
      "base\n",
      "pair\n",
      "primer\n",
      "template\n",
      "show\n",
      "influenza\n",
      "virus\n",
      "transcription\n",
      "require\n",
      "sequence\n",
      "match\n",
      "cellular\n",
      "RNA\n",
      "primer\n",
      "viral\n",
      "template\n",
      "Plotch\n",
      "et\n",
      "al.\n",
      "1981\n",
      "nidovirus\n",
      "system\n",
      "trim\n",
      "leader\n",
      "RNA\n",
      "fix\n",
      "template\n",
      "solely\n",
      "RNA\n",
      "protein\n",
      "interaction\n",
      "target\n",
      "nascent\n",
      "strand\n",
      "TRS\n",
      "base\n",
      "pair\n",
      "extremely\n",
      "puzzle\n",
      "Sequence\n",
      "datum\n",
      "sg\n",
      "RNA\n",
      "leader\n",
      "body\n",
      "junction\n",
      "arteriviruses\n",
      "difficult\n",
      "reconcile\n",
      "leader-primed\n",
      "transcription\n",
      "model\n",
      "porcine\n",
      "simian\n",
      "arteriviruses\n",
      "Meulenberg\n",
      "et\n",
      "al.\n",
      "1993\n",
      "Godeny\n",
      "et\n",
      "al.\n",
      "1998\n",
      "leader\n",
      "body\n",
      "junction\n",
      "sg\n",
      "RNAs\n",
      "map\n",
      "nucleotide\n",
      "upstream\n",
      "body\n",
      "TRS\n",
      "leave\n",
      "single\n",
      "nucleotide\n",
      "hold\n",
      "putative\n",
      "free\n",
      "leader\n",
      "template\n",
      "hypothetical\n",
      "trim\n",
      "hand\n",
      "finding\n",
      "datum\n",
      "explain\n",
      "readily\n",
      "discontinuous\n",
      "minus\n",
      "strand\n",
      "extension\n",
      "model\n",
      "Figure\n",
      "1C\n",
      "six-nucleotide\n",
      "duplex\n",
      "form\n",
      "body\n",
      "TRS\n",
      "complement\n",
      "3′\n",
      "end\n",
      "leaderless\n",
      "sg\n",
      "minus\n",
      "strand\n",
      "leader\n",
      "TRS\n",
      "genomic\n",
      "RNA\n",
      "template\n",
      "suffice\n",
      "position\n",
      "nascent\n",
      "minus\n",
      "strand\n",
      "properly\n",
      "subsequent\n",
      "elongation\n",
      "add\n",
      "complement\n",
      "leader\n",
      "sequence\n",
      "case\n",
      "nascent\n",
      "minus\n",
      "strand\n",
      "contain\n",
      "entire\n",
      "body\n",
      "TRS\n",
      "complement\n",
      "3′\n",
      "end\n",
      "moment\n",
      "strand\n",
      "transfer\n",
      "lead\n",
      "body\n",
      "TRS-derived\n",
      "leader\n",
      "body\n",
      "junction\n",
      "sequence\n",
      "sg\n",
      "mRNA\n",
      "molecule\n",
      "small\n",
      "numb\n",
      "transcript\n",
      "minus\n",
      "strand\n",
      "synthesis\n",
      "appear\n",
      "interrupt\n",
      "nucleotide\n",
      "+\n",
      "1\n",
      "body\n",
      "TRS\n",
      "copy\n",
      "strand\n",
      "transfer\n",
      "resume\n",
      "incorporate\n",
      "complement\n",
      "+\n",
      "1\n",
      "nucleotide\n",
      "leader\n",
      "TRS\n",
      "state\n",
      "postulate\n",
      "detection\n",
      "phenomenon\n",
      "determine\n",
      "level\n",
      "crossover\n",
      "–1\n",
      "+\n",
      "1\n",
      "position\n",
      "allow\n",
      "mutation\n",
      "introduce\n",
      "+\n",
      "1\n",
      "position\n",
      "body\n",
      "TRS\n",
      "leader\n",
      "TRS\n",
      "formally\n",
      "exclude\n",
      "trim\n",
      "activity\n",
      "degrade\n",
      "3′-terminal\n",
      "nucleotide\n",
      "minus\n",
      "strand\n",
      "strand\n",
      "transfer\n",
      "note\n",
      "discontinuous\n",
      "minus\n",
      "strand\n",
      "extension\n",
      "model\n",
      "Figure\n",
      "1C\n",
      "activity\n",
      "disturb\n",
      "proper\n",
      "position\n",
      "nascent\n",
      "minus\n",
      "strand\n",
      "leader\n",
      "template\n",
      "TRS–\n",
      "TRS\n",
      "duplex\n",
      "shorten\n",
      "nucleotide\n",
      "discontinuous\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "nidoviruses\n",
      "occupy\n",
      "special\n",
      "niche\n",
      "+\n",
      "RNA\n",
      "virus\n",
      "world\n",
      "mode\n",
      "sg\n",
      "RNA\n",
      "production\n",
      "clearly\n",
      "different\n",
      "+\n",
      "RNA\n",
      "virus\n",
      "resemble\n",
      "well-documented\n",
      "+\n",
      "RNA\n",
      "virus\n",
      "feature\n",
      "RNA\n",
      "recombination\n",
      "recent\n",
      "review\n",
      "Nagy\n",
      "Simon\n",
      "1997\n",
      "Aaziz\n",
      "Tepfer\n",
      "1999\n",
      "Worobey\n",
      "Holmes\n",
      "1999\n",
      "experimental\n",
      "evidence\n",
      "support\n",
      "RdRp\n",
      "template\n",
      "switch\n",
      "Kirkegaard\n",
      "Baltimore\n",
      "1986\n",
      "main\n",
      "mechanism\n",
      "RNA\n",
      "recombination\n",
      "Mechanistically\n",
      "template\n",
      "switch\n",
      "involve\n",
      "transfer\n",
      "nascent\n",
      "strand\n",
      "RNA\n",
      "template\n",
      "donor\n",
      "acceptor\n",
      "nidovirus\n",
      "discontinuous\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "involve\n",
      "transfer\n",
      "nascent\n",
      "RNA\n",
      "strand\n",
      "sg\n",
      "RNA\n",
      "site\n",
      "template\n",
      "Based\n",
      "datum\n",
      "currently\n",
      "available\n",
      "refer\n",
      "discontinuous\n",
      "minus\n",
      "strand\n",
      "extension\n",
      "model\n",
      "work\n",
      "model\n",
      "nidovirus\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "apply\n",
      "recombination\n",
      "term\n",
      "model\n",
      "Chang\n",
      "et\n",
      "al.\n",
      "1996\n",
      "Brian\n",
      "Spaan\n",
      "1997\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999a\n",
      "donor\n",
      "strand\n",
      "body\n",
      "genomic\n",
      "RNA\n",
      "template\n",
      "acceptor\n",
      "strand\n",
      "leader\n",
      "genomic\n",
      "RNA\n",
      "template\n",
      "nascent\n",
      "strand\n",
      "discontinuously\n",
      "synthesize\n",
      "minus\n",
      "strand\n",
      "Nagy\n",
      "Simon\n",
      "1997\n",
      "define\n",
      "main\n",
      "class\n",
      "RNA\n",
      "recombination\n",
      "similarity-essential\n",
      "similarity-non-essential\n",
      "similarity-assisted\n",
      "recombination\n",
      "define\n",
      "mechanism\n",
      "strand\n",
      "transfer\n",
      "determine\n",
      "sequence\n",
      "similarity\n",
      "parental\n",
      "RNAs\n",
      "additional\n",
      "RNA\n",
      "determinant\n",
      "present\n",
      "parental\n",
      "RNAs\n",
      "result\n",
      "present\n",
      "study\n",
      "strongly\n",
      "suggest\n",
      "nidovirus\n",
      "discontinuous\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "consider\n",
      "special\n",
      "case\n",
      "high-frequency\n",
      "similarity-assisted\n",
      "RNA\n",
      "recombination\n",
      "obvious\n",
      "function\n",
      "leader\n",
      "TRS\n",
      "ensure\n",
      "fidelity\n",
      "strand\n",
      "transfer\n",
      "base\n",
      "pair\n",
      "3′\n",
      "end\n",
      "nascent\n",
      "strand\n",
      "body\n",
      "TRS\n",
      "donor\n",
      "template\n",
      "additional\n",
      "sequence-specific\n",
      "function\n",
      "function\n",
      "apparently\n",
      "pause\n",
      "terminate\n",
      "nascent\n",
      "strand\n",
      "synthesis\n",
      "provide\n",
      "opportunity\n",
      "strand\n",
      "transfer\n",
      "addition\n",
      "body\n",
      "TRS-derived\n",
      "nucleotide\n",
      "play\n",
      "role\n",
      "reinitiation\n",
      "nascent\n",
      "strand\n",
      "synthesis\n",
      "acceptor\n",
      "template\n",
      "Given\n",
      "compact\n",
      "nature\n",
      "EAV\n",
      "TRS\n",
      "possible\n",
      "nucleotide\n",
      "fulfil\n",
      "multiple\n",
      "task\n",
      "sequence-specific\n",
      "function\n",
      "body\n",
      "TRS\n",
      "reveal\n",
      "study\n",
      "exert\n",
      "level\n",
      "primary\n",
      "sequence\n",
      "secondary\n",
      "structure\n",
      "numb\n",
      "+\n",
      "RNA\n",
      "virus\n",
      "RNA\n",
      "secondary\n",
      "structure\n",
      "motif\n",
      "locate\n",
      "proximity\n",
      "sg\n",
      "RNA\n",
      "promoter\n",
      "vital\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "alfalfa\n",
      "mosaic\n",
      "virus\n",
      "Haasnoot\n",
      "et\n",
      "al.\n",
      "2000\n",
      "turnip\n",
      "crinkle\n",
      "virus\n",
      "TCV\n",
      "Wang\n",
      "et\n",
      "al.\n",
      "1999\n",
      "barley\n",
      "yellow\n",
      "dwarf\n",
      "virus\n",
      "Koev\n",
      "et\n",
      "al.\n",
      "1999\n",
      "stem\n",
      "loop\n",
      "structure\n",
      "sg\n",
      "RNA\n",
      "promoter\n",
      "region\n",
      "template\n",
      "strand\n",
      "require\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "sg\n",
      "RNA1\n",
      "promoter\n",
      "virus\n",
      "especially\n",
      "interest\n",
      "contain\n",
      "stem\n",
      "loop\n",
      "domain\n",
      "secondary\n",
      "structure\n",
      "primary\n",
      "sequence\n",
      "important\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "domain\n",
      "act\n",
      "primary\n",
      "sequence\n",
      "secondary\n",
      "structure\n",
      "Koev\n",
      "et\n",
      "al.\n",
      "1999\n",
      "Similarly\n",
      "RNA\n",
      "secondary\n",
      "structure\n",
      "play\n",
      "minor\n",
      "role\n",
      "sequence-specific\n",
      "recognition\n",
      "BMV\n",
      "sg\n",
      "RNA\n",
      "promoter\n",
      "RdRp\n",
      "Adkins\n",
      "et\n",
      "al.\n",
      "1997\n",
      "Siegel\n",
      "et\n",
      "al.\n",
      "1997\n",
      "suggest\n",
      "previously\n",
      "RNA\n",
      "secondary\n",
      "structure\n",
      "body\n",
      "TRS\n",
      "region\n",
      "contribute\n",
      "attenuate\n",
      "potential\n",
      "determine\n",
      "relative\n",
      "portion\n",
      "nascent\n",
      "minus\n",
      "strand\n",
      "transfer\n",
      "leader\n",
      "TRS\n",
      "template\n",
      "Pasternak\n",
      "et\n",
      "al.\n",
      "2000\n",
      "present\n",
      "unknown\n",
      "EAV\n",
      "body\n",
      "TRSs\n",
      "RNA\n",
      "structural\n",
      "motif\n",
      "essential\n",
      "body\n",
      "TRS\n",
      "function\n",
      "recognize\n",
      "protein\n",
      "factor\n",
      "sequence-specific\n",
      "manner\n",
      "likely\n",
      "LB4\n",
      "Figure\n",
      "2\n",
      "TRS\n",
      "nucleotide\n",
      "substitute\n",
      "produce\n",
      "sg\n",
      "RNA7\n",
      "∼30-fold\n",
      "wild-type\n",
      "control\n",
      "fact\n",
      "sequence\n",
      "EAV\n",
      "genome\n",
      "match\n",
      "leader\n",
      "TRS\n",
      "perfectly\n",
      "sg\n",
      "mRNA\n",
      "synthesis\n",
      "argue\n",
      "recognition\n",
      "specific\n",
      "sequence\n",
      "Pasternak\n",
      "et\n",
      "al.\n",
      "2000\n",
      "probably\n",
      "mutagenesis\n",
      "RNA7\n",
      "body\n",
      "TRS\n",
      "disturb\n",
      "RNA\n",
      "structure\n",
      "necessary\n",
      "function\n",
      "example\n",
      "explain\n",
      "fact\n",
      "BU6C\n",
      "substitution\n",
      "reduce\n",
      "RNA7\n",
      "20-fold\n",
      "rescue\n",
      "mutation\n",
      "leader\n",
      "TRS\n",
      "wild-type\n",
      "RNA6\n",
      "body\n",
      "TRS\n",
      "contain\n",
      "C\n",
      "position\n",
      "protein\n",
      "factor\n",
      "involve\n",
      "sequence-specific\n",
      "TRS\n",
      "recognition\n",
      "expect\n",
      "recognize\n",
      "TRSs\n",
      "similarly\n",
      "RNA\n",
      "structure\n",
      "important\n",
      "recognition\n",
      "protein\n",
      "BU6C\n",
      "substitution\n",
      "probably\n",
      "disturb\n",
      "structural\n",
      "motif\n",
      "RNA7\n",
      "TRS\n",
      "present\n",
      "RNA6\n",
      "TRS\n",
      "hand\n",
      "conservation\n",
      "TRS\n",
      "arteriviruses\n",
      "suggest\n",
      "sequence-specific\n",
      "recognition\n",
      "study\n",
      "require\n",
      "distinguish\n",
      "possibility\n",
      "TCV\n",
      "satellite\n",
      "RNA\n",
      "recombination\n",
      "system\n",
      "hairpin\n",
      "structure\n",
      "acceptor\n",
      "strand\n",
      "donor\n",
      "acceptor\n",
      "homology\n",
      "region\n",
      "necessary\n",
      "template\n",
      "switch\n",
      "Nagy\n",
      "et\n",
      "al.\n",
      "1998\n",
      "hairpin\n",
      "postulate\n",
      "bind\n",
      "RdRp\n",
      "homology\n",
      "region\n",
      "target\n",
      "nascent\n",
      "strand\n",
      "crossover\n",
      "site\n",
      "TCV\n",
      "RdRp\n",
      "probably\n",
      "recognize\n",
      "secondary\n",
      "and/or\n",
      "tertiary\n",
      "structure\n",
      "hairpin\n",
      "individual\n",
      "nucleotide\n",
      "play\n",
      "important\n",
      "role\n",
      "Nagy\n",
      "Simon\n",
      "1998\n",
      "EAV\n",
      "leader\n",
      "TRS\n",
      "acceptor\n",
      "template\n",
      "predict\n",
      "reside\n",
      "loop\n",
      "extensive\n",
      "hairpin\n",
      "base\n",
      "pair\n",
      "interaction\n",
      "body\n",
      "TRS\n",
      "complement\n",
      "3′\n",
      "end\n",
      "nascent\n",
      "minus\n",
      "strand\n",
      "resemble\n",
      "certain\n",
      "antisense\n",
      "RNA-regulated\n",
      "control\n",
      "mechanism\n",
      "base\n",
      "interaction\n",
      "single-stranded\n",
      "tail\n",
      "hairpin\n",
      "loop\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999a\n",
      "reference\n",
      "possible\n",
      "EAV\n",
      "RdRp\n",
      "accessory\n",
      "protein\n",
      "bind\n",
      "stem\n",
      "long\n",
      "hairpin\n",
      "present\n",
      "leader\n",
      "TRS\n",
      "case\n",
      "leader\n",
      "TRS\n",
      "recognize\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "protein\n",
      "sequence-specific\n",
      "manner\n",
      "body\n",
      "TRS\n",
      "well\n",
      "candidate\n",
      "serve\n",
      "protein\n",
      "recognition\n",
      "site\n",
      "protein\n",
      "mediate\n",
      "pause\n",
      "nascent\n",
      "strand\n",
      "synthesis\n",
      "and/or\n",
      "nascent\n",
      "strand\n",
      "transfer\n",
      "resemble\n",
      "DNA-dependent\n",
      "RNA\n",
      "polymerase\n",
      "termination\n",
      "system\n",
      "specific\n",
      "DNA-binding\n",
      "terminator\n",
      "protein\n",
      "bind\n",
      "termination\n",
      "sequence\n",
      "Reeder\n",
      "Lang\n",
      "1997\n",
      "function\n",
      "HIV\n",
      "nucleocapsid\n",
      "protein\n",
      "promote\n",
      "minus\n",
      "strand\n",
      "strong-stop\n",
      "DNA\n",
      "transfer\n",
      "Guo\n",
      "et\n",
      "al.\n",
      "1997\n",
      "EAV\n",
      "replicase\n",
      "component\n",
      "nsp1\n",
      "recently\n",
      "show\n",
      "possess\n",
      "sg\n",
      "RNA\n",
      "synthesis-specific\n",
      "activity\n",
      "Tijms\n",
      "et\n",
      "al.\n",
      "2001\n",
      "good\n",
      "candidate\n",
      "regulatory\n",
      "role\n",
      "Residues\n",
      "predict\n",
      "form\n",
      "zinc\n",
      "finger\n",
      "structure\n",
      "nsp1\n",
      "show\n",
      "necessary\n",
      "sg\n",
      "RNA\n",
      "synthesis\n",
      "Interestingly\n",
      "zinc\n",
      "finger\n",
      "structure\n",
      "HIV\n",
      "nucleocapsid\n",
      "protein\n",
      "facilitate\n",
      "strand\n",
      "transfer\n",
      "Guo\n",
      "et\n",
      "al.\n",
      "2000\n",
      "Finally\n",
      "note\n",
      "RNA\n",
      "structure\n",
      "nascent\n",
      "strand\n",
      "influence\n",
      "pause\n",
      "strand\n",
      "transfer\n",
      "reinitiation\n",
      "illustrate\n",
      "fact\n",
      "stable\n",
      "hairpin\n",
      "structure\n",
      "nascent\n",
      "strand\n",
      "promote\n",
      "termination\n",
      "transcription\n",
      "Escherichia\n",
      "coli\n",
      "RNA\n",
      "polymerase\n",
      "Wilson\n",
      "von\n",
      "Hippel\n",
      "1995\n",
      "Site-directed\n",
      "mutagenesis\n",
      "EAV\n",
      "leader\n",
      "body\n",
      "TRSs\n",
      "carry\n",
      "describe\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999a\n",
      "mutant\n",
      "construct\n",
      "sequence\n",
      "Following\n",
      "vitro\n",
      "transcription\n",
      "infectious\n",
      "cDNA\n",
      "clone\n",
      "full-length\n",
      "EAV\n",
      "RNA\n",
      "introduce\n",
      "BHK-21\n",
      "cell\n",
      "electroporation\n",
      "describe\n",
      "van\n",
      "Dinten\n",
      "et\n",
      "al.\n",
      "1997\n",
      "Immunofluorescence\n",
      "assay\n",
      "EAV-specific\n",
      "antiserum\n",
      "perform\n",
      "14\n",
      "h\n",
      "post-transfection\n",
      "describe\n",
      "van\n",
      "der\n",
      "Meer\n",
      "et\n",
      "al.\n",
      "1998\n",
      "visualize\n",
      "nucleus\n",
      "cell\n",
      "count\n",
      "nuclear\n",
      "DNA\n",
      "stain\n",
      "5\n",
      "µg/ml\n",
      "Hoechst\n",
      "B2883\n",
      "Sigma\n",
      "Cells\n",
      "count\n",
      "Scion\n",
      "Image\n",
      "software\n",
      "Scion\n",
      "Corporation\n",
      "percentage\n",
      "transfect\n",
      "cell\n",
      "calculate\n",
      "basis\n",
      "numb\n",
      "cell\n",
      "positive\n",
      "EAV\n",
      "replicase\n",
      "component\n",
      "nsp3\n",
      "Pedersen\n",
      "et\n",
      "al.\n",
      "1999\n",
      "RNA\n",
      "analysis\n",
      "cell\n",
      "lyse\n",
      "14\n",
      "h\n",
      "post-transfection\n",
      "Intracellular\n",
      "RNA\n",
      "isolation\n",
      "perform\n",
      "acidic\n",
      "phenol\n",
      "method\n",
      "describe\n",
      "Pasternak\n",
      "et\n",
      "al.\n",
      "2000\n",
      "Total\n",
      "intracellular\n",
      "RNA\n",
      "resolve\n",
      "denature\n",
      "agarose\n",
      "formaldehyde\n",
      "gel\n",
      "Hybridization\n",
      "dry\n",
      "gel\n",
      "radioactively\n",
      "label\n",
      "oligonucleotide\n",
      "probe\n",
      "E154\n",
      "complementary\n",
      "3′\n",
      "end\n",
      "EAV\n",
      "genome\n",
      "recognize\n",
      "viral\n",
      "mRNA\n",
      "molecule\n",
      "genomic\n",
      "subgenomic\n",
      "phosphoimager\n",
      "quantitation\n",
      "individual\n",
      "band\n",
      "perform\n",
      "describe\n",
      "Pasternak\n",
      "et\n",
      "al.\n",
      "2000\n",
      "determine\n",
      "leader\n",
      "body\n",
      "junction\n",
      "sequence\n",
      "sg\n",
      "mRNA7\n",
      "mRNA7-specific\n",
      "RT\n",
      "PCRs\n",
      "carry\n",
      "describe\n",
      "van\n",
      "Marle\n",
      "et\n",
      "al.\n",
      "1999b\n",
      "antisense\n",
      "RT\n",
      "PCR\n",
      "primer\n",
      "RNA7\n",
      "body\n",
      "region\n",
      "sense\n",
      "PCR\n",
      "primer\n",
      "match\n",
      "leader\n",
      "sequence\n",
      "RT\n",
      "PCR\n",
      "product\n",
      "sequence\n",
      "directly\n",
      "describe\n",
      "Pasternak\n",
      "et\n",
      "al.\n",
      "2000\n",
      "leader-derived\n",
      "primer\n",
      "ABI\n",
      "PRISM\n",
      "™\n",
      "sequence\n",
      "kit\n",
      "Perkin\n",
      "Elmer\n",
      "ABI\n",
      "PRISM\n",
      "™\n",
      "310\n",
      "Genetic\n",
      "Analyser\n",
      "Perkin\n",
      "Elmer\n"
     ]
    }
   ],
   "source": [
    "# This cell tests the mechanisms used to remove stopwords,\n",
    "# punctuation, spaces, and extract the token's lemma.\n",
    "for token in doc:\n",
    "    if not token.is_stop and not token.is_punct and not token.is_space:\n",
    "        print(token.lemma_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are 95 occurrences of OOV tokens, which stands for 2.16% of the total amount of document's tokens.\n",
    "Note that relevant tokens, such as `coronavirus` are included in the language model vocabulary.\n",
    "\n",
    "## Latent Dirichlet Allocation (LDA)\n",
    "\n",
    "The following cells will perform topic modelling experiments using the LDA technique.\n",
    "The `scikit-learn` implementation of this model will be used."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div>\n",
       "        <style>\n",
       "            /* Turns off some styling */\n",
       "            progress {\n",
       "                /* gets rid of default border in Firefox and Opera. */\n",
       "                border: none;\n",
       "                /* Needs to be in here for Safari polyfill so background images work as expected. */\n",
       "                background-size: auto;\n",
       "            }\n",
       "            .progress-bar-interrupted, .progress-bar-interrupted::-webkit-progress-bar {\n",
       "                background: #F44336;\n",
       "            }\n",
       "        </style>\n",
       "      <progress value='44648' class='' max='44648' style='width:300px; height:20px; vertical-align: middle;'></progress>\n",
       "      100.00% [44648/44648 00:27<00:00]\n",
       "    </div>\n",
       "    "
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# First, let's process all documents texts.\n",
    "def process_papers_file_contents(papers):\n",
    "    texts = []\n",
    "    for paper in progress_bar(papers):\n",
    "        text = \" \\n \".join([ paragraph[\"text\"] for paragraph in paper._file_contents[\"body_text\"]])\n",
    "        \"\"\"\n",
    "        NB.: for development speed purposes, the only document's attributes\n",
    "        considered for the topic modelling were the title and the abstract.\n",
    "        Should the text be included in other experiments, the following line\n",
    "        should be modified to include `{paper.text}`.\n",
    "        \"\"\"\n",
    "        texts.append(f\"{paper.title} \\n {paper.abstract}\")\n",
    "    return texts\n",
    "\n",
    "docs = process_papers_file_contents(\n",
    "    papers=list(pageranks.keys()),\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> Sequence requirements for RNA strand transfer during nidovirus discontinuous subgenomic RNA synthesis \n",
      " Nidovirus subgenomic mRNAs contain a leader sequence derived from the 5′ end of the genome fused to different sequences (‘bodies’) derived from the 3′ end. Their generation involves a unique mechanism of discontinuous subgenomic RNA synthesis that resembles copy-choice RNA recombination. During this process, the nascent RNA strand is transferred from one site in the template to another, during either plus or minus strand synthesis, to yield subgenomic RNA molecules. Central to this process are transcription-regulating sequences (TRSs), which are present at both template sites and ensure the fidelity of strand transfer. Here we present results of a comprehensive co-variation mutagenesis study of equine arteritis virus TRSs, demonstrating that discontinuous RNA synthesis depends not only on base pairing between sense leader TRS and antisense body TRS, but also on the primary sequence of the body TRS. While the leader TRS merely plays a targeting role for strand transfer, the body TRS fulfils multiple functions. The sequences of mRNA leader–body junctions of TRS mutants strongly suggested that the discontinuous step occurs during minus strand synthesis.\n",
      ">> Crystal structure of murine sCEACAM1a[1,4]: a coronavirus receptor in the CEA family \n",
      " CEACAM1 is a member of the carcinoembryonic antigen (CEA) family. Isoforms of murine CEACAM1 serve as receptors for mouse hepatitis virus (MHV), a murine coronavirus. Here we report the crystal structure of soluble murine sCEACAM1a[1,4], which is composed of two Ig-like domains and has MHV neutralizing activity. Its N-terminal domain has a uniquely folded CC′ loop that encompasses key virus-binding residues. This is the first atomic structure of any member of the CEA family, and provides a prototypic architecture for functional exploration of CEA family members. We discuss the structural basis of virus receptor activities of murine CEACAM1 proteins, binding of Neisseria to human CEACAM1, and other homophilic and heterophilic interactions of CEA family members.\n",
      ">> Synthesis of a novel hepatitis C virus protein by ribosomal frameshift \n",
      " Hepatitis C virus (HCV) is an important human pathogen that affects ∼100 million people worldwide. Its RNA genome codes for a polyprotein, which is cleaved by viral and cellular proteases to produce at least 10 mature viral protein products. We report here the discovery of a novel HCV protein synthesized by ribosomal frameshift. This protein, which we named the F protein, is synthesized from the initiation codon of the polyprotein sequence followed by ribosomal frameshift into the −2/+1 reading frame. This ribosomal frameshift requires only codons 8–14 of the core protein‐coding sequence, and the shift junction is located at or near codon 11. An F protein analog synthesized in vitro reacted with the sera of HCV patients but not with the sera of hepatitis B patients, indicating the expression of the F protein during natural HCV infection. This unexpected finding may open new avenues for the development of anti‐HCV drugs.\n",
      ">> Structure of coronavirus main proteinase reveals combination of a chymotrypsin fold with an extra α-helical domain \n",
      " The key enzyme in coronavirus polyprotein processing is the viral main proteinase, M(pro), a protein with extremely low sequence similarity to other viral and cellular proteinases. Here, the crystal structure of the 33.1 kDa transmissible gastroenteritis (corona)virus M(pro) is reported. The structure was refined to 1.96 Å resolution and revealed three dimers in the asymmetric unit. The mutual arrangement of the protomers in each of the dimers suggests that M(pro) self-processing occurs in trans. The active site, comprised of Cys144 and His41, is part of a chymotrypsin-like fold that is connected by a 16 residue loop to an extra domain featuring a novel α-helical fold. Molecular modelling and mutagenesis data implicate the loop in substrate binding and elucidate S1 and S2 subsites suitable to accommodate the side chains of the P1 glutamine and P2 leucine residues of M(pro) substrates. Interactions involving the N-terminus and the α-helical domain stabilize the loop in the orientation required for trans-cleavage activity. The study illustrates that RNA viruses have evolved unprecedented variations of the classical chymotrypsin fold.\n",
      ">> Discontinuous and non-discontinuous subgenomic RNA transcription in a nidovirus \n",
      " Arteri-, corona-, toro- and roniviruses are evolutionarily related positive-strand RNA viruses, united in the order Nidovirales. The best studied nidoviruses, the corona- and arteriviruses, employ a unique transcription mechanism, which involves discontinuous RNA synthesis, a process resembling similarity-assisted copy-choice RNA recombination. During infection, multiple subgenomic (sg) mRNAs are transcribed from a mirror set of sg negative-strand RNA templates. The sg mRNAs all possess a short 5′ common leader sequence, derived from the 5′ end of the genomic RNA. The joining of the non-contiguous ‘leader’ and ‘body’ sequences presumably occurs during minus-strand synthesis. To study whether toroviruses use a similar transcription mechanism, we characterized the 5′ termini of the genome and the four sg mRNAs of Berne virus (BEV). We show that BEV mRNAs 3–5 lack a leader sequence. Surprisingly, however, RNA 2 does contain a leader, identical to the 5′-terminal 18 residues of the genome. Apparently, BEV combines discontinuous and non-discontinous RNA synthesis to produce its sg mRNAs. Our findings have important implications for the understanding of the mechanism and evolution of nidovirus transcription.\n"
     ]
    }
   ],
   "source": [
    "for doc in docs[:5]:\n",
    "    print('>>', doc)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Vectors storing the token occurrence count will be used as document representations.\n",
    "`tf-idf` vectors are purposefully not used because the document frequency normalization is already carried out by the LDA technique."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.feature_extraction.text import CountVectorizer\n",
    "\n",
    "def tokenizer(sentence):\n",
    "    tokens = []\n",
    "    for token in nlp(sentence, disable=[\"tagger\", \"parser\", \"ner\"]):\n",
    "        # Se descartan números, stopwords, puntuación, espacio y tokens de largo 1\n",
    "        if not (token.like_num or token.is_stop or token.is_punct or token.is_space or len(token)==1):\n",
    "            tokens.append(token.lemma_)\n",
    "    return tokens\n",
    "\n",
    "count_vectorizer = CountVectorizer(\n",
    "    tokenizer=tokenizer,\n",
    "    lowercase=True,\n",
    ")\n",
    "vectorized_docs = count_vectorizer.fit_transform(docs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<44648x164664 sparse matrix of type '<class 'numpy.int64'>'\n",
       "\twith 3168863 stored elements in Compressed Sparse Row format>"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vectorized_docs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'sequence': 137903,\n",
       " 'requirement': 130614,\n",
       " 'rna': 132466,\n",
       " 'strand': 143531,\n",
       " 'transfer': 151049,\n",
       " 'nidovirus': 105209,\n",
       " 'discontinuous': 51900,\n",
       " 'subgenomic': 144172,\n",
       " 'synthesis': 145952,\n",
       " 'mrnas': 100208,\n",
       " 'contain': 44122,\n",
       " 'leader': 89237,\n",
       " 'derive': 50233,\n",
       " '5′': 14413,\n",
       " 'end': 57299,\n",
       " 'genome': 67110,\n",
       " 'fuse': 65319,\n",
       " 'different': 51336,\n",
       " 'body': 33356,\n",
       " '3′': 12213,\n",
       " 'generation': 67003,\n",
       " 'involve': 83901,\n",
       " 'unique': 154307,\n",
       " 'mechanism': 95218,\n",
       " 'resemble': 130672,\n",
       " 'copy-choice': 44580,\n",
       " 'recombination': 129362,\n",
       " 'process': 124103,\n",
       " 'nascent': 103366,\n",
       " 'site': 140083,\n",
       " 'template': 147705,\n",
       " 'plus': 120219,\n",
       " 'minus': 98084,\n",
       " 'yield': 161147,\n",
       " 'molecule': 99191,\n",
       " 'central': 38846,\n",
       " 'transcription-regulating': 150990,\n",
       " 'trss': 152146,\n",
       " 'present': 123413,\n",
       " 'ensure': 57825,\n",
       " 'fidelity': 62774,\n",
       " 'result': 131035,\n",
       " 'comprehensive': 43284,\n",
       " 'co-variation': 42097,\n",
       " 'mutagenesis': 101415,\n",
       " 'study': 143846,\n",
       " 'equine': 58803,\n",
       " 'arteritis': 27290,\n",
       " 'virus': 157408,\n",
       " 'demonstrate': 49856,\n",
       " 'depend': 50113,\n",
       " 'base': 30282,\n",
       " 'pair': 114369,\n",
       " 'sense': 137741,\n",
       " 'tr': 150645,\n",
       " 'antisense': 25870,\n",
       " 'primary': 123675,\n",
       " 'merely': 95936,\n",
       " 'play': 120033,\n",
       " 'target': 146978,\n",
       " 'role': 132870,\n",
       " 'fulfil': 65086,\n",
       " 'multiple': 101048,\n",
       " 'function': 65174,\n",
       " 'mrna': 100180,\n",
       " 'junction': 85945,\n",
       " 'mutant': 101422,\n",
       " 'strongly': 143731,\n",
       " 'suggest': 144603,\n",
       " 'step': 143160,\n",
       " 'occur': 109894,\n",
       " 'crystal': 46502,\n",
       " 'structure': 143760,\n",
       " 'murine': 101325,\n",
       " 'sceacam1a[1,4': 136179,\n",
       " 'coronavirus': 44742,\n",
       " 'receptor': 129158,\n",
       " 'cea': 38422,\n",
       " 'family': 61565,\n",
       " 'ceacam1': 38439,\n",
       " 'member': 95674,\n",
       " 'carcinoembryonic': 36541,\n",
       " 'antigen': 25591,\n",
       " 'isoform': 84625,\n",
       " 'serve': 138302,\n",
       " 'mouse': 99958,\n",
       " 'hepatitis': 72687,\n",
       " 'mhv': 96830,\n",
       " 'report': 130503,\n",
       " 'soluble': 141273,\n",
       " 'compose': 43258,\n",
       " 'ig-like': 78856,\n",
       " 'domain': 53119,\n",
       " 'neutralize': 104777,\n",
       " 'activity': 20156,\n",
       " 'n-terminal': 102280,\n",
       " 'uniquely': 154308,\n",
       " 'fold': 63968,\n",
       " 'cc′': 37738,\n",
       " 'loop': 91270,\n",
       " 'encompass': 57268,\n",
       " 'key': 86819,\n",
       " 'virus-binding': 157450,\n",
       " 'residue': 130724,\n",
       " 'atomic': 28250,\n",
       " 'provide': 125180,\n",
       " 'prototypic': 125121,\n",
       " 'architecture': 26889,\n",
       " 'functional': 65184,\n",
       " 'exploration': 60664,\n",
       " 'discuss': 51946,\n",
       " 'structural': 143740,\n",
       " 'basis': 30350,\n",
       " 'protein': 124794,\n",
       " 'bind': 32023,\n",
       " 'neisseria': 104158,\n",
       " 'human': 76468,\n",
       " 'homophilic': 74865,\n",
       " 'heterophilic': 73052,\n",
       " 'interaction': 82822,\n",
       " 'novel': 108253,\n",
       " 'ribosomal': 131911,\n",
       " 'frameshift': 64629,\n",
       " 'hcv': 71889,\n",
       " 'important': 80567,\n",
       " 'pathogen': 115626,\n",
       " 'affect': 21208,\n",
       " '∼100': 164145,\n",
       " 'people': 117082,\n",
       " 'worldwide': 160213,\n",
       " 'code': 42260,\n",
       " 'polyprotein': 121213,\n",
       " 'cleave': 41273,\n",
       " 'viral': 157141,\n",
       " 'cellular': 38755,\n",
       " 'protease': 124718,\n",
       " 'produce': 124165,\n",
       " 'mature': 94670,\n",
       " 'product': 124176,\n",
       " 'discovery': 51916,\n",
       " 'synthesize': 145959,\n",
       " 'name': 102947,\n",
       " 'initiation': 82134,\n",
       " 'codon': 42296,\n",
       " 'follow': 64010,\n",
       " '−2/+1': 163897,\n",
       " 'read': 128930,\n",
       " 'frame': 64617,\n",
       " 'require': 130613,\n",
       " '8–14': 16592,\n",
       " 'core': 44619,\n",
       " 'protein‐coding': 124999,\n",
       " 'shift': 138877,\n",
       " 'locate': 90993,\n",
       " 'near': 103892,\n",
       " 'analog': 23672,\n",
       " 'vitro': 157828,\n",
       " 'react': 128892,\n",
       " 'serum': 138271,\n",
       " 'patient': 115814,\n",
       " 'indicate': 81100,\n",
       " 'expression': 60745,\n",
       " 'natural': 103491,\n",
       " 'infection': 81498,\n",
       " 'unexpected': 154177,\n",
       " 'find': 62977,\n",
       " 'open': 110957,\n",
       " 'new': 104841,\n",
       " 'avenue': 29030,\n",
       " 'development': 50674,\n",
       " 'anti‐hcv': 25991,\n",
       " 'drug': 53852,\n",
       " 'main': 93530,\n",
       " 'proteinase': 124958,\n",
       " 'reveal': 131339,\n",
       " 'combination': 42808,\n",
       " 'chymotrypsin': 40473,\n",
       " 'extra': 60831,\n",
       " 'α-helical': 162638,\n",
       " 'enzyme': 58242,\n",
       " 'm(pro': 92783,\n",
       " 'extremely': 60961,\n",
       " 'low': 91387,\n",
       " 'similarity': 139572,\n",
       " 'kda': 86635,\n",
       " 'transmissible': 151228,\n",
       " 'gastroenteritis': 66403,\n",
       " '(corona)virus': 428,\n",
       " 'refine': 129726,\n",
       " 'resolution': 130781,\n",
       " 'dimer': 51582,\n",
       " 'asymmetric': 28034,\n",
       " 'unit': 154314,\n",
       " 'mutual': 101460,\n",
       " 'arrangement': 27210,\n",
       " 'protomers': 125099,\n",
       " 'self-processing': 137392,\n",
       " 'trans': 150830,\n",
       " 'active': 20133,\n",
       " 'comprise': 43305,\n",
       " 'cys144': 47657,\n",
       " 'his41': 73872,\n",
       " 'chymotrypsin-like': 40474,\n",
       " 'connect': 43808,\n",
       " 'feature': 62087,\n",
       " 'molecular': 99167,\n",
       " 'model': 98943,\n",
       " 'datum': 48733,\n",
       " 'implicate': 80540,\n",
       " 'substrate': 144366,\n",
       " 'elucidate': 56868,\n",
       " 's1': 134696,\n",
       " 's2': 134813,\n",
       " 'subsite': 144329,\n",
       " 'suitable': 144637,\n",
       " 'accommodate': 19584,\n",
       " 'chain': 39281,\n",
       " 'p1': 113128,\n",
       " 'glutamine': 68285,\n",
       " 'p2': 113294,\n",
       " 'leucine': 89748,\n",
       " 'n-terminus': 102285,\n",
       " 'stabilize': 142740,\n",
       " 'orientation': 111641,\n",
       " 'trans-cleavage': 150865,\n",
       " 'illustrate': 79630,\n",
       " 'evolve': 60151,\n",
       " 'unprecedented': 154481,\n",
       " 'variation': 156011,\n",
       " 'classical': 41177,\n",
       " 'non-discontinuous': 106183,\n",
       " 'transcription': 150968,\n",
       " 'arteri-': 27279,\n",
       " 'corona-': 44691,\n",
       " 'toro-': 150388,\n",
       " 'roniviruses': 132919,\n",
       " 'evolutionarily': 60140,\n",
       " 'relate': 130081,\n",
       " 'positive-strand': 121600,\n",
       " 'unite': 154327,\n",
       " 'order': 111374,\n",
       " 'nidovirales': 105208,\n",
       " 'well': 159271,\n",
       " 'nidoviruses': 105214,\n",
       " 'arteriviruses': 27300,\n",
       " 'employ': 57119,\n",
       " 'similarity-assisted': 139573,\n",
       " 'sg': 138605,\n",
       " 'transcribe': 150951,\n",
       " 'mirror': 98255,\n",
       " 'set': 138338,\n",
       " 'negative-strand': 104105,\n",
       " 'possess': 121633,\n",
       " 'short': 138983,\n",
       " 'common': 42931,\n",
       " 'genomic': 67151,\n",
       " 'join': 85740,\n",
       " 'non-contiguous': 106091,\n",
       " 'presumably': 123483,\n",
       " 'minus-strand': 98089,\n",
       " 'toroviruses': 150399,\n",
       " 'use': 155167,\n",
       " 'similar': 139568,\n",
       " 'characterize': 39434,\n",
       " 'terminus': 147927,\n",
       " 'berne': 31299,\n",
       " 'bev': 31635,\n",
       " '3–5': 12175,\n",
       " 'lack': 88404,\n",
       " 'surprisingly': 145298,\n",
       " 'identical': 78240,\n",
       " '5′-terminal': 14490,\n",
       " 'apparently': 26496,\n",
       " 'combine': 42817,\n",
       " 'non-discontinous': 106182,\n",
       " 'finding': 62982,\n",
       " 'implication': 80542,\n",
       " 'understand': 154080,\n",
       " 'evolution': 60134,\n",
       " 'airborne': 21951,\n",
       " 'rhinovirus': 131722,\n",
       " 'detection': 50552,\n",
       " 'effect': 55987,\n",
       " 'ultraviolet': 153673,\n",
       " 'irradiation': 84340,\n",
       " 'semi-nested': 137590,\n",
       " 'rt-pcr': 133771,\n",
       " 'assay': 27824,\n",
       " 'background': 29717,\n",
       " 'cause': 37280,\n",
       " 'upper': 154802,\n",
       " 'respiratory': 130893,\n",
       " 'tract': 150710,\n",
       " 'asthma': 27945,\n",
       " 'exacerbation': 60207,\n",
       " 'possibly': 121644,\n",
       " 'death': 49151,\n",
       " 'method': 96458,\n",
       " 'transmission': 151234,\n",
       " 'dispute': 52258,\n",
       " 'aerosol': 21102,\n",
       " 'likely': 90246,\n",
       " 'adult': 20910,\n",
       " 'possible': 121642,\n",
       " 'develope': 50671,\n",
       " 'detect': 50535,\n",
       " 'aerosolized': 21120,\n",
       " 'extend': 60787,\n",
       " 'exist': 60406,\n",
       " 'technology': 147509,\n",
       " 'infectious': 81586,\n",
       " 'agent': 21489,\n",
       " 'nasal': 103341,\n",
       " 'specimen': 141843,\n",
       " 'small': 140649,\n",
       " 'chamber': 39325,\n",
       " 'experiment': 60582,\n",
       " 'conduct': 43568,\n",
       " 'decrease': 49386,\n",
       " 'concentration': 43413,\n",
       " 'determine': 50602,\n",
       " 'uv': 155373,\n",
       " 'rhinoviral': 131721,\n",
       " 'expose': 60701,\n",
       " 'dose': 53334,\n",
       " 'mj/m(2': 98567,\n",
       " 'collect': 42618,\n",
       " 'teflon': 147551,\n",
       " 'filter': 62924,\n",
       " 'recover': 129461,\n",
       " 'qiagen': 126812,\n",
       " 'avl': 29109,\n",
       " 'buff': 34589,\n",
       " 'qiaamp': 126809,\n",
       " 'kit': 87110,\n",
       " 'corp': 44836,\n",
       " 'valencia': 155831,\n",
       " 'california': 35836,\n",
       " 'gel': 66774,\n",
       " 'electrophoresis': 56648,\n",
       " 'obtain': 109813,\n",
       " 'positive': 121582,\n",
       " 'sample': 135356,\n",
       " 'tcid(50': 147316,\n",
       " 'great': 69347,\n",
       " 'upper-room': 154812,\n",
       " 'germicidal': 67414,\n",
       " 'application': 26543,\n",
       " 'inhibit': 82065,\n",
       " 'subsequent': 144310,\n",
       " 'conclusion': 43480,\n",
       " 'air': 21908,\n",
       " 'extraction': 60884,\n",
       " 'methodology': 96477,\n",
       " 'applicable': 26538,\n",
       " 'indoor': 81257,\n",
       " 'office': 110137,\n",
       " 'school': 136337,\n",
       " 'distinguish': 52336,\n",
       " 'inactivate': 80792,\n",
       " 'particle': 115350,\n",
       " 'discover': 51914,\n",
       " 'history': 73983,\n",
       " 'stomach': 143431,\n",
       " 'bacterium': 29866,\n",
       " 'recent': 129145,\n",
       " 'analysis': 23692,\n",
       " 'evolutionary': 60141,\n",
       " 'congruent': 43746,\n",
       " 'hypothesize': 77447,\n",
       " 'pattern': 115943,\n",
       " 'ancient': 23802,\n",
       " 'modern': 99003,\n",
       " 'population': 121408,\n",
       " 'migration': 97737,\n",
       " 'phylogenetic': 118961,\n",
       " 'tree': 151498,\n",
       " 'strain': 143492,\n",
       " 'helicobacter': 72336,\n",
       " 'pylorus': 126474,\n",
       " 'polyoma': 121150,\n",
       " 'jc': 85378,\n",
       " 'take': 146788,\n",
       " 'geographically': 67287,\n",
       " 'diverse': 52428,\n",
       " 'group': 69543,\n",
       " 'being': 30999,\n",
       " 'correlate': 44866,\n",
       " 'closely': 41523,\n",
       " 'relationship': 130093,\n",
       " 'prokaryotic-style': 124379,\n",
       " 'frameshifting': 64639,\n",
       " 'plant': 119823,\n",
       " 'translation': 151164,\n",
       " 'system': 146013,\n",
       " 'conservation': 43897,\n",
       " 'unusual': 154699,\n",
       " 'single-trna': 139851,\n",
       " 'slippage': 140533,\n",
       " 'event': 60027,\n",
       " 'signal': 139384,\n",
       " 'mobile': 98878,\n",
       " 'genetic': 67028,\n",
       " 'element': 56701,\n",
       " 'gene': 66868,\n",
       " 'prokaryote': 124376,\n",
       " 'eukaryote': 59797,\n",
       " 'typically': 152992,\n",
       " 'slippery': 140538,\n",
       " 'xxy': 160802,\n",
       " 'yyz': 161425,\n",
       " 'stimulatory': 143345,\n",
       " 'influence': 81794,\n",
       " 'host': 75101,\n",
       " 'translational': 151177,\n",
       " 'environment': 58177,\n",
       " 'identity': 78279,\n",
       " 'sequence-decoding': 137912,\n",
       " 'trnas': 151995,\n",
       " 'express': 60739,\n",
       " 'candidate': 36101,\n",
       " 'escherichia': 59218,\n",
       " 'coli': 42541,\n",
       " 'wheatgerm': 159549,\n",
       " 'extract': 60879,\n",
       " 'deplete': 50138,\n",
       " 'endogenous': 57417,\n",
       " 'supplement': 145102,\n",
       " 'prokaryotic': 124377,\n",
       " 'eukaryotic': 59799,\n",
       " 'trna': 151975,\n",
       " 'decode': 49337,\n",
       " 'aag': 18767,\n",
       " 'a-site': 17595,\n",
       " 'e.coli': 55035,\n",
       " 'trna(lys': 151980,\n",
       " 'promote': 124455,\n",
       " 'highly': 73596,\n",
       " 'ribosome': 131916,\n",
       " 'appear': 26508,\n",
       " 'adjacent': 20700,\n",
       " 'p-site': 112943,\n",
       " 'influential': 81800,\n",
       " 'conversely': 44394,\n",
       " 'asparaginyl-trna': 27753,\n",
       " 'dual': 54065,\n",
       " 'determinant': 50592,\n",
       " 'selection': 137190,\n",
       " 'single-': 139723,\n",
       " 'dual-trna': 54104,\n",
       " 'time': 149365,\n",
       " 'trna(asn': 151977,\n",
       " 'inherently': 82055,\n",
       " 'unslippery': 154584,\n",
       " 'induce': 81270,\n",
       " 'efficient': 56044,\n",
       " 'context': 44175,\n",
       " 'recruit': 129488,\n",
       " 'army': 27170,\n",
       " 'man': 93828,\n",
       " 'john': 85734,\n",
       " 'bunyan': 34684,\n",
       " 'memorable': 95779,\n",
       " 'phrase': 118911,\n",
       " 'fear': 62071,\n",
       " 'face': 61258,\n",
       " 'wear': 159128,\n",
       " 'surgical': 145272,\n",
       " 'mask': 94427,\n",
       " 'sseg': 142610,\n",
       " 'virulence': 157389,\n",
       " 'salmonella': 135259,\n",
       " 'golgi': 68747,\n",
       " 'network': 104405,\n",
       " 'intracellular': 83532,\n",
       " 'replication': 130445,\n",
       " 'bacterial': 29807,\n",
       " 'enterica': 57855,\n",
       " 'membrane-bound': 95701,\n",
       " 'compartment': 43079,\n",
       " 'call': 35848,\n",
       " 'salmonella-containing': 135260,\n",
       " 'vacuole': 155762,\n",
       " 'scvs': 136645,\n",
       " 'maturation': 94664,\n",
       " 'scv': 136638,\n",
       " 'show': 139054,\n",
       " 'selective': 137196,\n",
       " 'endocytic': 57393,\n",
       " 'pathway': 115798,\n",
       " 'invasion': 83816,\n",
       " 'epithelial': 58610,\n",
       " 'cell': 38544,\n",
       " 'perinuclear': 117440,\n",
       " 'location': 90994,\n",
       " 'majority': 93576,\n",
       " 'surround': 145306,\n",
       " 'membrane': 95688,\n",
       " 'dependent': 50120,\n",
       " 'pathogenicity': 115709,\n",
       " 'island': 84563,\n",
       " 'type': 152935,\n",
       " 'iii': 79179,\n",
       " 'secretion': 136952,\n",
       " 'effector': 56002,\n",
       " 'infect': 81449,\n",
       " 'associate': 27898,\n",
       " 'peripheral': 117476,\n",
       " 'punctate': 126172,\n",
       " 'able': 19000,\n",
       " 'multiply': 101120,\n",
       " 'furthermore': 65307,\n",
       " 'mutation': 101429,\n",
       " 'disruption': 52264,\n",
       " 'growth': 69594,\n",
       " 'co-localized': 41980,\n",
       " 'extensively': 60802,\n",
       " 'marker': 94273,\n",
       " 'trans-golgi': 150887,\n",
       " 'identify': 78274,\n",
       " 'golgi-targeting': 68759,\n",
       " 'region': 129835,\n",
       " 'localization': 90983,\n",
       " 'simultaneous': 139642,\n",
       " 'secretory': 136968,\n",
       " 'association': 27901,\n",
       " 'hla': 74246,\n",
       " 'class': 41163,\n",
       " 'severe': 138443,\n",
       " 'acute': 20233,\n",
       " 'syndrome': 145862,\n",
       " 'leukocyte': 89798,\n",
       " 'widely': 159716,\n",
       " 'strategy': 143562,\n",
       " 'search': 136818,\n",
       " 'etiology': 59694,\n",
       " 'disease': 51967,\n",
       " 'autoimmune': 28813,\n",
       " 'disorder': 52168,\n",
       " 'taiwan': 146760,\n",
       " 'epidemic': 58441,\n",
       " 'sars': 135640,\n",
       " 'health': 72059,\n",
       " 'care': 36638,\n",
       " 'worker': 160175,\n",
       " 'effort': 56049,\n",
       " 'establish': 59424,\n",
       " 'screen': 136589,\n",
       " 'program': 124308,\n",
       " 'high': 73381,\n",
       " 'risk': 132250,\n",
       " 'personal': 117726,\n",
       " 'distribution': 52352,\n",
       " 'ii': 79086,\n",
       " 'allele': 22580,\n",
       " 'case': 36829,\n",
       " 'control': 44302,\n",
       " 'examine': 60235,\n",
       " 'presence': 123402,\n",
       " 'susceptibly': 145417,\n",
       " 'resistance': 130749,\n",
       " 'hla-class': 74328,\n",
       " 'pcr-ssop': 116334,\n",
       " 'perform': 117347,\n",
       " 'probable': 123994,\n",
       " 'fever': 62493,\n",
       " 'exclude': 60319,\n",
       " 'late': 88876,\n",
       " 'non-infected': 106441,\n",
       " 'additional': 20497,\n",
       " 'normal': 107988,\n",
       " 'healthy': 72131,\n",
       " 'unrelated': 154526,\n",
       " 'taiwanese': 146764,\n",
       " 'woolf': 160143,\n",
       " 'haldane': 71042,\n",
       " 'odd': 110032,\n",
       " 'ratio': 128410,\n",
       " 'correct': 44853,\n",
       " 'p-value': 112960,\n",
       " 'pc': 116127,\n",
       " 'tail': 146733,\n",
       " 'fisher': 63196,\n",
       " 'exact': 60219,\n",
       " 'test': 148003,\n",
       " 'susceptibility': 145360,\n",
       " 'analyze': 23714,\n",
       " 'hla-b*4601': 74309,\n",
       " 'n.s': 102306,\n",
       " 'hla-b*5401': 74311,\n",
       " 'favor': 61859,\n",
       " 'select': 137182,\n",
       " 'compare': 43068,\n",
       " 'severity': 138462,\n",
       " 'significantly': 139433,\n",
       " 'densely': 49990,\n",
       " 'populate': 121405,\n",
       " 'genetically': 67038,\n",
       " 'southern': 141558,\n",
       " 'asian': 27595,\n",
       " 'spread': 142337,\n",
       " 'recently': 129148,\n",
       " 'indigenous': 81119,\n",
       " 'distinct': 52325,\n",
       " 'general': 66977,\n",
       " 'hla-b': 74298,\n",
       " 'frequency': 64795,\n",
       " 'increase': 80994,\n",
       " 'observe': 109766,\n",
       " 'significant': 139432,\n",
       " 'see': 137031,\n",
       " 'independent': 81055,\n",
       " 'need': 104058,\n",
       " 'double': 53405,\n",
       " 'propagation': 124525,\n",
       " 'guangdong': 69805,\n",
       " 'province': 125190,\n",
       " 'rest': 130976,\n",
       " 'china': 39886,\n",
       " 'world': 160195,\n",
       " 'puzzle': 126329,\n",
       " 'contagion': 44109,\n",
       " 'behavior': 30951,\n",
       " 'predict': 123111,\n",
       " 'future': 65392,\n",
       " 'outbreak': 112164,\n",
       " 'implement': 80520,\n",
       " 'effective': 55996,\n",
       " 'prophylactic': 124565,\n",
       " 'measure': 95181,\n",
       " 'standard': 142828,\n",
       " 'susceptible-infected-removed': 145394,\n",
       " 'sir': 139977,\n",
       " 'account': 19602,\n",
       " 'develop': 50668,\n",
       " 'superimpose': 145020,\n",
       " 'hong': 74908,\n",
       " 'kong': 87449,\n",
       " 'explore': 60670,\n",
       " 'situation': 140140,\n",
       " 'change': 39355,\n",
       " 'tropism': 152057,\n",
       " 'consequence': 43886,\n",
       " 'innocuous': 82261,\n",
       " 'generate': 67002,\n",
       " 'variant': 155998,\n",
       " 'property': 124557,\n",
       " 'reconcile': 129406,\n",
       " 'convenient': 44364,\n",
       " 'mild': 97788,\n",
       " 'protect': 124768,\n",
       " 'simultaneously': 139643,\n",
       " 'suffer': 144563,\n",
       " 'percentage': 117281,\n",
       " 'person': 117711,\n",
       " 'inconsistent': 80966,\n",
       " 'incomplete': 80943,\n",
       " 'reflect': 129732,\n",
       " 'appropriate': 26597,\n",
       " 'identification': 78257,\n",
       " 'finally': 62960,\n",
       " 'framework': 64641,\n",
       " 'fix': 63306,\n",
       " 'limit': 90289,\n",
       " 'allow': 22713,\n",
       " 'landmark': 88658,\n",
       " 'useful': 155177,\n",
       " 'monitor': 99316,\n",
       " 'precursor': 123078,\n",
       " 'large': 88743,\n",
       " 'reservoir': 130686,\n",
       " 'prompt': 124475,\n",
       " 'implementation': 80525,\n",
       " 'precautionary': 123002,\n",
       " 'weather': 159138,\n",
       " 'cool': 44445,\n",
       " 'cloak': 41467,\n",
       " 'hiv-1': 74018,\n",
       " 'sars-cov': 135676,\n",
       " 'anti-sars': 25210,\n",
       " 'febrile': 62104,\n",
       " 'illness': 79609,\n",
       " 'etiologically': 59691,\n",
       " 'link': 90412,\n",
       " 'sars-associated': 135657,\n",
       " 'coronaviridae': 44728,\n",
       " 'spike': 142034,\n",
       " 'believe': 31052,\n",
       " 'entry': 58066,\n",
       " 'facilitate': 61290,\n",
       " 'fusion': 65326,\n",
       " 'responsible': 130963,\n",
       " 'viral-induced': 157164,\n",
       " 'gp41': 69002,\n",
       " 'differ': 51328,\n",
       " 'length': 89529,\n",
       " 'homology': 74837,\n",
       " 'share': 138750,\n",
       " 'motif': 99896,\n",
       " 'construct': 44028,\n",
       " 'conformation': 43682,\n",
       " 'include': 80907,\n",
       " 'leucine/isoleucine': 89751,\n",
       " 'zipper-like': 161783,\n",
       " 'c-terminal': 35161,\n",
       " 'heptad': 72773,\n",
       " 'repeat': 130372,\n",
       " 'upstream': 154848,\n",
       " 'aromatic': 27184,\n",
       " 'residue-rich': 130730,\n",
       " 'juxtapose': 86034,\n",
       " 'transmembrane': 151201,\n",
       " 'segment': 137062,\n",
       " 'point': 120648,\n",
       " 'mode': 98937,\n",
       " 'action': 20101,\n",
       " 'anti-viral': 25335,\n",
       " 'adopt': 20804,\n",
       " 'fda': 62041,\n",
       " 'approve': 26607,\n",
       " 'enfuvirtide': 57666,\n",
       " 'synthetic': 145967,\n",
       " 'peptide': 117143,\n",
       " 'correspond': 44876,\n",
       " 'anti-aids': 24450,\n",
       " 'c34': 35366,\n",
       " 'anti': 24428,\n",
       " 'exert': 60373,\n",
       " 'inhibitory': 82097,\n",
       " 'conformational': 43690,\n",
       " 'activation': 20117,\n",
       " 'inhibitor': 82081,\n",
       " 'pathogenic': 115705,\n",
       " 'tetranucleotide': 148153,\n",
       " 'usage': 155141,\n",
       " 'profile': 124244,\n",
       " 'origin': 111661,\n",
       " 'question': 127162,\n",
       " 'single-stranded': 139847,\n",
       " 'ssrna': 142650,\n",
       " 'non-standard': 106987,\n",
       " 'approach': 26586,\n",
       " 'begin': 30922,\n",
       " 'vectorial': 156255,\n",
       " 'tetra-nucleotide': 148094,\n",
       " 'distance': 52307,\n",
       " 'vector': 156229,\n",
       " 'correlation': 44867,\n",
       " 'coefficient': 42323,\n",
       " 'devise': 50714,\n",
       " 'neighbor-joining': 104143,\n",
       " 'algorithm': 22404,\n",
       " 'multivariate': 101198,\n",
       " 'factor': 61318,\n",
       " 'embed': 56908,\n",
       " 'subsequently': 144312,\n",
       " 'classify': 41191,\n",
       " 'outcome': 112197,\n",
       " 'consistent': 43952,\n",
       " 'know': 87302,\n",
       " 'classification': 41185,\n",
       " 'conform': 43677,\n",
       " 'specific': 141816,\n",
       " 'word': 160147,\n",
       " 'overall': 112403,\n",
       " 'tgv': 148424,\n",
       " 'feline': 62241,\n",
       " 'fcov': 61953,\n",
       " 'close': 41501,\n",
       " 'dna': 52584,\n",
       " 'stage': 142773,\n",
       " 'display': 52219,\n",
       " 'remarkable': 130182,\n",
       " 'discrimination': 51937,\n",
       " 'cpg': 45565,\n",
       " 'upa': 154757,\n",
       " 'di-nucleotide': 50898,\n",
       " 'respectively': 130847,\n",
       " 'upg': 154773,\n",
       " 'cpa': 45538,\n",
       " 'potential': 122213,\n",
       " 'bias': 31834,\n",
       " 'powerful': 122317,\n",
       " 'congruence': 43745,\n",
       " 'prominent': 124442,\n",
       " 'replicase': 130432,\n",
       " 'recovery': 129467,\n",
       " 'microarray': 97105,\n",
       " 'constant': 43978,\n",
       " 'threat': 148998,\n",
       " 'pose': 121557,\n",
       " 'emerge': 56980,\n",
       " 'limitation': 90300,\n",
       " 'demand': 49792,\n",
       " 'technological': 147503,\n",
       " 'describe': 50343,\n",
       " 'microarray-based': 97109,\n",
       " 'platform': 120009,\n",
       " 'characterization': 39432,\n",
       " 'design': 50404,\n",
       " 'wide': 159704,\n",
       " 'range': 128188,\n",
       " 'conserve': 43910,\n",
       " '70mer': 15508,\n",
       " 'fully': 65127,\n",
       " 'reference': 129705,\n",
       " 'genbank': 66854,\n",
       " 'march': 94170,\n",
       " 'hybridization': 76813,\n",
       " 'previously': 123586,\n",
       " 'uncharacterized': 153861,\n",
       " 'isolate': 84636,\n",
       " 'cultivate': 46919,\n",
       " 'approximately': 26610,\n",
       " 'kb': 86597,\n",
       " 'unknown': 154387,\n",
       " 'clone': 41485,\n",
       " 'physically': 119005,\n",
       " 'hybridize': 76816,\n",
       " 'individual': 81176,\n",
       " 'array': 27211,\n",
       " 'fragment': 64595,\n",
       " 'confirm': 43634,\n",
       " 'direct': 51753,\n",
       " 'prove': 125158,\n",
       " 'rapid': 128260,\n",
       " 'tripeptidyl-peptidase': 151854,\n",
       " 'cln2': 41456,\n",
       " 'ubiquitous': 153320,\n",
       " 'sedolisin': 137027,\n",
       " 'serine-carboxyl': 138080,\n",
       " 'peptidase': 117135,\n",
       " 'sedolisins': 137028,\n",
       " 'defect': 49474,\n",
       " 'lead': 89230,\n",
       " 'fatal': 61800,\n",
       " 'neurodegenerative': 104565,\n",
       " 'late-infantile': 88888,\n",
       " 'neuronal': 104660,\n",
       " 'ceroid': 38983,\n",
       " 'lipofuscinosis': 90531,\n",
       " 'species': 141781,\n",
       " 'systematic': 146033,\n",
       " 'attempt': 28381,\n",
       " 'orthologs': 111794,\n",
       " 'numb': 109111,\n",
       " 'macaque': 93127,\n",
       " 'rat': 128372,\n",
       " 'dog': 52819,\n",
       " 'cow': 45400,\n",
       " 'clearly': 41261,\n",
       " 'fish': 63186,\n",
       " 'fugu': 65070,\n",
       " 'zebra': 161561,\n",
       " 'frog': 64880,\n",
       " 'xenopus': 160626,\n",
       " 'tropicalis': 152050,\n",
       " 'three-dimensional': 149017,\n",
       " 'build': 34604,\n",
       " 'mainly': 93540,\n",
       " 'pseudomonas': 125608,\n",
       " 'sp.': 141621,\n",
       " 'distribute': 52350,\n",
       " 'organism': 111562,\n",
       " 'life': 90073,\n",
       " 'cycle': 47421,\n",
       " 'accessible': 19564,\n",
       " 'completely': 43172,\n",
       " 'surprise': 145297,\n",
       " 'tripeptidase': 151848,\n",
       " 'constrain': 44021,\n",
       " 'pocket': 120578,\n",
       " 'anticipate': 25522,\n",
       " 'physiological': 119042,\n",
       " 'pollution': 120821,\n",
       " 'fatality': 61804,\n",
       " 'republic': 130587,\n",
       " 'ecologic': 55668,\n",
       " 'claim': 41127,\n",
       " 'live': 90698,\n",
       " 'mainland': 93535,\n",
       " 'november': 108259,\n",
       " 'vary': 156065,\n",
       " 'geographical': 67286,\n",
       " 'area': 26926,\n",
       " 'partially': 115330,\n",
       " 'explain': 60637,\n",
       " 'level': 89850,\n",
       " 'publicly': 126019,\n",
       " 'morbidity': 99685,\n",
       " 'mortality': 99806,\n",
       " 'utilize': 155314,\n",
       " 'evaluate': 59986,\n",
       " 'index': 81066,\n",
       " 'api': 26319,\n",
       " 'particulate': 115383,\n",
       " 'matt': 94652,\n",
       " 'sulfur': 144741,\n",
       " 'dioxide': 51680,\n",
       " 'nitrogen': 105423,\n",
       " 'carbon': 36458,\n",
       " 'monoxide': 99567,\n",
       " 'ground-level': 69532,\n",
       " 'ozone': 112765,\n",
       " 'fit': 63218,\n",
       " 'assess': 27858,\n",
       " 'long-term': 91191,\n",
       " 'short-term': 139011,\n",
       " 'exposure': 60714,\n",
       " 'dye': 54373,\n",
       " 'rate': 128386,\n",
       " 'increment': 81003,\n",
       " 'moderate': 98981,\n",
       " 'apis': 26340,\n",
       " 'rr': 133304,\n",
       " 'ci': 40495,\n",
       " '1.41–2.40': 4285,\n",
       " 'similarly': 139578,\n",
       " 'twice': 152743,\n",
       " 'die': 51248,\n",
       " '1.31–3.65': 4171,\n",
       " 'ambient': 23142,\n",
       " 'chinese': 39914,\n",
       " 'statistic': 143022,\n",
       " 'fallacy': 61511,\n",
       " 'uncontrolled': 153918,\n",
       " 'confound': 43702,\n",
       " 'possibility': 121638,\n",
       " 'detrimental': 50621,\n",
       " 'prognosis': 124300,\n",
       " 'deserve': 50389,\n",
       " 'investigation': 83867,\n",
       " 'young': 161277,\n",
       " 'virologist': 157331,\n",
       " 'work': 160155,\n",
       " 'singapore': 139711,\n",
       " 'height': 72260,\n",
       " 'pandemic': 114647,\n",
       " 'heterogeneous': 73018,\n",
       " 'nuclear': 108943,\n",
       " 'ribonucleoprotein': 131890,\n",
       " 'a1': 17903,\n",
       " 'regulate': 129922,\n",
       " 'cytoplasmic': 47909,\n",
       " 'hnrnp': 74554,\n",
       " 'pre-mrna': 122862,\n",
       " 'splice': 142203,\n",
       " 'nucleus': 109072,\n",
       " 'regulation': 129923,\n",
       " 'cytoplasm': 47904,\n",
       " 'participate': 115342,\n",
       " 'overexpression': 112442,\n",
       " 'accelerate': 19533,\n",
       " 'kinetic': 87051,\n",
       " 'dominant-negative': 53178,\n",
       " 'transport': 151306,\n",
       " 'delay': 49683,\n",
       " 'global': 68065,\n",
       " 'inhibition': 82071,\n",
       " 'preferential': 123196,\n",
       " 'defective-interfering': 49478,\n",
       " 'rnas': 132622,\n",
       " 'wild-type': 159785,\n",
       " 'complex': 43178,\n",
       " 'polymerase': 121093,\n",
       " 'nucleocapsid': 108982,\n",
       " 'contrast': 44270,\n",
       " 'fail': 61444,\n",
       " 'recruitment': 129494,\n",
       " 'importance': 80563,\n",
       " 'rna-dependent': 132500,\n",
       " 'p62': 113779,\n",
       " 'uba': 153237,\n",
       " 'interact': 82816,\n",
       " 'polyubiquitin': 121276,\n",
       " 'throughput': 149146,\n",
       " 'domain-interacting': 53133,\n",
       " 'proteome-wide': 125036,\n",
       " 'perspective': 117758,\n",
       " 'rabbit': 127758,\n",
       " 'reticulocyte': 131135,\n",
       " 'lysate': 92325,\n",
       " 'successfully': 144483,\n",
       " 'alzheimer': 23080,\n",
       " 'regulatory': 129931,\n",
       " 'easy': 55344,\n",
       " 'route': 133138,\n",
       " 'unfold': 154200,\n",
       " 'vaccinia': 155721,\n",
       " 'disrupt': 52262,\n",
       " 'microtubule': 97513,\n",
       " 'organization': 111582,\n",
       " 'centrosome': 38893,\n",
       " 'cytoskeleton': 47930,\n",
       " 'newly': 104871,\n",
       " 'assemble': 27845,\n",
       " 'accumulate': 19620,\n",
       " 'vicinity': 156931,\n",
       " 'microtubule-organizing': 97521,\n",
       " 'centre': 38869,\n",
       " ...}"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_vectorizer.vocabulary_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A sparse matrix is built with 38.882 rows, one for each document, and 149.389 columns, one for each token."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n",
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   49.8s remaining:   49.8s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   52.8s finished\n",
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 1 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   32.2s remaining:   32.2s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   33.8s finished\n",
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 2 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   26.6s remaining:   26.6s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   27.9s finished\n",
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 3 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   22.3s remaining:   22.3s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   23.4s finished\n",
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 4 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   19.6s remaining:   19.6s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   20.7s finished\n",
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 5 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   17.7s remaining:   17.7s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   18.8s finished\n",
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 6 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   16.9s remaining:   16.9s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   17.8s finished\n",
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 7 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   16.1s remaining:   16.1s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   17.1s finished\n",
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 8 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   17.7s remaining:   17.7s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   18.6s finished\n",
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 9 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   16.3s remaining:   16.3s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   17.0s finished\n",
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 10 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   12.8s remaining:   12.8s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   13.5s finished\n"
     ]
    }
   ],
   "source": [
    "from sklearn.decomposition import LatentDirichletAllocation\n",
    "\n",
    "# Se ajusta un modelo LDA para identificar tópicos\n",
    "lda = LatentDirichletAllocation(\n",
    "    n_components=10,  # number of topics\n",
    "    verbose=2, n_jobs=-1)\n",
    "lda = lda.fit(vectorized_docs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "LatentDirichletAllocation(batch_size=128, doc_topic_prior=None,\n",
       "                          evaluate_every=-1, learning_decay=0.7,\n",
       "                          learning_method='batch', learning_offset=10.0,\n",
       "                          max_doc_update_iter=100, max_iter=10,\n",
       "                          mean_change_tol=0.001, n_components=10, n_jobs=-1,\n",
       "                          perp_tol=0.1, random_state=None,\n",
       "                          topic_word_prior=None, total_samples=1000000.0,\n",
       "                          verbose=2)"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lda"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The execution of the following cells will display the most relevant tokens for each identified topic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def print_topic_words(topic_model, vectorizer, num_words):\n",
    "    feature_names = vectorizer.get_feature_names()\n",
    "    for topic_idx, topic in enumerate(topic_model.components_):\n",
    "        message = \"\\nTopic #%d: \" % topic_idx\n",
    "        message += \" \".join(\n",
    "            [feature_names[i] for i in topic.argsort()[:-num_words - 1:-1]])\n",
    "        print(message)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Topic #0: cell response mouse immune infection expression antibody induce human disease study antigen result gene cytokine increase effect role level protein\n",
      "\n",
      "Topic #1: patient group calve treatment study day disease clinical increase level associate effect high blood control age result lung significantly serum\n",
      "\n",
      "Topic #2: virus protein cell viral infection rna replication host coronavirus gene antiviral activity expression sars-cov human study bind membrane domain role\n",
      "\n",
      "Topic #3: virus strain sequence sample gene assay isolate detect coronavirus detection analysis study porcine pig test result pedv disease genome infectious\n",
      "\n",
      "Topic #4: de nan la en et des les el los las le un se chapter que infection con une del patient\n",
      "\n",
      "Topic #5: respiratory virus patient infection influenza child viral clinical study pneumonia acute human test result case detect sample age severe associate\n",
      "\n",
      "Topic #6: air study effect concentration temperature result method test reduce mask aerosol activity surface transmission water particle airborne extract control room\n",
      "\n",
      "Topic #7: protein structure sequence pathogen human method system new model study approach review molecular target analysis virus provide host disease application\n",
      "\n",
      "Topic #8: vaccine health disease infection mers-cov syndrome respiratory sars care infectious human development influenza control virus patient public emerge clinical outbreak\n",
      "\n",
      "Topic #9: covid-19 disease case model health datum outbreak epidemic study transmission china risk result population numb report spread time public estimate\n"
     ]
    }
   ],
   "source": [
    "print_topic_words(lda, count_vectorizer, 20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The dataset papers will be classified into the different previously modelled topics."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.\n",
      "[Parallel(n_jobs=4)]: Done   2 out of   4 | elapsed:   12.9s remaining:   12.9s\n",
      "[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:   13.5s finished\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array([[1.49689264e-02, 8.19863592e-04, 9.44715587e-01, 3.45768806e-02,\n",
       "        8.19740956e-04, 8.19741833e-04, 8.19767263e-04, 8.19944548e-04,\n",
       "        8.19776267e-04, 8.19771824e-04],\n",
       "       [3.33557314e-01, 1.25001908e-03, 5.17972921e-01, 1.25009814e-03,\n",
       "        1.25053607e-03, 4.61212638e-02, 1.25003355e-03, 9.48476030e-02,\n",
       "        1.25011700e-03, 1.25009486e-03],\n",
       "       [1.09907509e-03, 5.29271005e-02, 8.23947488e-01, 1.09928554e-03,\n",
       "        1.09895796e-03, 1.09928650e-03, 1.09932241e-03, 1.02005968e-01,\n",
       "        1.09919643e-03, 1.45243196e-02],\n",
       "       [8.92929571e-04, 8.92916485e-04, 5.86757920e-01, 8.93033253e-04,\n",
       "        8.92889038e-04, 8.92955232e-04, 5.82827871e-02, 3.48708643e-01,\n",
       "        8.92947900e-04, 8.92978955e-04],\n",
       "       [2.49737177e-02, 8.40394684e-04, 8.77864410e-01, 9.12787598e-02,\n",
       "        8.40361663e-04, 8.40461005e-04, 8.40389795e-04, 8.40540520e-04,\n",
       "        8.40479144e-04, 8.40485879e-04]])"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "docs_classified = lda.transform(vectorized_docs)\n",
    "docs_classified[:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finalmente, para cada uno de los temas identifiados, se imprimen los top-5 papers pertenecientes al tema, ordenados por su *pagerank*.\n",
    "Finally, the top-5 PageRank sorted papers belonging to each topic are displayed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Topic ID #0\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** A human in vitro model system for investigating genome-wide host responses to SARS coronavirus infection\n",
       "- **Authors:** Ng, Lisa FP; Hibberd, Martin L; Ooi, Eng-Eong; Tang, Kin-Fai; Neo, Soek-Ying; Tan, Jenny; Krishna Murthy, Karuturi R; Vega, Vinsensius B; Chia, Jer-Ming; Liu, Edison T; Ren, Ee-Chee\n",
       "- **Publish date/time:** 2004-09-09\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: The molecular basis of severe acute respiratory syndrome (SARS) coronavirus (CoV) induced pathology is still largely unclear. Many SARS patients suffer respiratory distress brought on by interstitial infiltration and frequently show peripheral blood lymphopenia and occasional leucopenia. One possible cause of this could be interstitial inflammation, following a localized host response. In this study, we therefore examine the immune response of SARS-CoV in human peripheral blood mononuclear cells (PBMCs) over the first 24 hours. METHODS: PBMCs from normal healthy donors were inoculated in vitro with SARS-CoV and the viral replication kinetics was studied by real-time quantitative assays. SARS-CoV specific gene expression changes were examined by high-density oligonucleotide array analysis. RESULTS: We observed that SARS-CoV was capable of infecting and replicating in PBMCs and the kinetics of viral replication was variable among the donors. SARS-CoV antibody binding assays indicated that SARS specific antibodies inhibited SARS-CoV viral replication. Array data showed monocyte-macrophage cell activation, coagulation pathway upregulation and cytokine production together with lung trafficking chemokines such as IL8 and IL17, possibly activated through the TLR9 signaling pathway; that mimicked clinical features of the disease. CONCLUSIONS: The identification of human blood mononuclear cells as a direct target of SARS-CoV in the model system described here provides a new insight into disease pathology and a tool for investigating the host response and mechanisms of pathogenesis."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** The involvement of survival signaling pathways in rubella-virus induced apoptosis\n",
       "- **Authors:** Cooray, Samantha; Jin, Li; Best, Jennifer M\n",
       "- **Publish date/time:** 2005-01-04\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** Rubella virus (RV) causes severe congenital defects when acquired during the first trimester of pregnancy. RV cytopathic effect has been shown to be due to caspase-dependent apoptosis in a number of susceptible cell lines, and it has been suggested that this apoptotic induction could be a causal factor in the development of such defects. Often the outcome of apoptotic stimuli is dependent on apoptotic, proliferative and survival signaling mechanisms in the cell. Therefore we investigated the role of phosphoinositide 3-kinase (PI3K)-Akt survival signaling and Ras-Raf-MEK-ERK proliferative signaling during RV-induced apoptosis in RK13 cells. Increasing levels of phosphorylated ERK, Akt and GSK3β were detected from 24–96 hours post-infection, concomitant with RV-induced apoptotic signals. Inhibition of PI3K-Akt signaling reduced cell viability, and increased the speed and magnitude of RV-induced apoptosis, suggesting that this pathway contributes to cell survival during RV infection. In contrast, inhibition of the Ras-Raf-MEK-ERK pathway impaired RV replication and growth and reduced RV-induced apoptosis, suggesting that the normal cellular growth is required for efficient virus production."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Protein secretion in Lactococcus lactis : an efficient way to increase the overall heterologous protein production\n",
       "- **Authors:** Le Loir, Yves; Azevedo, Vasco; Oliveira, Sergio C; Freitas, Daniela A; Miyoshi, Anderson; Bermúdez-Humarán, Luis G; Nouaille, Sébastien; Ribeiro, Luciana A; Leclercq, Sophie; Gabriel, Jane E; Guimaraes, Valeria D; Oliveira, Maricê N; Charlier, Cathy; Gautier, Michel; Langella, Philippe\n",
       "- **Publish date/time:** 2005-01-04\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** Lactococcus lactis, the model lactic acid bacterium (LAB), is a food grade and well-characterized Gram positive bacterium. It is a good candidate for heterologous protein delivery in foodstuff or in the digestive tract. L. lactis can also be used as a protein producer in fermentor. Many heterologous proteins have already been produced in L. lactis but only few reports allow comparing production yields for a given protein either produced intracellularly or secreted in the medium. Here, we review several works evaluating the influence of the localization on the production yields of several heterologous proteins produced in L. lactis. The questions of size limits, conformation, and proteolysis are addressed and discussed with regard to protein yields. These data show that i) secretion is preferable to cytoplasmic production; ii) secretion enhancement (by signal peptide and propeptide optimization) results in increased production yield; iii) protein conformation rather than protein size can impair secretion and thus alter production yields; and iv) fusion of a stable protein can stabilize labile proteins. The role of intracellular proteolysis on heterologous cytoplasmic proteins and precursors is discussed. The new challenges now are the development of food grade systems and the identification and optimization of host factors affecting heterologous protein production not only in L. lactis, but also in other LAB species."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Expression profile of immune response genes in patients with Severe Acute Respiratory Syndrome\n",
       "- **Authors:** Reghunathan, Renji; Jayapal, Manikandan; Hsu, Li-Yang; Chng, Hiok-Hee; Tai, Dessmon; Leung, Bernard P; Melendez, Alirio J\n",
       "- **Publish date/time:** 2005-01-18\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Severe acute respiratory syndrome (SARS) emerged in later February 2003, as a new epidemic form of life-threatening infection caused by a novel coronavirus. However, the immune-pathogenesis of SARS is poorly understood. To understand the host response to this pathogen, we investigated the gene expression profiles of peripheral blood mononuclear cells (PBMCs) derived from SARS patients, and compared with healthy controls. RESULTS: The number of differentially expressed genes was found to be 186 under stringent filtering criteria of microarray data analysis. Several genes were highly up-regulated in patients with SARS, such as, the genes coding for Lactoferrin, S100A9 and Lipocalin 2. The real-time PCR method verified the results of the gene array analysis and showed that those genes that were up-regulated as determined by microarray analysis were also found to be comparatively up-regulated by real-time PCR analysis. CONCLUSIONS: This differential gene expression profiling of PBMCs from patients with SARS strongly suggests that the response of SARS affected patients seems to be mainly an innate inflammatory response, rather than a specific immune response against a viral infection, as we observed a complete lack of cytokine genes usually triggered during a viral infection. Our study shows for the first time how the immune system responds to the SARS infection, and opens new possibilities for designing new diagnostics and treatments for this new life-threatening disease."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Synergistic inhibition of human cytomegalovirus replication by interferon-alpha/beta and interferon-gamma\n",
       "- **Authors:** Sainz, Bruno; LaMarca, Heather L; Garry, Robert F; Morris, Cindy A\n",
       "- **Publish date/time:** 2005-02-23\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Recent studies have shown that gamma interferon (IFN-γ) synergizes with the innate IFNs (IFN-α and IFN-β) to inhibit herpes simplex virus type 1 (HSV-1) replication in vitro. To determine whether this phenomenon is shared by other herpesviruses, we investigated the effects of IFNs on human cytomegalovirus (HCMV) replication. RESULTS: We have found that as with HSV-1, IFN-γ synergizes with the innate IFNs (IFN-α/β) to potently inhibit HCMV replication in vitro. While pre-treatment of human foreskin fibroblasts (HFFs) with IFN-α, IFN-β or IFN-γ alone inhibited HCMV plaque formation by ~30 to 40-fold, treatment with IFN-α and IFN-γ or IFN-β and IFN-γ inhibited HCMV plaque formation by 163- and 662-fold, respectively. The generation of isobole plots verified that the observed inhibition of HCMV plaque formation and replication in HFFs by IFN-α/β and IFN-γ was a synergistic interaction. Additionally, real-time PCR analyses of the HCMV immediate early (IE) genes (IE1 and IE2) revealed that IE mRNA expression was profoundly decreased in cells stimulated with IFN-α/β and IFN-γ (~5-11-fold) as compared to vehicle-treated cells. Furthermore, decreased IE mRNA expression was accompanied by a decrease in IE protein expression, as demonstrated by western blotting and immunofluorescence. CONCLUSION: These findings suggest that IFN-α/β and IFN-γ synergistically inhibit HCMV replication through a mechanism that may involve the regulation of IE gene expression. We hypothesize that IFN-γ produced by activated cells of the adaptive immune response may potentially synergize with endogenous type I IFNs to inhibit HCMV dissemination in vivo."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Topic ID #1\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Dynamic changes of serum SARS-Coronavirus IgG, pulmonary function and radiography in patients recovering from SARS after hospital discharge\n",
       "- **Authors:** Xie, Lixin; Liu, Youning; Fan, Baoxing; Xiao, Yueyong; Tian, Qing; Chen, Liangan; Zhao, Hong; Chen, Weijun\n",
       "- **Publish date/time:** 2005-01-08\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** OBJECTIVE: The intent of this study was to examine the recovery of individuals who had been hospitalized for severe acute respiratory syndrome (SARS) in the year following their discharge from the hospital. Parameters studied included serum levels of SARS coronavirus (SARS-CoV) IgG antibody, tests of lung function, and imaging data to evaluate changes in lung fibrosis. In addition, we explored the incidence of femoral head necrosis in some of the individuals recovering from SARS. METHODS: The subjects of this study were 383 clinically diagnosed SARS patients in Beijing, China. They were tested regularly for serum levels of SARS-CoV IgG antibody and lung function and were given chest X-rays and/or high resolution computerized tomography (HRCT) examinations at the Chinese PLA General Hospital during the 12 months that followed their release from the hospital. Those individuals who were found to have lung diffusion abnormities (transfer coefficient for carbon monoxide [D(L)CO] < 80% of predicted value [pred]) received regular lung function tests and HRCT examinations in the follow-up phase in order to document the changes in their lung condition. Some patients who complained of joint pain were given magnetic resonance imaging (MRI) examinations of their femoral heads. FINDINGS: Of all the subjects, 81.2% (311 of 383 patients) tested positive for serum SARS-CoV IgG. Of those testing positive, 27.3% (85 of 311 patients) were suffering from lung diffusion abnormities (D(L)CO < 80% pred) and 21.5% (67 of 311 patients) exhibited lung fibrotic changes. In the 12 month duration of this study, all of the 40 patients with lung diffusion abnormities who were examined exhibited some improvement of lung function and fibrosis detected by radiography. Of the individuals receiving MRI examinations, 23.1% (18 of 78 patients) showed signs of femoral head necrosis. INTERPRETATION: The lack of sero-positive SARS-CoV in some individuals suggests that there may have been some misdiagnosed cases among the subjects included in this study. Of those testing positive, the serum levels of SARS-CoV IgG antibody decreased significantly during the 12 months after hospital discharge. Additionally, we found that the individuals who had lung fibrosis showed some spontaneous recovery. Finally, some of the subjects developed femoral head necrosis."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Absence of association between angiotensin converting enzyme polymorphism and development of adult respiratory distress syndrome in patients with severe acute respiratory syndrome: a case control study\n",
       "- **Authors:** Chan, KC Allen; Tang, Nelson LS; Hui, David SC; Chung, Grace TY; Wu, Alan KL; Chim, Stephen SC; Chiu, Rossa WK; Lee, Nelson; Choi, KW; Sung, YM; Chan, Paul KS; Tong, YK; Lai, ST; Yu, WC; Tsang, Owen; Lo, YM Dennis\n",
       "- **Publish date/time:** 2005-04-09\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: It has been postulated that genetic predisposition may influence the susceptibility to SARS-coronavirus infection and disease outcomes. A recent study has suggested that the deletion allele (D allele) of the angiotensin converting enzyme (ACE) gene is associated with hypoxemia in SARS patients. Moreover, the ACE D allele has been shown to be more prevalent in patients suffering from adult respiratory distress syndrome (ARDS) in a previous study. Thus, we have investigated the association between ACE insertion/deletion (I/D) polymorphism and the progression to ARDS or requirement of intensive care in SARS patients. METHOD: One hundred and forty genetically unrelated Chinese SARS patients and 326 healthy volunteers were recruited. The ACE I/D genotypes were determined by polymerase chain reaction and agarose gel electrophoresis. RESULTS: There is no significant difference in the genotypic distributions and the allelic frequencies of the ACE I/D polymorphism between the SARS patients and the healthy control subjects. Moreover, there is also no evidence that ACE I/D polymorphism is associated with the progression to ARDS or the requirement of intensive care in the SARS patients. In multivariate logistic analysis, age is the only factor associated with the development of ARDS while age and male sex are independent factors associated with the requirement of intensive care. CONCLUSION: The ACE I/D polymorphism is not directly related to increased susceptibility to SARS-coronavirus infection and is not associated with poor outcomes after SARS-coronavirus infection."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Persistence of lung inflammation and lung cytokines with high-resolution CT abnormalities during recovery from SARS\n",
       "- **Authors:** Wang, Chun-Hua; Liu, Chien-Ying; Wan, Yung-Liang; Chou, Chun-Liang; Huang, Kuo-Hsiung; Lin, Horng-Chyuan; Lin, Shu-Min; Lin, Tzou-Yien; Chung, Kian Fan; Kuo, Han-Pin\n",
       "- **Publish date/time:** 2005-05-11\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: During the acute phase of severe acute respiratory syndrome (SARS), mononuclear cells infiltration, alveolar cell desquamation and hyaline membrane formation have been described, together with dysregulation of plasma cytokine levels. Persistent high-resolution computed tomography (HRCT) abnormalities occur in SARS patients up to 40 days after recovery. METHODS: To determine further the time course of recovery of lung inflammation, we investigated the HRCT and inflammatory profiles, and coronavirus persistence in bronchoalveolar lavage fluid (BALF) of 12 patients at recovery at 60 and 90 days. RESULTS: At 60 days, compared to normal controls, SARS patients had increased cellularity of BALF with increased alveolar macrophages (AM) and CD8 cells. HRCT scores were increased and correlated with T-cell numbers and their subpopulations, and inversely with CD4/CD8 ratio. TNF-α, IL-6, IL-8, RANTES and MCP-1 levels were increased. Viral particles in AM were detected by electron microscopy in 7 of 12 SARS patients with high HRCT score. On day 90, HRCT scores improved significantly in 10 of 12 patients, with normalization of BALF cell counts in 6 of 12 patients with repeat bronchoscopy. Pulse steroid therapy and prolonged fever were two independent factors associated with delayed resolution of pneumonitis, in this non-randomized, retrospective analysis. CONCLUSION: Resolution of pneumonitis is delayed in some patients during SARS recovery and may be associated with delayed clearance of coronavirus, Complete resolution may occur by 90 days or later."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** The effects of injection of bovine vaccine into a human digit: a case report\n",
       "- **Authors:** O'Neill, Jennifer K; Richards, Simon W; Ricketts, David M; Patterson, Marc H\n",
       "- **Publish date/time:** 2005-10-11\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: The incidence of needlestick injuries in farmers and veterinary surgeons is significant and the consequences of such an injection can be serious. CASE PRESENTATION: We report accidental injection of bovine vaccine into the base of the little finger. This resulted in increased pressure in the flexor sheath causing signs and symptoms of ischemia. Amputation of the digit was required despite repeated surgical debridement and decompression. CONCLUSION: There have been previous reports of injection of oil-based vaccines into the human hand resulting in granulomatous inflammation or sterile abscess and causing morbidity and tissue loss. Self-injection with veterinary vaccines is an occupational hazard for farmers and veterinary surgeons. Injection of vaccine into a closed compartment such as the human finger can have serious sequelae including loss of the injected digit. These injuries are not to be underestimated. Early debridement and irrigation of the injected area with decompression is likely to give the best outcome. Frequent review is necessary after the first procedure because repeat operations may be required."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Pneumothorax and mortality in the mechanically ventilated SARS patients: a prospective clinical study\n",
       "- **Authors:** Kao, Hsin-Kuo; Wang, Jia-Horng; Sung, Chun-Sung; Huang, Ying-Che; Lien, Te-Cheng\n",
       "- **Publish date/time:** 2005-06-22\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** INTRODUCTION: Pneumothorax often complicates the management of mechanically ventilated severe acute respiratory syndrome (SARS) patients in the isolation intensive care unit (ICU). We sought to determine whether pneumothoraces are induced by high ventilatory pressure or volume and if they are associated with mortality in mechanically ventilated SARS patients. METHODS: We conducted a prospective, clinical study. Forty-one mechanically ventilated SARS patients were included in our study. All SARS patients were sedated and received mechanical ventilation in the isolation ICU. RESULTS: The mechanically ventilated SARS patients were divided into two groups either with or without pneumothorax. Their demographic data, clinical characteristics, ventilatory variables such as positive end-expiratory pressure, peak inspiratory pressure, mean airway pressure, tidal volume, tidal volume per kilogram, respiratory rate and minute ventilation and the accumulated mortality rate at 30 days after mechanical ventilation were analyzed. There were no statistically significant differences in the pressures and volumes between the two groups, and the mortality was also similar between the groups. However, patients developing pneumothorax during mechanical ventilation frequently expressed higher respiratory rates on admission, and a lower PaO(2)/FiO(2 )ratio and higher PaCO(2 )level during hospitalization compared with those without pneumothorax. CONCLUSION: In our study, the SARS patients who suffered pneumothorax presented as more tachypnic on admission, and more pronounced hypoxemic and hypercapnic during hospitalization. These variables signaled a deterioration in respiratory function and could be indicators of developing pneumothorax during mechanical ventilation in the SARS patients. Meanwhile, meticulous respiratory therapy and monitoring were mandatory in these patients."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Topic ID #2\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Sequence requirements for RNA strand transfer during nidovirus discontinuous subgenomic RNA synthesis\n",
       "- **Authors:** Pasternak, Alexander O.; van den Born, Erwin; Spaan, Willy J.M.; Snijder, Eric J.\n",
       "- **Publish date/time:** 2001-12-17\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** Nidovirus subgenomic mRNAs contain a leader sequence derived from the 5′ end of the genome fused to different sequences (‘bodies’) derived from the 3′ end. Their generation involves a unique mechanism of discontinuous subgenomic RNA synthesis that resembles copy-choice RNA recombination. During this process, the nascent RNA strand is transferred from one site in the template to another, during either plus or minus strand synthesis, to yield subgenomic RNA molecules. Central to this process are transcription-regulating sequences (TRSs), which are present at both template sites and ensure the fidelity of strand transfer. Here we present results of a comprehensive co-variation mutagenesis study of equine arteritis virus TRSs, demonstrating that discontinuous RNA synthesis depends not only on base pairing between sense leader TRS and antisense body TRS, but also on the primary sequence of the body TRS. While the leader TRS merely plays a targeting role for strand transfer, the body TRS fulfils multiple functions. The sequences of mRNA leader–body junctions of TRS mutants strongly suggested that the discontinuous step occurs during minus strand synthesis."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Crystal structure of murine sCEACAM1a[1,4]: a coronavirus receptor in the CEA family\n",
       "- **Authors:** Tan, Kemin; Zelus, Bruce D.; Meijers, Rob; Liu, Jin-huan; Bergelson, Jeffrey M.; Duke, Norma; Zhang, Rongguang; Joachimiak, Andrzej; Holmes, Kathryn V.; Wang, Jia-huai\n",
       "- **Publish date/time:** 2002-05-01\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** CEACAM1 is a member of the carcinoembryonic antigen (CEA) family. Isoforms of murine CEACAM1 serve as receptors for mouse hepatitis virus (MHV), a murine coronavirus. Here we report the crystal structure of soluble murine sCEACAM1a[1,4], which is composed of two Ig-like domains and has MHV neutralizing activity. Its N-terminal domain has a uniquely folded CC′ loop that encompasses key virus-binding residues. This is the first atomic structure of any member of the CEA family, and provides a prototypic architecture for functional exploration of CEA family members. We discuss the structural basis of virus receptor activities of murine CEACAM1 proteins, binding of Neisseria to human CEACAM1, and other homophilic and heterophilic interactions of CEA family members."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Synthesis of a novel hepatitis C virus protein by ribosomal frameshift\n",
       "- **Authors:** Xu, Zhenming; Choi, Jinah; Yen, T.S.Benedict; Lu, Wen; Strohecker, Anne; Govindarajan, Sugantha; Chien, David; Selby, Mark J.; Ou, Jing‐hsiung\n",
       "- **Publish date/time:** 2001-07-16\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** Hepatitis C virus (HCV) is an important human pathogen that affects ∼100 million people worldwide. Its RNA genome codes for a polyprotein, which is cleaved by viral and cellular proteases to produce at least 10 mature viral protein products. We report here the discovery of a novel HCV protein synthesized by ribosomal frameshift. This protein, which we named the F protein, is synthesized from the initiation codon of the polyprotein sequence followed by ribosomal frameshift into the −2/+1 reading frame. This ribosomal frameshift requires only codons 8–14 of the core protein‐coding sequence, and the shift junction is located at or near codon 11. An F protein analog synthesized in vitro reacted with the sera of HCV patients but not with the sera of hepatitis B patients, indicating the expression of the F protein during natural HCV infection. This unexpected finding may open new avenues for the development of anti‐HCV drugs."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Structure of coronavirus main proteinase reveals combination of a chymotrypsin fold with an extra α-helical domain\n",
       "- **Authors:** Anand, Kanchan; Palm, Gottfried J.; Mesters, Jeroen R.; Siddell, Stuart G.; Ziebuhr, John; Hilgenfeld, Rolf\n",
       "- **Publish date/time:** 2002-07-01\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** The key enzyme in coronavirus polyprotein processing is the viral main proteinase, M(pro), a protein with extremely low sequence similarity to other viral and cellular proteinases. Here, the crystal structure of the 33.1 kDa transmissible gastroenteritis (corona)virus M(pro) is reported. The structure was refined to 1.96 Å resolution and revealed three dimers in the asymmetric unit. The mutual arrangement of the protomers in each of the dimers suggests that M(pro) self-processing occurs in trans. The active site, comprised of Cys144 and His41, is part of a chymotrypsin-like fold that is connected by a 16 residue loop to an extra domain featuring a novel α-helical fold. Molecular modelling and mutagenesis data implicate the loop in substrate binding and elucidate S1 and S2 subsites suitable to accommodate the side chains of the P1 glutamine and P2 leucine residues of M(pro) substrates. Interactions involving the N-terminus and the α-helical domain stabilize the loop in the orientation required for trans-cleavage activity. The study illustrates that RNA viruses have evolved unprecedented variations of the classical chymotrypsin fold."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Discontinuous and non-discontinuous subgenomic RNA transcription in a nidovirus\n",
       "- **Authors:** van Vliet, A.L.W.; Smits, S.L.; Rottier, P.J.M.; de Groot, R.J.\n",
       "- **Publish date/time:** 2002-12-01\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** Arteri-, corona-, toro- and roniviruses are evolutionarily related positive-strand RNA viruses, united in the order Nidovirales. The best studied nidoviruses, the corona- and arteriviruses, employ a unique transcription mechanism, which involves discontinuous RNA synthesis, a process resembling similarity-assisted copy-choice RNA recombination. During infection, multiple subgenomic (sg) mRNAs are transcribed from a mirror set of sg negative-strand RNA templates. The sg mRNAs all possess a short 5′ common leader sequence, derived from the 5′ end of the genomic RNA. The joining of the non-contiguous ‘leader’ and ‘body’ sequences presumably occurs during minus-strand synthesis. To study whether toroviruses use a similar transcription mechanism, we characterized the 5′ termini of the genome and the four sg mRNAs of Berne virus (BEV). We show that BEV mRNAs 3–5 lack a leader sequence. Surprisingly, however, RNA 2 does contain a leader, identical to the 5′-terminal 18 residues of the genome. Apparently, BEV combines discontinuous and non-discontinous RNA synthesis to produce its sg mRNAs. Our findings have important implications for the understanding of the mechanism and evolution of nidovirus transcription."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Topic ID #3\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Relationship of SARS-CoV to other pathogenic RNA viruses explored by tetranucleotide usage profiling\n",
       "- **Authors:** Yap, Yee Leng; Zhang, Xue Wu; Danchin, Antoine\n",
       "- **Publish date/time:** 2003-09-20\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: The exact origin of the cause of the Severe Acute Respiratory Syndrome (SARS) is still an open question. The genomic sequence relationship of SARS-CoV with 30 different single-stranded RNA (ssRNA) viruses of various families was studied using two non-standard approaches. Both approaches began with the vectorial profiling of the tetra-nucleotide usage pattern V for each virus. In approach one, a distance measure of a vector V, based on correlation coefficient was devised to construct a relationship tree by the neighbor-joining algorithm. In approach two, a multivariate factor analysis was performed to derive the embedded tetra-nucleotide usage patterns. These patterns were subsequently used to classify the selected viruses. RESULTS: Both approaches yielded relationship outcomes that are consistent with the known virus classification. They also indicated that the genome of RNA viruses from the same family conform to a specific pattern of word usage. Based on the correlation of the overall tetra-nucleotide usage patterns, the Transmissible Gastroenteritis Virus (TGV) and the Feline CoronaVirus (FCoV) are closest to SARS-CoV. Surprisingly also, the RNA viruses that do not go through a DNA stage displayed a remarkable discrimination against the CpG and UpA di-nucleotide (z = -77.31, -52.48 respectively) and selection for UpG and CpA (z = 65.79,49.99 respectively). Potential factors influencing these biases are discussed. CONCLUSION: The study of genomic word usage is a powerful method to classify RNA viruses. The congruence of the relationship outcomes with the known classification indicates that there exist phylogenetic signals in the tetra-nucleotide usage patterns, that is most prominent in the replicase open reading frames."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Bioinformatics analysis of SARS coronavirus genome polymorphism\n",
       "- **Authors:** Pavlović-Lažetić, Gordana M; Mitić, Nenad S; Beljanski, Miloš V\n",
       "- **Publish date/time:** 2004-05-25\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: We have compared 38 isolates of the SARS-CoV complete genome. The main goal was twofold: first, to analyze and compare nucleotide sequences and to identify positions of single nucleotide polymorphism (SNP), insertions and deletions, and second, to group them according to sequence similarity, eventually pointing to phylogeny of SARS-CoV isolates. The comparison is based on genome polymorphism such as insertions or deletions and the number and positions of SNPs. RESULTS: The nucleotide structure of all 38 isolates is presented. Based on insertions and deletions and dissimilarity due to SNPs, the dataset of all the isolates has been qualitatively classified into three groups each having their own subgroups. These are the A-group with \"regular\" isolates (no insertions / deletions except for 5' and 3' ends), the B-group of isolates with \"long insertions\", and the C-group of isolates with \"many individual\" insertions and deletions. The isolate with the smallest average number of SNPs, compared to other isolates, has been identified (TWH). The density distribution of SNPs, insertions and deletions for each group or subgroup, as well as cumulatively for all the isolates is also presented, along with the gene map for TWH. Since individual SNPs may have occurred at random, positions corresponding to multiple SNPs (occurring in two or more isolates) are identified and presented. This result revises some previous results of a similar type. Amino acid changes caused by multiple SNPs are also identified (for the annotated sequences, as well as presupposed amino acid changes for non-annotated ones). Exact SNP positions for the isolates in each group or subgroup are presented. Finally, a phylogenetic tree for the SARS-CoV isolates has been produced using the CLUSTALW program, showing high compatibility with former qualitative classification. CONCLUSIONS: The comparative study of SARS-CoV isolates provides essential information for genome polymorphism, indication of strain differences and variants evolution. It may help with the development of effective treatment."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Mutational dynamics of the SARS coronavirus in cell culture and human populations isolated in 2003\n",
       "- **Authors:** Vega, Vinsensius B; Ruan, Yijun; Liu, Jianjun; Lee, Wah Heng; Wei, Chia Lin; Se-Thoe, Su Yun; Tang, Kin Fai; Zhang, Tao; Kolatkar, Prasanna R; Ooi, Eng Eong; Ling, Ai Ee; Stanton, Lawrence W; Long, Philip M; Liu, Edison T\n",
       "- **Publish date/time:** 2004-09-06\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: The SARS coronavirus is the etiologic agent for the epidemic of the Severe Acute Respiratory Syndrome. The recent emergence of this new pathogen, the careful tracing of its transmission patterns, and the ability to propagate in culture allows the exploration of the mutational dynamics of the SARS-CoV in human populations. METHODS: We sequenced complete SARS-CoV genomes taken from primary human tissues (SIN3408, SIN3725V, SIN3765V), cultured isolates (SIN848, SIN846, SIN842, SIN845, SIN847, SIN849, SIN850, SIN852, SIN3408L), and five consecutive Vero cell passages (SIN2774_P1, SIN2774_P2, SIN2774_P3, SIN2774_P4, SIN2774_P5) arising from SIN2774 isolate. These represented individual patient samples, serial in vitro passages in cell culture, and paired human and cell culture isolates. Employing a refined mutation filtering scheme and constant mutation rate model, the mutation rates were estimated and the possible date of emergence was calculated. Phylogenetic analysis was used to uncover molecular relationships between the isolates. RESULTS: Close examination of whole genome sequence of 54 SARS-CoV isolates identified before 14(th )October 2003, including 22 from patients in Singapore, revealed the mutations engendered during human-to-Vero and Vero-to-human transmission as well as in multiple Vero cell passages in order to refine our analysis of human-to-human transmission. Though co-infection by different quasipecies in individual tissue samples is observed, the in vitro mutation rate of the SARS-CoV in Vero cell passage is negligible. The in vivo mutation rate, however, is consistent with estimates of other RNA viruses at approximately 5.7 × 10(-6 )nucleotide substitutions per site per day (0.17 mutations per genome per day), or two mutations per human passage (adjusted R-square = 0.4014). Using the immediate Hotel M contact isolates as roots, we observed that the SARS epidemic has generated four major genetic groups that are geographically associated: two Singapore isolates, one Taiwan isolate, and one North China isolate which appears most closely related to the putative SARS-CoV isolated from a palm civet. Non-synonymous mutations are centered in non-essential ORFs especially in structural and antigenic genes such as the S and M proteins, but these mutations did not distinguish the geographical groupings. However, no non-synonymous mutations were found in the 3CLpro and the polymerase genes. CONCLUSIONS: Our results show that the SARS-CoV is well adapted to growth in culture and did not appear to undergo specific selection in human populations. We further assessed that the putative origin of the SARS epidemic was in late October 2002 which is consistent with a recent estimate using cases from China. The greater sequence divergence in the structural and antigenic proteins and consistent deletions in the 3' – most portion of the viral genome suggest that certain selection pressures are interacting with the functional nature of these validated and putative ORFs."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** SARS Transmission Pattern in Singapore Reassessed by Viral Sequence Variation Analysis\n",
       "- **Authors:** Liu, Jianjun; Lim, Siew Lan; Ruan, Yijun; Ling, Ai Ee; Ng, Lisa F. P; Drosten, Christian; Liu, Edison T; Stanton, Lawrence W; Hibberd, Martin L\n",
       "- **Publish date/time:** 2005-02-22\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Epidemiological investigations of infectious disease are mainly dependent on indirect contact information and only occasionally assisted by characterization of pathogen sequence variation from clinical isolates. Direct sequence analysis of the pathogen, particularly at a population level, is generally thought to be too cumbersome, technically difficult, and expensive. We present here a novel application of mass spectrometry (MS)–based technology in characterizing viral sequence variations that overcomes these problems, and we apply it retrospectively to the severe acute respiratory syndrome (SARS) outbreak in Singapore. METHODS AND FINDINGS: The success rate of the MS-based analysis for detecting SARS coronavirus (SARS-CoV) sequence variations was determined to be 95% with 75 copies of viral RNA per reaction, which is sufficient to directly analyze both clinical and cultured samples. Analysis of 13 SARS-CoV isolates from the different stages of the Singapore outbreak identified nine sequence variations that could define the molecular relationship between them and pointed to a new, previously unidentified, primary route of introduction of SARS-CoV into the Singapore population. Our direct determination of viral sequence variation from a clinical sample also clarified an unresolved epidemiological link regarding the acquisition of SARS in a German patient. We were also able to detect heterogeneous viral sequences in primary lung tissues, suggesting a possible coevolution of quasispecies of virus within a single host. CONCLUSION: This study has further demonstrated the importance of improving clinical and epidemiological studies of pathogen transmission through the use of genetic analysis and has revealed the MS-based analysis to be a sensitive and accurate method for characterizing SARS-CoV genetic variations in clinical samples. We suggest that this approach should be used routinely during outbreaks of a wide variety of agents, in order to allow the most effective control."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection\n",
       "- **Authors:** Froissart, Remy; Roze, Denis; Uzest, Marilyne; Galibert, Lionel; Blanc, Stephane; Michalakis, Yannis\n",
       "- **Publish date/time:** 2005-03-01\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** Viral recombination can dramatically impact evolution and epidemiology. In viruses, the recombination rate depends on the frequency of genetic exchange between different viral genomes within an infected host cell and on the frequency at which such co-infections occur. While the recombination rate has been recently evaluated in experimentally co-infected cell cultures for several viruses, direct quantification at the most biologically significant level, that of a host infection, is still lacking. This study fills this gap using the cauliflower mosaic virus as a model. We distributed four neutral markers along the viral genome, and co-inoculated host plants with marker-containing and wild-type viruses. The frequency of recombinant genomes was evaluated 21 d post-inoculation. On average, over 50% of viral genomes recovered after a single host infection were recombinants, clearly indicating that recombination is very frequent in this virus. Estimates of the recombination rate show that all regions of the genome are equally affected by this process. Assuming that ten viral replication cycles occurred during our experiment—based on data on the timing of coat protein detection—the per base and replication cycle recombination rate was on the order of 2 × 10(−5) to 4 × 10(−5). This first determination of a virus recombination rate during a single multi-cellular host infection indicates that recombination is very frequent in the everyday life of this virus."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Topic ID #4\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** SARS, Mars and chocolate bars\n",
       "- **Authors:** Gannon, Frank\n",
       "- **Publish date/time:** 2005-01-01\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** nan"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** What does the peripheral blood tell you in SARS?\n",
       "- **Authors:** OPENSHAW, P J M\n",
       "- **Publish date/time:** 2004-04-01\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** nan"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Ménage à trois of bacterial and viral pulmonary pathogens delivers coup de grace to the lung\n",
       "- **Authors:** HUSSELL, T; WILLIAMS, A\n",
       "- **Publish date/time:** 2004-06-15\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** nan"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** The Cuyahoga Is Still Burning\n",
       "- **Authors:** Silbergeld, Ellen K.; Graham, Jay P.\n",
       "- **Publish date/time:** 2008-04-23\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** nan"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Dengue and Relative Bradycardia\n",
       "- **Authors:** Lateef, Aisha; Fisher, Dale Andrew; Tambyah, Paul Ananth\n",
       "- **Publish date/time:** 2007-04-23\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** nan"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Topic ID #5\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Detection of hepatitis C virus in the nasal secretions of an intranasal drug-user\n",
       "- **Authors:** McMahon, James M; Simm, Malgorzata; Milano, Danielle; Clatts, Michael\n",
       "- **Publish date/time:** 2004-05-07\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: One controversial source of infection for hepatitis C virus (HCV) involves the sharing of contaminated implements, such as straws or spoons, used to nasally inhale cocaine and other powdered drugs. An essential precondition for this mode of transmission is the presence of HCV in the nasal secretions of intranasal drug users. METHODS: Blood and nasal secretion samples were collected from five plasma-positive chronic intranasal drug users and tested for HCV RNA using RT-PCR. RESULTS: HCV was detected in all five blood samples and in the nasal secretions of the subject with the highest serum viral load. CONCLUSIONS: This study is the first to demonstrate the presence of HCV in nasal secretions. This finding has implications for potential transmission of HCV through contact with contaminated nasal secretions."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** A touchdown nucleic acid amplification protocol as an alternative to culture backup for immunofluorescence in the routine diagnosis of acute viral respiratory tract infections\n",
       "- **Authors:** Coyle, Peter V; Ong, Grace M; O'Neill, Hugh J; McCaughey, Conall; De Ornellas, Dennis; Mitchell, Frederick; Mitchell, Suzanne J; Feeney, Susan A; Wyatt, Dorothy E; Forde, Marian; Stockton, Joanne\n",
       "- **Publish date/time:** 2004-10-25\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Immunofluorescence and virus culture are the main methods used to diagnose acute respiratory virus infections. Diagnosing these infections using nucleic acid amplification presents technical challenges, one of which is facilitating the different optimal annealing temperatures needed for each virus. To overcome this problem we developed a diagnostic molecular strip which combined a generic nested touchdown protocol with in-house primer master-mixes that could recognise 12 common respiratory viruses. RESULTS: Over an 18 month period a total of 222 specimens were tested by both immunofluorescence and the molecular strip. The specimens came from 103 males (median age 3.5 y), 80 females (median age 9 y) and 5 quality assurance scheme specimens. Viruses were recovered from a number of specimen types including broncho-alveolar lavage, nasopharyngeal secretions, sputa, post-mortem lung tissue and combined throat and nasal swabs. Viral detection by IF was poor in sputa and respiratory swabs. A total of 99 viruses were detected in the study from 79 patients and 4 quality control specimens: 31 by immunofluorescence and 99 using the molecular strip. The strip consistently out-performed immunofluorescence with no loss of diagnostic specificity. CONCLUSIONS: The touchdown protocol with pre-dispensed primer master-mixes was suitable for replacing virus culture for the diagnosis of respiratory viruses which were negative by immunofluorescence. Results by immunofluorescence were available after an average of 4–12 hours while molecular strip results were available within 24 hours, considerably faster than viral culture. The combined strip and touchdown protocol proved to be a convenient and reliable method of testing for multiple viruses in a routine setting."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** A novel pancoronavirus RT-PCR assay: frequent detection of human coronavirus NL63 in children hospitalized with respiratory tract infections in Belgium\n",
       "- **Authors:** Moës, Elien; Vijgen, Leen; Keyaerts, Els; Zlateva, Kalina; Li, Sandra; Maes, Piet; Pyrc, Krzysztof; Berkhout, Ben; van der Hoek, Lia; Van Ranst, Marc\n",
       "- **Publish date/time:** 2005-02-01\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Four human coronaviruses are currently known to infect the respiratory tract: human coronaviruses OC43 (HCoV-OC43) and 229E (HCoV-229E), SARS associated coronavirus (SARS-CoV) and the recently identified human coronavirus NL63 (HCoV-NL63). In this study we explored the incidence of HCoV-NL63 infection in children diagnosed with respiratory tract infections in Belgium. METHODS: Samples from children hospitalized with respiratory diseases during the winter seasons of 2003 and 2004 were evaluated for the presence of HCoV-NL63 using a optimized pancoronavirus RT-PCR assay. RESULTS: Seven HCoV-NL63 positive samples were identified, six were collected during January/February 2003 and one at the end of February 2004. CONCLUSIONS: Our results support the notation that HCoV-NL63 can cause serious respiratory symptoms in children. Sequence analysis of the S gene showed that our isolates could be classified into two subtypes corresponding to the two prototype HCoV-NL63 sequences isolated in The Netherlands in 1988 and 2003, indicating that these two subtypes may currently be cocirculating."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Croup Is Associated with the Novel Coronavirus NL63\n",
       "- **Authors:** van der Hoek, Lia; Sure, Klaus; Ihorst, Gabriele; Stang, Alexander; Pyrc, Krzysztof; Jebbink, Maarten F; Petersen, Gudula; Forster, Johannes; Berkhout, Ben; Überla, Klaus\n",
       "- **Publish date/time:** 2005-08-23\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: The clinical relevance of infections with the novel human coronavirus NL63 (HCoV-NL63) has not been investigated systematically. We therefore determined its association with disease in young children with lower respiratory tract infection (LRTI). METHODS AND FINDINGS: Nine hundred forty-nine samples of nasopharyngeal secretions from children under 3 y of age with LRTIs were analysed by a quantitative HCoV-NL63-specific real-time PCR. The samples had been collected from hospitalised patients and outpatients from December 1999 to October 2001 in four different regions in Germany as part of the prospective population-based PRI.DE study and analysed for RNA from respiratory viruses. Forty-nine samples (5.2%), mainly derived from the winter season, were positive for HCoV-NL63 RNA. The viral RNA was more prevalent in samples from outpatients (7.9%) than from hospitalised patients (3.2%, p = 0.003), and co-infection with either respiratory syncytial virus or parainfluenza virus 3 was observed frequently. Samples in which only HCoV-NL63 RNA could be detected had a significantly higher viral load than samples containing additional respiratory viruses (median 2.1 × 10(6) versus 2.7 × 10(2) copies/ml, p = 0.0006). A strong association with croup was apparent: 43% of the HCoV-NL63-positive patients with high HCoV-NL63 load and absence of co-infection suffered from croup, compared to 6% in the HCoV-NL63-negative group, p < 0.0001. A significantly higher fraction (17.4%) of samples from croup patients than from non-croup patients (4.2%) contained HCoV-NL63 RNA. CONCLUSION: HCoV-NL63 infections occur frequently in young children with LRTI and show a strong association with croup, suggesting a causal relationship."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Time Course and Cellular Localization of SARS-CoV Nucleoprotein and RNA in Lungs from Fatal Cases of SARS\n",
       "- **Authors:** Nicholls, John M; Butany, Jagdish; Poon, Leo L. M; Chan, Kwok H; Beh, Swan Lip; Poutanen, Susan; Peiris, J. S. Malik; Wong, Maria\n",
       "- **Publish date/time:** 2006-01-03\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Cellular localization of severe acute respiratory syndrome coronavirus (SARS-CoV) in the lungs of patients with SARS is important in confirming the etiological association of the virus with disease as well as in understanding the pathogenesis of the disease. To our knowledge, there have been no comprehensive studies investigating viral infection at the cellular level in humans. METHODS AND FINDINGS: We collected the largest series of fatal cases of SARS with autopsy material to date by merging the pathological material from two regions involved in the 2003 worldwide SARS outbreak in Hong Kong, China, and Toronto, Canada. We developed a monoclonal antibody against the SARS-CoV nucleoprotein and used it together with in situ hybridization (ISH) to analyze the autopsy lung tissues of 32 patients with SARS from Hong Kong and Toronto. We compared the results of these assays with the pulmonary pathologies and the clinical course of illness for each patient. SARS-CoV nucleoprotein and RNA were detected by immunohistochemistry and ISH, respectively, primarily in alveolar pneumocytes and, less frequently, in macrophages. Such localization was detected in four of the seven patients who died within two weeks of illness onset, and in none of the 25 patients who died later than two weeks after symptom onset. CONCLUSIONS: The pulmonary alveolar epithelium is the chief target of SARS-CoV, with macrophages infected subsequently. Viral replication appears to be limited to the first two weeks after symptom onset, with little evidence of continued widespread replication after this period. If antiviral therapy is considered for future treatment, it should be focused on this two-week period of acute clinical disease."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Topic ID #6\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Airborne rhinovirus detection and effect of ultraviolet irradiation on detection by a semi-nested RT-PCR assay\n",
       "- **Authors:** Myatt, Theodore A; Johnston, Sebastian L; Rudnick, Stephen; Milton, Donald K\n",
       "- **Publish date/time:** 2003-01-13\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Rhinovirus, the most common cause of upper respiratory tract infections, has been implicated in asthma exacerbations and possibly asthma deaths. Although the method of transmission of rhinoviruses is disputed, several studies have demonstrated that aerosol transmission is a likely method of transmission among adults. As a first step in studies of possible airborne rhinovirus transmission, we developed methods to detect aerosolized rhinovirus by extending existing technology for detecting infectious agents in nasal specimens. METHODS: We aerosolized rhinovirus in a small aerosol chamber. Experiments were conducted with decreasing concentrations of rhinovirus. To determine the effect of UV irradiation on detection of rhinoviral aerosols, we also conducted experiments in which we exposed aerosols to a UV dose of 684 mJ/m(2). Aerosols were collected on Teflon filters and rhinovirus recovered in Qiagen AVL buffer using the Qiagen QIAamp Viral RNA Kit (Qiagen Corp., Valencia, California) followed by semi-nested RT-PCR and detection by gel electrophoresis. RESULTS: We obtained positive results from filter samples that had collected at least 1.3 TCID(50 )of aerosolized rhinovirus. Ultraviolet irradiation of airborne virus at doses much greater than those used in upper-room UV germicidal irradiation applications did not inhibit subsequent detection with the RT-PCR assay. CONCLUSION: The air sampling and extraction methodology developed in this study should be applicable to the detection of rhinovirus and other airborne viruses in the indoor air of offices and schools. This method, however, cannot distinguish UV inactivated virus from infectious viral particles."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** A systematic review of the effectiveness of antimicrobial rinse-free hand sanitizers for prevention of illness-related absenteeism in elementary school children\n",
       "- **Authors:** Meadows, Emily; Le Saux, Nicole\n",
       "- **Publish date/time:** 2004-11-01\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Absenteeism due to communicable illness is a major problem encountered by North American elementary school children. Although handwashing is a proven infection control measure, barriers exist in the school environment, which hinder compliance to this routine. Currently, alternative hand hygiene techniques are being considered, and one such technique is the use of antimicrobial rinse-free hand sanitizers. METHODS: A systematic review was conducted to examine the effectiveness of antimicrobial rinse-free hand sanitizer interventions in the elementary school setting. MEDLINE, EMBASE, Biological Abstract, CINAHL, HealthSTAR and Cochrane Controlled Trials Register were searched for both randomized and non-randomized controlled trials. Absenteeism due to communicable illness was the primary outcome variable. RESULTS: Six eligible studies, two of which were randomized, were identified (5 published studies, 1 published abstract). The quality of reporting was low. Due to a large amount of heterogeneity and low quality of reporting, no pooled estimates were calculated. There was a significant difference reported in favor of the intervention in all 5 published studies. CONCLUSIONS: The available evidence for the effectiveness of antimicrobial rinse-free hand sanitizer in the school environment is of low quality. The results suggest that the strength of the benefit should be interpreted with caution. Given the potential to reduce student absenteeism, teacher absenteeism, school operating costs, healthcare costs and parental absenteeism, a well-designed and analyzed trial is needed to optimize this hand hygiene technique."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Accuracy of parents in measuring body temperature with a tympanic thermometer\n",
       "- **Authors:** Robinson, Joan L; Jou, Hsing; Spady, Donald W\n",
       "- **Publish date/time:** 2005-01-11\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: It is now common for parents to measure tympanic temperatures in children. The objective of this study was to assess the diagnostic accuracy of these measurements. METHODS: Parents and then nurses measured the temperature of 60 children with a tympanic thermometer designed for home use (home thermometer). The reference standard was a temperature measured by a nurse with a model of tympanic thermometer commonly used in hospitals (hospital thermometer). A difference of ≥ 0.5 °C was considered clinically significant. A fever was defined as a temperature ≥ 38.5 °C. RESULTS: The mean absolute difference between the readings done by the parent and the nurse with the home thermometer was 0.44 ± 0.61 °C, and 33% of the readings differed by ≥ 0.5 °C. The mean absolute difference between the readings done by the parent with the home thermometer and the nurse with the hospital thermometer was 0.51 ± 0.63 °C, and 72 % of the readings differed by ≥ 0.5 °C. Using the home thermometer, parents detected fever with a sensitivity of 76% (95% CI 50–93%), a specificity of 95% (95% CI 84–99%), a positive predictive value of 87% (95% CI 60–98%), and a negative predictive value of 91% (95% CI 79–98 %). In comparing the readings the nurse obtained from the two different tympanic thermometers, the mean absolute difference was 0.24 ± 0.22 °C. Nurses detected fever with a sensitivity of 94% (95 % CI 71–100 %), a specificity of 88% (95% CI 75–96 %), a positive predictive value of 76% (95% CI 53–92%), and a negative predictive value of 97% (95%CI 87–100 %) using the home thermometer. The intraclass correlation coefficient for the three sets of readings was 0.80, and the consistency of readings was not affected by the body temperature. CONCLUSIONS: The readings done by parents with a tympanic thermometer designed for home use differed a clinically significant amount from the reference standard (readings done by nurses with a model of tympanic thermometer commonly used in hospitals) the majority of the time, and parents failed to detect fever about one-quarter of the time. Tympanic readings reported by parents should be interpreted with great caution."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** A new paradigm in respiratory hygiene: increasing the cohesivity of airway secretions to improve cough interaction and reduce aerosol dispersion\n",
       "- **Authors:** Zayas, Gustavo; Dimitry, John; Zayas, Ana; O'Brien, Darryl; King, Malcolm\n",
       "- **Publish date/time:** 2005-09-02\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Infectious respiratory diseases are transmitted to non-infected subjects when an infected person expels pathogenic microorganisms to the surrounding environment when coughing or sneezing. When the airway mucus layer interacts with high-speed airflow, droplets are expelled as aerosol; their concentration and size distribution may each play an important role in disease transmission. Our goal is to reduce the aerosolizability of respiratory secretions while interfering only minimally with normal mucus clearance using agents capable of increasing crosslinking in the mucin glycoprotein network. METHODS: We exposed mucus simulants (MS) to airflow in a simulated cough machine (SCM). The MS ranged from non-viscous, non-elastic substances (water) to MS of varying degrees of viscosity and elasticity. Mucociliary clearance of the MS was assessed on the frog palate, elasticity in the Filancemeter and the aerosol pattern in a \"bulls-eye\" target. The sample loaded was weighed before and after each cough maneuver. We tested two mucomodulators: sodium tetraborate (XL\"B\") and calcium chloride (XL \"C\"). RESULTS: Mucociliary transport was close to normal speed in viscoelastic samples compared to non-elastic, non-viscous or viscous-only samples. Spinnability ranged from 2.5 ± 0.6 to 50.9 ± 6.9 cm, and the amount of MS expelled from the SCM increased from 47 % to 96 % adding 1.5 μL to 150 μL of XL \"B\". Concurrently, particles were inversely reduced to almost disappear from the aerosolization pattern. CONCLUSION: The aerosolizability of MS was modified by increasing its cohesivity, thereby reducing the number of particles expelled from the SCM while interfering minimally with its clearance on the frog palate. An unexpected finding is that MS crosslinking increased \"expectoration\"."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** How long do nosocomial pathogens persist on inanimate surfaces? A systematic review\n",
       "- **Authors:** Kramer, Axel; Schwebke, Ingeborg; Kampf, Günter\n",
       "- **Publish date/time:** 2006-08-16\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Inanimate surfaces have often been described as the source for outbreaks of nosocomial infections. The aim of this review is to summarize data on the persistence of different nosocomial pathogens on inanimate surfaces. METHODS: The literature was systematically reviewed in MedLine without language restrictions. In addition, cited articles in a report were assessed and standard textbooks on the topic were reviewed. All reports with experimental evidence on the duration of persistence of a nosocomial pathogen on any type of surface were included. RESULTS: Most gram-positive bacteria, such as Enterococcus spp. (including VRE), Staphylococcus aureus (including MRSA), or Streptococcus pyogenes, survive for months on dry surfaces. Many gram-negative species, such as Acinetobacter spp., Escherichia coli, Klebsiella spp., Pseudomonas aeruginosa, Serratia marcescens, or Shigella spp., can also survive for months. A few others, such as Bordetella pertussis, Haemophilus influenzae, Proteus vulgaris, or Vibrio cholerae, however, persist only for days. Mycobacteria, including Mycobacterium tuberculosis, and spore-forming bacteria, including Clostridium difficile, can also survive for months on surfaces. Candida albicans as the most important nosocomial fungal pathogen can survive up to 4 months on surfaces. Persistence of other yeasts, such as Torulopsis glabrata, was described to be similar (5 months) or shorter (Candida parapsilosis, 14 days). Most viruses from the respiratory tract, such as corona, coxsackie, influenza, SARS or rhino virus, can persist on surfaces for a few days. Viruses from the gastrointestinal tract, such as astrovirus, HAV, polio- or rota virus, persist for approximately 2 months. Blood-borne viruses, such as HBV or HIV, can persist for more than one week. Herpes viruses, such as CMV or HSV type 1 and 2, have been shown to persist from only a few hours up to 7 days. CONCLUSION: The most common nosocomial pathogens may well survive or persist on surfaces for months and can thereby be a continuous source of transmission if no regular preventive surface disinfection is performed."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Topic ID #7\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Discovering human history from stomach bacteria\n",
       "- **Authors:** Disotell, Todd R\n",
       "- **Publish date/time:** 2003-04-28\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** Recent analyses of human pathogens have revealed that their evolutionary histories are congruent with the hypothesized pattern of ancient and modern human population migrations. Phylogenetic trees of strains of the bacterium Helicobacter pylori and the polyoma JC virus taken from geographically diverse groups of human beings correlate closely with relationships of the populations in which they are found."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Viral Discovery and Sequence Recovery Using DNA Microarrays\n",
       "- **Authors:** Wang, David; Urisman, Anatoly; Liu, Yu-Tsueng; Springer, Michael; Ksiazek, Thomas G; Erdman, Dean D; Mardis, Elaine R; Hickenbotham, Matthew; Magrini, Vincent; Eldred, James; Latreille, J. Phillipe; Wilson, Richard K; Ganem, Don; DeRisi, Joseph L\n",
       "- **Publish date/time:** 2003-11-17\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** Because of the constant threat posed by emerging infectious diseases and the limitations of existing approaches used to identify new pathogens, there is a great demand for new technological methods for viral discovery. We describe herein a DNA microarray-based platform for novel virus identification and characterization. Central to this approach was a DNA microarray designed to detect a wide range of known viruses as well as novel members of existing viral families; this microarray contained the most highly conserved 70mer sequences from every fully sequenced reference viral genome in GenBank. During an outbreak of severe acute respiratory syndrome (SARS) in March 2003, hybridization to this microarray revealed the presence of a previously uncharacterized coronavirus in a viral isolate cultivated from a SARS patient. To further characterize this new virus, approximately 1 kb of the unknown virus genome was cloned by physically recovering viral sequences hybridized to individual array elements. Sequencing of these fragments confirmed that the virus was indeed a new member of the coronavirus family. This combination of array hybridization followed by direct viral sequence recovery should prove to be a general strategy for the rapid identification and characterization of novel viruses and emerging infectious disease."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** A model of tripeptidyl-peptidase I (CLN2), a ubiquitous and highly conserved member of the sedolisin family of serine-carboxyl peptidases\n",
       "- **Authors:** Wlodawer, Alexander; Durell, Stewart R; Li, Mi; Oyama, Hiroshi; Oda, Kohei; Dunn, Ben M\n",
       "- **Publish date/time:** 2003-11-11\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Tripeptidyl-peptidase I, also known as CLN2, is a member of the family of sedolisins (serine-carboxyl peptidases). In humans, defects in expression of this enzyme lead to a fatal neurodegenerative disease, classical late-infantile neuronal ceroid lipofuscinosis. Similar enzymes have been found in the genomic sequences of several species, but neither systematic analyses of their distribution nor modeling of their structures have been previously attempted. RESULTS: We have analyzed the presence of orthologs of human CLN2 in the genomic sequences of a number of eukaryotic species. Enzymes with sequences sharing over 80% identity have been found in the genomes of macaque, mouse, rat, dog, and cow. Closely related, although clearly distinct, enzymes are present in fish (fugu and zebra), as well as in frogs (Xenopus tropicalis). A three-dimensional model of human CLN2 was built based mainly on the homology with Pseudomonas sp. 101 sedolisin. CONCLUSION: CLN2 is very highly conserved and widely distributed among higher organisms and may play an important role in their life cycles. The model presented here indicates a very open and accessible active site that is almost completely conserved among all known CLN2 enzymes. This result is somehow surprising for a tripeptidase where the presence of a more constrained binding pocket was anticipated. This structural model should be useful in the search for the physiological substrates of these enzymes and in the design of more specific inhibitors of CLN2."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Base-By-Base: Single nucleotide-level analysis of whole viral genome alignments\n",
       "- **Authors:** Brodie, Ryan; Smith, Alex J; Roper, Rachel L; Tcherepanov, Vasily; Upton, Chris\n",
       "- **Publish date/time:** 2004-07-14\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: With ever increasing numbers of closely related virus genomes being sequenced, it has become desirable to be able to compare two genomes at a level more detailed than gene content because two strains of an organism may share the same set of predicted genes but still differ in their pathogenicity profiles. For example, detailed comparison of multiple isolates of the smallpox virus genome (each approximately 200 kb, with 200 genes) is not feasible without new bioinformatics tools. RESULTS: A software package, Base-By-Base, has been developed that provides visualization tools to enable researchers to 1) rapidly identify and correct alignment errors in large, multiple genome alignments; and 2) generate tabular and graphical output of differences between the genomes at the nucleotide level. Base-By-Base uses detailed annotation information about the aligned genomes and can list each predicted gene with nucleotide differences, display whether variations occur within promoter regions or coding regions and whether these changes result in amino acid substitutions. Base-By-Base can connect to our mySQL database (Virus Orthologous Clusters; VOCs) to retrieve detailed annotation information about the aligned genomes or use information from text files. CONCLUSION: Base-By-Base enables users to quickly and easily compare large viral genomes; it highlights small differences that may be responsible for important phenotypic differences such as virulence. It is available via the Internet using Java Web Start and runs on Macintosh, PC and Linux operating systems with the Java 1.4 virtual machine."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Virology on the Internet: the time is right for a new journal\n",
       "- **Authors:** Garry, Robert F\n",
       "- **Publish date/time:** 2004-08-26\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** Virology Journal is an exclusively on-line, Open Access journal devoted to the presentation of high-quality original research concerning human, animal, plant, insect bacterial, and fungal viruses. Virology Journal will establish a strategic alternative to the traditional virology communication process."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Topic ID #8\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** A new recruit for the army of the men of death\n",
       "- **Authors:** Petsko, Gregory A\n",
       "- **Publish date/time:** 2003-06-27\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** The army of the men of death, in John Bunyan's memorable phrase, has a new recruit, and fear has a new face: a face wearing a surgical mask."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Association of HLA class I with severe acute respiratory syndrome coronavirus infection\n",
       "- **Authors:** Lin, Marie; Tseng, Hsiang-Kuang; Trejaut, Jean A; Lee, Hui-Lin; Loo, Jun-Hun; Chu, Chen-Chung; Chen, Pei-Jan; Su, Ying-Wen; Lim, Ken Hong; Tsai, Zen-Uong; Lin, Ruey-Yi; Lin, Ruey-Shiung; Huang, Chun-Hsiung\n",
       "- **Publish date/time:** 2003-09-12\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: The human leukocyte antigen (HLA) system is widely used as a strategy in the search for the etiology of infectious diseases and autoimmune disorders. During the Taiwan epidemic of severe acute respiratory syndrome (SARS), many health care workers were infected. In an effort to establish a screening program for high risk personal, the distribution of HLA class I and II alleles in case and control groups was examined for the presence of an association to a genetic susceptibly or resistance to SARS coronavirus infection. METHODS: HLA-class I and II allele typing by PCR-SSOP was performed on 37 cases of probable SARS, 28 fever patients excluded later as probable SARS, and 101 non-infected health care workers who were exposed or possibly exposed to SARS coronavirus. An additional control set of 190 normal healthy unrelated Taiwanese was also used in the analysis. RESULTS: Woolf and Haldane Odds ratio (OR) and corrected P-value (Pc) obtained from two tails Fisher exact test were used to show susceptibility of HLA class I or class II alleles with coronavirus infection. At first, when analyzing infected SARS patients and high risk health care workers groups, HLA-B*4601 (OR = 2.08, P = 0.04, Pc = n.s.) and HLA-B*5401 (OR = 5.44, P = 0.02, Pc = n.s.) appeared as the most probable elements that may be favoring SARS coronavirus infection. After selecting only a \"severe cases\" patient group from the infected \"probable SARS\" patient group and comparing them with the high risk health care workers group, the severity of SARS was shown to be significantly associated with HLA-B*4601 (P = 0.0008 or Pc = 0.0279). CONCLUSIONS: Densely populated regions with genetically related southern Asian populations appear to be more affected by the spreading of SARS infection. Up until recently, no probable SARS patients were reported among Taiwan indigenous peoples who are genetically distinct from the Taiwanese general population, have no HLA-B* 4601 and have high frequency of HLA-B* 1301. While increase of HLA-B* 4601 allele frequency was observed in the \"Probable SARS infected\" patient group, a further significant increase of the allele was seen in the \"Severe cases\" patient group. These results appeared to indicate association of HLA-B* 4601 with the severity of SARS infection in Asian populations. Independent studies are needed to test these results."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** The Virus That Changed My World\n",
       "- **Authors:** Poh Ng, Lisa Fong\n",
       "- **Publish date/time:** 2003-12-22\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** Personal account of a young virologist working in Singapore at the height of the 2003 SARS pandemic"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Pro/con clinical debate: Steroids are a key component in the treatment of SARS\n",
       "- **Authors:** Gomersall, Charles D; Kargel, Marcus J; Lapinsky, Stephen E\n",
       "- **Publish date/time:** 2004-01-26\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** SARS (severe acute respiratory syndrome) proved an enormous physical and emotional challenge to frontline health care workers throughout the world in late 2002 through to mid 2003. A large percentage of patients (many being health care workers themselves) became critically ill. Unfortunately, clinicians caring for these individuals did not have the advantage of previous experience or research data on which to base treatment decisions. As a result, at least early in the outbreak, a 'best guess approach' and/or anecdotes drove therapy. In many centres systemic steroids, which carry many potential downsides, became a mainstay of therapy. In this issue of Critical Care, two groups that have frontline experience of SARS debate the role of steroids. Let us hope and pray together that we never have the patient population needed to resolve the questions the two sides raise."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** 8th Annual Toronto Critical Care Medicine Symposium, 30 October–1 November 2003, Toronto, Ontario, Canada\n",
       "- **Authors:** Granton, Jeff; Granton, John\n",
       "- **Publish date/time:** 2004-01-02\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** nan"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Topic ID #9\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** A double epidemic model for the SARS propagation\n",
       "- **Authors:** Ng, Tuen Wai; Turinici, Gabriel; Danchin, Antoine\n",
       "- **Publish date/time:** 2003-09-10\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: An epidemic of a Severe Acute Respiratory Syndrome (SARS) caused by a new coronavirus has spread from the Guangdong province to the rest of China and to the world, with a puzzling contagion behavior. It is important both for predicting the future of the present outbreak and for implementing effective prophylactic measures, to identify the causes of this behavior. RESULTS: In this report, we show first that the standard Susceptible-Infected-Removed (SIR) model cannot account for the patterns observed in various regions where the disease spread. We develop a model involving two superimposed epidemics to study the recent spread of the SARS in Hong Kong and in the region. We explore the situation where these epidemics may be caused either by a virus and one or several mutants that changed its tropism, or by two unrelated viruses. This has important consequences for the future: the innocuous epidemic might still be there and generate, from time to time, variants that would have properties similar to those of SARS. CONCLUSION: We find that, in order to reconcile the existing data and the spread of the disease, it is convenient to suggest that a first milder outbreak protected against the SARS. Regions that had not seen the first epidemic, or that were affected simultaneously with the SARS suffered much more, with a very high percentage of persons affected. We also find regions where the data appear to be inconsistent, suggesting that they are incomplete or do not reflect an appropriate identification of SARS patients. Finally, we could, within the framework of the model, fix limits to the future development of the epidemic, allowing us to identify landmarks that may be useful to set up a monitoring system to follow the evolution of the epidemic. The model also suggests that there might exist a SARS precursor in a large reservoir, prompting for implementation of precautionary measures when the weather cools down."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Air pollution and case fatality of SARS in the People's Republic of China: an ecologic study\n",
       "- **Authors:** Cui, Yan; Zhang, Zuo-Feng; Froines, John; Zhao, Jinkou; Wang, Hua; Yu, Shun-Zhang; Detels, Roger\n",
       "- **Publish date/time:** 2003-11-20\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** BACKGROUND: Severe acute respiratory syndrome (SARS) has claimed 349 lives with 5,327 probable cases reported in mainland China since November 2002. SARS case fatality has varied across geographical areas, which might be partially explained by air pollution level. METHODS: Publicly accessible data on SARS morbidity and mortality were utilized in the data analysis. Air pollution was evaluated by air pollution index (API) derived from the concentrations of particulate matter, sulfur dioxide, nitrogen dioxide, carbon monoxide and ground-level ozone. Ecologic analysis was conducted to explore the association and correlation between air pollution and SARS case fatality via model fitting. Partially ecologic studies were performed to assess the effects of long-term and short-term exposures on the risk of dying from SARS. RESULTS: Ecologic analysis conducted among 5 regions with 100 or more SARS cases showed that case fatality rate increased with the increment of API (case fatality = - 0.063 + 0.001 * API). Partially ecologic study based on short-term exposure demonstrated that SARS patients from regions with moderate APIs had an 84% increased risk of dying from SARS compared to those from regions with low APIs (RR = 1.84, 95% CI: 1.41–2.40). Similarly, SARS patients from regions with high APIs were twice as likely to die from SARS compared to those from regions with low APIs. (RR = 2.18, 95% CI: 1.31–3.65). Partially ecologic analysis based on long-term exposure to ambient air pollution showed the similar association. CONCLUSION: Our studies demonstrated a positive association between air pollution and SARS case fatality in Chinese population by utilizing publicly accessible data on SARS statistics and air pollution indices. Although ecologic fallacy and uncontrolled confounding effect might have biased the results, the possibility of a detrimental effect of air pollution on the prognosis of SARS patients deserves further investigation."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Towards evidence-based, GIS-driven national spatial health information infrastructure and surveillance services in the United Kingdom\n",
       "- **Authors:** Boulos, Maged N Kamel\n",
       "- **Publish date/time:** 2004-01-28\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** The term \"Geographic Information Systems\" (GIS) has been added to MeSH in 2003, a step reflecting the importance and growing use of GIS in health and healthcare research and practices. GIS have much more to offer than the obvious digital cartography (map) functions. From a community health perspective, GIS could potentially act as powerful evidence-based practice tools for early problem detection and solving. When properly used, GIS can: inform and educate (professionals and the public); empower decision-making at all levels; help in planning and tweaking clinically and cost-effective actions, in predicting outcomes before making any financial commitments and ascribing priorities in a climate of finite resources; change practices; and continually monitor and analyse changes, as well as sentinel events. Yet despite all these potentials for GIS, they remain under-utilised in the UK National Health Service (NHS). This paper has the following objectives: (1) to illustrate with practical, real-world scenarios and examples from the literature the different GIS methods and uses to improve community health and healthcare practices, e.g., for improving hospital bed availability, in community health and bioterrorism surveillance services, and in the latest SARS outbreak; (2) to discuss challenges and problems currently hindering the wide-scale adoption of GIS across the NHS; and (3) to identify the most important requirements and ingredients for addressing these challenges, and realising GIS potential within the NHS, guided by related initiatives worldwide. The ultimate goal is to illuminate the road towards implementing a comprehensive national, multi-agency spatio-temporal health information infrastructure functioning proactively in real time. The concepts and principles presented in this paper can be also applied in other countries, and on regional (e.g., European Union) and global levels."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Descriptive review of geographic mapping of severe acute respiratory syndrome (SARS) on the Internet\n",
       "- **Authors:** Boulos, Maged N Kamel\n",
       "- **Publish date/time:** 2004-01-28\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** From geographic mapping at different scales to location-based alerting services, geoinformatics plays an important role in the study and control of global outbreaks like severe acute respiratory syndrome (SARS). This paper reviews several geographic mapping efforts of SARS on the Internet that employ a variety of techniques like choropleth rendering, graduated circles, graduated pie charts, buffering, overlay analysis and animation. The aim of these mapping services is to educate the public (especially travellers to potentially at-risk areas) and assist public health authorities in analysing the spatial and temporal trends and patterns of SARS and in assessing/revising current control measures."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "\n",
       "- **Title:** Recently published papers: all the usual suspects and carbon dioxide\n",
       "- **Authors:** Ball, Jonathan\n",
       "- **Publish date/time:** 2004-01-02\n",
       "- **Linked references:** 0\n",
       "- **Linked referenced by:** 0\n",
       "- **Abstract:** nan"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "docs_topics = docs_classified.argmax(1)\n",
    "topic_papers = defaultdict(list)\n",
    "all_papers = list(pageranks.keys())\n",
    "for idx, topic_id in enumerate(docs_topics):\n",
    "    topic_papers[topic_id].append(all_papers[idx])\n",
    "    \n",
    "for topic_id, papers in sorted(topic_papers.items(), key=lambda t: t[0]):\n",
    "    print(f\"Topic ID #{topic_id}\")\n",
    "    sorted_papers = sorted(papers, reverse=True, key=lambda p: pageranks[p])\n",
    "    for paper in sorted_papers[:5]:\n",
    "        paper_as_markdown(paper)\n",
    "    #print(\"\\n\", end=\"\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
