{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Representations, LDA\n",
    "\n",
    "En este notebook se calcularán representaciones de los papers en función del texto de éstos.\n",
    "Luego, a partir de dichas representaciones se realizará un modelado de tópicos usando la técnica LDA.\n",
    "Finalmente, se determinarán los papers más relevantes de cada tópico usando los puntajes PageRank de cada paper."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{<__main__.PDFPaper at 0x7f170915ae90>: 0.01430039297450863,\n",
       " <__main__.PDFPaper at 0x7f16d3a0fed0>: 0.006843458528155105,\n",
       " <__main__.PMCPaper at 0x7f16d3a216d0>: 0.004866234092719767,\n",
       " <__main__.PDFPaper at 0x7f16d3a3da50>: 0.004775174613208849,\n",
       " <__main__.PMCPaper at 0x7f170ca30750>: 0.00475030023872773,\n",
       " <__main__.PMCPaper at 0x7f170ca48c50>: 0.004457157589108286,\n",
       " <__main__.PDFPaper at 0x7f16d39e8710>: 0.0044072907443844,\n",
       " <__main__.PMCPaper at 0x7f16d3995250>: 0.004233732165218011,\n",
       " <__main__.PDFPaper at 0x7f16d39ad8d0>: 0.003923803811602406,\n",
       " <__main__.PDFPaper at 0x7f16d395c0d0>: 0.0037811442078439805,\n",
       " <__main__.PDFPaper at 0x7f16d3862a90>: 0.0036584608150057128,\n",
       " <__main__.PDFPaper at 0x7f16d3867c90>: 0.0034915974841629935,\n",
       " <__main__.PDFPaper at 0x7f16d38853d0>: 0.003391606502514324,\n",
       " <__main__.PMCPaper at 0x7f16d3831f90>: 0.0029271613198010357,\n",
       " <__main__.PDFPaper at 0x7f16d383d850>: 0.002651556917500903,\n",
       " <__main__.PDFPaper at 0x7f16d37d0c10>: 0.002460381954514867,\n",
       " <__main__.PMCPaper at 0x7f16d37ee990>: 0.002410994825091878,\n",
       " <__main__.PMCPaper at 0x7f16d3802fd0>: 0.0023625192359901306,\n",
       " <__main__.PDFPaper at 0x7f16d37ac990>: 0.0022487220961048154,\n",
       " <__main__.PDFPaper at 0x7f16d37bbd90>: 0.0022350635980696604,\n",
       " <__main__.PDFPaper at 0x7f16d3750910>: 0.00202560531991078,\n",
       " <__main__.PMCPaper at 0x7f16d372b250>: 0.0019719642356628327,\n",
       " <__main__.PMCPaper at 0x7f16d3730f10>: 0.001827014041600769,\n",
       " <__main__.PDFPaper at 0x7f16d36d7710>: 0.0017077327853978691,\n",
       " <__main__.PMCPaper at 0x7f16d36f0d50>: 0.001566568963988873,\n",
       " <__main__.PMCPaper at 0x7f16d3692250>: 0.0015039486579978862,\n",
       " <__main__.PMCPaper at 0x7f16d35dd450>: 0.0014982487994804952,\n",
       " <__main__.PDFPaper at 0x7f16d35e4f90>: 0.0014899273892289378,\n",
       " <__main__.PDFPaper at 0x7f16d35921d0>: 0.001478124798150249,\n",
       " <__main__.PDFPaper at 0x7f16d35a8a10>: 0.0013542168342878028,\n",
       " <__main__.PDFPaper at 0x7f16d35c8c10>: 0.0013102097794139872,\n",
       " <__main__.PDFPaper at 0x7f16d3554310>: 0.0012744169190402942,\n",
       " <__main__.PMCPaper at 0x7f16d3574650>: 0.0012479458593963946,\n",
       " <__main__.PMCPaper at 0x7f16d3523250>: 0.0012158796544376853,\n",
       " <__main__.PDFPaper at 0x7f16d34e4690>: 0.0011926302910788633,\n",
       " <__main__.PMCPaper at 0x7f16d34f5790>: 0.0011783559566485198,\n",
       " <__main__.PDFPaper at 0x7f16d34a70d0>: 0.001162453327040416,\n",
       " <__main__.PMCPaper at 0x7f16d345d790>: 0.0011570999463557678,\n",
       " <__main__.PMCPaper at 0x7f16d347d810>: 0.0011380676427415219,\n",
       " <__main__.PMCPaper at 0x7f16d3416790>: 0.001135419433398014,\n",
       " <__main__.PDFPaper at 0x7f16d33510d0>: 0.0011139290796710972,\n",
       " <__main__.PMCPaper at 0x7f16d337d750>: 0.001103512241204263,\n",
       " <__main__.PDFPaper at 0x7f16d332dfd0>: 0.0010894306367354308,\n",
       " <__main__.PMCPaper at 0x7f16d32e5290>: 0.0010529421153311514,\n",
       " <__main__.PMCPaper at 0x7f16d32fe4d0>: 0.0010510825747788062,\n",
       " <__main__.PDFPaper at 0x7f16d3051cd0>: 0.0010171660778103202,\n",
       " <__main__.PDFPaper at 0x7f16d3066110>: 0.0010133564873648906,\n",
       " <__main__.PMCPaper at 0x7f16d306d910>: 0.0009994613798101014,\n",
       " <__main__.PDFPaper at 0x7f16d300e410>: 0.000992867870433156,\n",
       " <__main__.PMCPaper at 0x7f16d3013d90>: 0.0009892878972661448,\n",
       " <__main__.PDFPaper at 0x7f16d30223d0>: 0.0009759471442975491,\n",
       " <__main__.PMCPaper at 0x7f16d2fe4690>: 0.0009436127381192216,\n",
       " <__main__.PMCPaper at 0x7f16d2fb3350>: 0.0009248321025645149,\n",
       " <__main__.PDFPaper at 0x7f16d2fbe390>: 0.0009229982686424677,\n",
       " <__main__.PDFPaper at 0x7f16d2f5df10>: 0.0009161417288003867,\n",
       " <__main__.PMCPaper at 0x7f16d2f86590>: 0.0009145306782065689,\n",
       " <__main__.PDFPaper at 0x7f16d2f86d90>: 0.0008908982176466145,\n",
       " <__main__.PMCPaper at 0x7f16d2f18250>: 0.0008903631764525719,\n",
       " <__main__.PDFPaper at 0x7f16d2f44850>: 0.0008860595620714116,\n",
       " <__main__.PDFPaper at 0x7f16d2ee3fd0>: 0.0008838179673871555,\n",
       " <__main__.PMCPaper at 0x7f16d2e97890>: 0.0008624471591148296,\n",
       " <__main__.PDFPaper at 0x7f16d2e83850>: 0.000855491204123645,\n",
       " <__main__.PDFPaper at 0x7f16d2e29b10>: 0.0008335017568958933,\n",
       " <__main__.PDFPaper at 0x7f16d2ddda10>: 0.0008332127595569157,\n",
       " <__main__.PDFPaper at 0x7f16d2e0ac50>: 0.0008299122804544944,\n",
       " <__main__.PDFPaper at 0x7f16d2d51310>: 0.00081929174857641,\n",
       " <__main__.PMCPaper at 0x7f16d2d7b4d0>: 0.0008115121113611667,\n",
       " <__main__.PDFPaper at 0x7f16d2d11a50>: 0.0008058625689859416,\n",
       " <__main__.PMCPaper at 0x7f16d2d1da50>: 0.0007966315678413592,\n",
       " <__main__.PMCPaper at 0x7f16d2d2a610>: 0.0007946038822407974,\n",
       " <__main__.PMCPaper at 0x7f16d2cdd7d0>: 0.0007945441950541879,\n",
       " <__main__.PMCPaper at 0x7f16d2cf5ad0>: 0.0007920820874137951,\n",
       " <__main__.PMCPaper at 0x7f16d2c96510>: 0.0007752292826181348,\n",
       " <__main__.PDFPaper at 0x7f16d2c73c50>: 0.0007702188574508303,\n",
       " <__main__.PDFPaper at 0x7f16d2c135d0>: 0.0007556815747099798,\n",
       " <__main__.PMCPaper at 0x7f16d2c2d8d0>: 0.0007489682662681983,\n",
       " <__main__.PMCPaper at 0x7f16d2c43210>: 0.0007392855851945979,\n",
       " <__main__.PMCPaper at 0x7f16d2bccc10>: 0.0007378677781884364,\n",
       " <__main__.PMCPaper at 0x7f16d2be75d0>: 0.0007373576762073199,\n",
       " <__main__.PMCPaper at 0x7f16d2b8c150>: 0.0007330927144377985,\n",
       " <__main__.PDFPaper at 0x7f16d2b9ef10>: 0.0007156056289693134,\n",
       " <__main__.PMCPaper at 0x7f16d2bc0ad0>: 0.0007095932632628967,\n",
       " <__main__.PDFPaper at 0x7f16d2b60f50>: 0.0007084911019024793,\n",
       " <__main__.PMCPaper at 0x7f16d2ac3110>: 0.0007065928837434289,\n",
       " <__main__.PDFPaper at 0x7f16d2a57c50>: 0.0007042137191091988,\n",
       " <__main__.PDFPaper at 0x7f16d2a6dc50>: 0.0007006260269954579,\n",
       " <__main__.PDFPaper at 0x7f16d2a23890>: 0.00069634024017567,\n",
       " <__main__.PDFPaper at 0x7f16d2a47cd0>: 0.0006929399112313394,\n",
       " <__main__.PDFPaper at 0x7f16d29f3d90>: 0.0006894031225736642,\n",
       " <__main__.PMCPaper at 0x7f16d28e5150>: 0.000681232840132905,\n",
       " <__main__.PDFPaper at 0x7f16d28f2d10>: 0.0006790589542772886,\n",
       " <__main__.PDFPaper at 0x7f16d28af390>: 0.0006734161998587223,\n",
       " <__main__.PDFPaper at 0x7f16d2850ad0>: 0.0006647694454991351,\n",
       " <__main__.PMCPaper at 0x7f16d2863190>: 0.0006598207503755178,\n",
       " <__main__.PMCPaper at 0x7f16d280d290>: 0.000657955601331927,\n",
       " <__main__.PDFPaper at 0x7f16d2827ad0>: 0.0006560787385988621,\n",
       " <__main__.PMCPaper at 0x7f16d2835bd0>: 0.0006480380159807599,\n",
       " <__main__.PDFPaper at 0x7f16d2792410>: 0.000646780860715142,\n",
       " <__main__.PDFPaper at 0x7f16d27bb090>: 0.000639637881789391,\n",
       " <__main__.PDFPaper at 0x7f16d27c0f50>: 0.0006274952252884479,\n",
       " <__main__.PDFPaper at 0x7f16d312a410>: 0.0006261600472889045,\n",
       " <__main__.PDFPaper at 0x7f16d30cbd90>: 0.0006198007978490753,\n",
       " <__main__.PMCPaper at 0x7f16d2565bd0>: 0.0006046534260491273,\n",
       " <__main__.PMCPaper at 0x7f16d2572550>: 0.0006038511151902394,\n",
       " <__main__.PMCPaper at 0x7f16d250aad0>: 0.0006021325693383526,\n",
       " <__main__.PMCPaper at 0x7f16d2522e10>: 0.0006009097092502836,\n",
       " <__main__.PDFPaper at 0x7f16d2528790>: 0.0005891688875274568,\n",
       " <__main__.PDFPaper at 0x7f16d248b150>: 0.0005825391233323523,\n",
       " <__main__.PMCPaper at 0x7f16d24a1d50>: 0.0005648950616473759,\n",
       " <__main__.PMCPaper at 0x7f16d2411b90>: 0.0005568367814457433,\n",
       " <__main__.PDFPaper at 0x7f16d2416450>: 0.0005556713445165165,\n",
       " <__main__.PMCPaper at 0x7f16d243fa10>: 0.0005495888329278636,\n",
       " <__main__.PDFPaper at 0x7f16d23db290>: 0.0005447305055799908,\n",
       " <__main__.PDFPaper at 0x7f16d23ee410>: 0.0005441817313985869,\n",
       " <__main__.PDFPaper at 0x7f16d23a6090>: 0.0005351876120323155,\n",
       " <__main__.PMCPaper at 0x7f16d23b3d50>: 0.0005325493869440276,\n",
       " <__main__.PDFPaper at 0x7f16d234fe10>: 0.000530403419622086,\n",
       " <__main__.PMCPaper at 0x7f16d22d8a90>: 0.0005247625480884806,\n",
       " <__main__.PMCPaper at 0x7f16d22eebd0>: 0.000521982384574816,\n",
       " <__main__.PMCPaper at 0x7f16d224c890>: 0.0005148165795061602,\n",
       " <__main__.PDFPaper at 0x7f16d225d110>: 0.0005145620101493629,\n",
       " <__main__.PDFPaper at 0x7f16d221cc90>: 0.0005092462488735188,\n",
       " <__main__.PDFPaper at 0x7f16d222e710>: 0.0005072461502303403,\n",
       " <__main__.PDFPaper at 0x7f16d223f390>: 0.0005068176671564511,\n",
       " <__main__.PMCPaper at 0x7f16d21ad990>: 0.0005060623066522507,\n",
       " <__main__.PMCPaper at 0x7f16d21b5250>: 0.0005034230552008859,\n",
       " <__main__.PMCPaper at 0x7f16d21c8f10>: 0.0005022630543079294,\n",
       " <__main__.PDFPaper at 0x7f16d2171290>: 0.000499502613378623,\n",
       " <__main__.PDFPaper at 0x7f16d217fa50>: 0.000499328752267253,\n",
       " <__main__.PMCPaper at 0x7f16d2125d50>: 0.0004959609560339257,\n",
       " <__main__.PDFPaper at 0x7f16d20ef2d0>: 0.0004897450022683636,\n",
       " <__main__.PDFPaper at 0x7f16d203c790>: 0.00048758090810729123,\n",
       " <__main__.PDFPaper at 0x7f16d1fea650>: 0.00048244299805614893,\n",
       " <__main__.PMCPaper at 0x7f16d1fa6f90>: 0.00047907377127173306,\n",
       " <__main__.PMCPaper at 0x7f16d1fb9c90>: 0.0004773103709314929,\n",
       " <__main__.PDFPaper at 0x7f16d1f4fc90>: 0.0004684734915979476,\n",
       " <__main__.PMCPaper at 0x7f16d1f86110>: 0.00046797898404784643,\n",
       " <__main__.PDFPaper at 0x7f16d1f1c290>: 0.0004663030467547613,\n",
       " <__main__.PMCPaper at 0x7f16d1ed0d50>: 0.0004657405148991084,\n",
       " <__main__.PMCPaper at 0x7f16d1edd050>: 0.00046436784598415294,\n",
       " <__main__.PDFPaper at 0x7f16d1ef00d0>: 0.0004590467536416714,\n",
       " <__main__.PMCPaper at 0x7f16d1e92c50>: 0.00045817607056336496,\n",
       " <__main__.PDFPaper at 0x7f16d1ea4fd0>: 0.0004553921059732964,\n",
       " <__main__.PMCPaper at 0x7f16d1e53750>: 0.00045519528909474075,\n",
       " <__main__.PMCPaper at 0x7f16d1e5ad10>: 0.0004496288600136698,\n",
       " <__main__.PDFPaper at 0x7f16d1e78d50>: 0.00044940081520011306,\n",
       " <__main__.PDFPaper at 0x7f16d1ddac50>: 0.00044496924349310423,\n",
       " <__main__.PDFPaper at 0x7f16d1df5110>: 0.00044293554842926884,\n",
       " <__main__.PDFPaper at 0x7f16d1da1350>: 0.00044015071282245616,\n",
       " <__main__.PDFPaper at 0x7f16d1da7710>: 0.00043806932471451783,\n",
       " <__main__.PMCPaper at 0x7f16d1d67f90>: 0.00043773400621849133,\n",
       " <__main__.PMCPaper at 0x7f16d1d139d0>: 0.0004366867467660519,\n",
       " <__main__.PDFPaper at 0x7f16d1d24610>: 0.0004350035885631844,\n",
       " <__main__.PDFPaper at 0x7f16d1c9e410>: 0.00043422740928562457,\n",
       " <__main__.PDFPaper at 0x7f16d1cbffd0>: 0.00043420453213629926,\n",
       " <__main__.PMCPaper at 0x7f16d1c5de10>: 0.0004279142322120074,\n",
       " <__main__.PDFPaper at 0x7f16d1c787d0>: 0.0004254776250717137,\n",
       " <__main__.PDFPaper at 0x7f16d1c0acd0>: 0.00042390596881693117,\n",
       " <__main__.PMCPaper at 0x7f16d1c3a990>: 0.0004237684603521243,\n",
       " <__main__.PDFPaper at 0x7f16d1b4d490>: 0.00042292432317527775,\n",
       " <__main__.PDFPaper at 0x7f16d1b5e590>: 0.0004227611379776034,\n",
       " <__main__.PMCPaper at 0x7f16d1b112d0>: 0.00042193338384230217,\n",
       " <__main__.PDFPaper at 0x7f16d1b39810>: 0.00042001389537824903,\n",
       " <__main__.PMCPaper at 0x7f16d1ae13d0>: 0.0004184854995311734,\n",
       " <__main__.PDFPaper at 0x7f16d1affe90>: 0.0004183772908898126,\n",
       " <__main__.PDFPaper at 0x7f16d1aa3690>: 0.0004173165207346538,\n",
       " <__main__.PDFPaper at 0x7f16d1ab0890>: 0.0004153893856551641,\n",
       " <__main__.PDFPaper at 0x7f16d1a4dd50>: 0.0004148693863784739,\n",
       " <__main__.PDFPaper at 0x7f16d1a797d0>: 0.00041330651562578085,\n",
       " <__main__.PMCPaper at 0x7f16d1a39810>: 0.00041095624963991115,\n",
       " <__main__.PMCPaper at 0x7f16d19dbb10>: 0.0004105771405029895,\n",
       " <__main__.PMCPaper at 0x7f16d19a71d0>: 0.0004096703202076548,\n",
       " <__main__.PMCPaper at 0x7f16d197d950>: 0.00040730078757216344,\n",
       " <__main__.PDFPaper at 0x7f16d19323d0>: 0.0004021581657030508,\n",
       " <__main__.PDFPaper at 0x7f16d18f0cd0>: 0.00040208328074349986,\n",
       " <__main__.PMCPaper at 0x7f16d1881dd0>: 0.0004007143369808989,\n",
       " <__main__.PDFPaper at 0x7f16d181e790>: 0.0003972334880323661,\n",
       " <__main__.PMCPaper at 0x7f16d17db110>: 0.0003958688084337522,\n",
       " <__main__.PMCPaper at 0x7f16d1808e90>: 0.00039585071074335875,\n",
       " <__main__.PMCPaper at 0x7f16d1747e90>: 0.0003953119067437758,\n",
       " <__main__.PDFPaper at 0x7f16d16d0850>: 0.0003925735273270921,\n",
       " <__main__.PDFPaper at 0x7f16d1700d50>: 0.0003901864155350756,\n",
       " <__main__.PDFPaper at 0x7f16d16ae4d0>: 0.0003837293081750516,\n",
       " <__main__.PDFPaper at 0x7f16d164e0d0>: 0.0003832385989404248,\n",
       " <__main__.PDFPaper at 0x7f16d1661b90>: 0.00038193054013608207,\n",
       " <__main__.PMCPaper at 0x7f16d1687590>: 0.00038096791624734494,\n",
       " <__main__.PMCPaper at 0x7f16d161d890>: 0.00038005628896004137,\n",
       " <__main__.PDFPaper at 0x7f16d162abd0>: 0.00037948353060592634,\n",
       " <__main__.PMCPaper at 0x7f16d15de310>: 0.000374047431414668,\n",
       " <__main__.PDFPaper at 0x7f16d158dc10>: 0.00037397408992330736,\n",
       " <__main__.PMCPaper at 0x7f16d15a9d50>: 0.00037371525840787387,\n",
       " <__main__.PDFPaper at 0x7f16d157a150>: 0.0003702239315740662,\n",
       " <__main__.PMCPaper at 0x7f16d152a710>: 0.0003687048894098682,\n",
       " <__main__.PDFPaper at 0x7f16d2787f90>: 0.0003667883695670463,\n",
       " <__main__.PDFPaper at 0x7f16d2727890>: 0.00036557413033334044,\n",
       " <__main__.PMCPaper at 0x7f16d2696bd0>: 0.00036500669532301083,\n",
       " <__main__.PMCPaper at 0x7f16d2654b50>: 0.00036448701482942126,\n",
       " <__main__.PMCPaper at 0x7f16d2665290>: 0.0003640300760647221,\n",
       " <__main__.PMCPaper at 0x7f16d2610710>: 0.00036386274208070657,\n",
       " <__main__.PMCPaper at 0x7f16d262fed0>: 0.0003615741089803069,\n",
       " <__main__.PDFPaper at 0x7f16d25daa10>: 0.00035977160242667386,\n",
       " <__main__.PMCPaper at 0x7f16d2604cd0>: 0.0003585972820221037,\n",
       " <__main__.PDFPaper at 0x7f16d2595f10>: 0.0003583038677495768,\n",
       " <__main__.PDFPaper at 0x7f16d25c69d0>: 0.0003580984388447288,\n",
       " <__main__.PDFPaper at 0x7f16d10a53d0>: 0.0003565624765662945,\n",
       " <__main__.PDFPaper at 0x7f16d10c4590>: 0.00035449028784663396,\n",
       " <__main__.PMCPaper at 0x7f16d1074310>: 0.0003533237104447509,\n",
       " <__main__.PMCPaper at 0x7f16d1074bd0>: 0.0003519005306798407,\n",
       " <__main__.PMCPaper at 0x7f16d10237d0>: 0.000350010023621395,\n",
       " <__main__.PMCPaper at 0x7f16d1041050>: 0.00034873108040890254,\n",
       " <__main__.PDFPaper at 0x7f16d0fe5210>: 0.00034724387423252053,\n",
       " <__main__.PDFPaper at 0x7f16d0f8b750>: 0.0003468288359237658,\n",
       " <__main__.PMCPaper at 0x7f16d0fb0810>: 0.0003467230287294634,\n",
       " <__main__.PMCPaper at 0x7f16d0fbb850>: 0.0003464876143046426,\n",
       " <__main__.PMCPaper at 0x7f16d0f4e850>: 0.0003463572018392057,\n",
       " <__main__.PDFPaper at 0x7f16d0f60d90>: 0.0003456934509133823,\n",
       " <__main__.PMCPaper at 0x7f16d0f2fd50>: 0.0003431327276099969,\n",
       " <__main__.PMCPaper at 0x7f16d0f3a110>: 0.00034245570646486526,\n",
       " <__main__.PDFPaper at 0x7f16d0f3aa90>: 0.0003417682743750433,\n",
       " <__main__.PDFPaper at 0x7f16d0e96bd0>: 0.0003402066454937178,\n",
       " <__main__.PDFPaper at 0x7f16d0eb61d0>: 0.00033706754214365563,\n",
       " <__main__.PMCPaper at 0x7f16d0e673d0>: 0.00033531948943444105,\n",
       " <__main__.PDFPaper at 0x7f16d0e72150>: 0.00033153056257501166,\n",
       " <__main__.PMCPaper at 0x7f16d0e10c50>: 0.0003304376832663519,\n",
       " <__main__.PMCPaper at 0x7f16d0e21ad0>: 0.00032727981400865755,\n",
       " <__main__.PMCPaper at 0x7f16d0e2e110>: 0.0003232311527687069,\n",
       " <__main__.PMCPaper at 0x7f16d0d8c0d0>: 0.00031980401241773235,\n",
       " <__main__.PMCPaper at 0x7f16d0d98050>: 0.0003158047160301163,\n",
       " <__main__.PMCPaper at 0x7f16d0da1950>: 0.0003157179351814598,\n",
       " <__main__.PDFPaper at 0x7f16d0d60d50>: 0.00031550105937644185,\n",
       " <__main__.PMCPaper at 0x7f16d0d86250>: 0.00031451607013879075,\n",
       " <__main__.PMCPaper at 0x7f16d0d2dcd0>: 0.00030986770919142,\n",
       " <__main__.PDFPaper at 0x7f16d0cef210>: 0.0003089434861046467,\n",
       " <__main__.PMCPaper at 0x7f16d0cf2f50>: 0.00030708535959045005,\n",
       " <__main__.PDFPaper at 0x7f16d0d088d0>: 0.00030549228169618985,\n",
       " <__main__.PDFPaper at 0x7f16d0cb31d0>: 0.00030532607290660054,\n",
       " <__main__.PMCPaper at 0x7f16d0cb7e90>: 0.00030384147282197196,\n",
       " <__main__.PDFPaper at 0x7f16d0c5f390>: 0.00030308887186184137,\n",
       " <__main__.PDFPaper at 0x7f16d0c85e10>: 0.000302073892136799,\n",
       " <__main__.PDFPaper at 0x7f16d0c46250>: 0.00030173569686790564,\n",
       " <__main__.PDFPaper at 0x7f16d0bccf90>: 0.000298316223337973,\n",
       " <__main__.PDFPaper at 0x7f16d0bd6310>: 0.0002976442572192202,\n",
       " <__main__.PDFPaper at 0x7f16d0bf3a10>: 0.0002975831221941094,\n",
       " <__main__.PMCPaper at 0x7f16d0b140d0>: 0.0002935853287175196,\n",
       " <__main__.PDFPaper at 0x7f16d0ad5610>: 0.00029317170944440584,\n",
       " <__main__.PDFPaper at 0x7f16d0af2890>: 0.0002924592157027167,\n",
       " <__main__.PDFPaper at 0x7f16d0b02750>: 0.00029194268738777203,\n",
       " <__main__.PDFPaper at 0x7f16d0a97a50>: 0.0002912055752285786,\n",
       " <__main__.PMCPaper at 0x7f16d0a31550>: 0.0002908202586299699,\n",
       " <__main__.PMCPaper at 0x7f16d09ffe50>: 0.0002908036148122207,\n",
       " <__main__.PDFPaper at 0x7f16d094dd50>: 0.0002903384113629453,\n",
       " <__main__.PMCPaper at 0x7f16d0981390>: 0.0002895807958817043,\n",
       " <__main__.PMCPaper at 0x7f16d091be90>: 0.0002889117915423663,\n",
       " <__main__.PDFPaper at 0x7f16d0905f50>: 0.0002886734155194033,\n",
       " <__main__.PDFPaper at 0x7f16d089d450>: 0.0002879313728591793,\n",
       " <__main__.PMCPaper at 0x7f16d08a21d0>: 0.00028782506748945095,\n",
       " <__main__.PDFPaper at 0x7f16d08a7410>: 0.0002863945287098629,\n",
       " <__main__.PMCPaper at 0x7f16d0854bd0>: 0.0002849542771402994,\n",
       " <__main__.PDFPaper at 0x7f16d0836350>: 0.0002847752223222118,\n",
       " <__main__.PDFPaper at 0x7f16d07936d0>: 0.0002843922860410307,\n",
       " <__main__.PMCPaper at 0x7f16d07c5a90>: 0.0002827923559671971,\n",
       " <__main__.PDFPaper at 0x7f16d075be10>: 0.0002827305624123837,\n",
       " <__main__.PMCPaper at 0x7f16d0766650>: 0.0002824114839739421,\n",
       " <__main__.PMCPaper at 0x7f16d06ca9d0>: 0.0002816757247909932,\n",
       " <__main__.PDFPaper at 0x7f16d06ff510>: 0.0002809156900486056,\n",
       " <__main__.PDFPaper at 0x7f16d06bf790>: 0.0002795373544017601,\n",
       " <__main__.PDFPaper at 0x7f16d0628150>: 0.00027935537722479286,\n",
       " <__main__.PDFPaper at 0x7f16d0647e50>: 0.00027873563985751914,\n",
       " <__main__.PDFPaper at 0x7f16d05e6650>: 0.0002786869205949422,\n",
       " <__main__.PDFPaper at 0x7f16d05591d0>: 0.0002785920283434551,\n",
       " <__main__.PDFPaper at 0x7f16d0570650>: 0.0002784145353022157,\n",
       " <__main__.PMCPaper at 0x7f16d0575910>: 0.0002780082765136287,\n",
       " <__main__.PMCPaper at 0x7f16d057ea50>: 0.0002760629116069421,\n",
       " <__main__.PDFPaper at 0x7f16d051b8d0>: 0.00027592341575896563,\n",
       " <__main__.PDFPaper at 0x7f16d04e9ed0>: 0.0002756578834353428,\n",
       " <__main__.PDFPaper at 0x7f16d04a3ed0>: 0.0002755096246333907,\n",
       " <__main__.PDFPaper at 0x7f16d04c6a50>: 0.00027532240463659874,\n",
       " <__main__.PDFPaper at 0x7f16d04728d0>: 0.0002745519213433774,\n",
       " <__main__.PDFPaper at 0x7f16d0428d90>: 0.0002740536978079684,\n",
       " <__main__.PMCPaper at 0x7f16d03d4a10>: 0.00027353733256063373,\n",
       " <__main__.PDFPaper at 0x7f16d0393bd0>: 0.00027335712287242264,\n",
       " <__main__.PMCPaper at 0x7f16d03c1710>: 0.00027312421522565254,\n",
       " <__main__.PMCPaper at 0x7f16d0290990>: 0.0002727479519250505,\n",
       " <__main__.PMCPaper at 0x7f16d029c210>: 0.00027215095598828764,\n",
       " <__main__.PMCPaper at 0x7f16d02b5490>: 0.00027116829271695155,\n",
       " <__main__.PMCPaper at 0x7f16d0268b90>: 0.00027041837424688367,\n",
       " <__main__.PDFPaper at 0x7f16d022cb50>: 0.0002684656165235148,\n",
       " <__main__.PDFPaper at 0x7f16d023c410>: 0.000268410668894332,\n",
       " <__main__.PDFPaper at 0x7f16d01e6e10>: 0.000268065937174305,\n",
       " <__main__.PMCPaper at 0x7f16d0203f90>: 0.00026790853735901777,\n",
       " <__main__.PDFPaper at 0x7f16d0190890>: 0.00026775390544414064,\n",
       " <__main__.PDFPaper at 0x7f16d01a6090>: 0.000267119400678792,\n",
       " <__main__.PDFPaper at 0x7f16d01bed50>: 0.00026616656400704856,\n",
       " <__main__.PDFPaper at 0x7f16d016a850>: 0.00026515023003133775,\n",
       " <__main__.PMCPaper at 0x7f16d0118810>: 0.0002644346599539269,\n",
       " <__main__.PDFPaper at 0x7f16d0127590>: 0.0002636521002539747,\n",
       " <__main__.PDFPaper at 0x7f16d0146590>: 0.0002635635719011798,\n",
       " <__main__.PDFPaper at 0x7f16d0096850>: 0.00026287576738882775,\n",
       " <__main__.PMCPaper at 0x7f16d00b23d0>: 0.0002627173835916355,\n",
       " <__main__.PDFPaper at 0x7f16d006df50>: 0.0002617432172297174,\n",
       " <__main__.PDFPaper at 0x7f16d001fd10>: 0.0002588239394760283,\n",
       " <__main__.PDFPaper at 0x7f16cffda390>: 0.0002587489692130701,\n",
       " <__main__.PDFPaper at 0x7f16cff8a090>: 0.00025869010144993456,\n",
       " <__main__.PDFPaper at 0x7f16cff95950>: 0.00025833366033375206,\n",
       " <__main__.PMCPaper at 0x7f16cffa2710>: 0.0002582746806230285,\n",
       " <__main__.PMCPaper at 0x7f16cffbff50>: 0.0002574854760623507,\n",
       " <__main__.PMCPaper at 0x7f16cff6da50>: 0.00025483661354673315,\n",
       " <__main__.PMCPaper at 0x7f16cff7ea90>: 0.0002547618634525274,\n",
       " <__main__.PDFPaper at 0x7f16cff24bd0>: 0.0002541601903185203,\n",
       " <__main__.PDFPaper at 0x7f16cfe6b5d0>: 0.0002538663809718913,\n",
       " <__main__.PMCPaper at 0x7f16cfe85310>: 0.00025195302904441733,\n",
       " <__main__.PMCPaper at 0x7f16cfe18850>: 0.0002519237140630412,\n",
       " <__main__.PDFPaper at 0x7f16cfdf61d0>: 0.00025148754596250914,\n",
       " <__main__.PDFPaper at 0x7f16cfd590d0>: 0.0002478270791306427,\n",
       " <__main__.PDFPaper at 0x7f16cfd6f790>: 0.0002474165409648137,\n",
       " <__main__.PDFPaper at 0x7f16cfd857d0>: 0.000247207894042123,\n",
       " <__main__.PDFPaper at 0x7f16cfd39150>: 0.0002466480466844716,\n",
       " <__main__.PDFPaper at 0x7f16cfce7410>: 0.0002466454818234022,\n",
       " <__main__.PDFPaper at 0x7f16cfc9b7d0>: 0.00024662051919335445,\n",
       " <__main__.PDFPaper at 0x7f16cfc752d0>: 0.00024619007073088006,\n",
       " <__main__.PDFPaper at 0x7f16cfc1f290>: 0.00024502946769754233,\n",
       " <__main__.PDFPaper at 0x7f16cfc3a210>: 0.0002448905156605865,\n",
       " <__main__.PMCPaper at 0x7f16cfbcf290>: 0.00024366531775685528,\n",
       " <__main__.PMCPaper at 0x7f16cfc00850>: 0.00024316431264598875,\n",
       " <__main__.PMCPaper at 0x7f16cfb8e710>: 0.00024261250960912992,\n",
       " <__main__.PDFPaper at 0x7f16cfb930d0>: 0.00024211188714876977,\n",
       " <__main__.PDFPaper at 0x7f16cfbc3650>: 0.00024197513484248485,\n",
       " <__main__.PDFPaper at 0x7f16cfb72050>: 0.0002413190601917342,\n",
       " <__main__.PMCPaper at 0x7f16cfa7fcd0>: 0.00024097366455966423,\n",
       " <__main__.PDFPaper at 0x7f16cfa16090>: 0.00024072936578021598,\n",
       " <__main__.PDFPaper at 0x7f16cfa2e090>: 0.0002404677516781183,\n",
       " <__main__.PMCPaper at 0x7f16cf9f4e10>: 0.00023951581780479699,\n",
       " <__main__.PMCPaper at 0x7f16cfa04d10>: 0.00023916245495489823,\n",
       " <__main__.PDFPaper at 0x7f16cf9a6f50>: 0.00023764293234574074,\n",
       " <__main__.PDFPaper at 0x7f16cf959750>: 0.0002374492212240599,\n",
       " <__main__.PMCPaper at 0x7f16cf976610>: 0.00023684911611228157,\n",
       " <__main__.PMCPaper at 0x7f16cf936490>: 0.00023647497396809872,\n",
       " <__main__.PMCPaper at 0x7f16cf8d8490>: 0.00023646828009700488,\n",
       " <__main__.PMCPaper at 0x7f16cf88ea10>: 0.0002355737972867354,\n",
       " <__main__.PDFPaper at 0x7f16cf8a8310>: 0.00023412340964635466,\n",
       " <__main__.PMCPaper at 0x7f16cf8b8d10>: 0.00023392628143214828,\n",
       " <__main__.PDFPaper at 0x7f16cf877bd0>: 0.00023345570591296508,\n",
       " <__main__.PMCPaper at 0x7f16cf817550>: 0.00023303197690634892,\n",
       " <__main__.PDFPaper at 0x7f16cf82da90>: 0.00023245591350206139,\n",
       " <__main__.PDFPaper at 0x7f16cf7d9f50>: 0.00023136645627485722,\n",
       " <__main__.PDFPaper at 0x7f16cf7f11d0>: 0.00023129482840352522,\n",
       " <__main__.PMCPaper at 0x7f16cf79f7d0>: 0.00023110229775953384,\n",
       " <__main__.PDFPaper at 0x7f16cf7a6ed0>: 0.00023099335994070273,\n",
       " <__main__.PDFPaper at 0x7f16cf6f7f90>: 0.00023098159303200035,\n",
       " <__main__.PDFPaper at 0x7f16cf698050>: 0.00023065224748142048,\n",
       " <__main__.PMCPaper at 0x7f16cf6ae990>: 0.0002306049660170762,\n",
       " <__main__.PMCPaper at 0x7f16cf67c9d0>: 0.00023050069671085434,\n",
       " <__main__.PMCPaper at 0x7f16cf621fd0>: 0.000229631840526212,\n",
       " <__main__.PMCPaper at 0x7f16cf5d4dd0>: 0.00022910869989585335,\n",
       " <__main__.PMCPaper at 0x7f16cf5a1d90>: 0.00022838668979057141,\n",
       " <__main__.PMCPaper at 0x7f16cf5b9290>: 0.00022838380004775992,\n",
       " <__main__.PDFPaper at 0x7f16cf562d90>: 0.00022837182033912746,\n",
       " <__main__.PMCPaper at 0x7f16cf571d90>: 0.00022795151735475653,\n",
       " <__main__.PMCPaper at 0x7f16cf5369d0>: 0.00022785508757051758,\n",
       " <__main__.PDFPaper at 0x7f16cf4f8650>: 0.0002275651185896232,\n",
       " <__main__.PDFPaper at 0x7f16cf5078d0>: 0.00022744886953499692,\n",
       " <__main__.PMCPaper at 0x7f16cf4b7350>: 0.0002272129743653834,\n",
       " <__main__.PDFPaper at 0x7f16cf44c050>: 0.00022660997905849225,\n",
       " <__main__.PDFPaper at 0x7f16cf3cead0>: 0.0002259339869867494,\n",
       " <__main__.PDFPaper at 0x7f16cf3ea0d0>: 0.00022589633174502843,\n",
       " <__main__.PMCPaper at 0x7f16cf38fc90>: 0.00022575434049845777,\n",
       " <__main__.PDFPaper at 0x7f16cf3bf250>: 0.00022486172555536704,\n",
       " <__main__.PDFPaper at 0x7f16cf3605d0>: 0.00022449756963507732,\n",
       " <__main__.PDFPaper at 0x7f16cf367e90>: 0.00022446669036972684,\n",
       " <__main__.PDFPaper at 0x7f16cf331890>: 0.00022434909139704734,\n",
       " <__main__.PDFPaper at 0x7f16cf2eb110>: 0.0002242623879811513,\n",
       " <__main__.PDFPaper at 0x7f16cf2895d0>: 0.00022396683895613701,\n",
       " <__main__.PMCPaper at 0x7f16cf28ef90>: 0.00022289391429553378,\n",
       " <__main__.PDFPaper at 0x7f16cf2a3cd0>: 0.00022276054454981444,\n",
       " <__main__.PDFPaper at 0x7f16cf2ac7d0>: 0.00022274875041832317,\n",
       " <__main__.PDFPaper at 0x7f16cf276850>: 0.00022243580404302733,\n",
       " <__main__.PMCPaper at 0x7f16cf20edd0>: 0.0002214845620548407,\n",
       " <__main__.PMCPaper at 0x7f16cf22ba50>: 0.00022113313551649178,\n",
       " <__main__.PMCPaper at 0x7f16cf1ccb10>: 0.00022030647462671627,\n",
       " <__main__.PMCPaper at 0x7f16cf1fbed0>: 0.00022025630151351842,\n",
       " <__main__.PMCPaper at 0x7f16cf1a93d0>: 0.0002197379772563975,\n",
       " <__main__.PDFPaper at 0x7f16cf1c0a90>: 0.0002193892634053792,\n",
       " <__main__.PMCPaper at 0x7f16cf10fd10>: 0.000218798039674078,\n",
       " <__main__.PMCPaper at 0x7f16cf046f90>: 0.0002179498275714,\n",
       " <__main__.PDFPaper at 0x7f16cefdb5d0>: 0.00021785321723361578,\n",
       " <__main__.PDFPaper at 0x7f16cef8b990>: 0.00021782914352237267,\n",
       " <__main__.PDFPaper at 0x7f16cefa1350>: 0.00021774811791491796,\n",
       " <__main__.PMCPaper at 0x7f16cef617d0>: 0.00021773692669149755,\n",
       " <__main__.PDFPaper at 0x7f16cef81a10>: 0.0002174166509239406,\n",
       " <__main__.PDFPaper at 0x7f16d14a1250>: 0.00021715962911827673,\n",
       " <__main__.PDFPaper at 0x7f16d14c0190>: 0.00021711761474159253,\n",
       " <__main__.PDFPaper at 0x7f16d144ccd0>: 0.0002168362843995743,\n",
       " <__main__.PDFPaper at 0x7f16d141c610>: 0.00021678089076607957,\n",
       " <__main__.PDFPaper at 0x7f16d1446390>: 0.00021653089636380782,\n",
       " <__main__.PDFPaper at 0x7f16d13d9350>: 0.0002164842173407933,\n",
       " <__main__.PMCPaper at 0x7f16d139e550>: 0.0002164734644074377,\n",
       " <__main__.PMCPaper at 0x7f16d135cd50>: 0.000215561796600032,\n",
       " <__main__.PMCPaper at 0x7f16d1367810>: 0.00021546537799672558,\n",
       " <__main__.PMCPaper at 0x7f16d1388450>: 0.0002154244424280375,\n",
       " <__main__.PMCPaper at 0x7f16d1315450>: 0.0002150168158751304,\n",
       " <__main__.PMCPaper at 0x7f16d1329510>: 0.00021477127144807105,\n",
       " <__main__.PDFPaper at 0x7f16d1338510>: 0.00021472374917014887,\n",
       " <__main__.PDFPaper at 0x7f16d128cd90>: 0.00021386577084167178,\n",
       " <__main__.PMCPaper at 0x7f16d12c84d0>: 0.00021378126120857706,\n",
       " <__main__.PDFPaper at 0x7f16d1270b90>: 0.00021341744017880064,\n",
       " <__main__.PDFPaper at 0x7f16d120d750>: 0.00021304243732895578,\n",
       " <__main__.PMCPaper at 0x7f16d11d7e90>: 0.0002125493132224725,\n",
       " <__main__.PMCPaper at 0x7f16d11e3750>: 0.0002122333626738104,\n",
       " <__main__.PDFPaper at 0x7f16d11e8790>: 0.00021220703870839978,\n",
       " <__main__.PDFPaper at 0x7f16d1203a50>: 0.00021211184500753805,\n",
       " <__main__.PMCPaper at 0x7f16d113d390>: 0.00021154795891389984,\n",
       " <__main__.PDFPaper at 0x7f16d10dbc10>: 0.00021073003290692394,\n",
       " <__main__.PDFPaper at 0x7f16d1106f50>: 0.00021071389009966413,\n",
       " <__main__.PMCPaper at 0x7f16ce7376d0>: 0.00020997001083838815,\n",
       " <__main__.PDFPaper at 0x7f16ce6d3990>: 0.000209876488631339,\n",
       " <__main__.PMCPaper at 0x7f16ce6f0710>: 0.00020978356809144973,\n",
       " <__main__.PDFPaper at 0x7f16ce700490>: 0.00020952690620906192,\n",
       " <__main__.PDFPaper at 0x7f16ce650850>: 0.00020818255592220942,\n",
       " <__main__.PDFPaper at 0x7f16ce66a650>: 0.00020811413711476184,\n",
       " <__main__.PDFPaper at 0x7f16ce61fc90>: 0.00020796915742643456,\n",
       " <__main__.PMCPaper at 0x7f16ce642ed0>: 0.00020721506960077392,\n",
       " <__main__.PDFPaper at 0x7f16ce647b10>: 0.00020699535773174065,\n",
       " <__main__.PMCPaper at 0x7f16ce5fc310>: 0.00020694361400700045,\n",
       " <__main__.PMCPaper at 0x7f16ce598890>: 0.00020680958204317688,\n",
       " <__main__.PMCPaper at 0x7f16ce55a5d0>: 0.00020663321450596547,\n",
       " <__main__.PDFPaper at 0x7f16ce567cd0>: 0.00020655606712900794,\n",
       " <__main__.PMCPaper at 0x7f16ce587890>: 0.00020628470228466452,\n",
       " <__main__.PDFPaper at 0x7f16ce511610>: 0.00020621293181397578,\n",
       " <__main__.PDFPaper at 0x7f16ce542990>: 0.0002059494588805719,\n",
       " <__main__.PDFPaper at 0x7f16ce43add0>: 0.00020484823240502235,\n",
       " <__main__.PMCPaper at 0x7f16ce3de310>: 0.0002044227663489204,\n",
       " <__main__.PMCPaper at 0x7f16ce3e6b10>: 0.00020368565801518994,\n",
       " <__main__.PDFPaper at 0x7f16ce38f450>: 0.00020351178614151484,\n",
       " <__main__.PDFPaper at 0x7f16ce399bd0>: 0.00020348349688661398,\n",
       " <__main__.PMCPaper at 0x7f16ce3c03d0>: 0.0002031202271155957,\n",
       " <__main__.PDFPaper at 0x7f16ce35a950>: 0.00020306966441186288,\n",
       " <__main__.PMCPaper at 0x7f16ce36b710>: 0.00020231867224797077,\n",
       " <__main__.PDFPaper at 0x7f16ce311d90>: 0.00020228962139610265,\n",
       " <__main__.PMCPaper at 0x7f16ce2d4410>: 0.00020226616756503126,\n",
       " <__main__.PDFPaper at 0x7f16ce294a10>: 0.00020226284008766126,\n",
       " <__main__.PDFPaper at 0x7f16ce256f50>: 0.0002018482737960969,\n",
       " <__main__.PMCPaper at 0x7f16ce1dd2d0>: 0.0002014354248288361,\n",
       " <__main__.PMCPaper at 0x7f16ce1f9c90>: 0.00020139858477767874,\n",
       " <__main__.PDFPaper at 0x7f16ce193a10>: 0.00020064765212291088,\n",
       " <__main__.PDFPaper at 0x7f16ce15efd0>: 0.0001998250199171679,\n",
       " <__main__.PDFPaper at 0x7f16ce186d50>: 0.00019977055446521676,\n",
       " <__main__.PMCPaper at 0x7f16ce121910>: 0.00019939332388031178,\n",
       " <__main__.PDFPaper at 0x7f16ce12be90>: 0.00019900640955834663,\n",
       " <__main__.PDFPaper at 0x7f16ce0d0390>: 0.00019798165373081635,\n",
       " <__main__.PDFPaper at 0x7f16ce0e7250>: 0.00019798089493826054,\n",
       " <__main__.PDFPaper at 0x7f16ce1070d0>: 0.00019780687877327444,\n",
       " <__main__.PMCPaper at 0x7f16ce09b7d0>: 0.00019708234986875602,\n",
       " <__main__.PMCPaper at 0x7f16ce0c3390>: 0.00019700063497259926,\n",
       " <__main__.PDFPaper at 0x7f16ce087490>: 0.00019614126996683263,\n",
       " <__main__.PDFPaper at 0x7f16ce01f890>: 0.00019545400867652103,\n",
       " <__main__.PDFPaper at 0x7f16cdfa5cd0>: 0.00019495512705435181,\n",
       " <__main__.PMCPaper at 0x7f16cdf78690>: 0.0001946885477711154,\n",
       " <__main__.PDFPaper at 0x7f16cdf18490>: 0.00019466364472546733,\n",
       " <__main__.PMCPaper at 0x7f16cdf25ed0>: 0.000194342756220983,\n",
       " <__main__.PDFPaper at 0x7f16cdeccb10>: 0.00019413982441690477,\n",
       " <__main__.PDFPaper at 0x7f16cdec7290>: 0.00019397270562900872,\n",
       " <__main__.PDFPaper at 0x7f16cde18ad0>: 0.00019269066857228694,\n",
       " <__main__.PDFPaper at 0x7f16cde3c0d0>: 0.00019259782099947974,\n",
       " <__main__.PMCPaper at 0x7f16cdde4410>: 0.0001919262892110946,\n",
       " <__main__.PDFPaper at 0x7f16cddae090>: 0.00019123442737420428,\n",
       " <__main__.PDFPaper at 0x7f16cdd745d0>: 0.000190814059193112,\n",
       " <__main__.PDFPaper at 0x7f16cdd43850>: 0.00019079327536444734,\n",
       " <__main__.PMCPaper at 0x7f16cdc56e90>: 0.000190573318427841,\n",
       " <__main__.PDFPaper at 0x7f16cdc65e10>: 0.00019036778061862236,\n",
       " <__main__.PMCPaper at 0x7f16cdc269d0>: 0.00019028753210356938,\n",
       " <__main__.PMCPaper at 0x7f16cdc314d0>: 0.0001901093098636023,\n",
       " <__main__.PDFPaper at 0x7f16cdc3b350>: 0.00019000732113819228,\n",
       " <__main__.PMCPaper at 0x7f16cdbd40d0>: 0.0001898658345176464,\n",
       " <__main__.PMCPaper at 0x7f16cdc06890>: 0.00018965736484040504,\n",
       " <__main__.PMCPaper at 0x7f16cdbabd10>: 0.000189506443134283,\n",
       " <__main__.PMCPaper at 0x7f16cdb4a250>: 0.0001893055950510399,\n",
       " <__main__.PDFPaper at 0x7f16cdb25a50>: 0.00018894776401899457,\n",
       " <__main__.PDFPaper at 0x7f16cdb458d0>: 0.00018854961010239524,\n",
       " <__main__.PMCPaper at 0x7f16cdaea5d0>: 0.00018788350669603018,\n",
       " <__main__.PMCPaper at 0x7f16cdaf44d0>: 0.00018758035950969422,\n",
       " <__main__.PMCPaper at 0x7f16cdaa9d90>: 0.0001873701511310468,\n",
       " <__main__.PMCPaper at 0x7f16cda453d0>: 0.0001872199881853465,\n",
       " <__main__.PDFPaper at 0x7f16cd9d9050>: 0.0001870629500748121,\n",
       " <__main__.PMCPaper at 0x7f16cd92be10>: 0.0001869394017779923,\n",
       " <__main__.PDFPaper at 0x7f16cd92f490>: 0.00018689736781957742,\n",
       " <__main__.PDFPaper at 0x7f16cd8d3ad0>: 0.00018676799907208416,\n",
       " <__main__.PMCPaper at 0x7f16cd8d9fd0>: 0.00018675168110120308,\n",
       " <__main__.PDFPaper at 0x7f16cd8f2290>: 0.00018673386108237287,\n",
       " <__main__.PMCPaper at 0x7f16cd8fe590>: 0.00018668004498128402,\n",
       " <__main__.PDFPaper at 0x7f16cd89acd0>: 0.00018633788706483672,\n",
       " <__main__.PDFPaper at 0x7f16cd862490>: 0.00018625335906947198,\n",
       " <__main__.PDFPaper at 0x7f16cd808390>: 0.00018596820712639674,\n",
       " <__main__.PDFPaper at 0x7f16cd7d7090>: 0.00018563399873431535,\n",
       " <__main__.PMCPaper at 0x7f16cd79a690>: 0.00018554919887469012,\n",
       " <__main__.PMCPaper at 0x7f16cd7ad0d0>: 0.0001854630349468355,\n",
       " <__main__.PDFPaper at 0x7f16cd70d1d0>: 0.00018501973144599123,\n",
       " <__main__.PMCPaper at 0x7f16cd73df50>: 0.00018499282693817994,\n",
       " <__main__.PMCPaper at 0x7f16cd6cdb50>: 0.00018498836099804,\n",
       " <__main__.PMCPaper at 0x7f16cd6e8750>: 0.00018451137186556283,\n",
       " <__main__.PDFPaper at 0x7f16cd68e790>: 0.00018442133868224077,\n",
       " <__main__.PMCPaper at 0x7f16cd6a7750>: 0.00018410513966901529,\n",
       " <__main__.PDFPaper at 0x7f16cd6bff90>: 0.00018406270577681901,\n",
       " <__main__.PMCPaper at 0x7f16cd60a410>: 0.00018387868305830616,\n",
       " <__main__.PMCPaper at 0x7f16cd615ed0>: 0.00018375925968985596,\n",
       " <__main__.PDFPaper at 0x7f16cd61e3d0>: 0.0001836362900780525,\n",
       " <__main__.PDFPaper at 0x7f16cd5e3b50>: 0.00018349440915467322,\n",
       " <__main__.PDFPaper at 0x7f16cd555750>: 0.00018320416419570574,\n",
       " <__main__.PDFPaper at 0x7f16cd562d90>: 0.00018309116901609417,\n",
       " <__main__.PMCPaper at 0x7f16cd57c250>: 0.0001830832500653821,\n",
       " <__main__.PMCPaper at 0x7f16cd52b790>: 0.0001830408491238421,\n",
       " <__main__.PMCPaper at 0x7f16cd539910>: 0.00018133671544913956,\n",
       " <__main__.PDFPaper at 0x7f16cd4c8850>: 0.00018111918382093513,\n",
       " <__main__.PDFPaper at 0x7f16cd4f4e50>: 0.00018094738013476152,\n",
       " <__main__.PDFPaper at 0x7f16cd4483d0>: 0.00018084774278799386,\n",
       " <__main__.PMCPaper at 0x7f16cd429a90>: 0.00018078778836053244,\n",
       " <__main__.PDFPaper at 0x7f16cd3e67d0>: 0.00018071980635870767,\n",
       " <__main__.PDFPaper at 0x7f16cd4050d0>: 0.00018067588763522787,\n",
       " <__main__.PDFPaper at 0x7f16cd3be350>: 0.0001805341463601377,\n",
       " <__main__.PDFPaper at 0x7f16cd368e50>: 0.0001805293575379552,\n",
       " <__main__.PDFPaper at 0x7f16cd30e6d0>: 0.00018021156040105878,\n",
       " <__main__.PDFPaper at 0x7f16cd318c10>: 0.0001800419088795042,\n",
       " <__main__.PMCPaper at 0x7f16cd329910>: 0.0001797510048987372,\n",
       " <__main__.PMCPaper at 0x7f16cd2e4dd0>: 0.00017940184146085264,\n",
       " <__main__.PMCPaper at 0x7f16cd2f0710>: 0.00017901888279790975,\n",
       " <__main__.PDFPaper at 0x7f16cd2bb5d0>: 0.00017901723097534287,\n",
       " <__main__.PDFPaper at 0x7f16cd24f9d0>: 0.00017873892710681141,\n",
       " <__main__.PDFPaper at 0x7f16cd282a90>: 0.000178589795174871,\n",
       " <__main__.PDFPaper at 0x7f16cd210550>: 0.00017841191987024658,\n",
       " <__main__.PMCPaper at 0x7f16cd238a10>: 0.0001783466217207593,\n",
       " <__main__.PDFPaper at 0x7f16cd243b10>: 0.00017828531043931301,\n",
       " <__main__.PDFPaper at 0x7f16cd1e5310>: 0.00017822310519602927,\n",
       " <__main__.PDFPaper at 0x7f16cd1bc790>: 0.00017818661646741427,\n",
       " <__main__.PMCPaper at 0x7f16cd1c6a90>: 0.00017805836786705044,\n",
       " <__main__.PMCPaper at 0x7f16cd176110>: 0.0001778666048815102,\n",
       " <__main__.PDFPaper at 0x7f16cd1817d0>: 0.00017779170548173345,\n",
       " <__main__.PMCPaper at 0x7f16cd0d0150>: 0.00017753484385219217,\n",
       " <__main__.PMCPaper at 0x7f16cd0ed9d0>: 0.0001774239292390327,\n",
       " <__main__.PMCPaper at 0x7f16cd0f6d90>: 0.00017682451411246526,\n",
       " <__main__.PMCPaper at 0x7f16cd0957d0>: 0.0001767203286319759,\n",
       " <__main__.PDFPaper at 0x7f16cd06f0d0>: 0.00017657611124275872,\n",
       " <__main__.PDFPaper at 0x7f16cd081790>: 0.00017613258516377045,\n",
       " <__main__.PDFPaper at 0x7f16cd00d490>: 0.00017594227753880466,\n",
       " <__main__.PMCPaper at 0x7f16cd02b890>: 0.00017583091759688078,\n",
       " <__main__.PDFPaper at 0x7f16cd043b50>: 0.0001745598666316535,\n",
       " <__main__.PDFPaper at 0x7f16ccfde990>: 0.00017451313450445925,\n",
       " <__main__.PDFPaper at 0x7f16ccffd990>: 0.00017417595898955,\n",
       " <__main__.PMCPaper at 0x7f16ccf92810>: 0.00017404432706146847,\n",
       " <__main__.PMCPaper at 0x7f16ccfbab50>: 0.0001740351891534126,\n",
       " <__main__.PDFPaper at 0x7f16ccf54f90>: 0.00017378388260566255,\n",
       " <__main__.PDFPaper at 0x7f16ccf09490>: 0.00017342277341600993,\n",
       " <__main__.PMCPaper at 0x7f16cced46d0>: 0.0001730411349992573,\n",
       " <__main__.PDFPaper at 0x7f16ccf06110>: 0.0001727952202847339,\n",
       " <__main__.PDFPaper at 0x7f16ccea5410>: 0.0001727795395871617,\n",
       " <__main__.PDFPaper at 0x7f16cceb6ad0>: 0.0001727535278044984,\n",
       " <__main__.PMCPaper at 0x7f16cce4d0d0>: 0.00017247425051774028,\n",
       " <__main__.PDFPaper at 0x7f16cce6f790>: 0.00017238169928313797,\n",
       " <__main__.PMCPaper at 0x7f16ccda4ad0>: 0.00017227462152556017,\n",
       " <__main__.PMCPaper at 0x7f16ccdc0850>: 0.00017214119810967916,\n",
       " <__main__.PMCPaper at 0x7f16ccd74dd0>: 0.00017183085429478128,\n",
       " <__main__.PMCPaper at 0x7f16ccd08ed0>: 0.0001717283418399297,\n",
       " <__main__.PMCPaper at 0x7f16ccc89450>: 0.0001715300345459935,\n",
       " <__main__.PMCPaper at 0x7f16ccc9ba90>: 0.00017143379764677924,\n",
       " <__main__.PDFPaper at 0x7f16cccae150>: 0.0001711900998744239,\n",
       " <__main__.PDFPaper at 0x7f16cccbe990>: 0.000171069133333748,\n",
       " <__main__.PDFPaper at 0x7f16ccc0c950>: 0.00017057465030395318,\n",
       " <__main__.PMCPaper at 0x7f16ccc214d0>: 0.0001704801202434956,\n",
       " <__main__.PDFPaper at 0x7f16ccc06050>: 0.00017039766135763522,\n",
       " <__main__.PMCPaper at 0x7f16ccb95290>: 0.00017035469626220303,\n",
       " <__main__.PDFPaper at 0x7f16ccb95bd0>: 0.0001702720562799738,\n",
       " <__main__.PDFPaper at 0x7f16ccb5b790>: 0.00016999361381989922,\n",
       " <__main__.PDFPaper at 0x7f16ccb64050>: 0.00016994272381810272,\n",
       " <__main__.PDFPaper at 0x7f16ccb81510>: 0.00016978586389334566,\n",
       " <__main__.PMCPaper at 0x7f16ccb31f10>: 0.000169763421648082,\n",
       " <__main__.PDFPaper at 0x7f16ccb43a50>: 0.0001692100864215554,\n",
       " <__main__.PDFPaper at 0x7f16ccaa5950>: 0.00016907033215819215,\n",
       " <__main__.PDFPaper at 0x7f16cca633d0>: 0.0001688058645624247,\n",
       " <__main__.PMCPaper at 0x7f16cca81250>: 0.00016790274149121843,\n",
       " <__main__.PDFPaper at 0x7f16cca0c490>: 0.00016750323568525075,\n",
       " <__main__.PMCPaper at 0x7f16cca364d0>: 0.00016721957678533656,\n",
       " <__main__.PMCPaper at 0x7f16cca42490>: 0.00016698579501635367,\n",
       " <__main__.PMCPaper at 0x7f16cc9e2110>: 0.0001668324054233595,\n",
       " <__main__.PDFPaper at 0x7f16cc959c50>: 0.0001666069293295534,\n",
       " <__main__.PMCPaper at 0x7f16cc9199d0>: 0.00016652017681168362,\n",
       " <__main__.PMCPaper at 0x7f16cc8d4290>: 0.0001663616156222542,\n",
       " <__main__.PDFPaper at 0x7f16cc8bf190>: 0.00016633127999498347,\n",
       " <__main__.PMCPaper at 0x7f16cc834450>: 0.00016629846912987113,\n",
       " <__main__.PDFPaper at 0x7f16cc7c93d0>: 0.0001662006156400302,\n",
       " <__main__.PMCPaper at 0x7f16cc7dc350>: 0.00016585033219512085,\n",
       " <__main__.PDFPaper at 0x7f16cc7f1dd0>: 0.00016574990567679815,\n",
       " <__main__.PDFPaper at 0x7f16cc7af510>: 0.00016570916349822616,\n",
       " <__main__.PDFPaper at 0x7f16cc7bf250>: 0.00016569425122517874,\n",
       " <__main__.PDFPaper at 0x7f16cc773450>: 0.00016563573976445807,\n",
       " <__main__.PDFPaper at 0x7f16cc70c990>: 0.0001655052456561727,\n",
       " <__main__.PMCPaper at 0x7f16cc746ad0>: 0.00016549792125857644,\n",
       " <__main__.PMCPaper at 0x7f16cc6f2a10>: 0.00016544080654207002,\n",
       " <__main__.PDFPaper at 0x7f16cc64b410>: 0.00016539097923163928,\n",
       " <__main__.PDFPaper at 0x7f16cc67dfd0>: 0.00016536329132334318,\n",
       " <__main__.PMCPaper at 0x7f16cc624c10>: 0.000165178656362472,\n",
       " <__main__.PMCPaper at 0x7f16cc631a10>: 0.00016517567670340153,\n",
       " <__main__.PMCPaper at 0x7f16cc5ce390>: 0.00016515878015799885,\n",
       " <__main__.PMCPaper at 0x7f16cc5edb10>: 0.00016501346514074787,\n",
       " <__main__.PDFPaper at 0x7f16cc5a28d0>: 0.00016478596296178312,\n",
       " <__main__.PDFPaper at 0x7f16cc56ba90>: 0.0001645858102418264,\n",
       " <__main__.PDFPaper at 0x7f16cc51d190>: 0.0001645760365190871,\n",
       " <__main__.PDFPaper at 0x7f16cc53a790>: 0.00016428239069521722,\n",
       " <__main__.PDFPaper at 0x7f16cc4d4310>: 0.00016402495477159975,\n",
       " <__main__.PMCPaper at 0x7f16cc4ed1d0>: 0.00016390941768787686,\n",
       " <__main__.PMCPaper at 0x7f16cc493910>: 0.00016390909802621375,\n",
       " <__main__.PDFPaper at 0x7f16cc44f350>: 0.00016376838358728515,\n",
       " <__main__.PDFPaper at 0x7f16cc46e3d0>: 0.00016355039433095392,\n",
       " <__main__.PDFPaper at 0x7f16cc3294d0>: 0.00016325332177489498,\n",
       " <__main__.PDFPaper at 0x7f16cc339510>: 0.00016317690046377224,\n",
       " <__main__.PDFPaper at 0x7f16cc2258d0>: 0.00016312114437094877,\n",
       " <__main__.PMCPaper at 0x7f16cc1cb5d0>: 0.0001628960418616601,\n",
       " <__main__.PDFPaper at 0x7f16cc205610>: 0.00016289310505534162,\n",
       " <__main__.PDFPaper at 0x7f16cc197990>: 0.0001628360564462394,\n",
       " <__main__.PDFPaper at 0x7f16cc150cd0>: 0.00016276476422177877,\n",
       " <__main__.PDFPaper at 0x7f16cc15ec10>: 0.00016274719729222821,\n",
       " <__main__.PMCPaper at 0x7f16cc120d90>: 0.00016253379600798465,\n",
       " <__main__.PMCPaper at 0x7f16cc139210>: 0.0001622795281623997,\n",
       " <__main__.PMCPaper at 0x7f16cc0cf250>: 0.0001621673030406872,\n",
       " <__main__.PMCPaper at 0x7f16cc107ad0>: 0.00016214093284692646,\n",
       " <__main__.PDFPaper at 0x7f16cc0ae710>: 0.000162004825652453,\n",
       " <__main__.PDFPaper at 0x7f16cc05ed10>: 0.00016195790022145547,\n",
       " <__main__.PMCPaper at 0x7f16cc0100d0>: 0.00016190723922682962,\n",
       " <__main__.PDFPaper at 0x7f16cc01e7d0>: 0.00016135414290605697,\n",
       " <__main__.PMCPaper at 0x7f16cc029150>: 0.0001609609922905855,\n",
       " <__main__.PDFPaper at 0x7f16cc041d10>: 0.00016080096657160913,\n",
       " <__main__.PMCPaper at 0x7f16cbfd49d0>: 0.00016078232339379367,\n",
       " <__main__.PMCPaper at 0x7f16cbfaa5d0>: 0.0001607795212444797,\n",
       " <__main__.PDFPaper at 0x7f16cbfbcad0>: 0.0001607254649874382,\n",
       " <__main__.PMCPaper at 0x7f16cbf5af90>: 0.00016064978672952795,\n",
       " <__main__.PDFPaper at 0x7f16cbf87b10>: 0.00016064668739316726,\n",
       " <__main__.PDFPaper at 0x7f16cbee8410>: 0.0001604260444321706,\n",
       " <__main__.PDFPaper at 0x7f16cbefe0d0>: 0.00016009944448066785,\n",
       " <__main__.PMCPaper at 0x7f16cbe8f310>: 0.00015993681380590028,\n",
       " <__main__.PMCPaper at 0x7f16cbe31710>: 0.0001597201585233504,\n",
       " <__main__.PMCPaper at 0x7f16cbdec910>: 0.00015965538276105902,\n",
       " <__main__.PMCPaper at 0x7f16cbda1790>: 0.00015959542914695932,\n",
       " <__main__.PDFPaper at 0x7f16cbd7dad0>: 0.00015958982472686077,\n",
       " <__main__.PMCPaper at 0x7f16cbd171d0>: 0.0001592193963805762,\n",
       " <__main__.PMCPaper at 0x7f16cbd27310>: 0.00015916349116943442,\n",
       " <__main__.PDFPaper at 0x7f16cbcda190>: 0.00015901465686160205,\n",
       " <__main__.PMCPaper at 0x7f16cbcf4410>: 0.0001584659354865718,\n",
       " <__main__.PDFPaper at 0x7f16cbc49550>: 0.00015835390062569646,\n",
       " <__main__.PDFPaper at 0x7f16cbc73190>: 0.0001582388294924452,\n",
       " <__main__.PMCPaper at 0x7f16cbc79950>: 0.00015813014773421427,\n",
       " <__main__.PDFPaper at 0x7f16cbc202d0>: 0.00015748094899593104,\n",
       " <__main__.PMCPaper at 0x7f16cbbce190>: 0.00015738823804830378,\n",
       " <__main__.PDFPaper at 0x7f16cbbe9610>: 0.00015691209518917256,\n",
       " <__main__.PDFPaper at 0x7f16cbbc09d0>: 0.00015661515616725784,\n",
       " <__main__.PMCPaper at 0x7f16cbb67ed0>: 0.00015650470066363499,\n",
       " <__main__.PMCPaper at 0x7f16cbafb210>: 0.00015633628468524554,\n",
       " <__main__.PDFPaper at 0x7f16cbab0ed0>: 0.00015609591208264457,\n",
       " <__main__.PDFPaper at 0x7f16cba48050>: 0.00015567935018134717,\n",
       " <__main__.PDFPaper at 0x7f16cba63c50>: 0.00015567451452200368,\n",
       " <__main__.PDFPaper at 0x7f16cba7b210>: 0.00015564827345125931,\n",
       " <__main__.PMCPaper at 0x7f16cb9db0d0>: 0.00015543343067705444,\n",
       " <__main__.PMCPaper at 0x7f16cb99e290>: 0.00015536055047228413,\n",
       " <__main__.PMCPaper at 0x7f16cb9b15d0>: 0.00015534948284813573,\n",
       " <__main__.PDFPaper at 0x7f16cb97ecd0>: 0.00015532505729976185,\n",
       " <__main__.PMCPaper at 0x7f16cb916b50>: 0.00015505776534633823,\n",
       " <__main__.PMCPaper at 0x7f16cb937a90>: 0.00015383145535242596,\n",
       " <__main__.PDFPaper at 0x7f16cb8e0850>: 0.00015354879263347056,\n",
       " <__main__.PMCPaper at 0x7f16cb88f750>: 0.0001535184070591294,\n",
       " <__main__.PDFPaper at 0x7f16cb8a36d0>: 0.00015346869205119744,\n",
       " <__main__.PDFPaper at 0x7f16cb856150>: 0.00015339743305126295,\n",
       " <__main__.PDFPaper at 0x7f16cb830310>: 0.00015299729614398692,\n",
       " <__main__.PMCPaper at 0x7f16cb799dd0>: 0.0001529885623570728,\n",
       " <__main__.PDFPaper at 0x7f16cb7af290>: 0.0001529564115827461,\n",
       " <__main__.PDFPaper at 0x7f16cb761650>: 0.0001527347191831339,\n",
       " <__main__.PMCPaper at 0x7f16cb70cd10>: 0.00015268346610629045,\n",
       " <__main__.PDFPaper at 0x7f16cb712750>: 0.00015263528258454526,\n",
       " <__main__.PMCPaper at 0x7f16cb6b0450>: 0.00015258550041605346,\n",
       " <__main__.PMCPaper at 0x7f16cb6498d0>: 0.00015250602139329055,\n",
       " <__main__.PDFPaper at 0x7f16cb60d2d0>: 0.0001523677431601473,\n",
       " <__main__.PDFPaper at 0x7f16cb61b610>: 0.00015219413646923253,\n",
       " <__main__.PDFPaper at 0x7f16cb629a10>: 0.00015176479714503233,\n",
       " <__main__.PMCPaper at 0x7f16cb5d5b90>: 0.00015152857899942642,\n",
       " <__main__.PDFPaper at 0x7f16cb5e5710>: 0.00015132783592769472,\n",
       " <__main__.PDFPaper at 0x7f16cb5b14d0>: 0.0001508798090711158,\n",
       " <__main__.PDFPaper at 0x7f16cb51b350>: 0.0001506453893608882,\n",
       " <__main__.PDFPaper at 0x7f16cb53f650>: 0.00015063829784997738,\n",
       " <__main__.PMCPaper at 0x7f16cb4f8250>: 0.00015057518762731927,\n",
       " <__main__.PDFPaper at 0x7f16cb494b10>: 0.00015046822354197333,\n",
       " <__main__.PDFPaper at 0x7f16cb4b0990>: 0.0001503130077519074,\n",
       " <__main__.PMCPaper at 0x7f16cb4bcad0>: 0.00015025965954000694,\n",
       " <__main__.PDFPaper at 0x7f16cb47a490>: 0.00015022820822851546,\n",
       " <__main__.PDFPaper at 0x7f16cb438a50>: 0.0001501063146732375,\n",
       " <__main__.PMCPaper at 0x7f16cb3d1cd0>: 0.0001499529610003624,\n",
       " <__main__.PMCPaper at 0x7f16cb388b50>: 0.00014978155591838718,\n",
       " <__main__.PMCPaper at 0x7f16cb359590>: 0.0001497705108490785,\n",
       " <__main__.PMCPaper at 0x7f16cb375990>: 0.00014971500670585077,\n",
       " <__main__.PDFPaper at 0x7f16cb3833d0>: 0.00014967025847487325,\n",
       " <__main__.PMCPaper at 0x7f16cb30f110>: 0.00014950520600667285,\n",
       " <__main__.PDFPaper at 0x7f16cb3190d0>: 0.00014945626916639895,\n",
       " <__main__.PMCPaper at 0x7f16cb2ab610>: 0.00014941260410190958,\n",
       " <__main__.PMCPaper at 0x7f16cb264950>: 0.0001492154683910264,\n",
       " <__main__.PMCPaper at 0x7f16cb2161d0>: 0.00014910186729749354,\n",
       " <__main__.PMCPaper at 0x7f16cb21b8d0>: 0.0001490438583145814,\n",
       " <__main__.PDFPaper at 0x7f16cb242c50>: 0.00014902041363842427,\n",
       " <__main__.PMCPaper at 0x7f16cb1e8a10>: 0.00014866543383879461,\n",
       " <__main__.PDFPaper at 0x7f16cb1f1cd0>: 0.0001486398683816646,\n",
       " <__main__.PMCPaper at 0x7f16cb11f4d0>: 0.0001485045853719988,\n",
       " <__main__.PDFPaper at 0x7f16cb130250>: 0.00014828419088384076,\n",
       " <__main__.PDFPaper at 0x7f16cb0e1310>: 0.0001482753307752459,\n",
       " <__main__.PDFPaper at 0x7f16cb107490>: 0.0001478546941161919,\n",
       " <__main__.PMCPaper at 0x7f16cb093050>: 0.00014774551459099586,\n",
       " <__main__.PDFPaper at 0x7f16cb09ab90>: 0.00014764904449531818,\n",
       " <__main__.PDFPaper at 0x7f16cb0bcb50>: 0.00014764295406300855,\n",
       " <__main__.PDFPaper at 0x7f16cb06d350>: 0.00014747962601686687,\n",
       " <__main__.PDFPaper at 0x7f16cb07d850>: 0.0001473964878626345,\n",
       " <__main__.PDFPaper at 0x7f16cb011850>: 0.00014729431540986195,\n",
       " <__main__.PMCPaper at 0x7f16cb035290>: 0.00014677295478293306,\n",
       " <__main__.PMCPaper at 0x7f16cb043190>: 0.00014648978013523335,\n",
       " <__main__.PMCPaper at 0x7f16cafd2c10>: 0.00014632566947982864,\n",
       " <__main__.PDFPaper at 0x7f16cafe2fd0>: 0.0001462067080789766,\n",
       " <__main__.PDFPaper at 0x7f16caf913d0>: 0.00014617532782171022,\n",
       " <__main__.PMCPaper at 0x7f16cafa9ed0>: 0.00014614645581740364,\n",
       " <__main__.PMCPaper at 0x7f16caf5ab90>: 0.00014609757779337094,\n",
       " <__main__.PMCPaper at 0x7f16caf7cf10>: 0.0001459802498954956,\n",
       " <__main__.PDFPaper at 0x7f16caf085d0>: 0.00014591998082544476,\n",
       " <__main__.PDFPaper at 0x7f16cae99590>: 0.0001457789745721174,\n",
       " <__main__.PDFPaper at 0x7f16caeb29d0>: 0.00014566316357691938,\n",
       " <__main__.PDFPaper at 0x7f16caec6bd0>: 0.00014564577478575706,\n",
       " <__main__.PDFPaper at 0x7f16cad90490>: 0.0001456183553354145,\n",
       " <__main__.PDFPaper at 0x7f16cadaa610>: 0.00014559850141848988,\n",
       " <__main__.PMCPaper at 0x7f16cad64110>: 0.00014549892047919136,\n",
       " <__main__.PMCPaper at 0x7f16cad6a850>: 0.00014536843827242152,\n",
       " <__main__.PMCPaper at 0x7f16cad17a10>: 0.0001453510471402598,\n",
       " <__main__.PDFPaper at 0x7f16cad25a50>: 0.0001450975756598851,\n",
       " <__main__.PDFPaper at 0x7f16cacd2890>: 0.00014487517939870627,\n",
       " <__main__.PMCPaper at 0x7f16caa43cd0>: 0.00014474951914257507,\n",
       " <__main__.PMCPaper at 0x7f16caa06c50>: 0.0001445458197086978,\n",
       " <__main__.PMCPaper at 0x7f16ca9a9e90>: 0.00014429508901131016,\n",
       " <__main__.PDFPaper at 0x7f16ca9b3c90>: 0.0001441750495253916,\n",
       " <__main__.PDFPaper at 0x7f16ca950990>: 0.00014407146249417646,\n",
       " <__main__.PMCPaper at 0x7f16ca9830d0>: 0.00014403432075860836,\n",
       " <__main__.PMCPaper at 0x7f16ca9168d0>: 0.00014347560057252634,\n",
       " <__main__.PDFPaper at 0x7f16ca936110>: 0.00014332578721347786,\n",
       " <__main__.PDFPaper at 0x7f16ca8e0c10>: 0.00014299900876973238,\n",
       " <__main__.PMCPaper at 0x7f16ca906dd0>: 0.0001428883982913362,\n",
       " <__main__.PDFPaper at 0x7f16ca8b23d0>: 0.00014235744726451356,\n",
       " <__main__.PMCPaper at 0x7f16ca849450>: 0.0001421321363417848,\n",
       " <__main__.PDFPaper at 0x7f16ca85bed0>: 0.0001417964814117037,\n",
       " <__main__.PDFPaper at 0x7f16ca87fad0>: 0.00014168780894535866,\n",
       " <__main__.PDFPaper at 0x7f16ca843710>: 0.0001416702625898981,\n",
       " <__main__.PDFPaper at 0x7f16ca7a1450>: 0.00014160369498274975,\n",
       " <__main__.PDFPaper at 0x7f16ca754c90>: 0.0001413238854988084,\n",
       " <__main__.PMCPaper at 0x7f16ca775610>: 0.00014131765112974617,\n",
       " <__main__.PDFPaper at 0x7f16ca713d50>: 0.00014117304073327753,\n",
       " <__main__.PMCPaper at 0x7f16ca6f6150>: 0.0001410469169774815,\n",
       " <__main__.PMCPaper at 0x7f16ca702290>: 0.00014104370062014178,\n",
       " <__main__.PDFPaper at 0x7f16ca6b3090>: 0.0001406666255014915,\n",
       " <__main__.PMCPaper at 0x7f16ca6bc590>: 0.00014022946871634348,\n",
       " <__main__.PDFPaper at 0x7f16ca6c60d0>: 0.00013939554497385912,\n",
       " <__main__.PDFPaper at 0x7f16ca65a110>: 0.00013938649256643455,\n",
       " <__main__.PMCPaper at 0x7f16ca678350>: 0.00013928468034716648,\n",
       " <__main__.PMCPaper at 0x7f16ca61b650>: 0.00013924053948744536,\n",
       " <__main__.PDFPaper at 0x7f16ca5c8350>: 0.00013916712509534269,\n",
       " <__main__.PMCPaper at 0x7f16ca5ef7d0>: 0.00013886640099737002,\n",
       " <__main__.PMCPaper at 0x7f16ca58ded0>: 0.00013863227137655675,\n",
       " <__main__.PDFPaper at 0x7f16ca559910>: 0.00013856332530912813,\n",
       " <__main__.PDFPaper at 0x7f16ca56b650>: 0.00013843315829940568,\n",
       " <__main__.PDFPaper at 0x7f16ca4ed410>: 0.00013829038035377073,\n",
       " <__main__.PDFPaper at 0x7f16ca49f790>: 0.00013796657514289102,\n",
       " <__main__.PMCPaper at 0x7f16cef2a550>: 0.0001378940325561377,\n",
       " <__main__.PMCPaper at 0x7f16ceed6ad0>: 0.00013782132032824106,\n",
       " <__main__.PMCPaper at 0x7f16ceeec150>: 0.00013749556323120642,\n",
       " <__main__.PDFPaper at 0x7f16ceeb0a50>: 0.00013731890003438848,\n",
       " <__main__.PDFPaper at 0x7f16cee7a310>: 0.0001371127180784227,\n",
       " <__main__.PDFPaper at 0x7f16cee84650>: 0.00013698803721063426,\n",
       " <__main__.PMCPaper at 0x7f16cedee150>: 0.0001368611375228577,\n",
       " <__main__.PDFPaper at 0x7f16cedfa3d0>: 0.0001368548212787505,\n",
       " <__main__.PMCPaper at 0x7f16cedad2d0>: 0.00013672775605094834,\n",
       " <__main__.PMCPaper at 0x7f16ced4fed0>: 0.00013650849083257562,\n",
       " <__main__.PDFPaper at 0x7f16ced02390>: 0.00013638838282468938,\n",
       " <__main__.PDFPaper at 0x7f16cecb4910>: 0.0001362112162123614,\n",
       " <__main__.PDFPaper at 0x7f16cec6e210>: 0.00013613846857993088,\n",
       " <__main__.PMCPaper at 0x7f16cec25710>: 0.00013607692905086364,\n",
       " <__main__.PDFPaper at 0x7f16cec03550>: 0.00013600192980407267,\n",
       " <__main__.PDFPaper at 0x7f16ceb695d0>: 0.00013587097608327818,\n",
       " <__main__.PMCPaper at 0x7f16ceb14150>: 0.00013583829801163186,\n",
       " <__main__.PDFPaper at 0x7f16ceadd2d0>: 0.00013578170250766404,\n",
       " <__main__.PMCPaper at 0x7f16cea9dfd0>: 0.0001353356174552047,\n",
       " <__main__.PMCPaper at 0x7f16ceaa2750>: 0.00013519185402524947,\n",
       " <__main__.PDFPaper at 0x7f16cea62750>: 0.00013516113924520536,\n",
       " <__main__.PDFPaper at 0x7f16cea23ad0>: 0.0001348082486980219,\n",
       " <__main__.PDFPaper at 0x7f16cea31bd0>: 0.0001347405174961483,\n",
       " <__main__.PDFPaper at 0x7f16cea46b50>: 0.0001345595537454169,\n",
       " <__main__.PDFPaper at 0x7f16ce9b9e90>: 0.00013401933832612577,\n",
       " <__main__.PMCPaper at 0x7f16ce9778d0>: 0.00013391491180366873,\n",
       " <__main__.PDFPaper at 0x7f16ce945150>: 0.00013390263950143302,\n",
       " <__main__.PDFPaper at 0x7f16ce862550>: 0.00013384253440592276,\n",
       " <__main__.PDFPaper at 0x7f16ce878290>: 0.00013378515988745108,\n",
       " <__main__.PMCPaper at 0x7f16ce80f190>: 0.00013361295618088598,\n",
       " <__main__.PMCPaper at 0x7f16ce845c50>: 0.00013325760768555535,\n",
       " <__main__.PDFPaper at 0x7f16ce801890>: 0.00013316857807462102,\n",
       " <__main__.PDFPaper at 0x7f16ce74b7d0>: 0.0001328721973725045,\n",
       " <__main__.PMCPaper at 0x7f16ce7708d0>: 0.0001325559506810036,\n",
       " <__main__.PMCPaper at 0x7f16ce780810>: 0.00013253277698284925,\n",
       " <__main__.PDFPaper at 0x7f16ce780d10>: 0.00013253108770726985,\n",
       " <__main__.PMCPaper at 0x7f16c9443990>: 0.00013230276738249267,\n",
       " <__main__.PDFPaper at 0x7f16c939b510>: 0.00013222974720647346,\n",
       " <__main__.PMCPaper at 0x7f16c93b6d10>: 0.0001322190952623152,\n",
       " <__main__.PMCPaper at 0x7f16c93509d0>: 0.00013194740986649918,\n",
       " <__main__.PMCPaper at 0x7f16c936e490>: 0.00013187526938631777,\n",
       " <__main__.PMCPaper at 0x7f16c9311b90>: 0.00013181681295254262,\n",
       " <__main__.PDFPaper at 0x7f16c92f3dd0>: 0.00013180690086557083,\n",
       " <__main__.PMCPaper at 0x7f16c929a5d0>: 0.00013179939539395577,\n",
       " <__main__.PDFPaper at 0x7f16c91dd690>: 0.00013171353582556227,\n",
       " <__main__.PDFPaper at 0x7f16c91fb150>: 0.00013168760556334476,\n",
       " <__main__.PMCPaper at 0x7f16c918ea50>: 0.00013150911442204503,\n",
       " <__main__.PMCPaper at 0x7f16c9196c10>: 0.0001309629968340377,\n",
       " <__main__.PMCPaper at 0x7f16c91c5590>: 0.00013066634503851324,\n",
       " <__main__.PDFPaper at 0x7f16c9156710>: 0.00013061977625872503,\n",
       " <__main__.PMCPaper at 0x7f16c9174e50>: 0.00013047488099326906,\n",
       " <__main__.PMCPaper at 0x7f16c9145dd0>: 0.00013031834349444294,\n",
       " <__main__.PDFPaper at 0x7f16c90bca10>: 0.00013011579841544724,\n",
       " <__main__.PMCPaper at 0x7f16c904bd10>: 0.00012988898188592778,\n",
       " <__main__.PMCPaper at 0x7f16c90709d0>: 0.00012970577062744186,\n",
       " <__main__.PMCPaper at 0x7f16c9084c90>: 0.00012963399974052638,\n",
       " <__main__.PMCPaper at 0x7f16c8fc7750>: 0.00012961418239980328,\n",
       " <__main__.PDFPaper at 0x7f16c8fd5c50>: 0.00012960057446105563,\n",
       " <__main__.PDFPaper at 0x7f16c8f9cfd0>: 0.00012959345356541217,\n",
       " <__main__.PDFPaper at 0x7f16c8f67a10>: 0.00012948094337984306,\n",
       " <__main__.PMCPaper at 0x7f16c8f35a10>: 0.0001292755871793678,\n",
       " <__main__.PMCPaper at 0x7f16c8ece5d0>: 0.00012913941535936763,\n",
       " <__main__.PMCPaper at 0x7f16c8ed8f10>: 0.00012875185738665156,\n",
       " <__main__.PMCPaper at 0x7f16c8f04750>: 0.0001287221070178837,\n",
       " <__main__.PDFPaper at 0x7f16c8eab810>: 0.00012867260927737438,\n",
       " <__main__.PMCPaper at 0x7f16c8ec3dd0>: 0.00012862591324806733,\n",
       " <__main__.PMCPaper at 0x7f16c8e6f710>: 0.00012861317713025653,\n",
       " <__main__.PMCPaper at 0x7f16c8e79310>: 0.0001285759562636644,\n",
       " <__main__.PDFPaper at 0x7f16c8e25ad0>: 0.0001285702647950959,\n",
       " <__main__.PMCPaper at 0x7f16c8dcdfd0>: 0.00012833693945744112,\n",
       " <__main__.PMCPaper at 0x7f16c8dfc790>: 0.00012797952388616306,\n",
       " <__main__.PDFPaper at 0x7f16c8dc0090>: 0.0001279529585804687,\n",
       " <__main__.PMCPaper at 0x7f16c8d35110>: 0.00012774036505302773,\n",
       " <__main__.PDFPaper at 0x7f16c8d412d0>: 0.00012770902032612375,\n",
       " <__main__.PDFPaper at 0x7f16c8cff210>: 0.000127642442722439,\n",
       " <__main__.PMCPaper at 0x7f16c8cb3250>: 0.00012740165918470543,\n",
       " <__main__.PMCPaper at 0x7f16c8c59190>: 0.0001273530413271133,\n",
       " <__main__.PDFPaper at 0x7f16c8c70210>: 0.000127294398969695,\n",
       " <__main__.PDFPaper at 0x7f16c8bdba10>: 0.0001270750232759964,\n",
       " <__main__.PDFPaper at 0x7f16c8bf1950>: 0.00012694852525116634,\n",
       " <__main__.PDFPaper at 0x7f16c8b88e10>: 0.0001268138946688978,\n",
       " <__main__.PMCPaper at 0x7f16c8bbcb50>: 0.0001263441050265766,\n",
       " <__main__.PMCPaper at 0x7f16c8a69150>: 0.00012620324551195628,\n",
       " <__main__.PDFPaper at 0x7f16c8a37910>: 0.00012609256520856925,\n",
       " <__main__.PDFPaper at 0x7f16c89d47d0>: 0.0001254999546818977,\n",
       " <__main__.PMCPaper at 0x7f16c89fa510>: 0.00012538404073019502,\n",
       " <__main__.PDFPaper at 0x7f16c89b0350>: 0.000125182217255126,\n",
       " <__main__.PMCPaper at 0x7f16c88970d0>: 0.00012517670103781182,\n",
       " <__main__.PMCPaper at 0x7f16c88ac150>: 0.0001250854788108161,\n",
       " <__main__.PMCPaper at 0x7f16c8847150>: 0.00012501165366211306,\n",
       " <__main__.PDFPaper at 0x7f16c88637d0>: 0.00012492793258792875,\n",
       " <__main__.PMCPaper at 0x7f16c8876110>: 0.00012492627131074438,\n",
       " <__main__.PDFPaper at 0x7f16c8831e10>: 0.0001248998371851858,\n",
       " <__main__.PDFPaper at 0x7f16c87ced90>: 0.00012480162200887432,\n",
       " <__main__.PMCPaper at 0x7f16c8806a10>: 0.0001247823301697809,\n",
       " <__main__.PMCPaper at 0x7f16c879add0>: 0.00012472370953274506,\n",
       " <__main__.PDFPaper at 0x7f16c87c6650>: 0.00012441290980886292,\n",
       " <__main__.PMCPaper at 0x7f16c875f550>: 0.00012436852789196124,\n",
       " <__main__.PMCPaper at 0x7f16c8769f90>: 0.0001242234942947281,\n",
       " <__main__.PDFPaper at 0x7f16c8722790>: 0.00012392198280072593,\n",
       " <__main__.PMCPaper at 0x7f16c86e3810>: 0.00012370386198059418,\n",
       " <__main__.PMCPaper at 0x7f16c86fbf90>: 0.00012369205424588977,\n",
       " <__main__.PDFPaper at 0x7f16c86bb790>: 0.0001236257672235121,\n",
       " <__main__.PDFPaper at 0x7f16c8652e50>: 0.00012360939010973654,\n",
       " <__main__.PDFPaper at 0x7f16c8679410>: 0.0001235929976081602,\n",
       " <__main__.PDFPaper at 0x7f16c8621c50>: 0.00012355391866135756,\n",
       " <__main__.PDFPaper at 0x7f16c85cf190>: 0.00012350019220883454,\n",
       " <__main__.PMCPaper at 0x7f16c85f69d0>: 0.0001234901434356241,\n",
       " <__main__.PDFPaper at 0x7f16c8577250>: 0.00012345156530651602,\n",
       " <__main__.PDFPaper at 0x7f16c851b490>: 0.00012336787288453043,\n",
       " <__main__.PDFPaper at 0x7f16c85383d0>: 0.00012318884388176402,\n",
       " <__main__.PDFPaper at 0x7f16c84d1650>: 0.0001228621099706913,\n",
       " <__main__.PDFPaper at 0x7f16c84d6c10>: 0.00012284876578605685,\n",
       " <__main__.PMCPaper at 0x7f16c8501a50>: 0.00012269932762916461,\n",
       " <__main__.PMCPaper at 0x7f16c845aed0>: 0.00012246799908561328,\n",
       " <__main__.PMCPaper at 0x7f16c846d4d0>: 0.00012240263715133237,\n",
       " <__main__.PMCPaper at 0x7f16c84255d0>: 0.00012230618387423946,\n",
       " <__main__.PMCPaper at 0x7f16c8432510>: 0.00012223290653295157,\n",
       " <__main__.PDFPaper at 0x7f16c8438a90>: 0.00012218192453756455,\n",
       " <__main__.PDFPaper at 0x7f16c83fc0d0>: 0.00012201162512960125,\n",
       " <__main__.PDFPaper at 0x7f16c83980d0>: 0.00012194308459838798,\n",
       " <__main__.PDFPaper at 0x7f16c83ba1d0>: 0.00012177411736337931,\n",
       " <__main__.PDFPaper at 0x7f16c835d290>: 0.00012167559082051784,\n",
       " <__main__.PMCPaper at 0x7f16c837b950>: 0.00012156868691516498,\n",
       " <__main__.PMCPaper at 0x7f16c8321410>: 0.00012154875013335013,\n",
       " <__main__.PMCPaper at 0x7f16c82fcdd0>: 0.00012153669453155377,\n",
       " <__main__.PDFPaper at 0x7f16c828b810>: 0.00012149686489979911,\n",
       " <__main__.PMCPaper at 0x7f16c829ee90>: 0.00012146550829016423,\n",
       " <__main__.PDFPaper at 0x7f16c8257d50>: 0.00012144416202084245,\n",
       " <__main__.PDFPaper at 0x7f16c81e5690>: 0.00012137035149102369,\n",
       " <__main__.PMCPaper at 0x7f16c81f1890>: 0.00012114205534165917,\n",
       " <__main__.PMCPaper at 0x7f16c8149090>: 0.00012112245083131616,\n",
       " <__main__.PDFPaper at 0x7f16c8117510>: 0.00012106354865218713,\n",
       " <__main__.PDFPaper at 0x7f16c813bcd0>: 0.00012104422922307437,\n",
       " <__main__.PMCPaper at 0x7f16c80a7610>: 0.00012103632360623645,\n",
       " <__main__.PDFPaper at 0x7f16c8048310>: 0.00012096585569889895,\n",
       " <__main__.PDFPaper at 0x7f16c80844d0>: 0.0001208282672828304,\n",
       " <__main__.PMCPaper at 0x7f16c7fc98d0>: 0.00012076784888921426,\n",
       " <__main__.PMCPaper at 0x7f16c7fd0110>: 0.00012065417107199831,\n",
       " <__main__.PDFPaper at 0x7f16c7fda7d0>: 0.00012044107964135877,\n",
       " <__main__.PDFPaper at 0x7f16c7fffc50>: 0.00012031200315720125,\n",
       " <__main__.PDFPaper at 0x7f16c7f90e90>: 0.00012029583183614977,\n",
       " <__main__.PDFPaper at 0x7f16c7f0e110>: 0.00012025591668875602,\n",
       " <__main__.PDFPaper at 0x7f16c7f18690>: 0.00012023584271800796,\n",
       " <__main__.PMCPaper at 0x7f16c7edccd0>: 0.00012006642927382601,\n",
       " <__main__.PMCPaper at 0x7f16c7ee7810>: 0.00011984947402832616,\n",
       " <__main__.PDFPaper at 0x7f16c7ef6750>: 0.00011977091852034765,\n",
       " <__main__.PMCPaper at 0x7f16c7e47bd0>: 0.0001197477912881639,\n",
       " <__main__.PDFPaper at 0x7f16c7e7a990>: 0.00011974594279312814,\n",
       " <__main__.PDFPaper at 0x7f16c7dcb110>: 0.00011968172897597906,\n",
       " <__main__.PMCPaper at 0x7f16c7d97290>: 0.00011927560379379365,\n",
       " <__main__.PDFPaper at 0x7f16c7d9cb90>: 0.00011924703930102553,\n",
       " <__main__.PDFPaper at 0x7f16c7d7a910>: 0.00011901836173004095,\n",
       " <__main__.PMCPaper at 0x7f16c7d1c450>: 0.00011888040537292146,\n",
       " <__main__.PMCPaper at 0x7f16c7d27450>: 0.00011882601528752474,\n",
       " <__main__.PMCPaper at 0x7f16c7d3ad10>: 0.00011875380925240905,\n",
       " <__main__.PMCPaper at 0x7f16c7d3f490>: 0.00011871167545886195,\n",
       " <__main__.PDFPaper at 0x7f16c7c89a90>: 0.0001186472293647305,\n",
       " <__main__.PDFPaper at 0x7f16c7cad590>: 0.00011862972906410727,\n",
       " <__main__.PMCPaper at 0x7f16c7c856d0>: 0.00011861153844731279,\n",
       " <__main__.PDFPaper at 0x7f16c7c107d0>: 0.00011859988733597889,\n",
       " <__main__.PDFPaper at 0x7f16c7c35a90>: 0.00011844153703259618,\n",
       " <__main__.PMCPaper at 0x7f16c7bd68d0>: 0.00011828963900106774,\n",
       " <__main__.PDFPaper at 0x7f16c7be3350>: 0.00011811708643805533,\n",
       " <__main__.PDFPaper at 0x7f16c7ba6b10>: 0.00011810150501689676,\n",
       " <__main__.PDFPaper at 0x7f16c7b4f4d0>: 0.00011809615368057094,\n",
       " <__main__.PDFPaper at 0x7f16c7b5d990>: 0.00011801696019130867,\n",
       " <__main__.PDFPaper at 0x7f16c7b76650>: 0.00011791815058010407,\n",
       " <__main__.PDFPaper at 0x7f16c7b1de90>: 0.0001178910683292386,\n",
       " <__main__.PDFPaper at 0x7f16c7ac88d0>: 0.00011774744307791021,\n",
       " <__main__.PDFPaper at 0x7f16c7a9d350>: 0.00011753348258790776,\n",
       " <__main__.PMCPaper at 0x7f16c7aace10>: 0.0001173598441365046,\n",
       " <__main__.PMCPaper at 0x7f16c7ac2cd0>: 0.0001173321458294304,\n",
       " <__main__.PMCPaper at 0x7f16c7a3c050>: 0.00011721583268361246,\n",
       " <__main__.PDFPaper at 0x7f16c79f3750>: 0.00011706845250563164,\n",
       " <__main__.PDFPaper at 0x7f16c7998850>: 0.00011696723835952886,\n",
       " <__main__.PDFPaper at 0x7f16c796aed0>: 0.00011695819285047515,\n",
       " <__main__.PMCPaper at 0x7f16c7851c50>: 0.00011675777061159258,\n",
       " <__main__.PDFPaper at 0x7f16c7881dd0>: 0.00011672177264289724,\n",
       " <__main__.PMCPaper at 0x7f16c782c950>: 0.00011662169253628703,\n",
       " <__main__.PDFPaper at 0x7f16c783f390>: 0.00011644068778123227,\n",
       " <__main__.PDFPaper at 0x7f16c7801410>: 0.00011643521989189424,\n",
       " <__main__.PMCPaper at 0x7f16c77b5990>: 0.00011641858570902224,\n",
       " <__main__.PDFPaper at 0x7f16c775e4d0>: 0.00011635076424127387,\n",
       " <__main__.PMCPaper at 0x7f16c7735550>: 0.00011617766202786888,\n",
       " <__main__.PMCPaper at 0x7f16c76d2490>: 0.00011606981214421557,\n",
       " <__main__.PMCPaper at 0x7f16c768b090>: 0.00011605059380730518,\n",
       " <__main__.PMCPaper at 0x7f16c76bfb10>: 0.00011604189852334906,\n",
       " <__main__.PDFPaper at 0x7f16c765b310>: 0.00011603694890027515,\n",
       " <__main__.PDFPaper at 0x7f16c7627390>: 0.00011600518593274435,\n",
       " <__main__.PMCPaper at 0x7f16c7642450>: 0.00011590279248307748,\n",
       " <__main__.PDFPaper at 0x7f16c7590050>: 0.00011585514657940877,\n",
       " <__main__.PDFPaper at 0x7f16c75b3650>: 0.00011582070289042052,\n",
       " <__main__.PMCPaper at 0x7f16c755e210>: 0.00011580563353609679,\n",
       " <__main__.PMCPaper at 0x7f16c7507bd0>: 0.00011579105427860895,\n",
       " <__main__.PDFPaper at 0x7f16c750b150>: 0.00011579105427860895,\n",
       " <__main__.PDFPaper at 0x7f16c7512810>: 0.00011578922929300049,\n",
       " <__main__.PDFPaper at 0x7f16c74d8e50>: 0.00011573056980892733,\n",
       " <__main__.PDFPaper at 0x7f16c74feb90>: 0.00011570625673473049,\n",
       " <__main__.PDFPaper at 0x7f16c74aaf10>: 0.00011563238662624935,\n",
       " <__main__.PMCPaper at 0x7f16c7447d10>: 0.00011555968285898208,\n",
       " <__main__.PMCPaper at 0x7f16c7407890>: 0.0001154173101635598,\n",
       " <__main__.PDFPaper at 0x7f16c740c0d0>: 0.00011517254789624744,\n",
       " <__main__.PMCPaper at 0x7f16c73cb0d0>: 0.0001150888847078743,\n",
       " <__main__.PDFPaper at 0x7f16c738e310>: 0.00011489044376360828,\n",
       " <__main__.PDFPaper at 0x7f16c73ba4d0>: 0.00011472851592305653,\n",
       " <__main__.PMCPaper at 0x7f16c7353110>: 0.00011462182028282384,\n",
       " <__main__.PMCPaper at 0x7f16c7367c90>: 0.00011450789479754036,\n",
       " <__main__.PDFPaper at 0x7f16c7317a10>: 0.00011447131618740008,\n",
       " <__main__.PMCPaper at 0x7f16c731ce50>: 0.00011435924621481252,\n",
       " <__main__.PMCPaper at 0x7f16c7333890>: 0.00011434202091213576,\n",
       " <__main__.PMCPaper at 0x7f16c72c8cd0>: 0.00011417151171578598,\n",
       " <__main__.PMCPaper at 0x7f16c72d0f90>: 0.00011413481898094303,\n",
       " <__main__.PMCPaper at 0x7f16c72a2610>: 0.00011389120372694169,\n",
       " <__main__.PDFPaper at 0x7f16c726fbd0>: 0.00011381021010747255,\n",
       " <__main__.PDFPaper at 0x7f16c721d1d0>: 0.00011376200003048839,\n",
       " <__main__.PMCPaper at 0x7f16c7229810>: 0.00011345479569935065,\n",
       " <__main__.PMCPaper at 0x7f16c71e4090>: 0.00011342055188608163,\n",
       " <__main__.PDFPaper at 0x7f16c71e4590>: 0.00011328956547068826,\n",
       " <__main__.PMCPaper at 0x7f16c71fad90>: 0.00011315181503468257,\n",
       " <__main__.PMCPaper at 0x7f16c71b9750>: 0.00011302829870343433,\n",
       " <__main__.PDFPaper at 0x7f16c71497d0>: 0.0001129343679402339,\n",
       " <__main__.PMCPaper at 0x7f16c70e5210>: 0.00011286517929625109,\n",
       " <__main__.PDFPaper at 0x7f16c70ea7d0>: 0.00011284475233499455,\n",
       " <__main__.PMCPaper at 0x7f16c708a2d0>: 0.00011284191821579588,\n",
       " <__main__.PMCPaper at 0x7f16c709ca50>: 0.00011283835629766634,\n",
       " <__main__.PDFPaper at 0x7f16c70a4c50>: 0.00011281993298625326,\n",
       " <__main__.PDFPaper at 0x7f16c7060290>: 0.00011277614345574169,\n",
       " <__main__.PDFPaper at 0x7f16c706fcd0>: 0.00011277432104797313,\n",
       " <__main__.PMCPaper at 0x7f16c7081610>: 0.00011244916877555096,\n",
       " <__main__.PMCPaper at 0x7f16c7016650>: 0.0001123483048669389,\n",
       " <__main__.PDFPaper at 0x7f16c70318d0>: 0.00011228175649604386,\n",
       " <__main__.PDFPaper at 0x7f16c6fdc050>: 0.00011224368564205484,\n",
       " <__main__.PDFPaper at 0x7f16c6fec190>: 0.00011223860272450325,\n",
       " ...}"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pickle\n",
    "\n",
    "class BasePaper:\n",
    "    def __init__(self, metadata_row, file_path):\n",
    "        self._metadata_row = metadata_row\n",
    "        self._file_path = file_path\n",
    "        self._file_contents = self._load_json_contents(file_path)\n",
    "        \n",
    "        self._referenced_by = []\n",
    "        self._references = []\n",
    "    \n",
    "    def __getstate__(self):\n",
    "        \"\"\"\n",
    "        Avoid RecursionErrors by not pickling references.\n",
    "        \"\"\"\n",
    "        state = self.__dict__.copy()\n",
    "        del state[\"_referenced_by\"]\n",
    "        del state[\"_references\"]\n",
    "        return state\n",
    "    \n",
    "    def __setstate__(self, state):\n",
    "        self.__dict__.update(state)\n",
    "        self._referenced_by = []\n",
    "        self._references = []\n",
    "        \n",
    "    @staticmethod\n",
    "    def _load_json_contents(path):\n",
    "        with open(path) as file:\n",
    "            contents = json.load(file)\n",
    "        return contents\n",
    "\n",
    "    @property\n",
    "    def title(self):\n",
    "        return self._metadata_row[\"title\"]\n",
    "        \n",
    "    @property\n",
    "    def authors(self):\n",
    "        return self._metadata_row[\"authors\"]\n",
    "        \n",
    "    @property\n",
    "    def publish_time(self):\n",
    "        return self._metadata_row[\"publish_time\"]\n",
    "        \n",
    "    @property\n",
    "    def abstract(self):\n",
    "        return self._metadata_row[\"abstract\"]\n",
    "        \n",
    "    @property\n",
    "    def bib_entries(self):\n",
    "        return self._file_contents[\"bib_entries\"]\n",
    "    \n",
    "    def register_reference(self, reference):\n",
    "        self._references.append(reference)\n",
    "        reference.register_referenced(self)\n",
    "    \n",
    "    def register_referenced(self, referenced):\n",
    "        self._referenced_by.append(referenced)\n",
    "    \n",
    "    \n",
    "class PDFPaper(BasePaper):\n",
    "    pass\n",
    "        \n",
    "\n",
    "class PMCPaper(BasePaper):\n",
    "    pass\n",
    "\n",
    "with open(\"./pageranks.pkl\", \"rb\") as dump_file:\n",
    "    pageranks = pickle.load(dump_file)\n",
    "\n",
    "pageranks"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Representaciones de papers\n",
    "\n",
    "A fin de construir las representaciones de los papers, se usarán las siguientes librerías:\n",
    "\n",
    "- spaCy: https://spacy.io/\n",
    "- scispaCy: https://allenai.github.io/scispacy/\n",
    "\n",
    "El modelo de lenguaje a usar será el denominado `en_core_sci_sm`, que corresponde a lenguaje biomédico con un vocabulario de aproximadamente 100.000 palabras.\n",
    "Existen modelos con vocabularios más grandes en caso de ser necesario."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: spacy in ./venv/lib/python3.7/site-packages (2.2.4)\n",
      "Requirement already satisfied: scispacy in ./venv/lib/python3.7/site-packages (0.2.4)\n",
      "Requirement already satisfied: srsly<1.1.0,>=1.0.2 in ./venv/lib/python3.7/site-packages (from spacy) (1.0.2)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in ./venv/lib/python3.7/site-packages (from spacy) (1.0.2)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.13.0 in ./venv/lib/python3.7/site-packages (from spacy) (2.23.0)\n",
      "Requirement already satisfied: catalogue<1.1.0,>=0.0.7 in ./venv/lib/python3.7/site-packages (from spacy) (1.0.0)\n",
      "Requirement already satisfied: numpy>=1.15.0 in ./venv/lib/python3.7/site-packages (from spacy) (1.18.2)\n",
      "Requirement already satisfied: wasabi<1.1.0,>=0.4.0 in ./venv/lib/python3.7/site-packages (from spacy) (0.6.0)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in ./venv/lib/python3.7/site-packages (from spacy) (2.0.3)\n",
      "Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in ./venv/lib/python3.7/site-packages (from spacy) (4.45.0)\n",
      "Requirement already satisfied: blis<0.5.0,>=0.4.0 in ./venv/lib/python3.7/site-packages (from spacy) (0.4.1)\n",
      "Requirement already satisfied: setuptools in ./venv/lib/python3.7/site-packages (from spacy) (40.8.0)\n",
      "Requirement already satisfied: plac<1.2.0,>=0.9.6 in ./venv/lib/python3.7/site-packages (from spacy) (1.1.3)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in ./venv/lib/python3.7/site-packages (from spacy) (3.0.2)\n",
      "Requirement already satisfied: thinc==7.4.0 in ./venv/lib/python3.7/site-packages (from spacy) (7.4.0)\n",
      "Requirement already satisfied: awscli in ./venv/lib/python3.7/site-packages (from scispacy) (1.18.45)\n",
      "Requirement already satisfied: conllu in ./venv/lib/python3.7/site-packages (from scispacy) (2.3.2)\n",
      "Requirement already satisfied: joblib in ./venv/lib/python3.7/site-packages (from scispacy) (0.14.1)\n",
      "Requirement already satisfied: nmslib>=1.7.3.6 in ./venv/lib/python3.7/site-packages (from scispacy) (2.0.6)\n",
      "Requirement already satisfied: scikit-learn>=0.20.3 in ./venv/lib/python3.7/site-packages (from scispacy) (0.22.2.post1)\n",
      "Requirement already satisfied: pysbd in ./venv/lib/python3.7/site-packages (from scispacy) (0.2.3)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in ./venv/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy) (2020.4.5.1)\n",
      "Requirement already satisfied: idna<3,>=2.5 in ./venv/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy) (2.9)\n",
      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in ./venv/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy) (1.25.9)\n",
      "Requirement already satisfied: chardet<4,>=3.0.2 in ./venv/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy) (3.0.4)\n",
      "Requirement already satisfied: importlib-metadata>=0.20; python_version < \"3.8\" in ./venv/lib/python3.7/site-packages (from catalogue<1.1.0,>=0.0.7->spacy) (1.6.0)\n",
      "Requirement already satisfied: s3transfer<0.4.0,>=0.3.0 in ./venv/lib/python3.7/site-packages (from awscli->scispacy) (0.3.3)\n",
      "Requirement already satisfied: colorama<0.4.4,>=0.2.5; python_version != \"3.4\" in ./venv/lib/python3.7/site-packages (from awscli->scispacy) (0.4.3)\n",
      "Requirement already satisfied: botocore==1.15.45 in ./venv/lib/python3.7/site-packages (from awscli->scispacy) (1.15.45)\n",
      "Requirement already satisfied: docutils<0.16,>=0.10 in ./venv/lib/python3.7/site-packages (from awscli->scispacy) (0.15.2)\n",
      "Requirement already satisfied: PyYAML<5.4,>=3.10; python_version != \"3.4\" in ./venv/lib/python3.7/site-packages (from awscli->scispacy) (5.3.1)\n",
      "Requirement already satisfied: rsa<=3.5.0,>=3.1.2 in ./venv/lib/python3.7/site-packages (from awscli->scispacy) (3.4.2)\n",
      "Requirement already satisfied: psutil in ./venv/lib/python3.7/site-packages (from nmslib>=1.7.3.6->scispacy) (5.7.0)\n",
      "Requirement already satisfied: pybind11>=2.2.3 in ./venv/lib/python3.7/site-packages (from nmslib>=1.7.3.6->scispacy) (2.5.0)\n",
      "Requirement already satisfied: scipy>=0.17.0 in ./venv/lib/python3.7/site-packages (from scikit-learn>=0.20.3->scispacy) (1.4.1)\n",
      "Requirement already satisfied: zipp>=0.5 in ./venv/lib/python3.7/site-packages (from importlib-metadata>=0.20; python_version < \"3.8\"->catalogue<1.1.0,>=0.0.7->spacy) (3.1.0)\n",
      "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in ./venv/lib/python3.7/site-packages (from botocore==1.15.45->awscli->scispacy) (0.9.5)\n",
      "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in ./venv/lib/python3.7/site-packages (from botocore==1.15.45->awscli->scispacy) (2.8.1)\n",
      "Requirement already satisfied: pyasn1>=0.1.3 in ./venv/lib/python3.7/site-packages (from rsa<=3.5.0,>=3.1.2->awscli->scispacy) (0.4.8)\n",
      "Requirement already satisfied: six>=1.5 in ./venv/lib/python3.7/site-packages (from python-dateutil<3.0.0,>=2.1->botocore==1.15.45->awscli->scispacy) (1.14.0)\n",
      "\u001b[33mYou are using pip version 19.0.3, however version 20.1b1 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n",
      "Collecting https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.2.4/en_core_sci_sm-0.2.4.tar.gz\n",
      "  Using cached https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.2.4/en_core_sci_sm-0.2.4.tar.gz\n",
      "Requirement already satisfied (use --upgrade to upgrade): en-core-sci-sm==0.2.4 from https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.2.4/en_core_sci_sm-0.2.4.tar.gz in ./venv/lib/python3.7/site-packages\n",
      "Requirement already satisfied: spacy>=2.2.1 in ./venv/lib/python3.7/site-packages (from en-core-sci-sm==0.2.4) (2.2.4)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.0.2)\n",
      "Requirement already satisfied: wasabi<1.1.0,>=0.4.0 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (0.6.0)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.13.0 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (2.23.0)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (2.0.3)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (3.0.2)\n",
      "Requirement already satisfied: catalogue<1.1.0,>=0.0.7 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.0.0)\n",
      "Requirement already satisfied: thinc==7.4.0 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (7.4.0)\n",
      "Requirement already satisfied: plac<1.2.0,>=0.9.6 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.1.3)\n",
      "Requirement already satisfied: blis<0.5.0,>=0.4.0 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (0.4.1)\n",
      "Requirement already satisfied: numpy>=1.15.0 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.18.2)\n",
      "Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (4.45.0)\n",
      "Requirement already satisfied: setuptools in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (40.8.0)\n",
      "Requirement already satisfied: srsly<1.1.0,>=1.0.2 in ./venv/lib/python3.7/site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.0.2)\n",
      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in ./venv/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.25.9)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in ./venv/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->en-core-sci-sm==0.2.4) (2020.4.5.1)\n",
      "Requirement already satisfied: idna<3,>=2.5 in ./venv/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->en-core-sci-sm==0.2.4) (2.9)\n",
      "Requirement already satisfied: chardet<4,>=3.0.2 in ./venv/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->en-core-sci-sm==0.2.4) (3.0.4)\n",
      "Requirement already satisfied: importlib-metadata>=0.20; python_version < \"3.8\" in ./venv/lib/python3.7/site-packages (from catalogue<1.1.0,>=0.0.7->spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.6.0)\n",
      "Requirement already satisfied: zipp>=0.5 in ./venv/lib/python3.7/site-packages (from importlib-metadata>=0.20; python_version < \"3.8\"->catalogue<1.1.0,>=0.0.7->spacy>=2.2.1->en-core-sci-sm==0.2.4) (3.1.0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mYou are using pip version 19.0.3, however version 20.1b1 is available.\r\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "!pip install spacy scispacy\n",
    "!pip install https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.2.4/en_core_sci_sm-0.2.4.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import scispacy\n",
    "import en_core_sci_sm\n",
    "\n",
    "# Cargamos el pipeline de lenguaje científico de scispacy\n",
    "nlp = en_core_sci_sm.load()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'continue'"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Seleccionamos un paper de muestra para observar el funcionamiento de spacy\n",
    "sample_paper = list(pageranks.keys())[0]\n",
    "sample_text = \"\\n\".join([ paragraph[\"text\"] for paragraph in sample_paper._file_contents[\"body_text\"]])\n",
    "sample_text\n",
    "\n",
    "doc = nlp(sample_text, disable=[\"tagger\", \"parser\", \"ner\"])\n",
    "doc[17].lemma_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Se observa el documento tokenizado por el pipeline de `spacy`.\n",
    "Lo interesante de usar `spacy` con el modelo entrenado es que entrega fácilmente representaciones vectoriales del documento y de los tokens que lo componen.\n",
    "\n",
    "Un aspecto relevante para el rendimiento de los modelos posteriores es la cantidad de tokens fuera del vocabulario.\n",
    "A continuación, se realizará una iteración sobre los tokens para detectarlos.\n",
    "Además, se imprimen los tokens terminados en `virus` indicando si se encuentran o no en el vocabulario."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "adenovirus found\n",
      "virus found\n",
      "virus found\n",
      "coronavirus found\n",
      "adenovirus found\n",
      "metapneumovirus found\n",
      "virus found\n",
      "virus found\n",
      "virus found\n",
      "coronavirus found\n",
      "virus found\n",
      "coronavirus found\n",
      "virus found\n",
      "Torovirus not found\n",
      "virus found\n",
      "virus found\n",
      "virus found\n",
      "torovirus not found\n",
      "coronavirus found\n",
      "virus found\n",
      "coronavirus found\n",
      "virus found\n",
      "coronavirus found\n",
      "virus found\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(95, 2.1610555050045495)"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "num_oov = 0\n",
    "for token in doc:\n",
    "    if token.is_oov and token.string != \"\\n\":\n",
    "        if token.string.endswith(\"virus\"):\n",
    "            print(token, \"not found\")\n",
    "        num_oov += 1\n",
    "    else:\n",
    "        if token.string.endswith(\"virus\"):\n",
    "            print(token, \"found\")\n",
    "num_oov, 100 * num_oov / len(doc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "outbreak \n",
      "atypical \n",
      "pneumonia \n",
      "guangdong \n",
      "province\n",
      "people\n",
      "republic \n",
      "china\n",
      "continued \n",
      "november\n",
      "2002\n",
      "reported \n",
      "affected \n",
      "792 \n",
      "people \n",
      "caused \n",
      "31 \n",
      "deaths\n",
      "1 \n",
      "adjacent \n",
      "hong \n",
      "kong\n",
      "surveillance \n",
      "severe \n",
      "atypical \n",
      "pneumonia \n",
      "heightened \n",
      "public \n",
      "hospital \n",
      "network \n",
      "hospital \n",
      "authority \n",
      "hong \n",
      "kong\n",
      "end \n",
      "february\n",
      "2003\n",
      "clusters \n",
      "patients \n",
      "pneumonia \n",
      "noted \n",
      "hong \n",
      "kong\n",
      "affected \n",
      "close \n",
      "contacts \n",
      "health-care \n",
      "workers\n",
      "disease \n",
      "respond \n",
      "empirical \n",
      "antimicrobial \n",
      "treatment \n",
      "acute \n",
      "community-acquired \n",
      "typical \n",
      "atypical \n",
      "pneumonia\n",
      "bacteriological \n",
      "virological \n",
      "pathogens \n",
      "known \n",
      "cause \n",
      "pneumonia \n",
      "identified\n",
      "new \n",
      "disorder \n",
      "called \n",
      "severe \n",
      "acute \n",
      "respiratory \n",
      "syndrome \n",
      "sars\n",
      "subsequently\n",
      "sars \n",
      "spread \n",
      "worldwide \n",
      "involve \n",
      "patients \n",
      "north \n",
      "america\n",
      "europe\n",
      "asian \n",
      "countries\n",
      "1 \n",
      "investigated \n",
      "patients \n",
      "hong \n",
      "kong \n",
      "try \n",
      "identify \n",
      "causal \n",
      "agent\n",
      "included \n",
      "study \n",
      "50 \n",
      "patients \n",
      "fitting \n",
      "modified \n",
      "definition \n",
      "sars \n",
      "admitted \n",
      "acute \n",
      "regional \n",
      "hospitals \n",
      "hong \n",
      "kong \n",
      "feb \n",
      "26 \n",
      "march \n",
      "26\n",
      "2003\n",
      "2 \n",
      "briefly\n",
      "case \n",
      "definition \n",
      "fever \n",
      "38ºc \n",
      "cough \n",
      "shortness \n",
      "breath\n",
      "new \n",
      "pulmonary \n",
      "infiltrates \n",
      "chest \n",
      "radiography\n",
      "history \n",
      "exposure \n",
      "patient \n",
      "sars \n",
      "absence \n",
      "response \n",
      "empirical \n",
      "antimicrobial \n",
      "coverage \n",
      "typical \n",
      "atypical \n",
      "pneumonia \n",
      "␤ \n",
      "lactams \n",
      "macrolides\n",
      "fluoroquinolones\n",
      "tetracyclines\n",
      "collected \n",
      "nasopharyngeal \n",
      "aspirates \n",
      "serum \n",
      "samples \n",
      "patients\n",
      "paired \n",
      "acute \n",
      "convalescent \n",
      "sera \n",
      "faeces \n",
      "available \n",
      "patients\n",
      "lungbiopsy \n",
      "tissue \n",
      "sample \n",
      "patient \n",
      "processed \n",
      "viral \n",
      "culture \n",
      "reverse-transcriptase \n",
      "pcr \n",
      "rt-pcr\n",
      "routine \n",
      "histopathological \n",
      "examination \n",
      "electron \n",
      "microscopy\n",
      "controls \n",
      "nasopharyngeal \n",
      "aspirates\n",
      "faeces \n",
      "sera \n",
      "submitted \n",
      "microbiological \n",
      "investigation \n",
      "diseases \n",
      "patients \n",
      "identities \n",
      "masked\n",
      "sars \n",
      "patients\n",
      "medical \n",
      "records \n",
      "reviewed \n",
      "retrospectively \n",
      "attending \n",
      "physicians \n",
      "clinical \n",
      "microbiologists\n",
      "routine \n",
      "haematological\n",
      "biochemical\n",
      "microbiological \n",
      "work-up \n",
      "including \n",
      "bacterial \n",
      "culture \n",
      "blood \n",
      "sputum\n",
      "serology\n",
      "nasopharyngeal \n",
      "aspirates \n",
      "virology\n",
      "nasopharyngeal \n",
      "aspirate \n",
      "assessed \n",
      "rapid \n",
      "immunoflourescent \n",
      "antigen \n",
      "detection \n",
      "influenza \n",
      "b\n",
      "parainfluenza \n",
      "types \n",
      "1\n",
      "2\n",
      "3\n",
      "respiratory \n",
      "syncytial \n",
      "virus \n",
      "adenovirus\n",
      "3 \n",
      "culture \n",
      "genetic \n",
      "sequencing \n",
      "coronavirus \n",
      "patients\n",
      "developed \n",
      "rt-pcr \n",
      "detect \n",
      "coronavirus \n",
      "sequence \n",
      "nasopharyngeal \n",
      "aspiration \n",
      "samples\n",
      "total \n",
      "rna \n",
      "clinical \n",
      "samples \n",
      "reverse \n",
      "transcribed \n",
      "random \n",
      "hexamers \n",
      "cdna \n",
      "amplified \n",
      "primers \n",
      "5ј-tacacacct \n",
      "cagcgttg-3ј \n",
      "5ј-cacgaacgtgacgaat-3ј \n",
      "presence \n",
      "2·5 \n",
      "mmol/l \n",
      "magnesium \n",
      "chloride \n",
      "94ºc \n",
      "8 \n",
      "min \n",
      "followed \n",
      "40 \n",
      "cycles \n",
      "94ºc \n",
      "1 \n",
      "min\n",
      "50ºc \n",
      "1 \n",
      "min\n",
      "72ºc \n",
      "1 \n",
      "min\n",
      "coronavirus-infected \n",
      "fetal \n",
      "rhesus \n",
      "kidney \n",
      "cells \n",
      "fixed \n",
      "acetone \n",
      "indirect \n",
      "immunofluorescence \n",
      "assay \n",
      "detect \n",
      "serological \n",
      "response \n",
      "virus\n",
      "find \n",
      "genetic \n",
      "sequence \n",
      "information \n",
      "unknown \n",
      "rna \n",
      "virus\n",
      "random \n",
      "rt-pcr \n",
      "assay\n",
      "total \n",
      "rna \n",
      "virus-infected \n",
      "virus-uninfected \n",
      "fetal \n",
      "rhesus \n",
      "kidney \n",
      "cells \n",
      "isolated\n",
      "rna \n",
      "samples \n",
      "reverse \n",
      "transcribed \n",
      "primer \n",
      "5ј-gccggagc \n",
      "tctgcagaattcnnnnnn-3ј\n",
      "n\n",
      "=\n",
      "t\n",
      "c\n",
      "g\n",
      "cdna \n",
      "amplified \n",
      "primer \n",
      "5ј-gccggagctctgcagaattc-3ј. \n",
      "unique \n",
      "pcr \n",
      "products \n",
      "size\n",
      "infected \n",
      "cell \n",
      "preparation \n",
      "cloned \n",
      "sequenced\n",
      "genetic \n",
      "homology \n",
      "compared \n",
      "genbank\n",
      "routine \n",
      "receipt \n",
      "inoculation \n",
      "samples \n",
      "biosafety \n",
      "level-2 \n",
      "laboratory\n",
      "laboratory \n",
      "procedures \n",
      "involving \n",
      "culture \n",
      "virus \n",
      "biosafety \n",
      "level-3 \n",
      "containment\n",
      "compared \n",
      "risk \n",
      "factors \n",
      "associated \n",
      "complicated \n",
      "uncomplicated \n",
      "disease \n",
      "2 \n",
      "test \n",
      "categorical \n",
      "variables\n",
      "continuous \n",
      "variables \n",
      "tested \n",
      "student\n",
      "t \n",
      "test\n",
      "p \n",
      "value \n",
      "0·05 \n",
      "taken \n",
      "significant\n",
      "spss \n",
      "version \n",
      "10.0\n",
      "analyses\n",
      "multivariate \n",
      "analysis \n",
      "number \n",
      "cases \n",
      "small \n",
      "meaningful \n",
      "results \n",
      "obtained\n",
      "sponsors \n",
      "study \n",
      "role \n",
      "study \n",
      "design\n",
      "data \n",
      "collection\n",
      "data \n",
      "analysis\n",
      "data \n",
      "interpretation\n",
      "writing \n",
      "report\n",
      "50 \n",
      "patients \n",
      "sars \n",
      "ethnic \n",
      "chinese\n",
      "represented \n",
      "different \n",
      "epidemiologically \n",
      "linked \n",
      "clusters \n",
      "sporadic \n",
      "cases \n",
      "fitting \n",
      "case \n",
      "definition\n",
      "admitted \n",
      "hospital \n",
      "mean \n",
      "5 \n",
      "days \n",
      "sd \n",
      "2·3\n",
      "onset \n",
      "symptoms\n",
      "median \n",
      "age \n",
      "42 \n",
      "years \n",
      "range \n",
      "23\n",
      "74\n",
      "female-to-male \n",
      "ratio \n",
      "1 \n",
      "1·3\n",
      "patients\n",
      "14 \n",
      "28\n",
      "health-care \n",
      "workers \n",
      "10\n",
      "history \n",
      "visit \n",
      "hospital \n",
      "major \n",
      "outbreak \n",
      "sars\n",
      "13 \n",
      "26\n",
      "household \n",
      "contacts \n",
      "12 \n",
      "24\n",
      "social \n",
      "contacts \n",
      "patients \n",
      "sars\n",
      "8\n",
      "recently \n",
      "travelled \n",
      "mainland \n",
      "china\n",
      "presenting \n",
      "complaint \n",
      "patients \n",
      "feverishness \n",
      "shortness \n",
      "breath\n",
      "cough \n",
      "myalgia \n",
      "present \n",
      "half \n",
      "patients \n",
      "anorexia \n",
      "n=5\n",
      "10\n",
      "reported\n",
      "initial \n",
      "examination\n",
      "auscultatory \n",
      "findings \n",
      "crepitations \n",
      "decreased \n",
      "air \n",
      "entry \n",
      "present \n",
      "19 \n",
      "38\n",
      "patients\n",
      "dry \n",
      "cough \n",
      "reported \n",
      "31 \n",
      "62\n",
      "patients\n",
      "radiological \n",
      "evidence \n",
      "consolidation \n",
      "time \n",
      "admission \n",
      "involving \n",
      "zone \n",
      "36\n",
      "zones \n",
      "13\n",
      "zones \n",
      "despite \n",
      "high \n",
      "fever\n",
      "49 \n",
      "98\n",
      "patients \n",
      "evidence \n",
      "leucocytosis\n",
      "peripheral \n",
      "blood \n",
      "tests \n",
      "lymphopenia \n",
      "present \n",
      "68\n",
      "leucopenia \n",
      "26\n",
      "thrombocytopenia \n",
      "40\n",
      "anaemia \n",
      "18\n",
      "table \n",
      "2\n",
      "alanine \n",
      "aminotransferase \n",
      "45\n",
      "350 \n",
      "u/l\n",
      "creatinine \n",
      "kinase \n",
      "141\n",
      "1379 \n",
      "u/l\n",
      "raised \n",
      "34\n",
      "26\n",
      "respectively\n",
      "routine \n",
      "microbiological \n",
      "investigation \n",
      "known \n",
      "viruses \n",
      "bacteria \n",
      "culture\n",
      "antigen \n",
      "detection\n",
      "pcr \n",
      "negative \n",
      "cases\n",
      "blood \n",
      "culture \n",
      "positive \n",
      "escherichia \n",
      "coli \n",
      "man \n",
      "aged \n",
      "74 \n",
      "years \n",
      "admitted \n",
      "intensive \n",
      "care\n",
      "finding \n",
      "attributed \n",
      "hospitalacquired \n",
      "urinary-tract \n",
      "infection\n",
      "klebsiella \n",
      "pneumoniae \n",
      "haemophilus \n",
      "influenzae \n",
      "isolated \n",
      "sputum \n",
      "samples \n",
      "patients \n",
      "admission\n",
      "oral \n",
      "levofloxacin \n",
      "500 \n",
      "mg \n",
      "24 \n",
      "h \n",
      "given \n",
      "patients\n",
      "amoxicillin-clavulanate \n",
      "given \n",
      "intravenously \n",
      "1·2 \n",
      "g \n",
      "8 \n",
      "h \n",
      "intervals \n",
      "orally \n",
      "375 \n",
      "mg \n",
      "times \n",
      "daily\n",
      "intravenous \n",
      "oral \n",
      "clarithromycin \n",
      "500 \n",
      "mg \n",
      "12 \n",
      "h \n",
      "given \n",
      "40 \n",
      "patients\n",
      "patients \n",
      "received \n",
      "oral \n",
      "oseltamivir \n",
      "75 \n",
      "mg \n",
      "twice \n",
      "daily\n",
      "patient\n",
      "intravenous \n",
      "ceftriaxone \n",
      "2 \n",
      "g \n",
      "24 \n",
      "h\n",
      "oral \n",
      "azithromycin \n",
      "500 \n",
      "mg \n",
      "24 \n",
      "h\n",
      "oral \n",
      "amantadine \n",
      "100 \n",
      "mg \n",
      "twice \n",
      "daily \n",
      "given \n",
      "empirical \n",
      "coverage \n",
      "typical \n",
      "atypical \n",
      "pneumonia\n",
      "time \n",
      "writing\n",
      "19 \n",
      "patients \n",
      "progressed \n",
      "severe \n",
      "disease \n",
      "oxygen \n",
      "desaturation \n",
      "requiring \n",
      "intensive \n",
      "care \n",
      "ventilatory \n",
      "support \n",
      "mean \n",
      "6·4 \n",
      "days\n",
      "mean \n",
      "time \n",
      "onset \n",
      "symptoms \n",
      "worsening \n",
      "8·3 \n",
      "days\n",
      "intravenous \n",
      "ribavirin \n",
      "8 \n",
      "mg/kg \n",
      "8 \n",
      "h \n",
      "7\n",
      "10 \n",
      "days \n",
      "steroid \n",
      "intravenous \n",
      "hydrocortisone \n",
      "100 \n",
      "mg \n",
      "6 \n",
      "h\n",
      "hydrocortisone \n",
      "200 \n",
      "mg \n",
      "8 \n",
      "h\n",
      "methylprednisolone \n",
      "1\n",
      "3 \n",
      "mg/kg \n",
      "24 \n",
      "h \n",
      "doses \n",
      "tailed \n",
      "2\n",
      "3 \n",
      "weeks\n",
      "given \n",
      "49 \n",
      "patients \n",
      "mean \n",
      "6·7 \n",
      "days \n",
      "onset \n",
      "symptoms\n",
      "patients \n",
      "given \n",
      "ribavirin \n",
      "steroids \n",
      "intubation \n",
      "ventilation \n",
      "intensive \n",
      "care\n",
      "consistent \n",
      "response \n",
      "terms \n",
      "resolution \n",
      "fever\n",
      "decreased \n",
      "respiratory \n",
      "support\n",
      "later \n",
      "radiological \n",
      "resolution\n",
      "fluctuating \n",
      "fever \n",
      "static \n",
      "requirement \n",
      "respiratory \n",
      "support\n",
      "risk \n",
      "factors \n",
      "associated \n",
      "severe \n",
      "complicated \n",
      "disease \n",
      "requiring \n",
      "intensive \n",
      "care \n",
      "ventilatory \n",
      "support \n",
      "older \n",
      "age\n",
      "severe \n",
      "lymphopenia\n",
      "impaired \n",
      "alanine \n",
      "aminotransferase\n",
      "delayed \n",
      "starting \n",
      "ribavirin \n",
      "steroid \n",
      "table \n",
      "3\n",
      "complicated \n",
      "cases \n",
      "treated \n",
      "ribavirin \n",
      "steroids \n",
      "admission \n",
      "intensive-care \n",
      "unit\n",
      "uncomplicated \n",
      "cases \n",
      "started \n",
      "ribavirin \n",
      "steroids \n",
      "general \n",
      "ward\n",
      "expected\n",
      "31 \n",
      "uncomplicated \n",
      "cases \n",
      "recovered \n",
      "improved\n",
      "patients \n",
      "complicated \n",
      "wbc\n",
      "=\n",
      "white-blood \n",
      "cell\n",
      "patients \n",
      "diabetes \n",
      "mellitus\n",
      "hypertrophic \n",
      "obstructive \n",
      "cardiomyopathy\n",
      "chronic \n",
      "active \n",
      "hepatitis \n",
      "b\n",
      "brain \n",
      "tumour\n",
      "†one \n",
      "patient \n",
      "essential \n",
      "hypertension\n",
      "‡desaturation \n",
      "requiring \n",
      "intensive-care \n",
      "support\n",
      "response \n",
      "defined \n",
      "resolution \n",
      "fever \n",
      "48 \n",
      "h\n",
      "decreased \n",
      "ventilatory \n",
      "support\n",
      "radiological \n",
      "improvement\n",
      "table \n",
      "3 \n",
      "risk \n",
      "factors \n",
      "associated \n",
      "severe \n",
      "disease \n",
      "requiring \n",
      "intensive \n",
      "care \n",
      "ventilatory \n",
      "support \n",
      "disease \n",
      "worsened\n",
      "death \n",
      "time \n",
      "writing\n",
      "50 \n",
      "patients \n",
      "monitored \n",
      "mean \n",
      "12 \n",
      "days \n",
      "sd \n",
      "6·1\n",
      "time \n",
      "writing\n",
      "virus \n",
      "isolates\n",
      "identified \n",
      "coronavirus\n",
      "isolated \n",
      "patients\n",
      "open \n",
      "lung \n",
      "biopsy \n",
      "sample \n",
      "male \n",
      "hong \n",
      "kong \n",
      "chinese \n",
      "resident \n",
      "aged \n",
      "53 \n",
      "years \n",
      "nasopharyngeal \n",
      "aspirate \n",
      "woman \n",
      "aged \n",
      "42 \n",
      "years \n",
      "good \n",
      "previous \n",
      "health\n",
      "man \n",
      "history \n",
      "10 \n",
      "h \n",
      "social \n",
      "contact \n",
      "chinese \n",
      "visitor \n",
      "coming \n",
      "guangzhou\n",
      "mainland \n",
      "china\n",
      "later \n",
      "died \n",
      "sars\n",
      "2 \n",
      "days \n",
      "exposure\n",
      "patient \n",
      "presented \n",
      "fever\n",
      "malaise\n",
      "myalgia\n",
      "headache\n",
      "crepitations \n",
      "present \n",
      "right \n",
      "lower \n",
      "zone \n",
      "corresponding \n",
      "alevolar \n",
      "shadow \n",
      "chest \n",
      "radiograph\n",
      "haematological \n",
      "investigation \n",
      "revealed \n",
      "lymphopenia \n",
      "0·7ϫ10 \n",
      "9 \n",
      "/l \n",
      "normal \n",
      "total \n",
      "white-cell \n",
      "platelet \n",
      "count\n",
      "alanine \n",
      "aminotransferase \n",
      "41 \n",
      "u/l\n",
      "creatinine \n",
      "phosphokinase \n",
      "405 \n",
      "u/l\n",
      "impaired\n",
      "despite \n",
      "combination \n",
      "oral \n",
      "azithromycin\n",
      "amantadine\n",
      "intravenous \n",
      "ceftriaxone\n",
      "increasing \n",
      "bilateral \n",
      "pulmonary \n",
      "infiltrates \n",
      "progressive \n",
      "oxygen \n",
      "desaturation\n",
      "open \n",
      "lung \n",
      "biopsy \n",
      "9 \n",
      "days \n",
      "admission\n",
      "histopathological \n",
      "examination \n",
      "showed \n",
      "mild \n",
      "interstitial \n",
      "inflammation \n",
      "scattered \n",
      "alveolar \n",
      "pneumocytes \n",
      "showing \n",
      "cytomegaly\n",
      "granular \n",
      "amphophilic \n",
      "cytoplasm\n",
      "enlarged \n",
      "nuclei \n",
      "prominent \n",
      "nucleoli\n",
      "cells \n",
      "showed \n",
      "inclusions \n",
      "typical \n",
      "herpes \n",
      "virus \n",
      "adenovirus \n",
      "infection\n",
      "required \n",
      "ventilation \n",
      "intensive \n",
      "care \n",
      "surgical \n",
      "procedure\n",
      "empirical \n",
      "intravenous \n",
      "ribavirin \n",
      "hydrocortisone \n",
      "given\n",
      "died \n",
      "20 \n",
      "days \n",
      "admission\n",
      "coronavirus \n",
      "rna \n",
      "detected \n",
      "nasopharyngeal \n",
      "aspirate\n",
      "lung \n",
      "biopsy \n",
      "samples\n",
      "postmortem \n",
      "lung \n",
      "samples\n",
      "significant \n",
      "rise \n",
      "antibody \n",
      "titre \n",
      "1/200 \n",
      "1/1600\n",
      "coronavirus \n",
      "isolate\n",
      "female \n",
      "patient \n",
      "coronavirus \n",
      "isolated \n",
      "history \n",
      "good \n",
      "health\n",
      "recently \n",
      "travelled \n",
      "guangzhou \n",
      "2 \n",
      "days\n",
      "presented \n",
      "fever \n",
      "diarrhoea \n",
      "5 \n",
      "days \n",
      "return \n",
      "hong \n",
      "kong\n",
      "physical \n",
      "examination \n",
      "showed \n",
      "crepitation \n",
      "right \n",
      "lower \n",
      "zone\n",
      "corresponding \n",
      "alveolar \n",
      "shadow \n",
      "chest \n",
      "radiograph\n",
      "investigation \n",
      "revealed \n",
      "leucopenia \n",
      "2·7ϫ10 \n",
      "9 \n",
      "/l\n",
      "lymphopenia \n",
      "0·6ϫ10 \n",
      "9 \n",
      "/l\n",
      "thrombocytopenia \n",
      "104ϫ10 \n",
      "9 \n",
      "/l\n",
      "despite \n",
      "empirical \n",
      "antimicrobial \n",
      "coverage \n",
      "amoxicillin-clavulanate\n",
      "clarithromycin\n",
      "oseltamivir\n",
      "worsened \n",
      "5 \n",
      "days \n",
      "admission \n",
      "required \n",
      "mechanical \n",
      "ventilation \n",
      "intensive \n",
      "care \n",
      "5 \n",
      "days\n",
      "gradually \n",
      "improved \n",
      "treatment \n",
      "ribavirin \n",
      "steroids\n",
      "nasopharyngeal \n",
      "aspirate \n",
      "positive \n",
      "rt-pcr \n",
      "coronavirus \n",
      "seroconverted \n",
      "titre \n",
      "50 \n",
      "1600 \n",
      "coronavirus \n",
      "isolate\n",
      "viruses \n",
      "isolated \n",
      "fetal \n",
      "rhesus \n",
      "kidney \n",
      "cells \n",
      "lung \n",
      "biopsy \n",
      "nasopharyngeal \n",
      "aspirate\n",
      "respectively\n",
      "patients\n",
      "initial \n",
      "cytopathic \n",
      "effect \n",
      "noted \n",
      "appearance \n",
      "rounded \n",
      "refractile \n",
      "cells \n",
      "appearing \n",
      "2\n",
      "4 \n",
      "days \n",
      "inoculation\n",
      "cytopathic \n",
      "effect \n",
      "progress \n",
      "initial \n",
      "culture \n",
      "tubes \n",
      "subsequent \n",
      "passage\n",
      "appeared \n",
      "24 \n",
      "h. \n",
      "virus \n",
      "isolates \n",
      "react \n",
      "routine \n",
      "panel \n",
      "reagents \n",
      "identify \n",
      "virus \n",
      "isolates\n",
      "including \n",
      "influenza \n",
      "b\n",
      "parainfluenza \n",
      "types \n",
      "1\n",
      "2\n",
      "3\n",
      "adenovirus\n",
      "respiratory \n",
      "syncytial \n",
      "virus \n",
      "dako\n",
      "glostrup\n",
      "denmark\n",
      "react \n",
      "rt-pcr \n",
      "assays \n",
      "influenza \n",
      "human \n",
      "metapneumovirus\n",
      "pcr \n",
      "assays \n",
      "mycoplasma\n",
      "virus \n",
      "ether \n",
      "sensitive\n",
      "shows \n",
      "enveloped \n",
      "virus\n",
      "electron \n",
      "microscopy \n",
      "negative \n",
      "stained \n",
      "3\n",
      "potassium \n",
      "phospho-tungstate\n",
      "ph \n",
      "7·0\n",
      "ultracentrifuged \n",
      "cell-culture \n",
      "extracts \n",
      "showed \n",
      "presence \n",
      "pleomorphic \n",
      "enveloped \n",
      "virus \n",
      "particles \n",
      "80\n",
      "90 \n",
      "nm \n",
      "range \n",
      "70\n",
      "130 \n",
      "nm\n",
      "diameter \n",
      "surface \n",
      "morphology \n",
      "compatible \n",
      "coronavirus \n",
      "figure \n",
      "1\n",
      "thin-section \n",
      "electron \n",
      "microscopy \n",
      "infected \n",
      "cells \n",
      "revealed \n",
      "virus \n",
      "particles \n",
      "55\n",
      "90 \n",
      "nm \n",
      "diameter \n",
      "smooth \n",
      "walled \n",
      "vesicles \n",
      "cytoplasm \n",
      "figure \n",
      "2\n",
      "b\n",
      "virus \n",
      "particles \n",
      "seen \n",
      "cell \n",
      "surface\n",
      "overall \n",
      "findings \n",
      "compatible \n",
      "coronavirus \n",
      "infection \n",
      "cells\n",
      "thin-section \n",
      "electron \n",
      "micrograph \n",
      "lung \n",
      "biopsy \n",
      "sample \n",
      "53-year-old \n",
      "male \n",
      "contained \n",
      "60\n",
      "90 \n",
      "nm \n",
      "viral \n",
      "particles \n",
      "cytoplasm \n",
      "desquamated \n",
      "cells\n",
      "viral \n",
      "particles \n",
      "similar \n",
      "size \n",
      "morphology \n",
      "observed \n",
      "cell \n",
      "cultured \n",
      "virus \n",
      "isolate \n",
      "patients \n",
      "figure \n",
      "2\n",
      "rt-pcr \n",
      "products \n",
      "generated \n",
      "random \n",
      "primer \n",
      "rt-pcr \n",
      "assay \n",
      "analysed\n",
      "unique \n",
      "bands \n",
      "found \n",
      "virus-infected \n",
      "samples \n",
      "cloned \n",
      "sequenced\n",
      "30 \n",
      "clones \n",
      "examined\n",
      "containing \n",
      "646 \n",
      "bp \n",
      "unknown \n",
      "origin \n",
      "identified\n",
      "sequence \n",
      "analysis \n",
      "dna \n",
      "fragment \n",
      "suggested \n",
      "sequence \n",
      "weak \n",
      "homology \n",
      "viruses \n",
      "family \n",
      "coronaviridae\n",
      "deducted \n",
      "aminoacid \n",
      "sequence \n",
      "215 \n",
      "aminoacids\n",
      "unknown \n",
      "sequence\n",
      "highest \n",
      "homology \n",
      "57\n",
      "rna \n",
      "polymerase \n",
      "bovine \n",
      "coronavirus \n",
      "murine \n",
      "hepatitis \n",
      "virus\n",
      "confirming \n",
      "virus \n",
      "belongs \n",
      "family \n",
      "coronaviridae\n",
      "phylogenetic \n",
      "analysis \n",
      "protein \n",
      "sequences \n",
      "showed \n",
      "virus\n",
      "closely \n",
      "related \n",
      "group \n",
      "ii \n",
      "coronaviruses\n",
      "distinct \n",
      "virus \n",
      "figure \n",
      "3\n",
      "based \n",
      "646 \n",
      "bp \n",
      "sequence \n",
      "isolate\n",
      "specific \n",
      "primers \n",
      "detecting \n",
      "new \n",
      "virus \n",
      "designed \n",
      "rt-pcr \n",
      "detection \n",
      "human \n",
      "pneumonia-associated \n",
      "35 \n",
      "50 \n",
      "recent \n",
      "serum \n",
      "samples \n",
      "patients \n",
      "sars \n",
      "evidence \n",
      "antibody \n",
      "coronavirus\n",
      "32 \n",
      "patients \n",
      "paired \n",
      "acute \n",
      "convalescent \n",
      "sera \n",
      "available\n",
      "seroconverted \n",
      "four-fold \n",
      "increase \n",
      "antibody \n",
      "titre \n",
      "virus\n",
      "pairs \n",
      "sera \n",
      "additional \n",
      "sars \n",
      "patients \n",
      "clusters \n",
      "outside \n",
      "study \n",
      "group \n",
      "tested \n",
      "provide \n",
      "wider \n",
      "sampling \n",
      "sars \n",
      "patients \n",
      "community \n",
      "seroconverted\n",
      "80 \n",
      "sera \n",
      "patients \n",
      "respiratory \n",
      "diseases \n",
      "200 \n",
      "blood \n",
      "donors \n",
      "detectable \n",
      "antibody\n",
      "seropositivity \n",
      "human \n",
      "pneumonia-associated \n",
      "coronavirus \n",
      "serum \n",
      "sample \n",
      "viral \n",
      "rna \n",
      "detection \n",
      "nasopharyngeal \n",
      "aspirates \n",
      "stools \n",
      "deemed \n",
      "evidence \n",
      "infection \n",
      "coronavirus\n",
      "45 \n",
      "50 \n",
      "patients \n",
      "evidence \n",
      "infection\n",
      "patients \n",
      "virological \n",
      "evidence \n",
      "coronavirus \n",
      "infection\n",
      "serum \n",
      "sample \n",
      "tested \n",
      "14 \n",
      "days \n",
      "onset \n",
      "clinical \n",
      "disease\n",
      "outbreak \n",
      "sars \n",
      "unusual \n",
      "ways\n",
      "especially \n",
      "appearance \n",
      "clusters \n",
      "patients \n",
      "pneumonia \n",
      "health-care \n",
      "workers \n",
      "family \n",
      "contacts\n",
      "series \n",
      "patients\n",
      "investigations \n",
      "conventional \n",
      "pathogens \n",
      "atypical \n",
      "pneumonia \n",
      "proved \n",
      "negative\n",
      "virus \n",
      "belonging \n",
      "family \n",
      "coronaviridae \n",
      "isolated \n",
      "lung \n",
      "biopsy \n",
      "nasopharyngeal \n",
      "aspirate \n",
      "sars \n",
      "patients \n",
      "patients \n",
      "sars \n",
      "serological \n",
      "response \n",
      "virus\n",
      "family \n",
      "coronaviridae \n",
      "includes \n",
      "genus \n",
      "coronavirus \n",
      "torovirus\n",
      "enveloped \n",
      "rna \n",
      "viruses \n",
      "cause \n",
      "disease \n",
      "human \n",
      "beings \n",
      "animals\n",
      "previously \n",
      "known \n",
      "human \n",
      "coronaviruses\n",
      "types \n",
      "229e \n",
      "oc43 \n",
      "major \n",
      "cause \n",
      "common \n",
      "cold\n",
      "6 \n",
      "occasionally \n",
      "cause \n",
      "pneumonia \n",
      "older \n",
      "adults\n",
      "neonates\n",
      "immunocompromised \n",
      "patients\n",
      "7\n",
      "8 \n",
      "coronaviruses \n",
      "reported \n",
      "important \n",
      "cause \n",
      "pneumonia \n",
      "military \n",
      "recruits\n",
      "accounting \n",
      "30\n",
      "cases \n",
      "studies\n",
      "9 \n",
      "human \n",
      "coronaviruses \n",
      "infect \n",
      "neurons\n",
      "viral \n",
      "rna \n",
      "detected \n",
      "brain \n",
      "patients \n",
      "multiple \n",
      "sclerosis\n",
      "10 \n",
      "hand\n",
      "animal \n",
      "coronaviruses \n",
      "eg\n",
      "porcine \n",
      "transmissible \n",
      "gastroenteritis \n",
      "virus\n",
      "murine \n",
      "hepatitis \n",
      "virus\n",
      "avian \n",
      "infectious \n",
      "bronchititis \n",
      "virus\n",
      "cause \n",
      "respiratory\n",
      "gastrointestinal\n",
      "neurological\n",
      "hepatic \n",
      "disease \n",
      "respective \n",
      "hosts\n",
      "11 \n",
      "phylogenetically\n",
      "human \n",
      "pneumonia-associated \n",
      "coronavirus \n",
      "closely \n",
      "related \n",
      "known \n",
      "human \n",
      "animal \n",
      "coronavirus \n",
      "torovirus\n",
      "based \n",
      "analysis \n",
      "646 \n",
      "bp \n",
      "fragment \n",
      "polymerase \n",
      "gene \n",
      "showed \n",
      "virus \n",
      "belongs \n",
      "antigenic \n",
      "group \n",
      "2 \n",
      "coronaviruses\n",
      "murine \n",
      "hepatitis \n",
      "virus \n",
      "bovine \n",
      "coronavirus\n",
      "viruses \n",
      "coronaviridae \n",
      "undergo \n",
      "heterologous \n",
      "recombination \n",
      "virus \n",
      "family \n",
      "genetic \n",
      "analysis \n",
      "parts \n",
      "genome \n",
      "needs \n",
      "nature \n",
      "new \n",
      "virus \n",
      "conclusively \n",
      "defined\n",
      "6 \n",
      "biological\n",
      "genetic\n",
      "clinical \n",
      "data \n",
      "taken \n",
      "new \n",
      "virus \n",
      "known \n",
      "human \n",
      "coronaviruses\n",
      "antibody \n",
      "previously \n",
      "recognised \n",
      "human \n",
      "229e \n",
      "oc43-like \n",
      "coronaviruses \n",
      "widespread \n",
      "human \n",
      "population\n",
      "12 \n",
      "lack \n",
      "serological \n",
      "reactivity \n",
      "novel \n",
      "pneumoniaassociated \n",
      "coronavirus \n",
      "patients \n",
      "implies \n",
      "little \n",
      "antigenic \n",
      "cross \n",
      "reactivity \n",
      "229e \n",
      "oc43 \n",
      "viruses\n",
      "patients \n",
      "clinically \n",
      "defined \n",
      "sars \n",
      "serological \n",
      "rt-pcr \n",
      "evidence \n",
      "infection \n",
      "virus\n",
      "contrast\n",
      "antibody \n",
      "viral \n",
      "rna \n",
      "detectable \n",
      "healthy \n",
      "controls\n",
      "32 \n",
      "patients \n",
      "acute \n",
      "convalescent \n",
      "sera \n",
      "available \n",
      "rising \n",
      "antibody \n",
      "titres \n",
      "human \n",
      "pneumonia-associated \n",
      "coronavirus\n",
      "strengthens \n",
      "contention \n",
      "recent \n",
      "infection \n",
      "virus \n",
      "necessary \n",
      "factor \n",
      "evolution \n",
      "sars\n",
      "addition\n",
      "pairs \n",
      "acute \n",
      "convalescent \n",
      "sera \n",
      "tested \n",
      "patients \n",
      "hospitals \n",
      "hong \n",
      "kong \n",
      "showed \n",
      "seroconversion \n",
      "virus\n",
      "patients \n",
      "sars \n",
      "serological \n",
      "virological \n",
      "evidence \n",
      "coronavirus \n",
      "infection\n",
      "need \n",
      "later \n",
      "convalescent \n",
      "sera \n",
      "tested \n",
      "define \n",
      "seroconvert \n",
      "subsequently\n",
      "concordance \n",
      "human \n",
      "pneumoniaassociated \n",
      "coronavirus \n",
      "clinical \n",
      "definition \n",
      "sars \n",
      "remarkable\n",
      "given \n",
      "clinical \n",
      "case \n",
      "definitions \n",
      "perfect\n",
      "evidence \n",
      "human-metapneumovirus \n",
      "infection\n",
      "rt-pcr \n",
      "rising \n",
      "antibody \n",
      "titre\n",
      "detected \n",
      "patients \n",
      "pathogen \n",
      "consistently \n",
      "detected\n",
      "highly \n",
      "likely \n",
      "coronavirus \n",
      "cause \n",
      "sars \n",
      "necessary \n",
      "prerequisite \n",
      "disease \n",
      "progression\n",
      "microbial \n",
      "non-microbial \n",
      "cofactors \n",
      "play \n",
      "progression \n",
      "disease \n",
      "remains \n",
      "investigated\n",
      "describe \n",
      "clinical \n",
      "presentation \n",
      "complications \n",
      "sars\n",
      "25\n",
      "patients \n",
      "coronaviral \n",
      "pneumonia \n",
      "upper-respiratory-tract \n",
      "symptoms\n",
      "expected \n",
      "atypical \n",
      "pneumonia\n",
      "respiratory \n",
      "symptoms \n",
      "positive \n",
      "auscultatory \n",
      "findings \n",
      "disproportionally \n",
      "mild \n",
      "compared \n",
      "chest \n",
      "radiographic \n",
      "findings\n",
      "gastrointestinal \n",
      "symptoms \n",
      "present \n",
      "10\n",
      "patients\n",
      "symptoms \n",
      "relevant \n",
      "viral \n",
      "rna \n",
      "detectable \n",
      "faeces \n",
      "patients \n",
      "coronaviruses \n",
      "associated \n",
      "diarrhoea \n",
      "animals \n",
      "human \n",
      "beings\n",
      "13 \n",
      "high \n",
      "incidence \n",
      "altered \n",
      "liver \n",
      "function\n",
      "leucopenia\n",
      "severe \n",
      "lymphopenia\n",
      "thrombocytopenia\n",
      "subsequent \n",
      "evolution \n",
      "adult \n",
      "respiratory \n",
      "distress \n",
      "syndrome \n",
      "suggests \n",
      "severe \n",
      "systemic \n",
      "inflammatory \n",
      "damage \n",
      "induced \n",
      "human \n",
      "pneumonia-associated \n",
      "coronavirus\n",
      "immunomodulation \n",
      "steroid \n",
      "treatment \n",
      "important \n",
      "complement \n",
      "empirical \n",
      "antiviral \n",
      "treatment \n",
      "ribavirin\n",
      "pertinent \n",
      "severe \n",
      "human \n",
      "disease \n",
      "associated \n",
      "avian \n",
      "influenza \n",
      "subtype \n",
      "h5n1\n",
      "virus \n",
      "crossed \n",
      "animals \n",
      "human \n",
      "beings\n",
      "postulated \n",
      "immunopathological \n",
      "component\n",
      "14 \n",
      "common \n",
      "h5n1 \n",
      "disease\n",
      "patients \n",
      "severe \n",
      "sars \n",
      "adults\n",
      "lymphopenia\n",
      "variables \n",
      "organ \n",
      "dysfunction \n",
      "respiratory \n",
      "tract\n",
      "15 \n",
      "window \n",
      "opportunity \n",
      "8 \n",
      "days \n",
      "exists \n",
      "onset \n",
      "symptoms \n",
      "respiratory \n",
      "failure\n",
      "severe \n",
      "complicated \n",
      "cases \n",
      "strongly \n",
      "associated \n",
      "underlying \n",
      "disease \n",
      "delayed \n",
      "use \n",
      "ribavirin \n",
      "steroid \n",
      "treatment\n",
      "clinical \n",
      "experience \n",
      "cases\n",
      "started \n",
      "combination \n",
      "treatment \n",
      "early \n",
      "subsequent \n",
      "cases\n",
      "generally \n",
      "uncomplicated \n",
      "time \n",
      "admission\n",
      "overall \n",
      "mortality \n",
      "time \n",
      "writing \n",
      "2\n",
      "use \n",
      "treatment \n",
      "regimen\n",
      "19 \n",
      "complicated \n",
      "cases \n",
      "shown \n",
      "notable \n",
      "response\n",
      "detailed \n",
      "analysis \n",
      "therapeutic \n",
      "response \n",
      "combination \n",
      "regimen \n",
      "impossible\n",
      "given \n",
      "heterogeneous \n",
      "dosing \n",
      "time \n",
      "starting \n",
      "treatment\n",
      "choice \n",
      "ribavirin \n",
      "empirical \n",
      "grounds\n",
      "cause \n",
      "identified\n",
      "reviewed \n",
      "in-vitro \n",
      "susceptibility \n",
      "human \n",
      "pneumonia-associated \n",
      "coronavirus \n",
      "antivirals \n",
      "better \n",
      "understood\n",
      "factor \n",
      "associated \n",
      "severe \n",
      "disease \n",
      "acquisition \n",
      "disease \n",
      "household \n",
      "contact\n",
      "people \n",
      "infected \n",
      "way \n",
      "higher \n",
      "dose \n",
      "duration \n",
      "viral \n",
      "exposure \n",
      "presence \n",
      "underlying \n",
      "diseases \n",
      "people \n",
      "exposed\n",
      "example\n",
      "social \n",
      "contact\n",
      "clinical \n",
      "description \n",
      "pertains \n",
      "largely \n",
      "severe \n",
      "cases \n",
      "admitted \n",
      "hospital\n",
      "presently \n",
      "data \n",
      "clinical \n",
      "spectrum \n",
      "emerging \n",
      "coronavirus \n",
      "infection \n",
      "community \n",
      "outpatients\n",
      "availability \n",
      "diagnostic \n",
      "tests \n",
      "describe \n",
      "help \n",
      "address \n",
      "questions\n",
      "addition\n",
      "allow \n",
      "questions \n",
      "addressed \n",
      "period \n",
      "virus \n",
      "shedding \n",
      "communicability\n",
      "convalescence\n",
      "presence \n",
      "virus \n",
      "body \n",
      "fluids \n",
      "excreta\n",
      "presence \n",
      "virus \n",
      "shedding \n",
      "incubation \n",
      "period\n",
      "epidemiological \n",
      "data \n",
      "present \n",
      "suggest \n",
      "virus \n",
      "spread \n",
      "droplets \n",
      "direct \n",
      "indirect \n",
      "contact\n",
      "airborne \n",
      "spread \n",
      "ruled \n",
      "finding \n",
      "infectious \n",
      "virus \n",
      "respiratory \n",
      "tract \n",
      "supports \n",
      "contention\n",
      "preliminary \n",
      "evidence \n",
      "suggests \n",
      "virus \n",
      "shed \n",
      "faeces\n",
      "detection \n",
      "viral \n",
      "rna \n",
      "prove \n",
      "virus \n",
      "viable \n",
      "transmissible\n",
      "viable \n",
      "virus \n",
      "detectable \n",
      "faeces\n",
      "potentially \n",
      "additional \n",
      "route \n",
      "transmission\n",
      "animal \n",
      "coronaviruses \n",
      "spread \n",
      "faecal-oral \n",
      "route\n",
      "11 \n",
      "samples \n",
      "patients \n",
      "sars \n",
      "readily \n",
      "distinguishable \n",
      "samples \n",
      "patients \n",
      "receipt \n",
      "laboratory\n",
      "initial \n",
      "processing \n",
      "samples \n",
      "biohazard \n",
      "level-2 \n",
      "containment\n",
      "culture \n",
      "virus \n",
      "biosafety \n",
      "level-3 \n",
      "containment\n",
      "containment \n",
      "measures \n",
      "proved \n",
      "successful \n",
      "far\n",
      "laboratory \n",
      "infections \n",
      "documented\n",
      "provided \n",
      "evidence \n",
      "virus \n",
      "coronavirus \n",
      "family \n",
      "causal \n",
      "agent \n",
      "sars\n",
      "remains \n",
      "possible \n",
      "viruses \n",
      "act \n",
      "opportunistic \n",
      "secondary \n",
      "invaders \n",
      "increase \n",
      "disease \n",
      "progression\n",
      "hypothesis \n",
      "needs \n",
      "investigated \n",
      "contributors \n",
      "j \n",
      "s \n",
      "m \n",
      "peiris \n",
      "k \n",
      "y \n",
      "yuen \n",
      "co-principal \n",
      "investigators\n",
      "jointly \n",
      "wrote \n",
      "report\n",
      "supervised \n",
      "virological \n",
      "clinical \n",
      "components \n",
      "study\n",
      "l \n",
      "l \n",
      "m \n",
      "poon \n",
      "obtained \n",
      "viral \n",
      "sequence \n",
      "data \n",
      "developed \n",
      "rt-pcr \n",
      "assay\n",
      "y \n",
      "guan\n",
      "k \n",
      "h \n",
      "chan\n",
      "w \n",
      "lim\n",
      "j \n",
      "m \n",
      "nicholls \n",
      "phylogenetic \n",
      "analysis\n",
      "clinical \n",
      "virology\n",
      "electron \n",
      "microscopy \n",
      "identify \n",
      "novel \n",
      "virus\n",
      "t \n",
      "k \n",
      "ng\n",
      "d \n",
      "n \n",
      "c \n",
      "tsang\n",
      "r \n",
      "yung\n",
      "w \n",
      "lim \n",
      "coordinated \n",
      "microbiological \n",
      "investigations \n",
      "analysed \n",
      "overall \n",
      "results\n",
      "researchers \n",
      "involved \n",
      "collection \n",
      "analysis \n",
      "clinical \n",
      "data\n",
      "72-year-old \n",
      "woman \n",
      "found \n",
      "elevated \n",
      "alkaline \n",
      "phosphatase \n",
      "alp\n",
      "concentration \n",
      "1500 \n",
      "iu/l \n",
      "normal \n",
      "80\n",
      "130 \n",
      "iu/l\n",
      "year \n",
      "ago \n",
      "routine \n",
      "laboratory \n",
      "evaluation \n",
      "outside \n",
      "hospital\n",
      "patient \n",
      "symptoms\n",
      "work-up \n",
      "year \n",
      "later\n",
      "undergoing \n",
      "follow-up \n",
      "magnetic \n",
      "resonance \n",
      "imaging \n",
      "mri\n",
      "scan \n",
      "previous \n",
      "stroke\n",
      "multiple \n",
      "mixed \n",
      "blastic \n",
      "lytic \n",
      "lesions \n",
      "seen \n",
      "skull \n",
      "suggestive \n",
      "paget\n",
      "disease \n",
      "figure\n",
      "skull \n",
      "enlarged \n",
      "slightly \n",
      "compressing \n",
      "frontal \n",
      "lobes \n",
      "bilaterally\n",
      "patient \n",
      "remained \n",
      "including \n",
      "normal \n",
      "hearing\n",
      "neurological \n",
      "deficit \n",
      "prior \n",
      "stroke\n",
      "serum \n",
      "alp \n",
      "1055 \n",
      "iu/l. \n",
      "liver \n",
      "function \n",
      "tests \n",
      "normal\n",
      "computed \n",
      "tomography \n",
      "scan \n",
      "ct\n",
      "confirmed \n",
      "pagetic \n",
      "nature \n",
      "lesions\n",
      "tc-99 \n",
      "bone \n",
      "scan \n",
      "showed \n",
      "intense \n",
      "isotope \n",
      "uptake \n",
      "skull\n",
      "areas \n",
      "uptake \n",
      "considered \n",
      "degenerative \n",
      "joint \n",
      "disease\n",
      "patient \n",
      "started \n",
      "oral \n",
      "bisphosphonate\n",
      "paget\n",
      "disease \n",
      "commonly \n",
      "seen \n",
      "elderly \n",
      "population \n",
      "elevated \n",
      "alp \n",
      "concentrations \n",
      "routine \n",
      "laboratory \n",
      "evaluation \n",
      "common \n",
      "presentation\n",
      "patients\n",
      "alp \n",
      "mildly \n",
      "elevated\n",
      "concentrations \n",
      "high\n",
      "skull \n",
      "involvement \n",
      "suspected\n",
      "treatment \n",
      "bisphosphonates \n",
      "mandatory \n",
      "prevent \n",
      "neurological \n",
      "complications\n"
     ]
    }
   ],
   "source": [
    "# A continuación se prueba el mecanismo para remover stopwords,\n",
    "# puntuación, espacios, y posteriormente extraer el lemma del token\n",
    "for token in doc:\n",
    "    if not token.is_stop and not token.is_punct and not token.is_space:\n",
    "        print(token.lemma_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "En total existen 95 tokens fuera del vocabulario, que representan el 2.16% del total de tokens del documento.\n",
    "Por otra parte, vemos como tokens relevantes, como `coronavirus` están incluidos en el vocabulario.\n",
    "\n",
    "## Latent Dirichlet Allocation (LDA)\n",
    "\n",
    "A continuación se realizarán los experimentos de modelado de tópicos usando la técnica LDA.\n",
    "Se usa la implementación de esta técnica incluida en la librería `scikit-learn`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "b93d7a804b624f65b474fecead8a4c09",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=0.0, max=38882.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[\"Coronavirus as a possible cause of severe acute respiratory syndrome \\n Summary Background An outbreak of severe acute respiratory syndrome (SARS) has been reported in Hong Kong. We investigated the viral cause and clinical presentation among 50 patients. Methods We analysed case notes and microbiological findings for 50 patients with SARS, representing more than five separate epidemiologically linked transmission clusters. We defined the clinical presentation and risk factors associated with severe disease and investigated the causal agents by chest radiography and laboratory testing of nasopharyngeal aspirates and sera samples. We compared the laboratory findings with those submitted for microbiological investigation of other diseases from patients whose identity was masked. Findings Patients' age ranged from 23 to 74 years. Fever, chills, myalgia, and cough were the most frequent complaints. When compared with chest radiographic changes, respiratory symptoms and auscultatory findings were disproportionally mild. Patients who were household contacts of other infected people and had older age, lymphopenia, and liver dysfunction were associated with severe disease. A virus belonging to the family Coronaviridae was isolated from two patients. By use of serological and reverse-transcriptase PCR specific for this virus, 45 of 50 patients with SARS, but no controls, had evidence of infection with this virus. Interpretation A coronavirus was isolated from patients with SARS that might be the primary agent associated with this disease. Serological and molecular tests specific for the virus permitted a definitive laboratory diagnosis to be made and allowed further investigation to define whether other cofactors play a part in disease progression.\",\n",
       " 'Severe Acute Respiratory Syndrome (SARS) \\n nan',\n",
       " 'Angiotensin-converting enzyme 2 is a functional receptor for the SARS coronavirus \\n Spike (S) proteins of coronaviruses, including the coronavirus that causes severe acute respiratory syndrome (SARS), associate with cellular receptors to mediate infection of their target cells(1,2). Here we identify a metallopeptidase, angiotensin-converting enzyme 2 (ACE2)(3,4), isolated from SARS coronavirus (SARS-CoV)-permissive Vero E6 cells, that efficiently binds the S1 domain of the SARS-CoV S protein. We found that a soluble form of ACE2, but not of the related enzyme ACE1, blocked association of the S1 domain with Vero E6 cells. 293T cells transfected with ACE2, but not those transfected with human immunodeficiency virus-1 receptors, formed multinucleated syncytia with cells expressing S protein. Furthermore, SARS-CoV replicated efficiently on ACE2-transfected but not mock-transfected 293T cells. Finally, anti-ACE2 but not anti-ACE1 antibody blocked viral replication on Vero E6 cells. Together our data indicate that ACE2 is a functional receptor for SARS-CoV. SUPPLEMENTARY INFORMATION: The online version of this article (doi:10.1038/nature02145) contains supplementary material, which is available to authorized users.',\n",
       " 'Coronavirus Pneumonia Following Autologous Bone Marrow Transplantation for Breast Cancer \\n Infectious bronchitis virus, otherwise known as coronavirus, can cause mild upper respiratory tract illnesses in children and adults. Rarely has coronavirus been linked, either by serology or nasal wash, to pneumonia. We report a case of a young woman who, following treatment for stage IIIA breast cancer using a high-dose chemotherapy regimen followed by autologous bone marrow and stem cell transplantation, developed respiratory failure and was found to have coronavirus pneumonia as diagnosed by electron microscopy from BAL fluid. We propose that coronavirus should be considered in the differential diagnosis of acute respiratory failure in cancer patients who have undergone high-dose chemotherapy and autologous hematopoietic support.',\n",
       " 'Coronavirus Infection in Acute Lower Respiratory Tract Disease of Infants \\n A serologic surveillance of lower respiratory tract disease in 417 hospitalized children under 18 months of age revealed infection with coronviruses (strains OC43 and/ or 229E) in 34 (8.2%). During the same interval, one of 13 control infants was infected. There were two distinct periods lasting six and 14 weeks, respectively, during which the incidence rose to as high as 18.9 % of patients with lower respiratory tract disease. The incidence of coronavirus infection in patients with pneumonia and bronchiolitis was higher than the incidences of adenoviruses, influenza, parainfluenza viruses types 1 and 2, and rhinoviruses, and lower only than the incidences of parainfluenza virus type 3 and respiratory syncytial virus. Coronoviruses serologically similar or identical to strain 229E were recovered from frozen nasal washes obtained during the acute phase of pneumonia in two children.',\n",
       " 'Further studies on human enteric coronaviruses \\n Comparisons were made between human enteric coronaviruses and the enteric coronaviruses of pigs and calves by negative staining. Examination of human intestinal organ culture fluids at various time intervals after inoculation with the human enteric coronavirus showed increasing numbers of particles in the fluids. Thin sections of the columnar epithelial cells of these explants showed a number of features consistent with the replication of known human and animal coronaviruses. Virus particles found in thin sections had a mean diameter of 68 nm. In addition, a structure was found in thin sections which has not been described previously. This structure may represent the viral nucleocapsid.',\n",
       " \"THE COMMON COLD \\n The common cold has intrigued physicians and the general public for centuries. It has been defined as an acute epidemic respiratory disease characterized by mild coryzal symptoms of rhinorrhea, nasal obstruction, and sneezing. The nasal discharge is usually copious and thin during the first 2 days of illness, then it generally becomes more viscous and purulent. 22 The disease is self-limited. Symptoms may persist for 2 days to more than 14 days; however, the cold may abort after only 1 day. Fever, cough, sore throat, or lacrimation may or may not be present. The common cold is of itself harmless, but bacterial invasion frequently follows the initial infection. It is these secondary invaders that may produce disorders of serious consequence. The common cold is the most frequent acute illness in the United States and throughout the industrialized world. About half the population gets at least one cold every year. 5 Colds account for 40% of all time lost from jobs among employed people (23 million days of work per year) and about 30% of absenteeism from schools (26 million school days per year). Estimates vary, but the average preschool child has somewhere between 4 and 10 colds per year, and the average adult has about two to four colds per year. The actual cost of caring for patients with colds in US physicians' offices is estimated to be $1.5 billion annually. 64 Seasonal patterns of infection can be identified for some of the various types of viruses that are responsible for outbreaks of the common cold. For example, available epidemiologic data suggest early fall and late spring are the most common times to find more outbreaks of rhinovirus. Respiratory syncytial virus (RSV) tends to follow winter and spring incidence, with a peak number of cases found mainly in January. Parainfluenza types 1 and 2 seem to peak during the autumn, whereas parainfluenza type 3 has an increased incidence during the late spring. Adenoviruses and coronaviruses tend to produce epidemics during the winter and spring (Fig. 1) This article presents data concerning the cause, pathogenesis, and treatment of the common cold, as well as discussion of the available diagnostic tests and their use in formulating differential diagnoses.\",\n",
       " 'A newly discovered human pneumovirus isolated from young children with respiratory tract disease \\n From 28 young children in the Netherlands, we isolated a paramyxovirus that was identified as a tentative new member of the Metapneumovirus genus based on virological data, sequence homology and gene constellation. Previously, avian pneumovirus was the sole member of this recently assigned genus, hence the provisional name for the newly discovered virus: human metapneumovirus. The clinical symptoms of the children from whom the virus was isolated were similar to those caused by human respiratory syncytial virus infection, ranging from upper respiratory tract disease to severe bronchiolitis and pneumonia. Serological studies showed that by the age of five years, virtually all children in the Netherlands have been exposed to human metapneumovirus and that the virus has been circulating in humans for at least 50 years.',\n",
       " 'Newly discovered coronavirus as the primary cause of severe acute respiratory syndrome \\n Summary Background The worldwide outbreak of severe acute respiratory syndrome (SARS) is associated with a newly discovered coronavirus, SARS-associated coronavirus (SARSCoV). We did clinical and experimental studies to assess the role of this virus in the cause of SARS. Methods We tested clinical and postmortem samples from 436 SARS patients in six countries for infection with SARSCoV, human metapneumovirus, and other respiratory pathogens. We infected four cynomolgus macaques (Macaca fascicularis) with SARS-CoV in an attempt to replicate SARS and did necropsies on day 6 after infection. Findings SARS-CoV infection was diagnosed in 329 (75%) of 436 patients fitting the case definition of SARS; human metapneumovirus was diagnosed in 41 (12%) of 335, and other respiratory pathogens were diagnosed only sporadically. SARS-CoV was, therefore, the most likely causal agent of SARS. The four SARS-CoV-infected macaques excreted SARS-CoV from nose, mouth, and pharynx from 2 days after infection. Three of four macaques developed diffuse alveolar damage, similar to that in SARS patients, and characterised by epithelial necrosis, serosanguineous exudate, formation of hyaline membranes, type 2 pneumocyte hyperplasia, and the presence of syncytia. SARS-CoV was detected in pneumonic areas by virus isolation and RT-PCR, and was localised to alveolar epithelial cells and syncytia by immunohistochemistry and transmission electron microscopy. Interpretation Replication in SARS-CoV-infected macaques of pneumonia similar to that in human beings with SARS, combined with the high prevalence of SARS-CoV infection in SARS patients, fulfill the criteria required to prove that SARS-CoV is the primary cause of SARS. Published online July 22, 2003 http://image.thelancet.com/extras/03art6318web.pdf',\n",
       " \"The Molecular Biology of Coronaviruses \\n Publisher Summary Coronaviruses have recently emerged as an important group of animal and human pathogens that share a distinctive replicative cycle. Some of the unique characteristics in the replication of coronaviruses include generation of a 3' coterminal-nested set of five or six subgenomic mRNAs, each of which appears to direct the synthesis of one protein. Two virus-specific RNA polymerase activities have been identified. Many of the distinctive features of coronavirus infection and coronavirus-induced diseases may result from the properties of the two coronavirus glycoproteins. The intracellular budding site, which may be important in the establishment and maintenance of persistent infections, appears to be due to the restricted intracytoplasmic migration of the E1 glycoprotein, which acts as a matrix-like transmembrane glycoprotein. E1 also exhibits distinctive behavior by self-aggregating on heating at 100°C in sodium dodecyl sulfate (SDS) and by its interaction with RNA in the viral nucleocapsid. The E1 of mouse hepatitis virus (MHV) is an O-linked glycoprotein, unlike most other viral glycoproteins. Thus, the coronavirus system may be a useful model for the study of synthesis, glycosylation, and transport of O-linked cellular glycoproteins.\"]"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from tqdm.notebook import tqdm\n",
    "\n",
    "# En primer lugar, se procesará el texto de todos los documentos.\n",
    "def process_papers_file_contents(papers):\n",
    "    texts = []\n",
    "    for paper in tqdm(papers):\n",
    "        text = \" \\n \".join([ paragraph[\"text\"] for paragraph in paper._file_contents[\"body_text\"]])\n",
    "        # OJO: por motivos de agilidad de desarrollo, sólo se modelaron los tópicos\n",
    "        # usando información del título y del astract. Si se desea incluir el cuerpo\n",
    "        # del paper se debe modificar la siguiente línea para incluir {text}.\n",
    "        texts.append(f\"{paper.title} \\n {paper.abstract}\")\n",
    "    return texts\n",
    "\n",
    "docs = process_papers_file_contents(\n",
    "    papers=list(pageranks.keys()),\n",
    ")\n",
    "docs[:10]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Para representar los papers se utilizarán vectores que cuentan la cantidad de ocurrencias de los distintos tokens en éstos.\n",
    "No se usan representaciones como `tf-idf` debido a que la técnica usada para modelar tópicos es LDA."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.feature_extraction.text import CountVectorizer\n",
    "\n",
    "def tokenizer(sentence):\n",
    "    tokens = []\n",
    "    for token in nlp(sentence, disable=[\"tagger\", \"parser\", \"ner\"]):\n",
    "        # Se descartan números, stopwords, puntuación, espacio y tokens de largo 1\n",
    "        if not (token.like_num or token.is_stop or token.is_punct or token.is_space or len(token)==1):\n",
    "            tokens.append(token.lemma_)\n",
    "    return tokens\n",
    "\n",
    "count_vectorizer = CountVectorizer(\n",
    "    tokenizer=tokenizer,\n",
    "    lowercase=True,\n",
    ")\n",
    "vectorized_docs = count_vectorizer.fit_transform(docs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<38882x149389 sparse matrix of type '<class 'numpy.int64'>'\n",
       "\twith 2797505 stored elements in Compressed Sparse Row format>"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vectorized_docs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'coronavirus': 40377,\n",
       " 'possible': 110379,\n",
       " 'cause': 33506,\n",
       " 'severe': 125651,\n",
       " 'acute': 17917,\n",
       " 'respiratory': 118791,\n",
       " 'syndrome': 132435,\n",
       " 'summary': 131458,\n",
       " 'background': 26581,\n",
       " 'outbreak': 101681,\n",
       " 'sars': 123140,\n",
       " 'report': 118430,\n",
       " 'hong': 67800,\n",
       " 'kong': 79214,\n",
       " 'investigate': 75880,\n",
       " 'viral': 142671,\n",
       " 'clinical': 37245,\n",
       " 'presentation': 112007,\n",
       " 'patient': 105033,\n",
       " 'method': 87402,\n",
       " 'analyse': 21076,\n",
       " 'case': 33096,\n",
       " 'note': 98054,\n",
       " 'microbiological': 88039,\n",
       " 'finding': 56943,\n",
       " 'represent': 118456,\n",
       " 'separate': 125091,\n",
       " 'epidemiologically': 52838,\n",
       " 'link': 81901,\n",
       " 'transmission': 137321,\n",
       " 'cluster': 37468,\n",
       " 'define': 44652,\n",
       " 'risk': 120047,\n",
       " 'factor': 55434,\n",
       " 'associate': 24913,\n",
       " 'disease': 46893,\n",
       " 'causal': 33491,\n",
       " 'agent': 19064,\n",
       " 'chest': 35727,\n",
       " 'radiography': 116082,\n",
       " 'laboratory': 80019,\n",
       " 'test': 134365,\n",
       " 'nasopharyngeal': 93803,\n",
       " 'aspirate': 24815,\n",
       " 'serum': 125496,\n",
       " 'sample': 122883,\n",
       " 'compare': 38834,\n",
       " 'submit': 130958,\n",
       " 'investigation': 75883,\n",
       " 'identity': 70791,\n",
       " 'mask': 85546,\n",
       " 'age': 18993,\n",
       " 'range': 116321,\n",
       " 'year': 146171,\n",
       " 'fever': 56508,\n",
       " 'chill': 35871,\n",
       " 'myalgia': 92116,\n",
       " 'cough': 40689,\n",
       " 'frequent': 58582,\n",
       " 'complaint': 38895,\n",
       " 'radiographic': 116074,\n",
       " 'change': 35412,\n",
       " 'symptom': 132328,\n",
       " 'auscultatory': 25599,\n",
       " 'disproportionally': 47155,\n",
       " 'mild': 88610,\n",
       " 'household': 68142,\n",
       " 'contact': 39773,\n",
       " 'infect': 73691,\n",
       " 'people': 106189,\n",
       " 'old': 100016,\n",
       " 'lymphopenia': 83571,\n",
       " 'live': 82161,\n",
       " 'dysfunction': 49180,\n",
       " 'virus': 142912,\n",
       " 'belong': 27813,\n",
       " 'family': 55651,\n",
       " 'coronaviridae': 40364,\n",
       " 'isolate': 76589,\n",
       " 'use': 140886,\n",
       " 'serological': 125385,\n",
       " 'reverse-transcriptase': 119234,\n",
       " 'pcr': 105471,\n",
       " 'specific': 128739,\n",
       " 'control': 39981,\n",
       " 'evidence': 54295,\n",
       " 'infection': 73735,\n",
       " 'interpretation': 75307,\n",
       " 'primary': 112218,\n",
       " 'molecular': 89888,\n",
       " 'permit': 106664,\n",
       " 'definitive': 44668,\n",
       " 'diagnosis': 45960,\n",
       " 'allow': 20183,\n",
       " 'cofactor': 38189,\n",
       " 'play': 108901,\n",
       " 'progression': 112833,\n",
       " 'nan': 93386,\n",
       " 'angiotensin-converting': 21390,\n",
       " 'enzyme': 52625,\n",
       " 'functional': 58939,\n",
       " 'receptor': 117198,\n",
       " 'spike': 128937,\n",
       " 'protein': 113264,\n",
       " 'include': 73190,\n",
       " 'cellular': 34870,\n",
       " 'mediate': 86354,\n",
       " 'target': 133445,\n",
       " 'cells(1,2': 34824,\n",
       " 'identify': 70786,\n",
       " 'metallopeptidase': 87263,\n",
       " '(ace2)(3,4': 210,\n",
       " '(sars-cov)-permissive': 1371,\n",
       " 'vero': 142198,\n",
       " 'e6': 49831,\n",
       " 'cell': 34666,\n",
       " 'efficiently': 50622,\n",
       " 'bind': 28673,\n",
       " 's1': 122273,\n",
       " 'domain': 47975,\n",
       " 'sars-cov': 123169,\n",
       " 'find': 56939,\n",
       " 'soluble': 128234,\n",
       " 'form': 58036,\n",
       " 'ace2': 17407,\n",
       " 'relate': 118045,\n",
       " 'ace1': 17406,\n",
       " 'block': 29593,\n",
       " 'association': 24916,\n",
       " 'transfect': 137147,\n",
       " 'human': 69166,\n",
       " 'immunodeficiency': 72458,\n",
       " 'virus-1': 142926,\n",
       " 'multinucleated': 91613,\n",
       " 'syncytia': 132409,\n",
       " 'express': 54893,\n",
       " 'furthermore': 59055,\n",
       " 'replicate': 118375,\n",
       " 'ace2-transfected': 17443,\n",
       " 'mock-transfected': 89653,\n",
       " 'finally': 56924,\n",
       " 'anti-ace2': 21766,\n",
       " 'anti-ace1': 21765,\n",
       " 'antibody': 22722,\n",
       " 'replication': 118378,\n",
       " 'datum': 43934,\n",
       " 'indicate': 73371,\n",
       " 'supplementary': 131750,\n",
       " 'information': 74097,\n",
       " 'online': 100476,\n",
       " 'version': 142253,\n",
       " 'article': 24407,\n",
       " 'doi:10.1038/nature02145': 47793,\n",
       " 'contain': 39819,\n",
       " 'material': 85656,\n",
       " 'available': 25905,\n",
       " 'authorize': 25668,\n",
       " 'user': 140901,\n",
       " 'pneumonia': 109277,\n",
       " 'follow': 57869,\n",
       " 'autologous': 25768,\n",
       " 'bone': 30011,\n",
       " 'marrow': 85454,\n",
       " 'transplantation': 137377,\n",
       " 'breast': 30462,\n",
       " 'cancer': 32412,\n",
       " 'infectious': 73814,\n",
       " 'bronchitis': 30682,\n",
       " 'know': 79081,\n",
       " 'upper': 140548,\n",
       " 'tract': 136840,\n",
       " 'illness': 72030,\n",
       " 'child': 35842,\n",
       " 'adult': 18534,\n",
       " 'rarely': 116441,\n",
       " 'serology': 125392,\n",
       " 'nasal': 93747,\n",
       " 'wash': 144260,\n",
       " 'young': 146411,\n",
       " 'woman': 145340,\n",
       " 'treatment': 137530,\n",
       " 'stage': 129615,\n",
       " 'iiia': 71645,\n",
       " 'high-dose': 66469,\n",
       " 'chemotherapy': 35690,\n",
       " 'regimen': 117814,\n",
       " 'stem': 129935,\n",
       " 'develope': 45711,\n",
       " 'failure': 55542,\n",
       " 'diagnose': 45957,\n",
       " 'electron': 51144,\n",
       " 'microscopy': 88320,\n",
       " 'bal': 26853,\n",
       " 'fluid': 57590,\n",
       " 'propose': 113107,\n",
       " 'consider': 39628,\n",
       " 'differential': 46320,\n",
       " 'undergo': 139848,\n",
       " 'hematopoietic': 65579,\n",
       " 'support': 131769,\n",
       " 'low': 82775,\n",
       " 'infant': 73664,\n",
       " 'serologic': 125384,\n",
       " 'surveillance': 131942,\n",
       " 'hospitalize': 67973,\n",
       " 'month': 90282,\n",
       " 'reveal': 119208,\n",
       " 'coronviruses': 40447,\n",
       " 'strain': 130264,\n",
       " 'oc43': 99575,\n",
       " 'and/': 21197,\n",
       " '229e': 8376,\n",
       " 'interval': 75376,\n",
       " 'distinct': 47227,\n",
       " 'period': 106522,\n",
       " 'last': 80500,\n",
       " 'week': 144503,\n",
       " 'respectively': 118748,\n",
       " 'incidence': 73162,\n",
       " 'rise': 120035,\n",
       " 'high': 66428,\n",
       " 'bronchiolitis': 30669,\n",
       " 'adenovirus': 18239,\n",
       " 'influenza': 74008,\n",
       " 'parainfluenza': 104259,\n",
       " 'type': 138853,\n",
       " 'rhinovirus': 119562,\n",
       " 'syncytial': 132411,\n",
       " 'coronoviruses': 40444,\n",
       " 'serologically': 125388,\n",
       " 'similar': 126701,\n",
       " 'identical': 70756,\n",
       " 'recover': 117473,\n",
       " 'freeze': 58551,\n",
       " 'obtain': 99545,\n",
       " 'phase': 107349,\n",
       " 'study': 130585,\n",
       " 'enteric': 52283,\n",
       " 'comparison': 38837,\n",
       " 'pig': 108272,\n",
       " 'calve': 32260,\n",
       " 'negative': 94447,\n",
       " 'stain': 129632,\n",
       " 'examination': 54429,\n",
       " 'intestinal': 75432,\n",
       " 'organ': 101103,\n",
       " 'culture': 42312,\n",
       " 'time': 135608,\n",
       " 'inoculation': 74435,\n",
       " 'show': 126223,\n",
       " 'increase': 73274,\n",
       " 'number': 98930,\n",
       " 'particle': 104623,\n",
       " 'thin': 135090,\n",
       " 'section': 124301,\n",
       " 'columnar': 38575,\n",
       " 'epithelial': 52964,\n",
       " 'explant': 54801,\n",
       " 'numb': 98927,\n",
       " 'feature': 56139,\n",
       " 'consistent': 39664,\n",
       " 'animal': 21455,\n",
       " 'mean': 86200,\n",
       " 'diameter': 46032,\n",
       " 'nm': 95802,\n",
       " 'addition': 18152,\n",
       " 'structure': 130509,\n",
       " 'describe': 45406,\n",
       " 'previously': 112159,\n",
       " 'nucleocapsid': 98808,\n",
       " 'common': 38708,\n",
       " 'cold': 38314,\n",
       " 'intrigue': 75745,\n",
       " 'physician': 107973,\n",
       " 'general': 60573,\n",
       " 'public': 114329,\n",
       " 'century': 34989,\n",
       " 'epidemic': 52804,\n",
       " 'characterize': 35486,\n",
       " 'coryzal': 40565,\n",
       " 'rhinorrhea': 119545,\n",
       " 'obstruction': 99531,\n",
       " 'sneeze': 127865,\n",
       " 'discharge': 46812,\n",
       " 'usually': 140971,\n",
       " 'copious': 40191,\n",
       " 'day': 43961,\n",
       " 'generally': 60593,\n",
       " 'viscous': 143185,\n",
       " 'purulent': 114604,\n",
       " 'self-limit': 124657,\n",
       " 'persist': 106749,\n",
       " 'abort': 16827,\n",
       " 'sore': 128389,\n",
       " 'throat': 135354,\n",
       " 'lacrimation': 80085,\n",
       " 'present': 111994,\n",
       " 'harmless': 64603,\n",
       " 'bacterial': 26665,\n",
       " 'invasion': 75833,\n",
       " 'frequently': 58591,\n",
       " 'initial': 74297,\n",
       " 'secondary': 124256,\n",
       " 'invader': 75813,\n",
       " 'produce': 112680,\n",
       " 'disorder': 47075,\n",
       " 'consequence': 39601,\n",
       " 'unite': 140110,\n",
       " 'state': 129763,\n",
       " 'industrialize': 73568,\n",
       " 'world': 145417,\n",
       " 'half': 64294,\n",
       " 'population': 110169,\n",
       " 'get': 61032,\n",
       " 'account': 17346,\n",
       " 'lose': 82734,\n",
       " 'job': 77599,\n",
       " 'employ': 51608,\n",
       " 'work': 145381,\n",
       " 'absenteeism': 16885,\n",
       " 'school': 123741,\n",
       " 'estimate': 53746,\n",
       " 'vary': 141701,\n",
       " 'average': 25947,\n",
       " 'preschool': 111957,\n",
       " 'actual': 17884,\n",
       " 'cost': 40602,\n",
       " 'care': 32926,\n",
       " 'office': 99843,\n",
       " 'annually': 21574,\n",
       " 'seasonal': 124186,\n",
       " 'pattern': 105148,\n",
       " 'responsible': 118854,\n",
       " 'example': 54437,\n",
       " 'epidemiologic': 52836,\n",
       " 'suggest': 131286,\n",
       " 'early': 49908,\n",
       " 'fall': 55598,\n",
       " 'late': 80521,\n",
       " 'spring': 129222,\n",
       " 'rsv': 121353,\n",
       " 'tend': 134162,\n",
       " 'winter': 145142,\n",
       " 'peak': 105792,\n",
       " 'mainly': 84721,\n",
       " 'january': 77207,\n",
       " 'autumn': 25868,\n",
       " 'fig.': 56832,\n",
       " 'concern': 39190,\n",
       " 'pathogenesis': 104932,\n",
       " 'discussion': 46878,\n",
       " 'diagnostic': 45966,\n",
       " 'formulate': 58089,\n",
       " 'newly': 95145,\n",
       " 'discover': 46847,\n",
       " 'pneumovirus': 109336,\n",
       " 'netherlands': 94709,\n",
       " 'paramyxovirus': 104336,\n",
       " 'tentative': 134207,\n",
       " 'new': 95119,\n",
       " 'member': 86685,\n",
       " 'metapneumovirus': 87312,\n",
       " 'genus': 60789,\n",
       " 'base': 27103,\n",
       " 'virological': 142836,\n",
       " 'sequence': 125154,\n",
       " 'homology': 67732,\n",
       " 'gene': 60472,\n",
       " 'constellation': 39701,\n",
       " 'avian': 25969,\n",
       " 'sole': 128183,\n",
       " 'recently': 117190,\n",
       " 'assign': 24889,\n",
       " 'provisional': 113636,\n",
       " 'virtually': 142885,\n",
       " 'expose': 54858,\n",
       " 'circulate': 36702,\n",
       " 'worldwide': 145430,\n",
       " 'sars-associated': 123153,\n",
       " 'sarscov': 123305,\n",
       " 'experimental': 54752,\n",
       " 'assess': 24875,\n",
       " 'role': 120613,\n",
       " 'postmortem': 110780,\n",
       " 'country': 40780,\n",
       " 'pathogen': 104868,\n",
       " 'cynomolgus': 42895,\n",
       " 'macaque': 84347,\n",
       " 'macaca': 84343,\n",
       " 'fascicularis': 55795,\n",
       " 'attempt': 25363,\n",
       " 'necropsy': 94377,\n",
       " 'fit': 57156,\n",
       " 'definition': 44665,\n",
       " 'sporadically': 129157,\n",
       " 'likely': 81752,\n",
       " 'sars-cov-infected': 123211,\n",
       " 'excrete': 54527,\n",
       " 'nose': 98002,\n",
       " 'mouth': 90636,\n",
       " 'pharynx': 107347,\n",
       " 'diffuse': 46364,\n",
       " 'alveolar': 20494,\n",
       " 'damage': 43737,\n",
       " 'characterise': 35479,\n",
       " 'necrosis': 94382,\n",
       " 'serosanguineous': 125426,\n",
       " 'exudate': 55127,\n",
       " 'formation': 58063,\n",
       " 'hyaline': 69451,\n",
       " 'membrane': 86697,\n",
       " 'pneumocyte': 109260,\n",
       " 'hyperplasia': 69852,\n",
       " 'presence': 111983,\n",
       " 'detect': 45584,\n",
       " 'pneumonic': 109310,\n",
       " 'area': 24024,\n",
       " 'isolation': 76601,\n",
       " 'rt-pcr': 121435,\n",
       " 'localise': 82419,\n",
       " 'immunohistochemistry': 72552,\n",
       " 'being': 27751,\n",
       " 'combine': 38605,\n",
       " 'prevalence': 112101,\n",
       " 'fulfill': 58848,\n",
       " 'criterium': 41517,\n",
       " 'require': 118530,\n",
       " 'prove': 113581,\n",
       " 'publish': 114353,\n",
       " 'july': 77802,\n",
       " 'http://image.thelancet.com/extras/03art6318web.pdf': 68864,\n",
       " 'biology': 29026,\n",
       " 'publisher': 114356,\n",
       " 'emerge': 51477,\n",
       " 'important': 72879,\n",
       " 'group': 62898,\n",
       " 'share': 125932,\n",
       " 'distinctive': 47230,\n",
       " 'replicative': 118407,\n",
       " 'cycle': 42760,\n",
       " 'unique': 140090,\n",
       " 'characteristic': 35480,\n",
       " 'generation': 60598,\n",
       " 'coterminal-nested': 40659,\n",
       " 'set': 125557,\n",
       " 'subgenomic': 130887,\n",
       " 'mrnas': 90847,\n",
       " 'appear': 23636,\n",
       " 'direct': 46697,\n",
       " 'synthesis': 132514,\n",
       " 'virus-specific': 143065,\n",
       " 'rna': 120254,\n",
       " 'polymerase': 109884,\n",
       " 'activity': 17850,\n",
       " 'coronavirus-induced': 40405,\n",
       " 'result': 118923,\n",
       " 'property': 113042,\n",
       " 'glycoprotein': 61917,\n",
       " 'intracellular': 75572,\n",
       " 'bud': 31015,\n",
       " 'site': 127168,\n",
       " 'establishment': 53696,\n",
       " 'maintenance': 84726,\n",
       " 'persistent': 106766,\n",
       " 'restrict': 118903,\n",
       " 'intracytoplasmic': 75593,\n",
       " 'migration': 88564,\n",
       " 'e1': 49724,\n",
       " 'act': 17747,\n",
       " 'matrix-like': 85728,\n",
       " 'transmembrane': 137292,\n",
       " 'exhibit': 54578,\n",
       " 'behavior': 27706,\n",
       " 'self-aggregating': 124536,\n",
       " 'heat': 65299,\n",
       " 'sodium': 128097,\n",
       " 'dodecyl': 47666,\n",
       " 'sulfate': 131366,\n",
       " 'sds': 124104,\n",
       " 'interaction': 74929,\n",
       " 'mouse': 90614,\n",
       " 'hepatitis': 65787,\n",
       " 'mhv': 87745,\n",
       " 'o-linked': 99244,\n",
       " 'unlike': 140174,\n",
       " 'system': 132570,\n",
       " 'useful': 140895,\n",
       " 'model': 89678,\n",
       " 'glycosylation': 61963,\n",
       " 'transport': 137385,\n",
       " 'propagate': 113013,\n",
       " 'non-bacterial': 96148,\n",
       " 'gastroenteritis': 60052,\n",
       " 'abstract': 16908,\n",
       " 'fæcal': 59201,\n",
       " 'suspension': 132054,\n",
       " 'large': 80398,\n",
       " 'examine': 54430,\n",
       " 'inoculate': 74434,\n",
       " 'embryo': 51439,\n",
       " 'intestinal-organ': 75435,\n",
       " 'embryo-kidney': 51443,\n",
       " 'monolayer': 90169,\n",
       " 'villous': 142601,\n",
       " 'epithelium': 52985,\n",
       " 'detach': 45572,\n",
       " 'see': 124353,\n",
       " 'medium': 86429,\n",
       " 'multiply': 91697,\n",
       " 'demonstrate': 44950,\n",
       " 'indirect': 73409,\n",
       " 'immunofluorescence': 72509,\n",
       " 'convalescent': 40026,\n",
       " 'load': 82384,\n",
       " 'community': 38745,\n",
       " 'coronavirus-associated': 40390,\n",
       " 'prospective': 113161,\n",
       " 'temporal': 134116,\n",
       " 'radiological': 116103,\n",
       " 'manage': 85023,\n",
       " 'standard': 129659,\n",
       " 'protocol': 113511,\n",
       " 'ribavirin': 119682,\n",
       " 'corticosteroid': 40535,\n",
       " 'poor': 110133,\n",
       " 'outcome': 101712,\n",
       " 'usefulness': 140898,\n",
       " 'initially': 74305,\n",
       " 'improve': 72937,\n",
       " 'recurrent': 117540,\n",
       " 'sd': 124030,\n",
       " 'watery': 144345,\n",
       " 'diarrhoea': 46098,\n",
       " 'worsen': 145453,\n",
       " 'improvement': 72939,\n",
       " 'pulmonary': 114450,\n",
       " 'lesion': 81203,\n",
       " 'appearance': 23637,\n",
       " 'spontaneous': 129149,\n",
       " 'pneumomediastinum': 109275,\n",
       " 'distress': 47248,\n",
       " 'ards': 24013,\n",
       " 'quantitative': 115251,\n",
       " 'rt': 121410,\n",
       " 'admission': 18410,\n",
       " 'chronic': 36346,\n",
       " 'treat': 137522,\n",
       " 'lamivudine': 80269,\n",
       " 'independent': 73332,\n",
       " 'significant': 126574,\n",
       " 'p=0.001': 103346,\n",
       " 'faeces': 55520,\n",
       " 'seroconversion': 125360,\n",
       " 'shift': 126054,\n",
       " 'infiltrate': 73913,\n",
       " 'invert': 75867,\n",
       " 'viral-load': 142698,\n",
       " 'profile': 112749,\n",
       " 'unrelated': 140294,\n",
       " 'uncontrolled': 139733,\n",
       " 'immunopathological': 72624,\n",
       " 'http://image.thelancet.com/extras/03art4432web.pdf': 68861,\n",
       " 'clone': 37354,\n",
       " 'matrix': 85724,\n",
       " 'turkey': 138576,\n",
       " 'rhinotracheitis': 119559,\n",
       " 'order': 100961,\n",
       " 'different': 46314,\n",
       " 'biochemical': 28836,\n",
       " 'fusion': 59073,\n",
       " 'trtv': 138145,\n",
       " 'subfamily': 130869,\n",
       " 'pneumovirinae': 109335,\n",
       " 'paramyxoviridae': 104333,\n",
       " 'generate': 60597,\n",
       " 'polycistronic': 109784,\n",
       " 'cdna': 34480,\n",
       " 'k/3': 77997,\n",
       " 'bv/85-infected': 31269,\n",
       " 'oligonucleotide': 100133,\n",
       " 'primer': 112253,\n",
       " 'correspond': 40489,\n",
       " 'region': 117822,\n",
       " 'cdnas': 34492,\n",
       " 'adjacent': 18341,\n",
       " 'begin': 27679,\n",
       " '3′': 10851,\n",
       " 'end': 51777,\n",
       " 'i.e.': 70280,\n",
       " 'partial': 104598,\n",
       " '3′-m-f-5′.': 10894,\n",
       " 'unexpected': 139970,\n",
       " 'r': 115529,\n",
       " 'species': 128708,\n",
       " '3′-m-sh-g-f-5′': 10895,\n",
       " 'sh': 125846,\n",
       " 'small': 127675,\n",
       " 'hydrophobic': 69563,\n",
       " 'attachment': 25344,\n",
       " 'instead': 74609,\n",
       " 'resemble': 118585,\n",
       " 'morbillivirus': 90372,\n",
       " 'near': 94259,\n",
       " 'oligo': 100057,\n",
       " '5′': 12788,\n",
       " 'space': 128588,\n",
       " 'nucleotide': 98864,\n",
       " 'apart': 23393,\n",
       " 'basis': 27162,\n",
       " 'chain': 35344,\n",
       " 'reaction': 116959,\n",
       " 'genomic': 60723,\n",
       " 'template': 134098,\n",
       " 'product': 112690,\n",
       " 'kb': 78431,\n",
       " 'size': 127312,\n",
       " 'sa/91': 122535,\n",
       " '/78': 1901,\n",
       " 'uk/8544/85': 139361,\n",
       " 'sa/2381/88': 122534,\n",
       " 'grow': 62932,\n",
       " 'chicken': 35771,\n",
       " 'tracheal': 136815,\n",
       " 'perform': 106430,\n",
       " 'trtv-3bv': 138146,\n",
       " 'upstream': 140593,\n",
       " '1.14-kb': 3441,\n",
       " 'expect': 54713,\n",
       " 'situate': 127212,\n",
       " 'f.': 55218,\n",
       " 'absence': 16880,\n",
       " 'plus': 109069,\n",
       " 'demonstration': 44952,\n",
       " 'm-f': 84058,\n",
       " 'dicistronic': 46193,\n",
       " 'conclusion': 39215,\n",
       " 'genome': 60691,\n",
       " \"3'-m-f-5′.\": 9491,\n",
       " 'terminus': 134296,\n",
       " 'confirm': 39362,\n",
       " 'mrna': 90821,\n",
       " 'map': 85229,\n",
       " 'encode': 51746,\n",
       " 'amino': 20741,\n",
       " 'acid': 17565,\n",
       " 'residue': 118635,\n",
       " 'morbilliviruses': 90373,\n",
       " 'paramyxoviruses': 104341,\n",
       " 'approximately': 23730,\n",
       " 'closely': 37391,\n",
       " 'pneumoviruses': 109338,\n",
       " 'rapid': 116388,\n",
       " 'entry': 52468,\n",
       " 'pneumovirus-like': 109337,\n",
       " 'electronmicroscopy': 51164,\n",
       " 'causative': 33505,\n",
       " 'repiratory': 118350,\n",
       " 'determine': 45646,\n",
       " 'induce': 73524,\n",
       " 'actinomycin': 17793,\n",
       " 'polyadenylated': 109743,\n",
       " 'band': 26931,\n",
       " 'probably': 112527,\n",
       " 'radiolabelling': 116099,\n",
       " '[35s]methionine': 15187,\n",
       " '[3h]glucosamine': 15197,\n",
       " 'polypeptide': 109951,\n",
       " 'mock-infected': 89651,\n",
       " 'virion': 142785,\n",
       " 'nucleocapsids': 98817,\n",
       " 'lentil-lectin': 81129,\n",
       " 'chromatography': 36307,\n",
       " 'immunoprecipitation': 72653,\n",
       " 'glycosylated': 61962,\n",
       " 'gp': 62306,\n",
       " 'gp82': 62436,\n",
       " '82k': 14339,\n",
       " 'gp68': 62427,\n",
       " 'gp53': 62413,\n",
       " 'gp15': 62369,\n",
       " 'p43': 103093,\n",
       " 'p40': 103058,\n",
       " 'p35': 102967,\n",
       " 'analogous': 21067,\n",
       " 'glycopolypeptide': 61912,\n",
       " 'hn': 67440,\n",
       " 'precursor': 111674,\n",
       " 'f0': 55230,\n",
       " 'cleavage': 37151,\n",
       " 'f1': 55232,\n",
       " 'f2': 55265,\n",
       " 'phosphorylate': 107762,\n",
       " '200k': 7861,\n",
       " '22k': 8395,\n",
       " '97k': 14984,\n",
       " 'tunicamycin': 138538,\n",
       " 'inhibit': 74241,\n",
       " 'little': 82147,\n",
       " 'affect': 18810,\n",
       " 'glycans': 61825,\n",
       " '79k': 14030,\n",
       " 'weight': 144536,\n",
       " 'aminopeptidase': 20777,\n",
       " 'hcv': 65060,\n",
       " 'serogroup': 125375,\n",
       " 'hcv-229e': 65063,\n",
       " 'hcv-oc43': 65077,\n",
       " 'infections(1': 73794,\n",
       " 'cell-surface': 34778,\n",
       " 'metalloprotease': 87266,\n",
       " 'lung': 83307,\n",
       " 'kidney': 78735,\n",
       " 'cells(2–5': 34827,\n",
       " 'monoclonal': 90102,\n",
       " 'rb': 116592,\n",
       " 'fibroblast': 56688,\n",
       " 'immunoprecipitated': 72651,\n",
       " 'enzymatic': 52619,\n",
       " 'hcv-229e-resistant': 65064,\n",
       " 'murine': 91868,\n",
       " 'susceptible': 131995,\n",
       " 'transfection': 137151,\n",
       " 'complementary': 38918,\n",
       " 'dna': 47470,\n",
       " 'n.': 92779,\n",
       " 'contrast': 39954,\n",
       " 'expression': 54899,\n",
       " 'enhance': 52148,\n",
       " 'mutant': 91958,\n",
       " 'lack': 80079,\n",
       " 'catalytic': 33328,\n",
       " 'render': 118206,\n",
       " 'virus-binding': 142949,\n",
       " 'lie': 81583,\n",
       " 'active': 17830,\n",
       " 'molecule': 89910,\n",
       " 'identification': 70772,\n",
       " 'exist': 54589,\n",
       " 'hcov-229e': 64980,\n",
       " 'hcov-oc43': 65009,\n",
       " '(sars)-associated': 1346,\n",
       " '4': 10957,\n",
       " 'hcov-nl63': 65000,\n",
       " 'discovery': 46849,\n",
       " '7-month-old': 13666,\n",
       " 'suffer': 131248,\n",
       " 'conjunctivitis': 39502,\n",
       " 'complete': 38930,\n",
       " 'recombinant': 117368,\n",
       " 'vitro': 143255,\n",
       " 'host': 67979,\n",
       " 'notable': 98044,\n",
       " 'tertiary': 134340,\n",
       " 'monkey': 90031,\n",
       " 'llc-mk2': 82267,\n",
       " 'line': 81845,\n",
       " 'n-terminal': 92760,\n",
       " 'fragment': 58393,\n",
       " 'screen': 123967,\n",
       " 'specimen': 128764,\n",
       " 'individual': 73443,\n",
       " 'additional': 18153,\n",
       " 'hcov-nl63-infected': 65001,\n",
       " 'widely': 144994,\n",
       " 'spread': 129210,\n",
       " 'doi:10.1038/nm1024': 47824,\n",
       " 'epidemiological': 52837,\n",
       " 'determinant': 45636,\n",
       " 'health': 65212,\n",
       " 'authority': 25666,\n",
       " 'especially': 53599,\n",
       " 'asia': 24615,\n",
       " 'pacific': 103639,\n",
       " 'seek': 124364,\n",
       " 'effective': 50575,\n",
       " 'public-health': 114333,\n",
       " 'intervention': 75399,\n",
       " 'continue': 39905,\n",
       " 'epidemiology': 52847,\n",
       " 'april': 23749,\n",
       " 'integrate': 74732,\n",
       " 'database': 43912,\n",
       " 'construct': 39732,\n",
       " 'source': 128461,\n",
       " 'demographic': 44939,\n",
       " 'variable': 141628,\n",
       " 'key': 78637,\n",
       " 'distribution': 47254,\n",
       " 'onset': 100491,\n",
       " 'death': 44313,\n",
       " 'measure': 86240,\n",
       " 'fatality': 55871,\n",
       " 'rate': 116498,\n",
       " 'patients’age': 105113,\n",
       " 'exponential': 54842,\n",
       " 'growth': 62941,\n",
       " 'encouragement': 51759,\n",
       " 'hospital': 67919,\n",
       " 'rapidly': 116401,\n",
       " 'trace': 136804,\n",
       " 'suspect': 132043,\n",
       " 'quarantine': 115276,\n",
       " 'monitor': 90022,\n",
       " 'travel': 137488,\n",
       " 'incubation': 73303,\n",
       " 'cl': 36985,\n",
       " '5.2–7.7': 12111,\n",
       " 'long': 82565,\n",
       " '9.8–16.8': 14666,\n",
       " '35.2–52.4': 10295,\n",
       " 'assume': 24935,\n",
       " 'parametric': 104320,\n",
       " 'non-parametric': 96823,\n",
       " 'yield': 146290,\n",
       " '4.0–9.6': 11137,\n",
       " '45.3–64.7': 11510,\n",
       " 'course': 40822,\n",
       " 'strongly': 130481,\n",
       " 'alter': 20413,\n",
       " 'short': 126156,\n",
       " 'wide': 144982,\n",
       " 'place': 108651,\n",
       " 'http://image.thelancet.com/extras/03art4453web.pdf': 68862,\n",
       " 'conserve': 39625,\n",
       " 'proteome': 113465,\n",
       " 'sars-coronavirus': 123162,\n",
       " 'split-off': 129108,\n",
       " 'lineage': 81855,\n",
       " 'organization': 101154,\n",
       " 'strategy': 130324,\n",
       " 'predict': 111707,\n",
       " 'putative': 114622,\n",
       " 'open': 100582,\n",
       " 'read': 116988,\n",
       " 'frame': 58411,\n",
       " 'nest': 94683,\n",
       " 'experimentally': 54758,\n",
       " '4382-': 11440,\n",
       " 'replicase': 118366,\n",
       " 'polyprotein': 109993,\n",
       " 'cleave': 37157,\n",
       " 'subunit': 131148,\n",
       " 'proteinase': 113420,\n",
       " 'bring': 30569,\n",
       " 'total': 136590,\n",
       " 'phylogenetic': 107928,\n",
       " 'analysis': 21080,\n",
       " 'distantly': 47216,\n",
       " 'torovirus': 136548,\n",
       " 'outgroup': 101738,\n",
       " 'despite': 45518,\n",
       " 'distant': 47214,\n",
       " 'homolog': 67725,\n",
       " 'process': 112621,\n",
       " 'recognize': 117359,\n",
       " 'mechanism': 86274,\n",
       " 'completely': 38933,\n",
       " 'perspective': 106816,\n",
       " 'design': 45465,\n",
       " 'antiviral': 23175,\n",
       " 'aim': 19432,\n",
       " 'koch': 79140,\n",
       " 'postulate': 110834,\n",
       " 'globally': 61565,\n",
       " 'probable': 112520,\n",
       " 'provide': 113600,\n",
       " 'proof': 112996,\n",
       " 'scv': 124008,\n",
       " 'aetiological': 18751,\n",
       " 'understand': 139884,\n",
       " 'aetiology': 18754,\n",
       " 'expedite': 54731,\n",
       " 'development': 45714,\n",
       " 'therapy': 134941,\n",
       " 'vaccine': 141340,\n",
       " 'concise': 39210,\n",
       " 'doi:10.1038/423240a': 47774,\n",
       " 'resource': 118711,\n",
       " 'national': 93848,\n",
       " 'center': 34929,\n",
       " 'biotechnology': 29230,\n",
       " 'ncbi': 94012,\n",
       " 'suite': 131323,\n",
       " 'biological': 29009,\n",
       " 'genbank': 60459,\n",
       " 'nucleic': 98794,\n",
       " 'pubmed': 114360,\n",
       " 'citation': 36816,\n",
       " 'life': 81595,\n",
       " 'science': 123835,\n",
       " 'journal': 77701,\n",
       " 'entrez': 52459,\n",
       " 'search': 124171,\n",
       " 'retrieval': 119082,\n",
       " 'operation': 100624,\n",
       " 'e-utilities': 49696,\n",
       " 'serve': 125525,\n",
       " 'programme': 112817,\n",
       " 'interface': 75020,\n",
       " 'augment': 25519,\n",
       " 'web': 144470,\n",
       " 'application': 23666,\n",
       " 'custom': 42466,\n",
       " 'implementation': 72840,\n",
       " 'blast': 29526,\n",
       " 'program': 112808,\n",
       " 'optimize': 100760,\n",
       " 'specialize': 128698,\n",
       " 'release': 118077,\n",
       " 'past': 104782,\n",
       " 'management': 85025,\n",
       " 'refseq': 117776,\n",
       " 'element': 51216,\n",
       " 'download': 48334,\n",
       " 'variation': 141654,\n",
       " 'service': 125529,\n",
       " 'api': 23457,\n",
       " 'magic-blast': 84636,\n",
       " 'quickblastp': 115405,\n",
       " 'update': 140513,\n",
       " 'viewer': 142552,\n",
       " 'page': 103705,\n",
       " 'osiris': 101455,\n",
       " 'pubchem': 114326,\n",
       " 'access': 17306,\n",
       " 'home': 67631,\n",
       " 'www.ncbi.nlm.nih.gov': 145635,\n",
       " 'pathology': 104988,\n",
       " 'fatal': 55867,\n",
       " 'novel': 98128,\n",
       " 'global': 61546,\n",
       " 'impact': 72775,\n",
       " 'unclear': 139693,\n",
       " 'post-mortem': 110594,\n",
       " 'tissue': 135806,\n",
       " 'die': 46235,\n",
       " 'february': 56161,\n",
       " 'march': 85308,\n",
       " 'biopsy': 29144,\n",
       " 'histology': 66928,\n",
       " 'virology': 142844,\n",
       " 'autopsy': 25834,\n",
       " 'pathological': 104979,\n",
       " 'light': 81694,\n",
       " 'recent': 117187,\n",
       " 'universal': 140131,\n",
       " 'morphological': 90442,\n",
       " 'bronchial': 30654,\n",
       " 'denudation': 45108,\n",
       " 'loss': 82737,\n",
       " 'cilium': 36605,\n",
       " 'squamous': 129299,\n",
       " 'metaplasia': 87304,\n",
       " 'giant-cell': 61273,\n",
       " 'pronounce': 112987,\n",
       " 'macrophage': 84474,\n",
       " 'alveolus': 20508,\n",
       " 'interstitium': 75359,\n",
       " 'haemophagocytosis': 64216,\n",
       " 'pneumocytes': 109261,\n",
       " 'cytomegaly': 43166,\n",
       " 'granular': 62600,\n",
       " 'amphophilic': 20905,\n",
       " 'cytoplasm': 43194,\n",
       " 'atrophy': 25323,\n",
       " 'white': 144867,\n",
       " 'pulp': 114464,\n",
       " 'spleen': 129072,\n",
       " 'epithelial-cell': 52966,\n",
       " 'proliferation': 112883,\n",
       " 'contention': 39860,\n",
       " 'cytokine': 43109,\n",
       " 'dysregulation': 49222,\n",
       " 'partly': 104672,\n",
       " 'severity': 125668,\n",
       " 'acknowledge': 17665,\n",
       " 'http://image.thelancet.com/extras/03art4347web.pdf': 68860,\n",
       " 'trend': 137593,\n",
       " 'eids': 50851,\n",
       " 'burden': 31147,\n",
       " 'economy': 50290,\n",
       " 'health(1–3': 65214,\n",
       " 'emergence': 51479,\n",
       " 'think': 135099,\n",
       " 'drive': 48564,\n",
       " 'largely': 80422,\n",
       " 'socio-economic': 128031,\n",
       " 'environmental': 52580,\n",
       " 'ecological': 50271,\n",
       " 'factors(1–9': 55482,\n",
       " 'comparative': 38828,\n",
       " 'explicitly': 54810,\n",
       " 'linkage': 81904,\n",
       " 'spatial': 128637,\n",
       " 'eid': 50834,\n",
       " 'event': 54245,\n",
       " 'origin': 101228,\n",
       " 'non-random': 96955,\n",
       " 'significantly': 126575,\n",
       " 'bias': 28495,\n",
       " ...}"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_vectorizer.vocabulary_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A sparse matrix is built with 38.882 rows, one for each document, and 149.389 columns, one for each token."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n",
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   33.0s finished\n",
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 1 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   24.6s finished\n",
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 2 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   21.3s finished\n",
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 3 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   19.1s finished\n",
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 4 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   17.9s finished\n",
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 5 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   15.7s finished\n",
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 6 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   14.6s finished\n",
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 7 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   13.9s finished\n",
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 8 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   15.0s finished\n",
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 9 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   13.8s finished\n",
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 10 of max_iter: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   12.8s finished\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "LatentDirichletAllocation(batch_size=128, doc_topic_prior=None,\n",
       "                          evaluate_every=-1, learning_decay=0.7,\n",
       "                          learning_method='batch', learning_offset=10.0,\n",
       "                          max_doc_update_iter=100, max_iter=10,\n",
       "                          mean_change_tol=0.001, n_components=10, n_jobs=None,\n",
       "                          perp_tol=0.1, random_state=None,\n",
       "                          topic_word_prior=None, total_samples=1000000.0,\n",
       "                          verbose=2)"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.decomposition import LatentDirichletAllocation\n",
    "\n",
    "# Se ajusta un modelo LDA para identificar tópicos\n",
    "lda = LatentDirichletAllocation(\n",
    "    n_components=10, # number of topics\n",
    "    verbose=2,\n",
    ")\n",
    "lda = lda.fit(vectorized_docs)\n",
    "lda"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A continuación se mostrarán los tokens más relevantes de los diferentes temas identificados."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Topic #0: sequence protein structure antibody epitope analysis bind peptide study genome domain result method acid gene structural region identify base interaction\n",
      "\n",
      "Topic #1: virus vaccine human viral influenza disease development pathogen assay drug detection new method antiviral target potential infection use test system\n",
      "\n",
      "Topic #2: health disease public study care risk infectious nan system research control include review outbreak pandemic global patient influenza infection need\n",
      "\n",
      "Topic #3: patient respiratory infection virus child clinical viral pneumonia study influenza acute result case test de age severe associate year sample\n",
      "\n",
      "Topic #4: model case covid-19 transmission outbreak disease epidemic datum china spread numb estimate result study time rate infection coronavirus contact control\n",
      "\n",
      "Topic #5: infection mers-cov virus mouse respiratory vaccine syndrome middle east coronavirus response disease lung study antibody day animal result infect high\n",
      "\n",
      "Topic #6: cell response infection immune expression disease mouse role increase study induce viral level effect mechanism receptor human gene result function\n",
      "\n",
      "Topic #7: virus strain sample study sequence isolate coronavirus bat calve diarrhea cat detect pedv group infection analysis porcine antibody result gene\n",
      "\n",
      "Topic #8: sars de sars-cov et la des les acute en syndrome severe le coronavirus serum patient codon respiratory antibody infection nan\n",
      "\n",
      "Topic #9: protein virus cell viral rna replication gene coronavirus host membrane activity expression infection study bind sequence human result genome fusion\n"
     ]
    }
   ],
   "source": [
    "def print_topic_words(topic_model, vectorizer, num_words):\n",
    "    feature_names = vectorizer.get_feature_names()\n",
    "    for topic_idx, topic in enumerate(topic_model.components_):\n",
    "        message = \"\\nTopic #%d: \" % topic_idx\n",
    "        message += \" \".join([feature_names[i]\n",
    "                             for i in topic.argsort()[:-num_words - 1:-1]])\n",
    "        print(message)\n",
    "\n",
    "print_topic_words(lda, count_vectorizer, 20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ahora corresponde clasificar los diferentes papers en los temas identificados en el paso anterior."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR! Session/line number was not unique in database. History logging moved to new session 63\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n",
      "[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   14.6s finished\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array([[6.89740189e-04, 6.89802180e-04, 6.89828066e-04, 7.06973626e-01,\n",
       "        1.71530305e-01, 6.89808919e-04, 6.89793022e-04, 8.00286583e-02,\n",
       "        3.73286408e-02, 6.89798120e-04],\n",
       "       [1.42857889e-02, 1.42858855e-02, 1.42875152e-02, 1.42933597e-02,\n",
       "        1.42866198e-02, 1.42901097e-02, 1.42860976e-02, 1.42860173e-02,\n",
       "        8.71412087e-01, 1.42865195e-02],\n",
       "       [9.43747076e-04, 9.43621468e-04, 9.43632205e-04, 9.43586700e-04,\n",
       "        9.43548655e-04, 9.43617014e-04, 7.14733276e-02, 9.43674895e-04,\n",
       "        2.64524996e-01, 6.57396248e-01],\n",
       "       [3.68712454e-02, 1.29894830e-03, 1.29892554e-03, 6.96339470e-01,\n",
       "        1.29901699e-03, 1.29915550e-03, 2.57695754e-01, 1.29937567e-03,\n",
       "        1.29889846e-03, 1.29921047e-03],\n",
       "       [1.25010851e-03, 1.25013402e-03, 1.25015393e-03, 8.58983662e-01,\n",
       "        1.25024100e-03, 1.25028197e-03, 1.25016220e-03, 9.45583853e-02,\n",
       "        3.77066875e-02, 1.25018315e-03]])"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "docs_classified = lda.transform(vectorized_docs)\n",
    "docs_classified[:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finalmente, para cada uno de los temas identifiados, se imprimen los top-5 papers pertenecientes al tema, ordenados por su *pagerank*."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Topic ID #0\n",
      "Title: An efficient method to make human monoclonal antibodies from memory B cells: potent neutralization of SARS coronavirus\n",
      "Authors: Traggiai, Elisabetta; Becker, Stephan; Subbarao, Kanta; Kolesnikova, Larissa; Uematsu, Yasushi; Gismondo, Maria Rita; Murphy, Brian R; Rappuoli, Rino; Lanzavecchia, Antonio\n",
      "Publish time: 2004-07-11\n",
      "Abstract: Passive serotherapy can confer immediate protection against microbial infection, but methods to rapidly generate human neutralizing monoclonal antibodies are not yet available. We have developed an improved method for Epstein-Barr virus transformation of human B cells. We used this method to analyze the memory repertoire of a patient who recovered from severe acute respiratory syndrome coronavirus (SARS-CoV) infection and to isolate monoclonal antibodies specific for different viral proteins, including 35 antibodies with in vitro neutralizing activity ranging from 10(−8)M to 10(−11)M. One such antibody confers protection in vivo in a mouse model of SARS-CoV infection. These results show that it is possible to interrogate the memory repertoire of immune donors to rapidly and efficiently isolate neutralizing antibodies that have been selected in the course of natural infection. SUPPLEMENTARY INFORMATION: The online version of this article (doi:10.1038/nm1080) contains supplementary material, which is available to authorized users.\n",
      "\n",
      "Title: Antigenic structure of the E2 glycoprotein from transmissible gastroenteritis coronavirus\n",
      "Authors: Correa, Isabel; Jiménez, Gustavo; Suñé, Carlos; Bullido, María J.; Enjuanes, Luis\n",
      "Publish time: 1988-04-30\n",
      "Abstract: Abstract The antigenic structure of transmissible gastroenteritis (TGE) virus E2 glycoprotein has been defined at three levels: antigenic sites, antigenic subsites and epitopes. Four antigenic sites (A, B, C and D) were defined by competitive radioimmunoassay (RIA) using monoclonal antibodies (MAbs) selected from 9 fusions. About 20% (197) of the hybridomas specific for TGE virus produced neutralizing MAbs specific for site A, which was one of the antigenically dominant determinants. Site A was differentiated in three antigenic subsites: a, b and c, by characterization of 11 MAb resistant (mar) mutants, that were defined by 8, 3, and 3 MAbs, respectively. These subsites were further subdivided in epitopes. A total of 11 epitopes were defined in E2 glycoprotein, eight of which were critical for virus neutralization. Neutralizing MAbs were obtained only when native virus was used to immunize mice, although to produce hybridomas mice immunizations were made with antigen in the native, denatured, or mixtures of native and denatured form. All neutralizing MAbs reacted to conformational epitopes. The antigenic structure of the E2-glycoprotein has been defined with murine MAbs, but the antigenic sites were relevant in the swine, the natural host of the virus, because porcine sera reacted against these sites. MAbs specific for TGE virus site C reacted to non-immune porcine sera. This reactivity was not directed against porcine immunoglobulins. These results indicated that TGE virus contains epitope(s) also present in some non-immunoglobulin component of porcine serum.\n",
      "\n",
      "Title: Molecular basis of binding between novel human coronavirus MERS-CoV and its receptor CD26\n",
      "Authors: Lu, Guangwen; Hu, Yawei; Wang, Qihui; Qi, Jianxun; Gao, Feng; Li, Yan; Zhang, Yanfang; Zhang, Wei; Yuan, Yuan; Bao, Jinku; Zhang, Buchang; Shi, Yi; Yan, Jinghua; Gao, George F.\n",
      "Publish time: 2013-07-07\n",
      "Abstract: The newly emergent Middle East respiratory syndrome coronavirus (MERS-CoV) can cause severe pulmonary disease in humans(1,2), representing the second example of a highly pathogenic coronavirus, the first being SARS-CoV(3). CD26 (also known as dipeptidyl peptidase 4, DPP4) was recently identified as the cellular receptor for MERS-CoV(4). The engagement of the MERS-CoV spike protein with CD26 mediates viral attachment to host cells and virus–cell fusion, thereby initiating infection. Here we delineate the molecular basis of this specific interaction by presenting the first crystal structures of both the free receptor binding domain (RBD) of the MERS-CoV spike protein and its complex with CD26. Furthermore, binding between the RBD and CD26 is measured using real-time surface plasmon resonance with a dissociation constant of 16.7 nM. The viral RBD is composed of a core subdomain homologous to that of the SARS-CoV spike protein, and a unique strand-dominated external receptor binding motif that recognizes blades IV and V of the CD26 β-propeller. The atomic details at the interface between the two binding entities reveal a surprising protein–protein contact mediated mainly by hydrophilic residues. Sequence alignment indicates, among betacoronaviruses, a possible structural conservation for the region homologous to the MERS-CoV RBD core, but a high variation in the external receptor binding motif region for virus-specific pathogenesis such as receptor recognition. SUPPLEMENTARY INFORMATION: The online version of this article (doi:10.1038/nature12328) contains supplementary material, which is available to authorized users.\n",
      "\n",
      "Title: ZCURVE_CoV: a new system to recognize protein coding genes in coronavirus genomes, and its applications in analyzing SARS-CoV genomes\n",
      "Authors: Chen, Ling-Ling; Ou, Hong-Yu; Zhang, Ren; Zhang, Chun-Ting\n",
      "Publish time: 2003-07-25\n",
      "Abstract: Abstract A new system to recognize protein coding genes in the coronavirus genomes, specially suitable for the SARS-CoV genomes, has been proposed in this paper. Compared with some existing systems, the new program package has the merits of simplicity, high accuracy, reliability, and quickness. The system ZCURVE_CoV has been run for each of the 11 newly sequenced SARS-CoV genomes. Consequently, six genomes not annotated previously have been annotated, and some problems of previous annotations in the remaining five genomes have been pointed out and discussed. In addition to the polyprotein chain ORFs 1a and 1b and the four genes coding for the major structural proteins, spike (S), small envelop (E), membrane (M), and nuleocaspid (N), respectively, ZCURVE_CoV also predicts 5–6 putative proteins in length between 39 and 274 amino acids with unknown functions. Some single nucleotide mutations within these putative coding sequences have been detected and their biological implications are discussed. A web service is provided, by which a user can obtain the annotated result immediately by pasting the SARS-CoV genome sequences into the input window on the web site (http://tubic.tju.edu.cn/sars/). The software ZCURVE_CoV can also be downloaded freely from the web address mentioned above and run in computers under the platforms of Windows or Linux.\n",
      "\n",
      "Title: Molecular mimicry in T cell-mediated autoimmunity: Viral peptides activate human T cell clones specific for myelin basic protein\n",
      "Authors: Wucherpfennig, Kai W; Strominger, Jack L\n",
      "Publish time: 1995-03-10\n",
      "Abstract: Abstract Structural similarity between viral T cell epitopes and self-peptides could lead to the induction of an autoaggressive T cell response. Based on the structural requirements for both MHC class 11 binding and TCR recognition of an immunodominant myelin basic protein (MBP) peptide, criteria for a data base search were developed in which the degeneracy of amino acid side chains required for MHC class 11 binding and the conservation of those required for T cell activation were considered. A panel of 129 peptides that matched the molecular mimicry motif was tested on seven MBP-specific T cell clones from multiple sclerosis patients. Seven viral and one bacterial peptide efficiently activated three of these clones. Only one peptide could have been identified as a molecular mimic by sequence alignment. The observation that a single T cell receptor can recognize quite distinct but structurally related peptides from multiple pathogens has important implications for understanding the pathogenesis of autoimmunity.\n",
      "\n",
      "\n",
      "Topic ID #1\n",
      "Title: Koch's postulates fulfilled for SARS virus\n",
      "Authors: Fouchier, Ron A. M.; Kuiken, Thijs; Schutten, Martin; van Amerongen, Geert; van Doornum, Gerard J. J.; van den Hoogen, Bernadette G.; Peiris, Malik; Lim, Wilina; Stöhr, Klaus; Osterhaus, Albert D. M. E.\n",
      "Publish time: 2003\n",
      "Abstract: Severe acute respiratory syndrome (SARS) has recently emerged as a new human disease, resulting globally in 435 deaths from 6,234 probable cases (as of 3 May 2003). Here we provide proof from experimental infection of cynomolgus macaques (Macaca fascicularis) that the newly discovered SARS-associated coronavirus (SCV) is the aetiological agent of this disease. Our understanding of the aetiology of SARS will expedite the development of diagnostic tests, antiviral therapies and vaccines, and may allow a more concise case definition for this emerging disease. SUPPLEMENTARY INFORMATION: The online version of this article (doi:10.1038/423240a) contains supplementary material, which is available to authorized users.\n",
      "\n",
      "Title: Recent advances in the discovery and delivery of vaccine adjuvants\n",
      "Authors: O'Hagan, Derek T.; Valiante, Nicholas M.\n",
      "Publish time: 2003\n",
      "Abstract: Adjuvant design has historically had a touch of alchemy at its heart due to its reliance on the complex biology of innate immune activation. However, a new mechanistic understanding of innate immunity, combined with new adjuvant and delivery platforms for exploiting this knowledge, has led to significant advances recently. Although many challenges remain, the field is moving rapidly and the proper tools and methodologies are in place for the use of traditional drug discovery engines in guiding the development of vaccine adjuvants. In this review, we outline the current trends in immune potentiator, delivery system and adjuvant design that will shape the vaccines of the future.\n",
      "\n",
      "Title: Coronaviruses post-SARS: Update on replication and pathogenesis\n",
      "Authors: Perlman, Stanley; Netland, Jason\n",
      "Publish time: 2009-05-11\n",
      "Abstract: While coronaviruses were first identified nearly 60 years ago, they received notoriety in 2003 when one of their members was identified as the etiological agent of the Severe Acute Respiratory Syndrome (SARS-CoV) 1. Previously these viruses were known to be important agents of respiratory and enteric infections of domestic and companion animals and to cause approximately 15% of all cases of the common cold. This Review focuses on recent advances in our understanding of the mechanisms of coronavirus replication, interactions with the host immune response and disease pathogenesis and highlights the recent identification of numerous novel coronaviruses and the propensity of this virus family to cross species barriers.\n",
      "\n",
      "Title: Using siRNA in prophylactic and therapeutic regimens against SARS coronavirus in Rhesus macaque\n",
      "Authors: Li, Bao-jian; Tang, Qingquan; Cheng, Du; Qin, Chuan; Xie, Frank Y; Wei, Qiang; Xu, Jun; Liu, Yijia; Zheng, Bo-jian; Woodle, Martin C; Zhong, Nanshan; Lu, Patrick Y\n",
      "Publish time: 2005-08-21\n",
      "Abstract: Development of therapeutic agents for severe acute respiratory syndrome (SARS) viral infection using short interfering RNA (siRNA) inhibitors exemplifies a powerful new means to combat emerging infectious diseases. Potent siRNA inhibitors of SARS coronavirus (SCV) in vitro were further evaluated for efficacy and safety in a rhesus macaque (Macaca mulatta) SARS model using clinically viable delivery while comparing three dosing regimens. Observations of SARS-like symptoms, measurements of SCV RNA presence and lung histopathology and immunohistochemistry consistently showed siRNA-mediated anti-SARS efficacy by either prophylactic or therapeutic regimens. The siRNAs used provided relief from SCV infection–induced fever, diminished SCV viral levels and reduced acute diffuse alveoli damage. The 10–40 mg/kg accumulated dosages of siRNA did not show any sign of siRNA-induced toxicity. These results suggest that a clinical investigation is warranted and illustrate the prospects for siRNA to enable a massive reduction in development time for new targeted therapeutic agents. SUPPLEMENTARY INFORMATION: The online version of this article (doi:10.1038/nm1280) contains supplementary material, which is available to authorized users.\n",
      "\n",
      "Title: Viral Discovery and Sequence Recovery Using DNA Microarrays\n",
      "Authors: Wang, David; Urisman, Anatoly; Liu, Yu-Tsueng; Springer, Michael; Ksiazek, Thomas G; Erdman, Dean D; Mardis, Elaine R; Hickenbotham, Matthew; Magrini, Vincent; Eldred, James; Latreille, J. Phillipe; Wilson, Richard K; Ganem, Don; DeRisi, Joseph L\n",
      "Publish time: 2003-11-17\n",
      "Abstract: Because of the constant threat posed by emerging infectious diseases and the limitations of existing approaches used to identify new pathogens, there is a great demand for new technological methods for viral discovery. We describe herein a DNA microarray-based platform for novel virus identification and characterization. Central to this approach was a DNA microarray designed to detect a wide range of known viruses as well as novel members of existing viral families; this microarray contained the most highly conserved 70mer sequences from every fully sequenced reference viral genome in GenBank. During an outbreak of severe acute respiratory syndrome (SARS) in March 2003, hybridization to this microarray revealed the presence of a previously uncharacterized coronavirus in a viral isolate cultivated from a SARS patient. To further characterize this new virus, approximately 1 kb of the unknown virus genome was cloned by physically recovering viral sequences hybridized to individual array elements. Sequencing of these fragments confirmed that the virus was indeed a new member of the coronavirus family. This combination of array hybridization followed by direct viral sequence recovery should prove to be a general strategy for the rapid identification and characterization of novel viruses and emerging infectious disease.\n",
      "\n",
      "\n",
      "Topic ID #2\n",
      "Title: Database resources of the National Center for Biotechnology Information\n",
      "Authors: nan\n",
      "Publish time: 2018-01-04\n",
      "Abstract: The National Center for Biotechnology Information (NCBI) provides a large suite of online resources for biological information and data, including the GenBank(®) nucleic acid sequence database and the PubMed database of citations and abstracts for published life science journals. The Entrez system provides search and retrieval operations for most of these data from 39 distinct databases. The E-utilities serve as the programming interface for the Entrez system. Augmenting many of the Web applications are custom implementations of the BLAST program optimized to search specialized data sets. New resources released in the past year include PubMed Data Management, RefSeq Functional Elements, genome data download, variation services API, Magic-BLAST, QuickBLASTp, and Identical Protein Groups. Resources that were updated in the past year include the genome data viewer, a human genome resources page, Gene, virus variation, OSIRIS, and PubChem. All of these resources can be accessed through the NCBI home page at www.ncbi.nlm.nih.gov.\n",
      "\n",
      "Title: Global trends in emerging infectious diseases\n",
      "Authors: Jones, Kate E.; Patel, Nikkita G.; Levy, Marc A.; Storeygard, Adam; Balk, Deborah; Gittleman, John L.; Daszak, Peter\n",
      "Publish time: 2008-02-01\n",
      "Abstract: Emerging infectious diseases (EIDs) are a significant burden on global economies and public health(1–3). Their emergence is thought to be driven largely by socio-economic, environmental and ecological factors(1–9), but no comparative study has explicitly analysed these linkages to understand global temporal and spatial patterns of EIDs. Here we analyse a database of 335 EID ‘events’ (origins of EIDs) between 1940 and 2004, and demonstrate non-random global patterns. EID events have risen significantly over time after controlling for reporting bias, with their peak incidence (in the 1980s) concomitant with the HIV pandemic. EID events are dominated by zoonoses (60.3% of EIDs): the majority of these (71.8%) originate in wildlife (for example, severe acute respiratory virus, Ebola virus), and are increasing significantly over time. We find that 54.3% of EID events are caused by bacteria or rickettsia, reflecting a large number of drug-resistant microbes in our database. Our results confirm that EID origins are significantly correlated with socio-economic, environmental and ecological factors, and provide a basis for identifying regions where new EIDs are most likely to originate (emerging disease ‘hotspots’). They also reveal a substantial risk of wildlife zoonotic and vector-borne EIDs originating at lower latitudes where reporting effort is low. We conclude that global resources to counter disease emergence are poorly allocated, with the majority of the scientific and surveillance effort focused on countries from where the next important EID is least likely to originate.\n",
      "\n",
      "Title: The challenge of emerging and re-emerging infectious diseases\n",
      "Authors: Morens, David M.; Folkers, Gregory K.; Fauci, Anthony S.\n",
      "Publish time: 2004-07-08\n",
      "Abstract: Infectious diseases have for centuries ranked with wars and famine as major challenges to human progress and survival. They remain among the leading causes of death and disability worldwide. Against a constant background of established infections, epidemics of new and old infectious diseases periodically emerge, greatly magnifying the global burden of infections. Studies of these emerging infections reveal the evolutionary properties of pathogenic microorganisms and the dynamic relationships between microorganisms, their hosts and the environment.\n",
      "\n",
      "Title: Infectious Diseases Society of America/American Thoracic Society Consensus Guidelines on the Management of Community-Acquired Pneumonia in Adults\n",
      "Authors: Mandell, Lionel A.; Wunderink, Richard G.; Anzueto, Antonio; Bartlett, John G.; Campbell, G. Douglas; Dean, Nathan C.; Dowell, Scott F.; File, Thomas M.; Musher, Daniel M.; Niederman, Michael S.; Torres, Antonio; Whitney, Cynthia G.\n",
      "Publish time: 2007-03-01\n",
      "Abstract: nan\n",
      "\n",
      "Title: World Health Organization\n",
      "Authors: Ogbu, Uzor C.; Arah, Onyebuchi A.\n",
      "Publish time: 2017-12-31\n",
      "Abstract: This article is reproduced from the previous edition, volume 6, pp. 634–640, © 2008, Elsevier Inc.\n",
      "\n",
      "\n",
      "Topic ID #3\n",
      "Title: Coronavirus as a possible cause of severe acute respiratory syndrome\n",
      "Authors: Peiris, JSM; Lai, ST; Poon, LLM; Guan, Y; Yam, LYC; Lim, W; Nicholls, J; Yee, WKS; Yan, WW; Cheung, MT; Cheng, VCC; Chan, KH; Tsang, DNC; Yung, RWH; Ng, TK; Yuen, KY\n",
      "Publish time: 2003-04-19\n",
      "Abstract: Summary Background An outbreak of severe acute respiratory syndrome (SARS) has been reported in Hong Kong. We investigated the viral cause and clinical presentation among 50 patients. Methods We analysed case notes and microbiological findings for 50 patients with SARS, representing more than five separate epidemiologically linked transmission clusters. We defined the clinical presentation and risk factors associated with severe disease and investigated the causal agents by chest radiography and laboratory testing of nasopharyngeal aspirates and sera samples. We compared the laboratory findings with those submitted for microbiological investigation of other diseases from patients whose identity was masked. Findings Patients' age ranged from 23 to 74 years. Fever, chills, myalgia, and cough were the most frequent complaints. When compared with chest radiographic changes, respiratory symptoms and auscultatory findings were disproportionally mild. Patients who were household contacts of other infected people and had older age, lymphopenia, and liver dysfunction were associated with severe disease. A virus belonging to the family Coronaviridae was isolated from two patients. By use of serological and reverse-transcriptase PCR specific for this virus, 45 of 50 patients with SARS, but no controls, had evidence of infection with this virus. Interpretation A coronavirus was isolated from patients with SARS that might be the primary agent associated with this disease. Serological and molecular tests specific for the virus permitted a definitive laboratory diagnosis to be made and allowed further investigation to define whether other cofactors play a part in disease progression.\n",
      "\n",
      "Title: Coronavirus Pneumonia Following Autologous Bone Marrow Transplantation for Breast Cancer\n",
      "Authors: Folz, Rodney J.; Elkordy, Maha A.\n",
      "Publish time: 1999-03-31\n",
      "Abstract: Infectious bronchitis virus, otherwise known as coronavirus, can cause mild upper respiratory tract illnesses in children and adults. Rarely has coronavirus been linked, either by serology or nasal wash, to pneumonia. We report a case of a young woman who, following treatment for stage IIIA breast cancer using a high-dose chemotherapy regimen followed by autologous bone marrow and stem cell transplantation, developed respiratory failure and was found to have coronavirus pneumonia as diagnosed by electron microscopy from BAL fluid. We propose that coronavirus should be considered in the differential diagnosis of acute respiratory failure in cancer patients who have undergone high-dose chemotherapy and autologous hematopoietic support.\n",
      "\n",
      "Title: Coronavirus Infection in Acute Lower Respiratory Tract Disease of Infants\n",
      "Authors: McIntosh, Kenneth; Chao, Ru Kwa; Krause, Helen E.; Wasil, Raymond; Mocega, Hilda E.; Mufson, Maurice A.\n",
      "Publish time: 2004-09-01\n",
      "Abstract: A serologic surveillance of lower respiratory tract disease in 417 hospitalized children under 18 months of age revealed infection with coronviruses (strains OC43 and/ or 229E) in 34 (8.2%). During the same interval, one of 13 control infants was infected. There were two distinct periods lasting six and 14 weeks, respectively, during which the incidence rose to as high as 18.9 % of patients with lower respiratory tract disease. The incidence of coronavirus infection in patients with pneumonia and bronchiolitis was higher than the incidences of adenoviruses, influenza, parainfluenza viruses types 1 and 2, and rhinoviruses, and lower only than the incidences of parainfluenza virus type 3 and respiratory syncytial virus. Coronoviruses serologically similar or identical to strain 229E were recovered from frozen nasal washes obtained during the acute phase of pneumonia in two children.\n",
      "\n",
      "Title: THE COMMON COLD\n",
      "Authors: Kirkpatrick, George L.\n",
      "Publish time: 1996-12-01\n",
      "Abstract: The common cold has intrigued physicians and the general public for centuries. It has been defined as an acute epidemic respiratory disease characterized by mild coryzal symptoms of rhinorrhea, nasal obstruction, and sneezing. The nasal discharge is usually copious and thin during the first 2 days of illness, then it generally becomes more viscous and purulent. 22 The disease is self-limited. Symptoms may persist for 2 days to more than 14 days; however, the cold may abort after only 1 day. Fever, cough, sore throat, or lacrimation may or may not be present. The common cold is of itself harmless, but bacterial invasion frequently follows the initial infection. It is these secondary invaders that may produce disorders of serious consequence. The common cold is the most frequent acute illness in the United States and throughout the industrialized world. About half the population gets at least one cold every year. 5 Colds account for 40% of all time lost from jobs among employed people (23 million days of work per year) and about 30% of absenteeism from schools (26 million school days per year). Estimates vary, but the average preschool child has somewhere between 4 and 10 colds per year, and the average adult has about two to four colds per year. The actual cost of caring for patients with colds in US physicians' offices is estimated to be $1.5 billion annually. 64 Seasonal patterns of infection can be identified for some of the various types of viruses that are responsible for outbreaks of the common cold. For example, available epidemiologic data suggest early fall and late spring are the most common times to find more outbreaks of rhinovirus. Respiratory syncytial virus (RSV) tends to follow winter and spring incidence, with a peak number of cases found mainly in January. Parainfluenza types 1 and 2 seem to peak during the autumn, whereas parainfluenza type 3 has an increased incidence during the late spring. Adenoviruses and coronaviruses tend to produce epidemics during the winter and spring (Fig. 1) This article presents data concerning the cause, pathogenesis, and treatment of the common cold, as well as discussion of the available diagnostic tests and their use in formulating differential diagnoses.\n",
      "\n",
      "Title: A newly discovered human pneumovirus isolated from young children with respiratory tract disease\n",
      "Authors: van den Hoogen, Bernadette G.; de Jong, Jan C.; Groen, Jan; Kuiken, Thijs; de Groot, Ronald; Fouchier, Ron A.M.; Osterhaus, Albert D.M.E.\n",
      "Publish time: 2001\n",
      "Abstract: From 28 young children in the Netherlands, we isolated a paramyxovirus that was identified as a tentative new member of the Metapneumovirus genus based on virological data, sequence homology and gene constellation. Previously, avian pneumovirus was the sole member of this recently assigned genus, hence the provisional name for the newly discovered virus: human metapneumovirus. The clinical symptoms of the children from whom the virus was isolated were similar to those caused by human respiratory syncytial virus infection, ranging from upper respiratory tract disease to severe bronchiolitis and pneumonia. Serological studies showed that by the age of five years, virtually all children in the Netherlands have been exposed to human metapneumovirus and that the virus has been circulating in humans for at least 50 years.\n",
      "\n",
      "\n",
      "Topic ID #4\n",
      "Title: Epidemiological determinants of spread of causal agent of severe acute respiratory syndrome in Hong Kong\n",
      "Authors: Donnelly, Christl A; Ghani, Azra C; Leung, Gabriel M; Hedley, Anthony J; Fraser, Christophe; Riley, Steven; Abu-Raddad, Laith J; Ho, Lai-Ming; Thach, Thuan-Quoc; Chau, Patsy; Chan, King-Pan; Lam, Tai-Hing; Tse, Lai-Yin; Tsang, Thomas; Liu, Shao-Haei; Kong, James HB; Lau, Edith MC; Ferguson, Neil M; Anderson, Roy M\n",
      "Publish time: 2003-05-24\n",
      "Abstract: Summary Background Health authorities worldwide, especially in the Asia Pacific region, are seeking effective public-health interventions in the continuing epidemic of severe acute respiratory syndrome (SARS). We assessed the epidemiology of SARS in Hong Kong. Methods We included 1425 cases reported up to April 28, 2003. An integrated database was constructed from several sources containing information on epidemiological, demographic, and clinical variables. We estimated the key epidemiological distributions: infection to onset, onset to admission, admission to death, and admission to discharge. We measured associations between the estimated case fatality rate and patients’age and the time from onset to admission. Findings After the initial phase of exponential growth, the rate of confirmed cases fell to less than 20 per day by April 28. Public-health interventions included encouragement to report to hospital rapidly after the onset of clinical symptoms, contact tracing for confirmed and suspected cases, and quarantining, monitoring, and restricting the travel of contacts. The mean incubation period of the disease is estimated to be 6.4 days (95% Cl 5.2–7.7). The mean time from onset of clinical symptoms to admission to hospital varied between 3 and 5 days, with longer times earlier in the epidemic. The estimated case fatality rate was 13.2% (9.8–16.8) for patients younger than 60 years and 43.3% (35.2–52.4) for patients aged 60 years or older assuming a parametric γ distribution. A non-parametric method yielded estimates of 6.8% (4.0–9.6) and 55.0% (45.3–64.7), respectively. Case clusters have played an important part in the course of the epidemic. Interpretation Patients’age was strongly associated with outcome. The time between onset of symptoms and admission to hospital did not alter outcome, but shorter intervals will be important to the wider population by restricting the infectious period before patients are placed in quarantine. Published online May 7, 2003 http://image.thelancet.com/extras/03art4453web.pdf\n",
      "\n",
      "Title: Coronaviruses\n",
      "Authors: Monto, Arnold S.; Cowling, Benjamin J.; Peiris, J. S. Malik\n",
      "Publish time: 2014-02-27\n",
      "Abstract: Coronaviruses of humans were first identified more than 60 years ago from individuals with respiratory infections, mainly mild. Two different viruses, 229E and OC43 were initially recognized. Because of difficulty in isolating them using standard techniques, many of the early studies of their occurrence were seroepidemiologic. They were confirmed to be worldwide in distribution, and, in the North Temperate Zone, mainly occurring in the winter season. With the development of the reverse transcriptase polymerase chain reaction (PCR) technique, two additional distinct viruses have been identified, HKU1 and NL63. The four viruses have now been recognized as important in the etiology of common respiratory infections, second only to the rhinoviruses. In 2002, a previously unrecognized betacoronavirus emerged from a zoonotic reservoir in Southern China and spread during the following year to several major cities of the world. The resulting illness was termed Severe Acute Respiratory Syndrome (SARS) because of its potential lethality. More than 8,000 probable cases were reported during 2003, mainly from Hong Kong and mainland China, producing social and economic disruption in those areas affected. A constant feature of the outbreak was the importance of nosocomial spread. In spite of an estimated basic reproductive number higher than influenza, the outbreak was ended, in large part because of control of in-hospital transmission. In 2012, another betacoronavirus has emerged in the Arabian peninsula which is producing a somewhat similar illness, termed Middle East Respiratory Syndrome (MERS), also marked by extensive nosocomial transmission. The outcome of this emergence is currently unknown.\n",
      "\n",
      "Title: Different Epidemic Curves for Severe Acute Respiratory Syndrome Reveal Similar Impacts of Control Measures\n",
      "Authors: Wallinga, Jacco; Teunis, Peter\n",
      "Publish time: 2004-09-15\n",
      "Abstract: Severe acute respiratory syndrome (SARS) has been the first severe contagious disease to emerge in the 21st century. The available epidemic curves for SARS show marked differences between the affected regions with respect to the total number of cases and epidemic duration, even for those regions in which outbreaks started almost simultaneously and similar control measures were implemented at the same time. The authors developed a likelihood-based estimation procedure that infers the temporal pattern of effective reproduction numbers from an observed epidemic curve. Precise estimates for the effective reproduction numbers were obtained by applying this estimation procedure to available data for SARS outbreaks that occurred in Hong Kong, Vietnam, Singapore, and Canada in 2003. The effective reproduction numbers revealed that epidemics in the various affected regions were characterized by markedly similar disease transmission potentials and similar levels of effectiveness of control measures. In controlling SARS outbreaks, timely alerts have been essential: Delaying the institution of control measures by 1 week would have nearly tripled the epidemic size and would have increased the expected epidemic duration by 4 weeks.\n",
      "\n",
      "Title: A familial cluster of pneumonia associated with the 2019 novel coronavirus indicating person-to-person transmission: a study of a family cluster\n",
      "Authors: Chan, Jasper Fuk-Woo; Yuan, Shuofeng; Kok, Kin-Hang; To, Kelvin Kai-Wang; Chu, Hin; Yang, Jin; Xing, Fanfan; Liu, Jieling; Yip, Cyril Chik-Yan; Poon, Rosana Wing-Shan; Tsoi, Hoi-Wah; Lo, Simon Kam-Fai; Chan, Kwok-Hung; Poon, Vincent Kwok-Man; Chan, Wan-Mui; Ip, Jonathan Daniel; Cai, Jian-Piao; Cheng, Vincent Chi-Chung; Chen, Honglin; Hui, Christopher Kim-Ming; Yuen, Kwok-Yung\n",
      "Publish time: 2020-02-21\n",
      "Abstract: Summary Background An ongoing outbreak of pneumonia associated with a novel coronavirus was reported in Wuhan city, Hubei province, China. Affected patients were geographically linked with a local wet market as a potential source. No data on person-to-person or nosocomial transmission have been published to date. Methods In this study, we report the epidemiological, clinical, laboratory, radiological, and microbiological findings of five patients in a family cluster who presented with unexplained pneumonia after returning to Shenzhen, Guangdong province, China, after a visit to Wuhan, and an additional family member who did not travel to Wuhan. Phylogenetic analysis of genetic sequences from these patients were done. Findings From Jan 10, 2020, we enrolled a family of six patients who travelled to Wuhan from Shenzhen between Dec 29, 2019 and Jan 4, 2020. Of six family members who travelled to Wuhan, five were identified as infected with the novel coronavirus. Additionally, one family member, who did not travel to Wuhan, became infected with the virus after several days of contact with four of the family members. None of the family members had contacts with Wuhan markets or animals, although two had visited a Wuhan hospital. Five family members (aged 36–66 years) presented with fever, upper or lower respiratory tract symptoms, or diarrhoea, or a combination of these 3–6 days after exposure. They presented to our hospital (The University of Hong Kong-Shenzhen Hospital, Shenzhen) 6–10 days after symptom onset. They and one asymptomatic child (aged 10 years) had radiological ground-glass lung opacities. Older patients (aged >60 years) had more systemic symptoms, extensive radiological ground-glass lung changes, lymphopenia, thrombocytopenia, and increased C-reactive protein and lactate dehydrogenase levels. The nasopharyngeal or throat swabs of these six patients were negative for known respiratory microbes by point-of-care multiplex RT-PCR, but five patients (four adults and the child) were RT-PCR positive for genes encoding the internal RNA-dependent RNA polymerase and surface Spike protein of this novel coronavirus, which were confirmed by Sanger sequencing. Phylogenetic analysis of these five patients' RT-PCR amplicons and two full genomes by next-generation sequencing showed that this is a novel coronavirus, which is closest to the bat severe acute respiatory syndrome (SARS)-related coronaviruses found in Chinese horseshoe bats. Interpretation Our findings are consistent with person-to-person transmission of this novel coronavirus in hospital and family settings, and the reports of infected travellers in other geographical regions. Funding The Shaw Foundation Hong Kong, Michael Seak-Kan Tong, Respiratory Viral Research Foundation Limited, Hui Ming, Hui Hoy and Chow Sin Lan Charity Fund Limited, Marina Man-Wai Lee, the Hong Kong Hainan Commercial Association South China Microbiology Research Fund, Sanming Project of Medicine (Shenzhen), and High Level-Hospital Program (Guangdong Health Commission).\n",
      "\n",
      "Title: Severe Acute Respiratory Syndrome (SARS) in Singapore: Clinical Features of Index Patient and Initial Contacts\n",
      "Authors: Hsu, Li-Yang; Lee, Cheng-Chuan; Green, Justin A.; Ang, Brenda; Paton, Nicholas I.; Lee, Lawrence; Villacian, Jorge S.; Lim, Poh-Lian; Earnest, Arul; Leo, Yee-Sin\n",
      "Publish time: 2003-06-10\n",
      "Abstract: Severe acute respiratory syndrome (SARS) is an emerging viral infectious disease. One of the largest outbreaks of SARS to date began in Singapore in March 2003. We describe the clinical, laboratory, and radiologic features of the index patient and the patient’s initial contacts affected with probable SARS.\n",
      "\n",
      "\n",
      "Topic ID #5\n",
      "Title: Newly discovered coronavirus as the primary cause of severe acute respiratory syndrome\n",
      "Authors: Kuiken, Thijs; Fouchier, Ron AM; Schutten, Martin; Rimmelzwaan, Guus F; van Amerongen, Geert; van Riel, Debby; Laman, Jon D; de Jong, Ton; van Doornum, Gerard; Lim, Wilina; Ling, Ai Ee; Chan, Paul KS; Tam, John S; Zambon, Maria C; Gopal, Robin; Drosten, Christian; van der Werf, Sylvie; Escriou, Nicolas; Manuguerra, Jean-Claude; Stöhr, Klaus; Peiris, J S Malik; Osterhaus, Albert DME\n",
      "Publish time: 2003-07-26\n",
      "Abstract: Summary Background The worldwide outbreak of severe acute respiratory syndrome (SARS) is associated with a newly discovered coronavirus, SARS-associated coronavirus (SARSCoV). We did clinical and experimental studies to assess the role of this virus in the cause of SARS. Methods We tested clinical and postmortem samples from 436 SARS patients in six countries for infection with SARSCoV, human metapneumovirus, and other respiratory pathogens. We infected four cynomolgus macaques (Macaca fascicularis) with SARS-CoV in an attempt to replicate SARS and did necropsies on day 6 after infection. Findings SARS-CoV infection was diagnosed in 329 (75%) of 436 patients fitting the case definition of SARS; human metapneumovirus was diagnosed in 41 (12%) of 335, and other respiratory pathogens were diagnosed only sporadically. SARS-CoV was, therefore, the most likely causal agent of SARS. The four SARS-CoV-infected macaques excreted SARS-CoV from nose, mouth, and pharynx from 2 days after infection. Three of four macaques developed diffuse alveolar damage, similar to that in SARS patients, and characterised by epithelial necrosis, serosanguineous exudate, formation of hyaline membranes, type 2 pneumocyte hyperplasia, and the presence of syncytia. SARS-CoV was detected in pneumonic areas by virus isolation and RT-PCR, and was localised to alveolar epithelial cells and syncytia by immunohistochemistry and transmission electron microscopy. Interpretation Replication in SARS-CoV-infected macaques of pneumonia similar to that in human beings with SARS, combined with the high prevalence of SARS-CoV infection in SARS patients, fulfill the criteria required to prove that SARS-CoV is the primary cause of SARS. Published online July 22, 2003 http://image.thelancet.com/extras/03art6318web.pdf\n",
      "\n",
      "Title: Lung pathology of fatal severe acute respiratory syndrome\n",
      "Authors: Nicholls, John M; Poon, Leo LM; Lee, Kam C; Ng, Wai F; Lai, Sik T; Leung, Chung Y; Chu, Chung M; Hui, Pak K; Mak, Kong L; Lim, Wilna; Yan, Kin W; Chan, Kwok H; Tsang, Ngai C; Guan, Yi; Yuen, Kwok Y; Malik Peiris, JS\n",
      "Publish time: 2003-05-24\n",
      "Abstract: Summary Background Severe acute respiratory syndrome (SARS) is a novel infectious disease with global impact. A virus from the family Coronaviridae has been identified as the cause, but the pathogenesis is still unclear. Methods Post-mortem tissue samples from six patients who died from SARS in February and March, 2003, and an open lung biopsy from one of these patients were studied by histology and virology. Only one full autopsy was done. Evidence of infection with the SARS-associated coronavirus (SARS-CoV) and human metapneumovirus was sought by reverse-transcriptase PCR and serology. Pathological samples were examined by light and electron microscopy and immunohistochemistry. Findings All six patients had serological evidence of recent infection with SARS-CoV. Diffuse alveolar damage was common but not universal. Morphological changes identified were bronchial epithelial denudation, loss of cilia, and squamous metaplasia. Secondary bacterial pneumonia was present in one case. A giant-cell infiltrate was seen in four patients, with a pronounced increase in macrophages in the alveoli and the interstitium of the lung. Haemophagocytosis was present in two patients. The alveolar pneumocytes also showed cytomegaly with granular amphophilic cytoplasm. The patient for whom full autopsy was done had atrophy of the white pulp of the spleen. Electron microscopy revealed viral particles in the cytoplasm of epithelial cells corresponding to coronavirus. Interpretation SARS is associated with epithelial-cell proliferation and an increase in macrophages in the lung. The presence of haemophagocytosis supports the contention that cytokine dysregulation may account, at least partly, for the severity of the clinical disease. The case definition of SARS should acknowledge the range of lung pathology associated with this disease. Published online May 16, 2003 http://image.thelancet.com/extras/03art4347web.pdf\n",
      "\n",
      "Title: Re-emergence of fatal human influenza A subtype H5N1 disease\n",
      "Authors: Peiris, JSM; Yu, WC; Leung, CW; Cheung, CY; Ng, WF; Nicholls, JM; Ng, TK; Chan, KH; Lai, ST; Lim, WL; Yuen, KY; Guan, Y\n",
      "Publish time: 2004-02-21\n",
      "Abstract: Summary Human disease associated with influenza A subtype H5N1 reemerged in January, 2003, for the first time since an outbreak in Hong Kong in 1997. Patients with H5N1 disease had unusually high serum concentrations of chemokines (eg, interferon induced protein-10 [IP-10] and monokine induced by interferon γ [MIG]). Taken together with a previous report that H5N1 influenza viruses induce large amounts of proinflam-matory cytokines from macrophage cultures in vitro, our findings suggest that cytokine dysfunction contributes to the pathogenesis of H5N1 disease. Development of vaccines against influenza A (H5N1) virus should be made a priority.\n",
      "\n",
      "Title: A DNA vaccine induces SARS coronavirus neutralization and protective immunity in mice\n",
      "Authors: Yang, Zhi-yong; Kong, Wing-pui; Huang, Yue; Roberts, Anjeanette; Murphy, Brian R.; Subbarao, Kanta; Nabel, Gary J.\n",
      "Publish time: 2004\n",
      "Abstract: Public health measures have successfully identified and contained outbreaks of the severe acute respiratory syndrome (SARS) coronavirus (SARS-CoV)(1,2,3,4,5), but concerns remain over the possibility of future recurrences. Finding a vaccine for this virus therefore remains a high priority. Here, we show that a DNA vaccine encoding the spike (S) glycoprotein of the SARS-CoV induces T cell and neutralizing antibody responses, as well as protective immunity, in a mouse model. Alternative forms of S were analysed by DNA immunization. These expression vectors induced robust immune responses mediated by CD4 and CD8 cells, as well as significant antibody titres, measured by enzyme-linked immunosorbent assay. Moreover, antibody responses in mice vaccinated with an expression vector encoding a form of S that includes its transmembrane domain elicited neutralizing antibodies. Viral replication was reduced by more than six orders of magnitude in the lungs of mice vaccinated with these S plasmid DNA expression vectors, and protection was mediated by a humoral but not a T-cell-dependent immune mechanism. Gene-based vaccination for the SARS-CoV elicits effective immune responses that generate protective immunity in an animal model. SUPPLEMENTARY INFORMATION: The online version of this article (doi:10.1038/nature02463) contains supplementary material, which is available to authorized users.\n",
      "\n",
      "Title: Middle East respiratory syndrome coronavirus neutralising serum antibodies in dromedary camels: a comparative serological study\n",
      "Authors: Reusken, Chantal BEM; Haagmans, Bart L; Müller, Marcel A; Gutierrez, Carlos; Godeke, Gert-Jan; Meyer, Benjamin; Muth, Doreen; Raj, V Stalin; Vries, Laura Smits-De; Corman, Victor M; Drexler, Jan-Felix; Smits, Saskia L; El Tahir, Yasmin E; De Sousa, Rita; van Beek, Janko; Nowotny, Norbert; van Maanen, Kees; Hidalgo-Hermoso, Ezequiel; Bosch, Berend-Jan; Rottier, Peter; Osterhaus, Albert; Gortázar-Schmidt, Christian; Drosten, Christian; Koopmans, Marion PG\n",
      "Publish time: 2013-10-31\n",
      "Abstract: Summary Background A new betacoronavirus—Middle East respiratory syndrome coronavirus (MERS-CoV)—has been identified in patients with severe acute respiratory infection. Although related viruses infect bats, molecular clock analyses have been unable to identify direct ancestors of MERS-CoV. Anecdotal exposure histories suggest that patients had been in contact with dromedary camels or goats. We investigated possible animal reservoirs of MERS-CoV by assessing specific serum antibodies in livestock. Methods We took sera from animals in the Middle East (Oman) and from elsewhere (Spain, Netherlands, Chile). Cattle (n=80), sheep (n=40), goats (n=40), dromedary camels (n=155), and various other camelid species (n=34) were tested for specific serum IgG by protein microarray using the receptor-binding S1 subunits of spike proteins of MERS-CoV, severe acute respiratory syndrome coronavirus, and human coronavirus OC43. Results were confirmed by virus neutralisation tests for MERS-CoV and bovine coronavirus. Findings 50 of 50 (100%) sera from Omani camels and 15 of 105 (14%) from Spanish camels had protein-specific antibodies against MERS-CoV spike. Sera from European sheep, goats, cattle, and other camelids had no such antibodies. MERS-CoV neutralising antibody titres varied between 1/320 and 1/2560 for the Omani camel sera and between 1/20 and 1/320 for the Spanish camel sera. There was no evidence for cross-neutralisation by bovine coronavirus antibodies. Interpretation MERS-CoV or a related virus has infected camel populations. Both titres and seroprevalences in sera from different locations in Oman suggest widespread infection. Funding European Union, European Centre For Disease Prevention and Control, Deutsche Forschungsgemeinschaft.\n",
      "\n",
      "\n",
      "Topic ID #6\n",
      "Title: Dipeptidyl peptidase 4 is a functional receptor for the emerging human coronavirus-EMC\n",
      "Authors: Raj, V. Stalin; Mou, Huihui; Smits, Saskia L.; Dekkers, Dick H. W.; Müller, Marcel A.; Dijkman, Ronald; Muth, Doreen; Demmers, Jeroen A. A.; Zaki, Ali; Fouchier, Ron A. M.; Thiel, Volker; Drosten, Christian; Rottier, Peter J. M.; Osterhaus, Albert D. M. E.; Bosch, Berend Jan; Haagmans, Bart L.\n",
      "Publish time: 2013-03-13\n",
      "Abstract: Most human coronaviruses cause mild upper respiratory tract disease but may be associated with more severe pulmonary disease in immunocompromised individuals(1). However, SARS coronavirus caused severe lower respiratory disease with nearly 10% mortality and evidence of systemic spread(2). Recently, another coronavirus (human coronavirus-Erasmus Medical Center (hCoV-EMC)) was identified in patients with severe and sometimes lethal lower respiratory tract infection(3,4). Viral genome analysis revealed close relatedness to coronaviruses found in bats(5). Here we identify dipeptidyl peptidase 4 (DPP4; also known as CD26) as a functional receptor for hCoV-EMC. DPP4 specifically co-purified with the receptor-binding S1 domain of the hCoV-EMC spike protein from lysates of susceptible Huh-7 cells. Antibodies directed against DPP4 inhibited hCoV-EMC infection of primary human bronchial epithelial cells and Huh-7 cells. Expression of human and bat (Pipistrellus pipistrellus) DPP4 in non-susceptible COS-7 cells enabled infection by hCoV-EMC. The use of the evolutionarily conserved DPP4 protein from different species as a functional receptor provides clues about the host range potential of hCoV-EMC. In addition, it will contribute critically to our understanding of the pathogenesis and epidemiology of this emerging human coronavirus, and may facilitate the development of intervention strategies. SUPPLEMENTARY INFORMATION: The online version of this article (doi:10.1038/nature12005) contains supplementary material, which is available to authorized users.\n",
      "\n",
      "Title: Kawasaki Disease\n",
      "Authors: Kato, Hirohisa\n",
      "Publish time: 2007\n",
      "Abstract: Kawasaki disease is an acute systemic vasculitis syndrome in infants and young children of unknown etiology, which mainly affects small and medium-sized arteries, particularly the coronary artery. Clinical features include an acute febrile illness associated with mucocutaneous involvements and lymphadenopathy. The coronary aneurysms develop in about 10% to 20% of patients, some of which may cause myocardial infarction or lead to ischemic heart disease. Combinations of high-dose gamma globulin and aspirin are the essential treatments for acute Kawasaki disease patients. Bypass surgery and the catheter intervention are the optional treatments for the coronary artery sequelae of Kawasaki disease. Some patients with coronary artery sequelae may develop into adult coronary artery disease.\n",
      "\n",
      "Title: Angiotensin-converting enzyme 2 protects from severe acute lung failure\n",
      "Authors: Imai, Yumiko; Kuba, Keiji; Rao, Shuan; Huan, Yi; Guo, Feng; Guan, Bin; Yang, Peng; Sarao, Renu; Wada, Teiji; Leong-Poi, Howard; Crackower, Michael A.; Fukamizu, Akiyoshi; Hui, Chi-Chung; Hein, Lutz; Uhlig, Stefan; Slutsky, Arthur S.; Jiang, Chengyu; Penninger, Josef M.\n",
      "Publish time: 2005\n",
      "Abstract: Acute respiratory distress syndrome (ARDS), the most severe form of acute lung injury, is a devastating clinical syndrome with a high mortality rate (30–60%) (refs 1–3). Predisposing factors for ARDS are diverse(1,3) and include sepsis, aspiration, pneumonias and infections with the severe acute respiratory syndrome (SARS) coronavirus(4,5). At present, there are no effective drugs for improving the clinical outcome of ARDS(1,2,3). Angiotensin-converting enzyme (ACE) and ACE2 are homologues with different key functions in the renin–angiotensin system(6,7,8). ACE cleaves angiotensin I to generate angiotensin II, whereas ACE2 inactivates angiotensin II and is a negative regulator of the system. ACE2 has also recently been identified as a potential SARS virus receptor and is expressed in lungs(9,10). Here we report that ACE2 and the angiotensin II type 2 receptor (AT(2)) protect mice from severe acute lung injury induced by acid aspiration or sepsis. However, other components of the renin–angiotensin system, including ACE, angiotensin II and the angiotensin II type 1a receptor (AT(1)a), promote disease pathogenesis, induce lung oedemas and impair lung function. We show that mice deficient for Ace show markedly improved disease, and also that recombinant ACE2 can protect mice from severe acute lung injury. Our data identify a critical function for ACE2 in acute lung injury, pointing to a possible therapy for a syndrome affecting millions of people worldwide every year. SUPPLEMENTARY INFORMATION: The online version of this article (doi:10.1038/nature03712) contains supplementary material, which is available to authorized users.\n",
      "\n",
      "Title: Cellular autophagy: surrender, avoidance and subversion by microorganisms\n",
      "Authors: Kirkegaard, Karla; Taylor, Matthew P.; Jackson, William T.\n",
      "Publish time: 2004\n",
      "Abstract: Intracellular bacteria and viruses must survive the vigorous antimicrobial responses of their hosts to replicate successfully. The cellular process of autophagy — in which compartments bound by double membranes engulf portions of the cytosol and then mature to degrade their cytoplasmic contents — is likely to be one such host-cell response. Several lines of evidence show that both bacteria and viruses are vulnerable to autophagic destruction and that successful pathogens have evolved strategies to avoid autophagy, or to actively subvert its components, to promote their own replication. The molecular mechanisms of the avoidance and subversion of autophagy by microorganisms will be the subject of much future research, not only to study their roles in the replication of these microorganisms, but also because they will provide — as bacteria and viruses so often have — unique tools to study the cellular process itself.\n",
      "\n",
      "Title: The interferon response circuit: Induction and suppression by pathogenic viruses\n",
      "Authors: Haller, Otto; Kochs, Georg; Weber, Friedemann\n",
      "Publish time: 2006-01-05\n",
      "Abstract: Abstract Type I interferons (IFN-α/β) are potent antiviral cytokines and modulators of the adaptive immune system. They are induced by viral infection or by double-stranded RNA (dsRNA), a by-product of viral replication, and lead to the production of a broad range of antiviral proteins and immunoactive cytokines. Viruses, in turn, have evolved multiple strategies to counter the IFN system which would otherwise stop virus growth early in infection. Here we discuss the current view on the balancing act between virus-induced IFN responses and the viral counterplayers.\n",
      "\n",
      "\n",
      "Topic ID #7\n",
      "Title: Further studies on human enteric coronaviruses\n",
      "Authors: Caul, E. O.; Egglestone, S. I.\n",
      "Publish time: 1977\n",
      "Abstract: Comparisons were made between human enteric coronaviruses and the enteric coronaviruses of pigs and calves by negative staining. Examination of human intestinal organ culture fluids at various time intervals after inoculation with the human enteric coronavirus showed increasing numbers of particles in the fluids. Thin sections of the columnar epithelial cells of these explants showed a number of features consistent with the replication of known human and animal coronaviruses. Virus particles found in thin sections had a mean diameter of 68 nm. In addition, a structure was found in thin sections which has not been described previously. This structure may represent the viral nucleocapsid.\n",
      "\n",
      "Title: CORONAVIRUS PROPAGATED FROM PATIENT WITH NON-BACTERIAL GASTROENTERITIS\n",
      "Authors: Caul, E.O.; Clarke, S.K.R.\n",
      "Publish time: 1975-11-15\n",
      "Abstract: Abstract A fæcal suspension from a patient with gastroenteritis, which contained large numbers of coronavirus particles when examined by electron microscopy, was inoculated into human embryo intestinal-organ cultures and primary human embryo-kidney monolayers. In the organ cultures, the villous epithelium became detached and large numbers of coronavirus particles were seen by electron microscopy both in the medium and in thin sections of the intestinal epithelial cells, where the virus appeared to be multiplying. In organ cultures and in human embryo-kidney cells, intracytoplasmic virus was demonstrated by indirect immunofluorescence with convalescent serum from the patient.\n",
      "\n",
      "Title: Comparative full-length genome sequence analysis of 14 SARS coronavirus isolates and common mutations associated with putative origins of infection\n",
      "Authors: Ruan, Yijun; Wei, Chia Lin; Ling, Ai Ee; Vega, Vinsensius B; Thoreau, Herve; Se Thoe, Su Yun; Chia, Jer-Ming; Ng, Patrick; Chiu, Kuo Ping; Lim, Landri; Zhang, Tao; Chan, Kwai Peng; Lin Ean, Lynette Oon; Ng, Mah Lee; Leo, Sin Yee; Ng, Lisa FP; Ren, Ee Chee; Stanton, Lawrence W; Long, Philip M; Liu, Edison T\n",
      "Publish time: 2003-05-24\n",
      "Abstract: Summary Background The cause of severe acute respiratory syndrome (SARS) has been identified as a new coronavirus. Whole genome sequence analysis of various isolates might provide an indication of potential strain differences of this new virus. Moreover, mutation analysis will help to develop effective vaccines. Methods We sequenced the entire SARS viral genome of cultured isolates from the index case (SIN2500) presenting in Singapore, from three primary contacts (SIN2774, SIN2748, and SIN2677), and one secondary contact (SIN2679). These sequences were compared with the isolates from Canada (TOR2), Hong Kong (CUHK-W1 and HKU39849), Hanoi (URBANI), Guangzhou (GZ01), and Beijing (BJ01, BJ02, BJ03, BJ04). Findings We identified 129 sequence variations among the 14 isolates, with 16 recurrent variant sequences. Common variant sequences at four loci define two distinct genotypes of the SARS virus. One genotype was linked with infections originating in Hotel M in Hong Kong, the second contained isolates from Hong Kong, Guangzhou, and Beijing with no association with Hotel M (p<0.0001). Moreover, other common sequence variants further distinguished the geographical origins of the isolates, especially between Singapore and Beijing. Interpretation Despite the recent onset of the SARS epidemic, genetic signatures are emerging that partition the worldwide SARS viral isolates into groups on the basis of contact source history and geography. These signatures can be used to trace sources of infection. In addition, a common variant associated with a non-conservative aminoacid change in the S1 region of the spike protein, suggests that immunological pressures might be starting to influence the evolution of the SARS virus in human populations. Published online May 9, 2003 http://image.thelancet.com/extras/03art4454web.pdf\n",
      "\n",
      "Title: Phylogenetic analysis of a highly conserved region of the polymerase gene from 11 coronaviruses and development of a consensus polymerase chain reaction assay\n",
      "Authors: Stephensen, Charles B.; Casebolt, Donald B.; Gangopadhyay, Nupur N.\n",
      "Publish time: 1999-04-30\n",
      "Abstract: Abstract Viruses in the genus Coronavirus are currently placed in three groups based on antigenic cross-reactivity and sequence analysis of structural protein genes. Consensus polymerase chain reaction (PCR) primers were used to obtain cDNA, then cloned and sequenced a highly conserved 922 nucleotide region in open reading frame (ORF) 1b of the polymerase (pol) gene from eight coronaviruses. These sequences were compared with published sequences for three additional coronaviruses. In this comparison, it was found that nucleotide substitution frequencies (per 100 nucleotides) varied from 46.40 to 50.13 when viruses were compared among the traditional coronavirus groups and, with one exception (the human coronavirus (HCV) 229E), varied from 2.54 to 15.89 when compared within these groups. (The substitution frequency for 229E, as compared to other members of the same group, varied from 35.37 to 35.72.) Phylogenetic analysis of these pol gene sequences resulted in groupings which correspond closely with the previously described groupings, including recent data which places the two avian coronaviruses—infectious bronchitis virus (IBV) of chickens and turkey coronavirus (TCV)—in the same group [Guy, J.S., Barnes, H.J., Smith L.G., Breslin, J., 1997. Avian Dis. 41:583–590]. A single pair of degenerate primers was identified which amplify a 251 bp region from coronaviruses of all three groups using the same reaction conditions. This consensus PCR assay for the genus Coronavirus may be useful in identifying as yet unknown coronaviruses.\n",
      "\n",
      "Title: Feline Infectious Peritonitis\n",
      "Authors: Andrew, Stacy E.\n",
      "Publish time: 2000-09-30\n",
      "Abstract: Feline infectious peritonitis (FIP) occurs most frequently in young animals from multicat households. Ocular manifestations of FIP are extremely common with the noneffusive form of the disease. All cats suspected of having FIP should undergo a thorough ophthalmic examination. Uveitis (inflarrunation of the iris, ciliary body, and choroid) is the primary clinical sign of ocular FIP. Treatment with anti-inflammatory medications is only supportive and aimed at disease palliation.\n",
      "\n",
      "\n",
      "Topic ID #8\n",
      "Title: Severe Acute Respiratory Syndrome (SARS)\n",
      "Authors: Peiris, J.S.M\n",
      "Publish time: 2003-12-31\n",
      "Abstract: nan\n",
      "\n",
      "Title: Severe Acute Respiratory Syndrome\n",
      "Authors: Hui, David S.C.; Sung, Joseph J.Y.\n",
      "Publish time: 2003-07-31\n",
      "Abstract: nan\n",
      "\n",
      "Title: Glycyrrhizin, an active component of liquorice roots, and replication of SARS-associated coronavirus\n",
      "Authors: Cinatl, J; Morgenstern, B; Bauer, G; Chandra, P; Rabenau, H; Doerr, HW\n",
      "Publish time: 2003-06-14\n",
      "Abstract: Summary The outbreak of SARS warrants the search for antiviral compounds to treat the disease. At present, no specific treatment has been identified for SARS-associated coronavirus infection. We assessed the antiviral potential of ribavirin, 6-azauridine, pyrazofurin, mycophenolic acid, and glycyrrhizin against two clinical isolates of coronavirus (FFM-1 and FFM-2) from patients with SARS admitted to the clinical centre of Frankfurt University, Germany. Of all the compounds, glycyrrhizin was the most active in inhibiting replication of the SARS-associated virus. Our findings suggest that glycyrrhizin should be assessed for treatment of SARS.\n",
      "\n",
      "Title: In-flight transmission of Severe Acute Respiratory Syndrome (SARS): A Case Report\n",
      "Authors: Wilder-Smith, Annelies; Leong, Hoe Nam; Villacian, Jorge S.\n",
      "Publish time: 2006-03-08\n",
      "Abstract: nan\n",
      "\n",
      "Title: Analysis of synonymous codon usage in SARS Coronavirus and other viruses in the Nidovirales\n",
      "Authors: Gu, Wanjun; Zhou, Tong; Ma, Jianmin; Sun, Xiao; Lu, Zuhong\n",
      "Publish time: 2004-05-31\n",
      "Abstract: Abstract In this study, we calculated the codon usage bias in severe acute respiratory syndrome Coronavirus (SARSCoV) and performed a comparative analysis of synonymous codon usage patterns in SARSCoV and 10 other evolutionary related viruses in the Nidovirales. Although there is a significant variation in codon usage bias among different SARSCoV genes, codon usage bias in SARSCoV is a little slight, which is mainly determined by the base compositions on the third codon position. By comparing synonymous codon usage patterns in different viruses, we observed that synonymous codon usage pattern in these virus genes was virus specific and phylogenetically conserved, but it was not host specific. Phylogenetic analysis based on codon usage pattern suggested that SARSCoV was diverged far from all three known groups of Coronavirus. Compositional constraints could explain most of the variation of synonymous codon usage among these virus genes, while gene function is also correlated to synonymous codon usages to a certain extent. However, translational selection and gene length have no effect on the variations of synonymous codon usage in these virus genes.\n",
      "\n",
      "\n",
      "Topic ID #9\n",
      "Title: Angiotensin-converting enzyme 2 is a functional receptor for the SARS coronavirus\n",
      "Authors: Li, Wenhui; Moore, Michael J.; Vasilieva, Natalya; Sui, Jianhua; Wong, Swee Kee; Berne, Michael A.; Somasundaran, Mohan; Sullivan, John L.; Luzuriaga, Katherine; Greenough, Thomas C.; Choe, Hyeryun; Farzan, Michael\n",
      "Publish time: 2003\n",
      "Abstract: Spike (S) proteins of coronaviruses, including the coronavirus that causes severe acute respiratory syndrome (SARS), associate with cellular receptors to mediate infection of their target cells(1,2). Here we identify a metallopeptidase, angiotensin-converting enzyme 2 (ACE2)(3,4), isolated from SARS coronavirus (SARS-CoV)-permissive Vero E6 cells, that efficiently binds the S1 domain of the SARS-CoV S protein. We found that a soluble form of ACE2, but not of the related enzyme ACE1, blocked association of the S1 domain with Vero E6 cells. 293T cells transfected with ACE2, but not those transfected with human immunodeficiency virus-1 receptors, formed multinucleated syncytia with cells expressing S protein. Furthermore, SARS-CoV replicated efficiently on ACE2-transfected but not mock-transfected 293T cells. Finally, anti-ACE2 but not anti-ACE1 antibody blocked viral replication on Vero E6 cells. Together our data indicate that ACE2 is a functional receptor for SARS-CoV. SUPPLEMENTARY INFORMATION: The online version of this article (doi:10.1038/nature02145) contains supplementary material, which is available to authorized users.\n",
      "\n",
      "Title: The Molecular Biology of Coronaviruses\n",
      "Authors: Sturman, Lawrence S.; Holmes, Kathryn V.\n",
      "Publish time: 1983-12-31\n",
      "Abstract: Publisher Summary Coronaviruses have recently emerged as an important group of animal and human pathogens that share a distinctive replicative cycle. Some of the unique characteristics in the replication of coronaviruses include generation of a 3' coterminal-nested set of five or six subgenomic mRNAs, each of which appears to direct the synthesis of one protein. Two virus-specific RNA polymerase activities have been identified. Many of the distinctive features of coronavirus infection and coronavirus-induced diseases may result from the properties of the two coronavirus glycoproteins. The intracellular budding site, which may be important in the establishment and maintenance of persistent infections, appears to be due to the restricted intracytoplasmic migration of the E1 glycoprotein, which acts as a matrix-like transmembrane glycoprotein. E1 also exhibits distinctive behavior by self-aggregating on heating at 100°C in sodium dodecyl sulfate (SDS) and by its interaction with RNA in the viral nucleocapsid. The E1 of mouse hepatitis virus (MHV) is an O-linked glycoprotein, unlike most other viral glycoproteins. Thus, the coronavirus system may be a useful model for the study of synthesis, glycosylation, and transport of O-linked cellular glycoproteins.\n",
      "\n",
      "Title: Cloning and sequencing of the matrix protein (M) gene of turkey rhinotracheitis virus reveal a gene order different from that of respiratory syncytial virus\n",
      "Authors: Yu, Q.; Davis, P.J.; Li, J.; Cavanagh, D.\n",
      "Publish time: 1992-02-29\n",
      "Abstract: Abstract Several biochemical properties and the sequence of the fusion glycoprotein (F) have indicated that turkey rhinotracheitis virus (TRTV) is a pneumovirus, subfamily Pneumovirinae of the Paramyxoviridae family. As TRTV was known to generate polycistronic mRNAs, cDNA was generated from TRTV strain U K/3 BV/85-infected Vero cell mRNAs using an oligonucleotide primer corresponding to a region of the F gene. Sequencing of four cDNAs revealed that the gene adjacent to the beginning (3′ end) of the F gene was that for the matrix (M) protein, i.e., that TRTV had the partial gene order 3′-M-F-5′. This was unexpected as human respiratory syncytial (RS) virus, the type species of the genus Pneumovirus, has the partial gene order 3′-M-SH-G-F-5′, where SH and G are the small hydrophobic protein and attachment glycoprotein, respectively. Instead TRTV resembled the Morbillivirus and Paramyxovirus genera of the Paramyxoviridae (subfamily Paramyxoviridae) which have the partial gene order 3′-M-F-5′. Two further oligonucleotides, one corresponding to a sequence near the end of the M gene and the other (oligo B) to a sequence near the beginning of the F gene, with their 5′ ends spaced 300 nucleotides apart on the basis of the cDNA sequence, were used in a polymerase chain reaction (PCR) using genomic RNA as template. Only a PCR product of 0.3 kb was obtained. The same sized product was also obtained using these oligonucleotides and genomic RNA from three other TRTV strains (SA/91 /78, UK/8544/85, and SA/2381/88) which had been grown in chicken tracheal organ cultures. In addition PCR was performed using genomic RNA from TRTV-3BV and SA/2381/88 with oligo B and another oligonucleotide near the 5′ end of the gene upstream from M, spaced 1141 nucleotides apart on the basis of the sequence data. Only a 1.14-kb PCR product was obtained. Larger products would have been expected if another gene had been situated between M and F. The absence of such larger products, plus the demonstration that infected cells contained M-F dicistronic mRNAs, supported the conclusion that in the TRTV genome the M gene is adjacent to the F gene in the order 3'-M-F-5′. The 5′ termini of the M and F mRNAs were confirmed by mRNA mapping. The TRTV M gene encoded a protein of 254 amino acids, very similar to that of RS virus (256 residues; 37% amino acid identity) but very different from that of the morbilliviruses and paramyxoviruses (approximately 350 residues). Thus, on the basis of the sequences of both the M and the F genes, TRTV is more closely related to the pneumoviruses than to the paramyxoviruses and morbilliviruses.\n",
      "\n",
      "Title: Coronavirus Spike Proteins in Viral Entry and Pathogenesis\n",
      "Authors: Gallagher, Thomas M.; Buchmeier, Michael J.\n",
      "Publish time: 2001-01-20\n",
      "Abstract: nan\n",
      "\n",
      "Title: Pneumovirus-like characteristics of the mRNA and proteins of turkey rhinotracheitis virus\n",
      "Authors: Cavanagh, David; Barrett, Thomas\n",
      "Publish time: 1988-10-31\n",
      "Abstract: Abstract Electronmicroscopy has indicated that turkey rhinotracheitis virus (TRTV), the causative agent of an acute repiratory disease in turkeys, is a member of the Paramyxoviridae family. To determine if TRTV belongs to one of the three defined genera of this family (Paramyxovirus, Morbillivirus and Pneumovirus) we have analysed the RNA and proteins induced during replication of TRTV in Vero cells. Following replication in the presence of actinomycin D 10 polyadenylated RNA bands, ranging in M r from 0.22 to 2.0 × 106, were detected in infected cells; some bands probably contained 2 or more RNA species. Viral proteins were studied after radiolabelling in the presence of [35S]methionine and [3H]glucosamine. Comparison of the polypeptides in mock-infected and infected cells, virions and nucleocapsids and after lentil-lectin chromatography and immunoprecipitation revealed seven virus-specific polypeptides (p), some of which were glycosylated (gp): gp82 (M r 82K), gp68, gp53, gp15, p43, p40 and p35. These are considered to be analogous to the large glycopolypeptide (HN, H and G), fusion protein precursor F0, the F protein cleavage products F1 and F2, nucleocapsid (N), phosphorylated (P) and matrix (M) polypeptides, respectively, of the Paramyxoviridae. Two other polypeptides (M r 200K and 22K) were also detected, as was a glycopolypeptide of M r 97K, probably related to gp82. Tunicamycin inhibited glycosylation of gp53 and gp15 but gp82 was little affected, most glycans still being present on a glycopolypeptide of approximately 79K. This finding, indicating that gp82 has mostly O-linked glycans, considered with the mRNA profile and the molecular weight of the N protein shows that of the three genera in this family, TRTV most closely resembles the Pneumovirus genus.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from collections import defaultdict\n",
    "import numpy as np\n",
    "\n",
    "\n",
    "def display_paper(paper):\n",
    "    if isinstance(paper, list):\n",
    "        for elem in paper:\n",
    "            display_paper(elem)\n",
    "            print(\"\\n\", end=\"\")\n",
    "    else:\n",
    "        print(f\"\"\"Title: {paper.title}\n",
    "Authors: {paper.authors}\n",
    "Publish time: {paper.publish_time}\n",
    "Abstract: {paper.abstract}\"\"\")\n",
    "        \n",
    "        \n",
    "docs_topics = docs_classified.argmax(1)\n",
    "topic_papers = defaultdict(list)\n",
    "all_papers = list(pageranks.keys())\n",
    "for idx, topic_id in enumerate(docs_topics):\n",
    "    topic_papers[topic_id].append(all_papers[idx])\n",
    "for topic_id, papers in sorted(topic_papers.items(), key=lambda t: t[0]):\n",
    "    print(f\"Topic ID #{topic_id}\")\n",
    "    sorted_papers = sorted(papers, reverse=True, key=lambda p: pageranks[p])\n",
    "    display_paper(sorted_papers[:5])\n",
    "    print(\"\\n\", end=\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
